<!doctype html>
<!-- This site was created with Hugo Blox. https://hugoblox.com -->
<!-- Last Published: July 7, 2025 --><html lang="en-us" dir="ltr"
      data-wc-theme-default="system">
  
<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <meta name="generator" content="Hugo Blox Builder 0.3.1" />

  
  












  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Jiyuan Liu" />

  
  
  
    
  
  <meta name="description" content="Expected Value of Kinetic State Probability Vector This document explains how to calculate the expected value of the kinetic state probability vector $\pi_{ng}$ , where this expectation is taken twice: first conditional on a latent variable $z_n$ , and then averaged over the possible values of $z_n$ for a given cell.
" />

  
  <link rel="alternate" hreflang="en-us" href="http://localhost:1313/post/11b.velocity_velovi/velovi_expected_state_n_latent_time/" />

  
  
  
  
    
    <link rel="stylesheet" href="/css/themes/emerald.min.css" />
  

  
  
    
    <link href="/dist/wc.min.css" rel="stylesheet" />
  

  
  
  

  
    
    <link href="/css/custom.min.9a8c1508a999d285edca71446056519c3773544bdbe9fd11396354c75238a016.css" rel="stylesheet" />
  

  <script>
     
    window.hbb = {
       defaultTheme: document.documentElement.dataset.wcThemeDefault,
       setDarkTheme: () => {
        document.documentElement.classList.add("dark");
        document.documentElement.style.colorScheme = "dark";
      },
       setLightTheme: () => {
        document.documentElement.classList.remove("dark");
        document.documentElement.style.colorScheme = "light";
      }
    }

    console.debug(`Default Hugo Blox Builder theme is ${window.hbb.defaultTheme}`);

    if ("wc-color-theme" in localStorage) {
      localStorage.getItem("wc-color-theme") === "dark" ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
    } else {
      window.hbb.defaultTheme === "dark" ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
      if (window.hbb.defaultTheme === "system") {
        window.matchMedia("(prefers-color-scheme: dark)").matches ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
      }
    }
  </script>

  <script>
    
    document.addEventListener('DOMContentLoaded', function () {
      
      let checkboxes = document.querySelectorAll('li input[type=\'checkbox\'][disabled]');
      checkboxes.forEach(e => {
        e.parentElement.parentElement.classList.add('task-list');
      });

      
      const liNodes = document.querySelectorAll('.task-list li');
      liNodes.forEach(nodes => {
        let textNodes = Array.from(nodes.childNodes).filter(node => node.nodeType === 3 && node.textContent.trim().length > 1);
        if (textNodes.length > 0) {
          const span = document.createElement('label');
          textNodes[0].after(span);  
          span.appendChild(nodes.querySelector('input[type=\'checkbox\']'));
          span.appendChild(textNodes[0]);
        }
      });
    });
  </script>

  
  
  




































  
  

  
  <link rel="icon" type="image/png" href="/media/icon_hu16496738209820128384.png" />
  <link rel="apple-touch-icon" type="image/png" href="/media/icon_hu17763857046462676418.png" />

  <link rel="canonical" href="http://localhost:1313/post/11b.velocity_velovi/velovi_expected_state_n_latent_time/" />

  
  
  
  
  
  
  
  
    
    
  
  

  
  
    
    
  
  <meta property="twitter:card" content="summary" />
  
    <meta property="twitter:site" content="@GetResearchDev" />
    <meta property="twitter:creator" content="@GetResearchDev" />
  
  <meta property="og:site_name" content="Jiyuan Liu" />
  <meta property="og:url" content="http://localhost:1313/post/11b.velocity_velovi/velovi_expected_state_n_latent_time/" />
  <meta property="og:title" content="Jiyuan Liu" />
  <meta property="og:description" content="Expected Value of Kinetic State Probability Vector
This document explains how to calculate the expected value of the kinetic state probability vector $\pi_{ng}$
, where this expectation is taken twice: first conditional on a latent variable $z_n$
, and then averaged over the possible values of $z_n$
 for a given cell." /><meta property="og:image" content="http://localhost:1313/media/icon_hu6671943710606299486.png" />
    <meta property="twitter:image" content="http://localhost:1313/media/icon_hu6671943710606299486.png" /><meta property="og:locale" content="en-us" />
  
    
    
  

  



  


  <title>Jiyuan Liu</title>

  
  
  
  
  
    
    
  
  
  <style>
    @font-face {
      font-family: 'Inter var';
      font-style: normal;
      font-weight: 100 900;
      font-display: swap;
      src: url(/dist/font/Inter.var.woff2) format(woff2);
    }
  </style>

  

  
  


  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
    
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  







<link type="text/css" rel="stylesheet" href="/dist/pagefind/pagefind-ui.be766eb419317a14ec769d216e9779bfe8f3737c80e780f4ba0dafb57a41a482.css" integrity="sha256-vnZutBkxehTsdp0hbpd5v&#43;jzc3yA54D0ug2vtXpBpII=" />


<script src="/dist/pagefind/pagefind-ui.87693d7c6f2b3b347ce359d0ede762c033419f0a32b22ce508c335a81d841f1b.js" integrity="sha256-h2k9fG8rOzR841nQ7ediwDNBnwoysizlCMM1qB2EHxs="></script>


<script>window.hbb.pagefind = {"baseUrl":"/"};</script>

<style>
  html.dark {
    --pagefind-ui-primary: #eeeeee;
    --pagefind-ui-text: #eeeeee;
    --pagefind-ui-background: #152028;
    --pagefind-ui-border: #152028;
    --pagefind-ui-tag: #152028;
  }
</style>

<script>
  window.addEventListener('DOMContentLoaded', (event) => {
    new PagefindUI({
      element: "#search",
      showSubResults: true,
      baseUrl: window.hbb.pagefind.baseUrl,
      bundlePath: window.hbb.pagefind.baseUrl + "pagefind/",
    });
  });
  document.addEventListener('DOMContentLoaded', () => {
    let element = document.getElementById('search');
    let trigger = document.getElementById('search_toggle');

    if (trigger) {
      trigger.addEventListener('click', () => {
        element.classList.toggle('hidden');
        element.querySelector("input").value = ""
        element.querySelector("input").focus()

        if (!element.classList.contains('hidden')) {
          let clear_trigger = document.querySelector('.pagefind-ui__search-clear');

          if (clear_trigger && !clear_trigger.hasAttribute('listenerOnClick')) {
            clear_trigger.setAttribute('listenerOnClick', 'true');

            clear_trigger.addEventListener('click', () => {
              element.classList.toggle('hidden');
            });
          }
        }

      });
    }
  });
</script>










  
  
  <link type="text/css" rel="stylesheet" href="/dist/lib/katex/katex.min.505d5f829022bb7b4f24dfee0aa1141cd7bba67afe411d1240335f820960b5c3.css" integrity="sha256-UF1fgpAiu3tPJN/uCqEUHNe7pnr&#43;QR0SQDNfgglgtcM=" />
  
  
  <script defer src="/dist/lib/katex/katex.min.dc84b296ec3e884de093158f760fd9d45b6c7abe58b5381557f4e138f46a58ae.js" integrity="sha256-3ISyluw&#43;iE3gkxWPdg/Z1Ftser5YtTgVV/ThOPRqWK4="></script>
  
  
  
  
  <script defer src="/js/katex-renderer.6579ec9683211cfb952064aedf3a3baea5eeb17a061775b32b70917474637c80.js" integrity="sha256-ZXnsloMhHPuVIGSu3zo7rqXusXoGF3WzK3CRdHRjfIA="></script>
  
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  






  
  
  
  
  
  
  
  <script
    defer
    src="/js/hugo-blox-en.min.js"
    integrity=""
  ></script>

  
  








  
    
      
      <script async defer src="https://buttons.github.io/buttons.js"></script>

      
    
  




</head>

  <body class="dark:bg-hb-dark dark:text-white page-wrapper" id="top">
    <div id="page-bg"></div>
    <div class="page-header sticky top-0 z-30">
      
      
      
        
        
        
          <header id="site-header" class="header">
  <nav class="navbar px-3 flex justify-left">
    <div class="order-0 h-100">
      
      <a class="navbar-brand" href="/" title="Jiyuan Liu">
        Home
      </a>
    </div>
    
    <input id="nav-toggle" type="checkbox" class="hidden" />
    <label
      for="nav-toggle"
      class="order-3 cursor-pointer flex items-center lg:hidden text-dark dark:text-white lg:order-1">
      <svg id="show-button" class="h-6 fill-current block" viewBox="0 0 20 20">
        <title>Open Menu</title>
        <path d="M0 3h20v2H0V3z m0 6h20v2H0V9z m0 6h20v2H0V0z"></path>
      </svg>
      <svg id="hide-button" class="h-6 fill-current hidden" viewBox="0 0 20 20">
        <title>Close Menu</title>
        <polygon
          points="11 9 22 9 22 11 11 11 11 22 9 22 9 11 -2 11 -2 9 9 9 9 -2 11 -2"
          transform="rotate(45 10 10)"></polygon>
      </svg>
    </label>
    

    
    
    <ul
      id="nav-menu"
      class="navbar-nav order-3 hidden lg:flex w-full pb-6 lg:order-1 lg:w-auto lg:space-x-2 lg:pb-0 xl:space-x-8 justify-left
      ">
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/"
        >Bio</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/#papers"
        >Papers</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/#talks"
        >Talks</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/post"
        >Blogs</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/experience/"
        >Experience</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/projects/"
        >Projects</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/teaching/"
        >Learning</a
        >
      </li>
      
      
      
    </ul>

    <div class="order-1 ml-auto flex items-center md:order-2 lg:ml-0">

      
      
      
      <button
        aria-label="search"
        class="text-black hover:text-primary  inline-block px-3 text-xl dark:text-white"
        id="search_toggle">
        <svg xmlns="http://www.w3.org/2000/svg" height="16" width="16" viewBox="0 0 512 512" fill="currentColor"><path d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"/></svg>
      </button>
      

      
      
      <div class="px-3 text-black hover:text-primary-700 dark:text-white dark:hover:text-primary-300
            [&.active]:font-bold [&.active]:text-black/90 dark:[&.active]:text-white">
        <button class="theme-toggle mt-1" accesskey="t" title="appearance">
          <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
               fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
               stroke-linejoin="round" class="dark:hidden">
            <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
          </svg>
          <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
               fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
               stroke-linejoin="round" class=" dark:block [&:not(dark)]:hidden">
            <circle cx="12" cy="12" r="5"></circle>
            <line x1="12" y1="1" x2="12" y2="3"></line>
            <line x1="12" y1="21" x2="12" y2="23"></line>
            <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
            <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
            <line x1="1" y1="12" x2="3" y2="12"></line>
            <line x1="21" y1="12" x2="23" y2="12"></line>
            <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
            <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
          </svg>
        </button>
      </div>
      

      
      

      
      
    </div>
  </nav>
</header>


<div id="search" class="hidden p-3"></div>


        
      
    </div>
    <div class="page-body  my-10">
      





<div class="mx-auto flex max-w-screen-xl">
  



<aside class="hb-sidebar-container max-lg:[transform:translate3d(0,-100%,0)] lg:hidden xl:block">
  
  <div class="px-4 pt-4 lg:hidden">
    
    
  </div>
  <div class="hb-scrollbar lg:h-[calc(100vh-var(--navbar-height))]">
    <ul class="flex flex-col gap-1 lg:hidden">
      
      
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/event/"
    
  >Recent &amp; Upcoming Talks
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/event/example/"
    
  >Example Talk
    </a>
              
            </li></ul>
      </div></li>
        <li class="open"><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/"
    
  >Computational Blogs
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11c.velocity_steady_state/"
    
  >üß¨ Math derivation for steady-state RNA velocity model
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19c.bayes_variational_infer_elbo/"
    
  >üéì Math Derivation for ELBO/KL in Bayesian Inference and VAEs
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19b.scrnaseq_impute_saver/"
    
  >üßÆ Math Derivation for scRNAseq Imputation (SAVER)
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19a.conjugate_prior_bayes/"
    
  >üìä Math Derivation in Bayesian Inference-- Bypassing Marginals Using Conjugate Priors
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19.ememwithmapvariationalbayesian/"
    
  >üß¨ EM algorithm, EM with max a posterior, full Bayesian inferences (variational Bayesian) which might be used in auto-coder
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/18.what_make_neural_network_powerful_popular/"
    
  >üß¨ what makes neural network so powerful and popular
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/17.smooth.tests.of.goodness.of.fit/"
    
  >üß¨ Smooth tests of goodness of Fit used in Stijn Hawinkel ‚Äôs plos one paper
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/16.how.good.is.negative.binominal/"
    
  >üß¨ how good is negative binomial, why AI works for biology
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/15.use.seurat.for.bulk.rnaseq/"
    
  >üß¨ Seurat to analysis large sample size of bulk RNAseq. Discuss Wixcox is better for clinical samples
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/13z.scvi_integrate_scrna_spatial/"
    
  >üß¨ Deep Neural Network to integrate scRNAseq and spatial genomics
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/14.spatial_genomics_presentation/"
    
  >üß¨ Spatial genomics presentation
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/12.basian_abundance_statistics/"
    
  >üß¨ Bayesian abundance statistics in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/13.yolov8/"
    
  >üß¨ Yolov8
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11a.velocity_unraveled/"
    
  >üß¨ Math Derivation of CME-defined Stochastic Model of RNA Velocity
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11.velocity_reasoning/"
    
  >üß¨ Velocity reasoning and discussion
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/10.cca_reasoning/"
    
  >üß¨ CCA reasoning and integration discussion
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/09.sctransform/"
    
  >üß¨ sctransfrom reasoning
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/08.clinical_relevance/"
    
  >üß¨ Clinical relevance evaluation in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/07.chord_diagram/"
    
  >üß¨ Expressive cord diagram in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/06.noise_or_true_stem_cell/"
    
  >üß¨ Determine whether it‚Äôs noise or true stem like cells in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/05.xensoft_caki_paper/"
    
  >üß¨ quantify xenograft and host gene expression from the same sample
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/04.huangcong_oncogene/"
    
  >üß¨ counteract treatment-specific background in differential expressed gene analysis (huangcong paper)
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/03.well-based_scrnaseq_herv_quantification/"
    
  >üß¨ A snakemake pipline-- quantify hERV, transgenes and genome from well-based single-cell RNAseq
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/02.snakemake_quantify-herv_10xscrnaseq/"
    
  >üß¨ A snakemake pipline-- quantify hERV, transgenes and genome from 10x scRNAseq
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/01a.repeatmasker/"
    
  >üõ†Ô∏è Generate and validate ERV gtf that matches the latest reference genome
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/01.bulkrnaseq_allow_herv_quantification/"
    
  >üß¨ Quantify hERV and transgenes from Bulk-RNAseq
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/00a.seqwins/"
    
  >‚ö° SeqWins, An R Package for Flexible Base Trimming and Comprehensive FASTQ Analysis on Windows
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/00.model_aerosol_depostion/"
    
  >üí® Modeling the deposition of infectious agents suggests that aerosolized drugs offer enhanced specificity
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.get-started/"
    
  >üéâ Easily create your own simple yet highly customizable blog
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.second-brain/"
    
  >üß† Sharpen your thinking with a second brain
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.data-visualization/"
    
  >üìà Communicate your results effectively with the best data visualizations
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.teach-courses/"
    
  >üë©üèº‚Äçüè´ Teach academic courses
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.project-management/"
    
  >‚úÖ Manage your projects
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_derive_time-varying_alpha_model/"
    
  >velovi_derive_time-varying_alpha_model
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_elbo_kl_breakdown/"
    
  >velovi_elbo_kl_breakdown
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_elbo_term_with_expectation/"
    
  >velovi_elbo_term_with_expectation
    </a>
              
            </li><li class="flex flex-col open"><a
    class="hb-sidebar-custom-link
      sidebar-active-item bg-primary-100 font-semibold text-primary-800 dark:bg-primary-300 dark:text-primary-900"
    href="/post/11b.velocity_velovi/velovi_expected_state_n_latent_time/"
    
  >velovi_expected_state_n_latent_time
    </a>
  
    <ul class="hb-sidebar-mobile-toc"><li>
              <a
                href="#understanding-the-components"
                class="hb-docs-link"
              >Understanding the Components</a>
            </li>
          <li>
              <a
                href="#step-1-calculate-the-inner-expectation"
                class="hb-docs-link"
              >Step 1: Calculate the Inner Expectation</a>
            </li>
          <li>
              <a
                href="#step-2-calculate-the-outer-expectation"
                class="hb-docs-link"
              >Step 2: Calculate the Outer Expectation</a>
            </li>
          <li>
              <a
                href="#derive-the-inner-expectation-for-time"
                class="hb-docs-link"
              >Derive the Inner Expectation for Time</a>
            </li>
          <li>
              <a
                href="#numeric-example-for-expected-time"
                class="hb-docs-link"
              >Numeric Example for Expected Time</a>
            </li>
          <li>
              <a
                href="#summary-table"
                class="hb-docs-link"
              >Summary Table</a>
            </li>
          <li>
              <a
                href="#1-recall-the-general-expectation-for-a-discrete-variable"
                class="hb-docs-link"
              >1. Recall the General Expectation for a Discrete Variable</a>
            </li>
          <li>
              <a
                href="#2-applying-to-our-velocity-case"
                class="hb-docs-link"
              >2. Applying to Our Velocity Case</a>
            </li>
          <li>
              <a
                href="#numeric-example-for-cell-1-gene-a-2"
                class="hb-docs-link"
              >Numeric Example for Cell 1, Gene A</a>
            </li>
          <li>
              <a
                href="#get-posterior-predictive-velocity-distribution"
                class="hb-docs-link"
              >Get Posterior Predictive Velocity Distribution</a>
            </li>
          </ul>
  
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_model_n_inference/"
    
  >velovi_model_n_inference
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_parameter_initialize_n_learn/"
    
  >velovi_parameter_initialize_n_learn
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_posterior_predictive_distribution/"
    
  >velovi_posterior_predictive_distribution
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_reparameterization_trick/"
    
  >velovi_reparameterization_trick
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_uncertainty_calculation/"
    
  >velovi_uncertainty_calculation
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/"
    
  >learning
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/visualize_distribution/"
    
  >Dirichlet Distribution
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/stanford_ml_em/"
    
  >Lecture 14 - Expectation-Maximization Algorithms | Stanford CS229 Machine Learning
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/linear_algebra/"
    
  >linear algebra
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/other/home_improvement/"
    
  >linear algebra
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/method_of_charateristics/"
    
  >Method of characteristics
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/method_of_seperation_of_variables/"
    
  >Separation of Variables
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/js/"
    
  >Learn JavaScript
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/python/"
    
  >Learn Python
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/projects/"
    
  >Projects
    </a></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/"
    
  >Projects
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/pandas/"
    
  >Pandas
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/pytorch/"
    
  >PyTorch
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/scikit/"
    
  >scikit-learn
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/experience/"
    
  >Experience
    </a></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/"
    
  >Publications
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/preprint/"
    
  >An example preprint / working paper
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/journal-article/"
    
  >An example journal article
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/conference-paper/"
    
  >An example conference paper
    </a>
              
            </li></ul>
      </div></li>
    </ul>

    <div class="max-xl:hidden h-0 w-64 shrink-0"></div></div>

</aside>
  

<nav class="hb-toc order-last hidden w-64 shrink-0 xl:block print:hidden px-4" aria-label="table of contents">
  











  <div class="hb-scrollbar text-sm [hyphens:auto] sticky top-16 overflow-y-auto pr-4 pt-6 max-h-[calc(100vh-var(--navbar-height)-env(safe-area-inset-bottom))] -mr-4 rtl:-ml-4"><p class="mb-4 font-semibold tracking-tight">On this page</p><ul>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#understanding-the-components">Understanding the Components</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#step-1-calculate-the-inner-expectation">Step 1: Calculate the Inner Expectation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#numeric-example-for-cell-1-gene-a">Numeric Example for Cell 1, Gene A</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#step-2-calculate-the-outer-expectation">Step 2: Calculate the Outer Expectation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#monte-carlo-approximation">Monte Carlo Approximation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#numeric-example-continued">Numeric Example Continued</a>
      </li></ul><ul>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#derive-the-inner-expectation-for-time">Derive the Inner Expectation for Time</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#numeric-example-for-expected-time">Numeric Example for Expected Time</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#what-we-assume-we-have-after-velovi-training">What We Assume We Have (after VeloVI training)</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#step-1-calculate-the-inner-expectation-1">Step 1: Calculate the Inner Expectation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#recall-the-inner-expectation">Recall the Inner Expectation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#numeric-example-for-cell-1-gene-a-1">Numeric Example for Cell 1, Gene A</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#step-2-monte-carlo-approximation-calculates-the-outer-expectation">Step 2: Monte Carlo Approximation Calculates the Outer Expectation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#summary-table">Summary Table</a>
      </li></ul><ul>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#1-recall-the-general-expectation-for-a-discrete-variable">1. Recall the General Expectation for a Discrete Variable</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#2-applying-to-our-velocity-case">2. Applying to Our Velocity Case</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#numeric-example-for-cell-1-gene-a-2">Numeric Example for Cell 1, Gene A</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#calculations-for-each-state-k-and-its-corresponding-hahahugoshortcode13s261hbhb">Calculations for each state k and its corresponding HAHAHUGOSHORTCODE13s261HBHB:</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#calculate-the-weighted-sum">Calculate the weighted sum:</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#get-posterior-predictive-velocity-distribution">Get Posterior Predictive Velocity Distribution</a>
      </li></ul>

  
  
    
    
  



    












  </div>
  </nav>


  <article class="w-full break-words flex min-h-[calc(100vh-var(--navbar-height))] min-w-0 justify-center pb-8 pr-[calc(env(safe-area-inset-right)-1.5rem)]">
    <main class="w-full min-w-0 max-w-6xl px-6 pt-4 md:px-12">

      

      <h1 class="mt-2 text-4xl font-bold tracking-tight text-slate-900 dark:text-slate-100"></h1>

      <div class="mt-4 mb-16">
      <div class="text-gray-500 dark:text-gray-300 text-sm flex items-center flex-wrap gap-y-2">
        

        
        <span class="mx-1">¬∑</span>
        <span class="mx-1">
          15 min read
        </span>
        
        </div>

        <div class="mt-3">
          





        </div>
      </div>



      
      
      

      
      

      <div class="prose prose-slate lg:prose-xl dark:prose-invert">
        <h1 id="expected-value-of-kinetic-state-probability-vector">Expected Value of Kinetic State Probability Vector</h1>
<p>This document explains how to calculate the expected value of the kinetic state probability vector $\pi_{ng}$
, where this expectation is taken twice: first conditional on a latent variable $z_n$
, and then averaged over the possible values of $z_n$
 for a given cell.</p>
<p>The main expression we&rsquo;re evaluating is:</p>

$$
\mathbb{E}_{q_\phi(z_n|u_n,s_n)} \left[ \mathbb{E}_{q_\phi(\pi_{ng}|z_n)} [\pi_{ng}] \right] \tag{1}
$$


<p>This represents VeloVI&rsquo;s average belief about the kinetic state probabilities ($\pi_{ng}$
) for gene $g$
 in cell $n$
, considering all possible latent representations ($z_n$
) that could explain the observed data ($u_n, s_n$
).</p>
<h2 id="understanding-the-components">Understanding the Components</h2>
<ul>
<li><strong>$\pi_{ng}$
</strong>: The vector of probabilities for the 4 kinetic states: $(\pi_{ng,1}, \pi_{ng,2}, \pi_{ng,3}, \pi_{ng,4})$
</li>
<li><strong>$q_\phi(\pi_{ng}|z_n)$
</strong>: The approximate posterior distribution for $\pi_{ng}$
 given a specific latent variable $z_n$
. This is a Dirichlet distribution with concentration parameters $\alpha_{q,ng}(z_n) = (\alpha_{q,ng,1}(z_n), \ldots, \alpha_{q,ng,K}(z_n))$
</li>
<li><strong>$q_\phi(z_n|u_n,s_n)$
</strong>: The approximate posterior distribution for the latent variable $z_n$
 given the observed counts. This is a Gaussian distribution $\mathcal{N}(\mu_{z_n}, \Sigma_{z_n})$
</li>
</ul>
<h2 id="step-1-calculate-the-inner-expectation">Step 1: Calculate the Inner Expectation</h2>
<p>The inner expectation $\mathbb{E}_{q_\phi(\pi_{ng}|z_n)} [\pi_{ng}]$
 is the expected value of a Dirichlet-distributed random variable.</p>
<p>If $\pi_{ng} \sim \text{Dirichlet}(\alpha_{q,ng,1}(z_n), \ldots, \alpha_{q,ng,K}(z_n))$
, then the expected value of its $i$
-th component is:</p>

$$
\mathbb{E}_{q_\phi(\pi_{ng}|z_n)} [\pi_{ng,i}] = \frac{\alpha_{q,ng,i}(z_n)}{\sum_{j=1}^K \alpha_{q,ng,j}(z_n)} \tag{2}
$$


<p>The result of the inner expectation is a vector of expected probabilities:</p>

$$
\bar{\pi}_{ng}(z_n) = \left( \frac{\alpha_{q,ng,1}(z_n)}{\sum_j \alpha_{q,ng,j}(z_n)}, \ldots, \frac{\alpha_{q,ng,K}(z_n)}{\sum_j \alpha_{q,ng,j}(z_n)} \right) \tag{3}
$$


<h3 id="numeric-example-for-cell-1-gene-a">Numeric Example for Cell 1, Gene A</h3>
<p>Let&rsquo;s assume we sample $z_1^* = (0.6558, -0.3779)$
.</p>
<p>The encoder outputs Dirichlet concentration parameters for Gene A:
$\alpha_{q,1A}(z_1^*) = (5.0, 0.1, 0.2, 0.1)$
</p>
<p>Calculate the expected $\pi_{1A}$
 for this $z_1^*$
:</p>
<ul>
<li>Sum of concentrations: $5.0 + 0.1 + 0.2 + 0.1 = 5.4$
</li>
<li>$\mathbb{E}[\pi_{1A,1}|z_1^*] = 5.0/5.4 \approx 0.9259$
</li>
<li>$\mathbb{E}[\pi_{1A,2}|z_1^*] = 0.1/5.4 \approx 0.0185$
</li>
<li>$\mathbb{E}[\pi_{1A,3}|z_1^*] = 0.2/5.4 \approx 0.0370$
</li>
<li>$\mathbb{E}[\pi_{1A,4}|z_1^*] = 0.1/5.4 \approx 0.0185$
</li>
</ul>
<p>So $\bar{\pi}_{1A}(z_1^*) = (0.9259, 0.0185, 0.0370, 0.0185)$
.</p>
<h2 id="step-2-calculate-the-outer-expectation">Step 2: Calculate the Outer Expectation</h2>
<p>Now we need to average $\bar{\pi}_{ng}(z_n)$
 over the distribution $q_\phi(z_n|u_n,s_n) = \mathcal{N}(\mu_{z_n}, \Sigma_{z_n})$
. Since $\bar{\pi}_{ng}(z_n)$
 involves a neural network mapping, this expectation is intractable and we use Monte Carlo approximation.</p>
<h3 id="monte-carlo-approximation">Monte Carlo Approximation</h3>
<p>For each sample $m \in \{1, \ldots, M\}$
:</p>
<ol>
<li>Draw random noise: $\epsilon_z^{(m)} \sim \mathcal{N}(0, I_d)$
</li>
<li>Calculate sample: $z_n^{*(m)} = \mu_{z_n} + \sqrt{\Sigma_{z_n}} \odot \epsilon_z^{(m)}$
 (reparameterization trick)</li>
</ol>
<h3 id="numeric-example-continued">Numeric Example Continued</h3>
<p><strong>Known parameters for Cell 1:</strong></p>
<ul>
<li>$\mu_{z_1} = (0.5, -0.3)$
</li>
<li>$\log\Sigma_{z_1} = (-0.5, -0.5) \rightarrow \Sigma_{z_1} = \text{diag}(0.6065, 0.6065)$
</li>
</ul>
<p><strong>Sample 1:</strong></p>
<ul>
<li>$\epsilon_z^{(1)} = (0.2, -0.1)$
</li>
<li>$z_1^{*(1)} = (0.5, -0.3) + (0.7788, 0.7788) \odot (0.2, -0.1) = (0.6558, -0.3779)$
</li>
</ul>
<p><strong>Sample 2:</strong></p>
<ul>
<li>$\epsilon_z^{(2)} = (-0.5, 0.3)$
</li>
<li>$z_1^{*(2)} = (0.5, -0.3) + (0.7788, 0.7788) \odot (-0.5, 0.3) = (0.1106, -0.0663)$
</li>
</ul>
<p>For $z_1^{*(2)}$
, assume encoder outputs $\alpha_{q,1A}(z_1^{*(2)}) = (4.0, 0.3, 0.1, 0.1)$
:</p>
<ul>
<li>Sum of concentrations: $4.0 + 0.3 + 0.1 + 0.1 = 4.5$
</li>
<li>$\bar{\pi}_{1A}^{(2)} \approx (0.8889, 0.0667, 0.0222, 0.0222)$
</li>
</ul>
<p><strong>Final average (with M=2):</strong></p>

$$
\mathbb{E}_{q_\phi(z_n|u_n,s_n)} [\bar{\pi}_{ng}(z_n)] \approx \frac{1}{M} \sum_{m=1}^M \bar{\pi}_{ng}^{(m)} \tag{4}
$$


<ul>
<li>Component 1: $(0.9259 + 0.8889)/2 = 0.9074$
</li>
<li>Component 2: $(0.0185 + 0.0667)/2 = 0.0426$
</li>
<li>Component 3: $(0.0370 + 0.0222)/2 = 0.0296$
</li>
<li>Component 4: $(0.0185 + 0.0222)/2 = 0.0204$
</li>
</ul>
<p><strong>Final result:</strong> $\approx (0.9074, 0.0426, 0.0296, 0.0204)$
</p>
<hr>
<h1 id="expected-latent-time-calculation">Expected Latent Time Calculation</h1>
<p>Now we calculate the average latent time $t_{ng}$
 for gene $g$
 in cell $n$
, accounting for uncertainty in both the cell&rsquo;s latent representation and the kinetic state.</p>
<p>The expression is:</p>

$$
\mathbb{E}_{q_\phi(z_n|u_n,s_n)} \left[ \mathbb{E}_{q_\phi(\pi_{ng}|z_n)} [t^{(k_{ng})}_{ng}] \right] \tag{5}
$$


<p>where $t^{(k_{ng})}_{ng}$
 represents the latent time for gene $g$
 in cell $n$
 if it&rsquo;s in kinetic state $k_{ng}$
.</p>
<h2 id="derive-the-inner-expectation-for-time">Derive the Inner Expectation for Time</h2>
<p>The inner expectation over kinetic states is:</p>

$$
\mathbb{E}_{q_\phi(\pi_{ng}|z_n)} [t^{(k_{ng})}_{ng}] = \sum_{k=1}^K P(k_{ng} = k | z_n) \cdot t_{ng}^{(k)}(z_n) \tag{6}
$$


<p>Where:</p>
<ul>
<li>$P(k_{ng} = k | z_n) = \frac{\alpha_{q,ng,k}(z_n)}{\sum_{j=1}^K \alpha_{q,ng,j}(z_n)}$
</li>
<li>$t_{ng}^{(k)}(z_n) = \rho_{ng}^{(k)}(z_n) \times t_g^s$
 (decoder output times switching time)</li>
</ul>
<p>In our expression $\mathbb{E}_{q_\phi(\pi_{ng}|z_n)} [t^{(k_{ng})}_{ng}]$
:</p>
<ul>
<li>
<p><strong>The random variable:</strong> The random variable is $t^{(k_{ng})}_{ng}$
. This variable&rsquo;s value <em>depends</em> on the discrete kinetic state $k_{ng}$
.</p>
</li>
<li>
<p><strong>The Possible Values of the Random Variable</strong></p>
<ul>
<li>The kinetic state $k_{ng}$
 can take on $K$
 discrete values (1, 2, 3, 4 for VeloVI&rsquo;s four states).</li>
<li>For each specific state $k$
, the corresponding value of our random variable is $t_{ng}^{(k)}(z_n)$
.</li>
<li>This notation $t_{ng}^{(k)}(z_n)$
 explicitly shows that the latent time is specific to state $k$
 and is determined by $z_n$
.</li>
</ul>
</li>
<li>
<p><strong>The Probability Distribution</strong></p>
<ul>
<li>The expectation is taken with respect to $q_\phi(\pi_{ng}|z_n)$
.</li>
<li>This distribution tells us the probabilities of being in each kinetic state $k$
.</li>
<li>Specifically, $\pi_{ng}$
 is a vector of probabilities for the states, and $q_\phi(\pi_{ng}|z_n)$
 is a Dirichlet distribution that <em>governs</em> these probabilities.</li>
</ul>
</li>
</ul>
<p>We know that $k_{ng} \sim \text{Categorical}(\pi_{ng})$
. This means, if we knew the exact value of $\pi_{ng}$
, then $P(k_{ng}=k|\pi_{ng}) = \pi_{ng,k}$
.</p>
<p>However, $\pi_{ng}$
 itself is a random variable distributed according to $q_\phi(\pi_{ng}|z_n)$
.</p>
<p>So, to get the marginal probability $P(k_{ng}=k|z_n)$
, we need to integrate over $\pi_{ng}$
:</p>

$$
P(k_{ng}=k|z_n) = \int P(k_{ng}=k|\pi_{ng}) q_\phi(\pi_{ng}|z_n) d\pi_{ng} \tag{6.1}
$$



$$
P(k_{ng}=k|z_n) = \int \pi_{ng,k} q_\phi(\pi_{ng}|z_n) d\pi_{ng} \tag{6.2}
$$


<p>This integral is simply the definition of the expected value of the $k$
-th component of $\pi_{ng}$
 under the distribution $q_\phi(\pi_{ng}|z_n)$
:</p>

$$
P(k_{ng}=k|z_n) = \mathbb{E}_{q_\phi(\pi_{ng}|z_n)} [\pi_{ng,k}] \tag{6.3}
$$


<p>For a Dirichlet distribution $\text{Dir}(\alpha_{q,ng,1}(z_n), \ldots, \alpha_{q,ng,K}(z_n))$
, this expectation has a closed form:</p>

$$
\mathbb{E}_{q_\phi(\pi_{ng}|z_n)} [\pi_{ng,k}] = \frac{\alpha_{q,ng,k}(z_n)}{\sum_{j=1}^K \alpha_{q,ng,j}(z_n)} \tag{6.4}
$$


<p>So, $P(k_{ng}=k|z_n)$
 is precisely this fraction.</p>
<p>Therefore:</p>

$$
\mathbb{E}_{q_\phi(\pi_{ng}|z_n)} [t^{(k_{ng})}_{ng}] = \sum_{k=1}^K \left( t_{ng}^{(k)}(z_n) \cdot P(k_{ng}=k|z_n) \right) \tag{6}
$$


<p>Substituting equation (4) into equation (6):</p>

$$
\mathbb{E}_{q_\phi(\pi_{ng}|z_n)} [t^{(k_{ng})}_{ng}] = \sum_{k=1}^K \left( t_{ng}^{(k)}(z_n) \cdot \frac{\alpha_{q,ng,k}(z_n)}{\sum_{j=1}^K \alpha_{q,ng,j}(z_n)} \right) \tag{6.5}
$$


<p>This is exactly the equation we use in practice, explicitly showing that the expectation is a <strong>weighted sum</strong> where:</p>
<ul>
<li>Each possible latent time $t_{ng}^{(k)}(z_n)$
 is weighted by</li>
<li>The probability of its corresponding state $k$
 occurring, as determined by the approximate posterior $q_\phi(\pi_{ng}|z_n)$
.</li>
</ul>
<h2 id="numeric-example-for-expected-time">Numeric Example for Expected Time</h2>
<p>This expression asks for the average latent time $t_{ng}$
 for gene g in cell n, taking into account both:</p>
<ul>
<li>The uncertainty in the cell&rsquo;s underlying latent representation $z_n$
</li>
<li>The uncertainty in which kinetic state $k_{ng}$
 gene g is in</li>
</ul>
<p>Similar to the previous expectation, this involves an intractable integral, so we will use Monte Carlo sampling for approximation.</p>
<p>The expression is:</p>

$$\mathbb{E}_{q_\phi(z_n|u_n,s_n)} \left[\mathbb{E}_{q_\phi(\pi_{ng}|z_n)} \left[t(k_{ng})_{ng}\right]\right] \tag{1}$$


<p>Where $t(k_{ng})_{ng}$
 represents the latent time for gene g in cell n if it&rsquo;s in kinetic state $k_{ng}$
. Note that this latent time $t_{ng}$
 is itself a function of $z_n$
 and the specific kinetic state k.</p>
<p>Let&rsquo;s break down the calculation in steps with a numeric example for Cell 1 and Gene A.</p>
<h3 id="what-we-assume-we-have-after-velovi-training">What We Assume We Have (after VeloVI training)</h3>
<hr>
<p><strong>Observed Data for Cell 1, Gene A:</strong> $u_{1A} = 0.8, s_{1A} = 0.2$
</p>
<p><strong>Learned Biophysical Parameters (Œ∏) for Gene A:</strong></p>
<ul>
<li>Switching Time $(t_A^s) = 8.0$
</li>
<li>(Other Œ∏ like Œ±,Œ≤,Œ≥ are not directly used in this specific calculation of expected time, but they would be used to calculate expected RNA levels from this time)</li>
</ul>
<p><strong>Learned Neural Network Parameters (œÜ) for Encoder and Decoder:</strong></p>
<p><em>Encoder outputs for</em> $z_1$
 <em>(given</em> $u_1, s_1$
*):* Defines $q_\phi(z_1|u_1,s_1) = \mathcal{N}(\mu_{z_1}, \Sigma_{z_1})$
</p>
<ul>
<li>$\mu_{z_1} = (0.5, -0.3)$
</li>
<li>$\Sigma_{z_1} = \text{diag}(e^{-0.5}, e^{-0.5}) \approx \text{diag}(0.6065, 0.6065)$
 (for $z_n$
 as 2D)</li>
</ul>
<p><em>Encoder outputs for</em> $\pi_{1A}$
 <em>(given</em> $z_1$
*):* Defines $q_\phi(\pi_{1A}|z_1) = \text{Dirichlet}(\alpha_{q,1A}(z_1))$
</p>
<p><em>Decoder outputs for</em> $\rho_{1A}^{(k)}$
 <em>(given</em> $z_1$
*):* For each state k, the decoder outputs a scaled latent time $\rho_{1A}^{(k)}$
, where $t_{1A}^{(k)} = \rho_{1A}^{(k)} \times t_A^s$
.</p>
<h3 id="step-1-calculate-the-inner-expectation-1">Step 1: Calculate the Inner Expectation</h3>
<hr>
<h4 id="recall-the-inner-expectation"><strong>Recall the Inner Expectation</strong></h4>

$$\mathbb{E}_{q_\phi(\pi_{ng}|z_n)} \left[t(k_{ng})_{ng}\right] \tag{2}$$


<p>This inner expectation is over the kinetic states $k_{ng}$
, weighted by their probabilities. The latent time $t(k_{ng})_{ng}$
 is specific to the state $k_{ng}$
. We can write this expectation as a sum:</p>

$$\mathbb{E}_{q_\phi(\pi_{ng}|z_n)} \left[t(k_{ng})_{ng}\right] = \sum_{k=1}^K P(k_{ng} = k|z_n) \cdot t_{ng}^{(k)}(z_n) \tag{3}$$


<p>Where:</p>
<ul>
<li>K is the number of kinetic states (4 in VeloVI)</li>
<li>$P(k_{ng} = k|z_n)$
 is the probability that gene g in cell n is in state k, given $z_n$
. This is the k-th component of the expected $\pi_{ng}$
 vector from the Dirichlet distribution:</li>
</ul>

$$P(k_{ng} = k|z_n) = \frac{\alpha_{q,ng,k}(z_n)}{\sum_{j=1}^K \alpha_{q,ng,j}(z_n)} \tag{4}$$


<ul>
<li>$t_{ng}^{(k)}(z_n)$
 is the latent time for gene g in cell n if it is in state k. This is a deterministic output of the decoder, given $z_n$
: $t_{ng}^{(k)}(z_n) = \text{Decoder}_k(z_n) \times t_g^s = \rho_{ng}^{(k)}(z_n) \times t_g^s$
.</li>
</ul>
<p>The result of this inner expectation will be a single scalar value, let&rsquo;s call it $\bar{t}_{ng}(z_n)$
, which is still a function of $z_n$
.</p>
<h4 id="numeric-example-for-cell-1-gene-a-1"><strong>Numeric Example for Cell 1, Gene A</strong></h4>
<p>Here we take Cell 1, Gene A for a Numeric Example: assume we have a sample $z_1^* = (0.6558, -0.3779)$
 (we&rsquo;ll show how to get this in Step 2).</p>
<ol>
<li>
<p><strong>From</strong> $z_1^*$
**, get Dirichlet concentration parameters:**</p>
<ul>
<li>Assume encoder outputs: $\alpha_{q,1A}(z_1^*) = (5.0, 0.1, 0.2, 0.1)$
</li>
<li>Sum of concentrations = 5.4</li>
</ul>
</li>
<li>
<p><strong>Calculate</strong> $P(k_{1A} = k|z_1^*)$
 <strong>for each state:</strong></p>
<ul>
<li>$P(k_{1A} = 1|z_1^*) = 5.0/5.4 \approx 0.9259$
</li>
<li>$P(k_{1A} = 2|z_1^*) = 0.1/5.4 \approx 0.0185$
</li>
<li>$P(k_{1A} = 3|z_1^*) = 0.2/5.4 \approx 0.0370$
</li>
<li>$P(k_{1A} = 4|z_1^*) = 0.1/5.4 \approx 0.0185$
</li>
</ul>
</li>
<li>
<p><strong>From</strong> $z_1^*$
**, get latent times** $t_{1A}^{(k)}(z_1^*)$
 <strong>for each state:</strong>
(These are specific outputs from the decoder for this $z_1^*$
)</p>
<ul>
<li>Assume decoder outputs $\rho_{1A}^{(1)}(z_1^*) = 0.5$
 (for Induction)
$t_{1A}^{(1)}(z_1^*) = 0.5 \times t_A^s = 0.5 \times 8.0 = 4.0$
</li>
<li>Assume decoder outputs $\rho_{1A}^{(2)}(z_1^*) = 0.3$
 (for Repression)
$t_{1A}^{(2)}(z_1^*) = 0.3 \times t_A^s = 0.3 \times 8.0 = 2.4$
</li>
<li>Assume decoder outputs $\rho_{1A}^{(3)}(z_1^*) = 0.8$
 (for Steady-State Induction)
$t_{1A}^{(3)}(z_1^*) = 0.8 \times t_A^s = 0.8 \times 8.0 = 6.4$
</li>
<li>Assume decoder outputs $\rho_{1A}^{(4)}(z_1^*) = 0.1$
 (for Steady-State Repression)
$t_{1A}^{(4)}(z_1^*) = 0.1 \times t_A^s = 0.1 \times 8.0 = 0.8$
</li>
</ul>
</li>
<li>
<p><strong>Calculate the weighted sum</strong> $\bar{t}_{1A}(z_1^*)$
**:**</p>
</li>
</ol>

$$\bar{t}_{1A}(z_1^*) = (0.9259 \times 4.0) + (0.0185 \times 2.4) + (0.0370 \times 6.4) + (0.0185 \times 0.8) \tag{5}$$



$$\bar{t}_{1A}(z_1^*) = 3.7036 + 0.0444 + 0.2368 + 0.0148 \tag{6}$$



$$\bar{t}_{1A}(z_1^*) \approx 3.9996 \text{ (This is a scalar value for this specific } z_1^*) \tag{7}$$


<h3 id="step-2-monte-carlo-approximation-calculates-the-outer-expectation">Step 2: Monte Carlo Approximation Calculates the Outer Expectation</h3>
<hr>

$$\mathbb{E}_{q_\phi(z_n|u_n,s_n)} \left[\bar{t}_{ng}(z_n)\right] \tag{8}$$


<p>Now we need to average this $\bar{t}_{ng}(z_n)$
 value over the distribution of $z_n$
. Since $\bar{t}_{ng}(z_n)$
 is a complex, non-linear function of $z_n$
 (involving neural network outputs), we use Monte Carlo approximation.</p>
<ol>
<li>
<p><strong>Draw M samples of</strong> $z_n^*$
 <strong>from</strong> $q_\phi(z_n|u_n,s_n)$
**:**</p>
<p>Recall $q_\phi(z_1|u_1,s_1) = \mathcal{N}(\mu_{z_1}, \Sigma_{z_1})$
 with $\mu_{z_1} = (0.5, -0.3)$
 and $\Sigma_{z_1} = \text{diag}(0.6065, 0.6065)$
.</p>
</li>
<li>
<p><strong>For each sample</strong> $m \in \{1, ..., M\}$
**:**</p>
<ul>
<li>Draw a random noise vector $\varepsilon_z^{(m)} \sim \mathcal{N}(0, I_d)$
</li>
<li>Calculate $z_n^{*(m)} = \mu_{z_n} + \sqrt{\Sigma_{z_n}} \odot \varepsilon_z^{(m)}$
 (This is the reparameterization trick)</li>
</ul>
</li>
</ol>
<p>Just two samples for brevity, usually M=1000+:</p>
<p><em>Sample 1:</em></p>
<ul>
<li>$\varepsilon_z^{(1)} = (0.2, -0.1)$
</li>
<li>$z_1^{*(1)} = (0.5, -0.3) + (\sqrt{0.6065}, \sqrt{0.6065}) \odot (0.2, -0.1) = (0.6558, -0.3779)$
</li>
</ul>
<p><em>Sample 2:</em></p>
<ul>
<li>$\varepsilon_z^{(2)} = (-0.5, 0.3)$
</li>
<li>$z_1^{*(2)} = (0.5, -0.3) + (\sqrt{0.6065}, \sqrt{0.6065}) \odot (-0.5, 0.3) = (0.1106, -0.0663)$
</li>
</ul>
<ol start="3">
<li>
<p><strong>For each sampled</strong> $z_n^{*(m)}$
**, calculate** $\bar{t}_{ng}(z_n^{*(m)})$
**:**</p>
<p>Follow the steps for the inner expectation (from above) for each $z_n^{*(m)}$
.</p>
</li>
</ol>
<p><em>For</em> $z_1^{*(1)} = (0.6558, -0.3779)$
*:*
We already calculated: $\bar{t}_{1A}^{(1)} \approx 3.9996$
</p>
<p><em>For</em> $z_1^{*(2)} = (0.1106, -0.0663)$
*:*</p>
<ul>
<li>Assume encoder outputs $\alpha_{q,1A}(z_1^{*(2)}) = (4.0, 0.3, 0.1, 0.1)$
 (differs from first sample). Sum = 4.5</li>
<li>$P(k_{1A} = 1|z_1^{*(2)}) = 4.0/4.5 \approx 0.8889$
</li>
<li>$P(k_{1A} = 2|z_1^{*(2)}) = 0.3/4.5 \approx 0.0667$
</li>
<li>$P(k_{1A} = 3|z_1^{*(2)}) = 0.1/4.5 \approx 0.0222$
</li>
<li>$P(k_{1A} = 4|z_1^{*(2)}) = 0.1/4.5 \approx 0.0222$
</li>
</ul>
<p>Assume decoder outputs (different for this $z_1^{*(2)}$
):</p>
<ul>
<li>$\rho_{1A}^{(1)}(z_1^{*(2)}) = 0.4 \rightarrow t_{1A}^{(1)}(z_1^{*(2)}) = 0.4 \times 8.0 = 3.2$
</li>
<li>$\rho_{1A}^{(2)}(z_1^{*(2)}) = 0.2 \rightarrow t_{1A}^{(2)}(z_1^{*(2)}) = 0.2 \times 8.0 = 1.6$
</li>
<li>$\rho_{1A}^{(3)}(z_1^{*(2)}) = 0.7 \rightarrow t_{1A}^{(3)}(z_1^{*(2)}) = 0.7 \times 8.0 = 5.6$
</li>
<li>$\rho_{1A}^{(4)}(z_1^{*(2)}) = 0.05 \rightarrow t_{1A}^{(4)}(z_1^{*(2)}) = 0.05 \times 8.0 = 0.4$
</li>
</ul>
<p>Calculate weighted sum $\bar{t}_{1A}(z_1^{*(2)})$
 for this sample:</p>

$$\bar{t}_{1A}(z_1^{*(2)}) = (0.8889 \times 3.2) + (0.0667 \times 1.6) + (0.0222 \times 5.6) + (0.0222 \times 0.4) \tag{9}$$



$$\bar{t}_{1A}(z_1^{*(2)}) = 2.8445 + 0.1067 + 0.1243 + 0.0089 \tag{10}$$



$$\bar{t}_{1A}(z_1^{*(2)}) \approx 3.0844 \tag{11}$$


<ol start="4">
<li><strong>Average the results:</strong></li>
</ol>
<p>The overall expectation is approximated by the average of these M scalar values:</p>

$$\mathbb{E}_{q_\phi(z_n|u_n,s_n)} [\bar{t}_{ng}(z_n)] \approx \frac{1}{M} \sum_{m=1}^M \bar{t}_{ng}^{(m)} \tag{12}$$


<p>For our example with M=2:</p>
<p>Average latent time for Gene A in Cell 1:</p>

$$\frac{3.9996 + 3.0844}{2} = 3.542 \tag{13}$$


<p>This final scalar value (e.g., 3.542) represents VeloVI&rsquo;s average estimated latent time for Gene A in Cell 1, taking into account all the uncertainties modeled by the variational distributions.</p>
<h2 id="summary-table">Summary Table</h2>
<table>
  <thead>
      <tr>
          <th>Item</th>
          <th>Status</th>
          <th>Role</th>
          <th>Example Value</th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td>$u_n, s_n$
</td>
          <td>Known (observed)</td>
          <td>Input data</td>
          <td>Cell 1 observations</td>
      </tr>
      <tr>
          <td>$\phi$
</td>
          <td>Learned &amp; Fixed</td>
          <td>Neural network parameters</td>
          <td>(Internal to networks)</td>
      </tr>
      <tr>
          <td>$\mu_{z_n}, \Sigma_{z_n}$
</td>
          <td>Calculated by encoder</td>
          <td>Parameters of $q_\phi(z_n\|u_n,s_n)$
</td>
          <td>$(0.5, -0.3)$
, $\text{diag}(0.6065, 0.6065)$
</td>
      </tr>
      <tr>
          <td>$z_n^{*(m)}$
</td>
          <td>Sampled</td>
          <td>Sample from $q_\phi(z_n\|u_n,s_n)$
</td>
          <td>$(0.6558, -0.3779)$
</td>
      </tr>
      <tr>
          <td>$\alpha_{q,ng}(z_n^{*(m)})$
</td>
          <td>Calculated by encoder</td>
          <td>Dirichlet parameters</td>
          <td>$(5.0, 0.1, 0.2, 0.1)$
</td>
      </tr>
      <tr>
          <td>$\bar{\pi}_{ng}(z_n^{*(m)})$
</td>
          <td>Calculated</td>
          <td>Inner expectation result</td>
          <td>$(0.9259, 0.0185, 0.0370, 0.0185)$
</td>
      </tr>
      <tr>
          <td>Final average</td>
          <td>Calculated</td>
          <td>Overall approximation</td>
          <td>$(0.9074, 0.0426, 0.0296, 0.0204)$
</td>
      </tr>
  </tbody>
</table>
<h1 id="calculate-expected-velocity-for-a-sampled-hahahugoshortcode13s214hbhb">Calculate Expected Velocity for a sampled $z_n$
</h1>
<blockquote>
<p><strong>Note: Posterior Velocity Distribution, i.e. the full posterior predictive velocity distribution, is just the distribution of all the expected velocity values for the sample $z_n^*$
 (drawn from $q_\phi(z_n|u_n, s_n)$
).</strong></p>
</blockquote>
<hr>

$$\mathbb{E}_{q_\phi(\pi_{ng}|z_n)} \left[v^{(g)}(t(k_{ng})_{ng}, k_{ng})\right] \tag{1}$$


<p>This term represents the expected velocity for gene g in cell n, given a specific latent variable $z_n$
, averaging over the uncertainty in its kinetic state $k_{ng}$
.</p>
<p>We will use the same logic as the previous expectation for latent time, as the structure is identical: we are calculating the expectation of a quantity ($v^{(g)}$
) that depends on a discrete random variable ($k_{ng}$
), whose probabilities are determined by the Dirichlet distribution $q_\phi(\pi_{ng}|z_n)$
.</p>
<h2 id="1-recall-the-general-expectation-for-a-discrete-variable">1. Recall the General Expectation for a Discrete Variable</h2>
<p>If X is a discrete random variable that can take on values $x_1, x_2, \ldots, x_K$
 with corresponding probabilities $P(X=x_1), P(X=x_2), \ldots, P(X=x_K)$
, then:</p>

$$\mathbb{E}[X] = \sum_{i=1}^K x_i \cdot P(X=x_i) \tag{2}$$


<h2 id="2-applying-to-our-velocity-case">2. Applying to Our Velocity Case</h2>
<p><strong>The Random Variable:</strong> Our random variable is $v^{(g)}(t(k_{ng})_{ng}, k_{ng})$
. Its value depends on the specific kinetic state $k_{ng}$
.</p>
<p><strong>The Possible Values:</strong> For each state $k \in \{1,2,3,4\}$
, the value of the velocity is $v^{(g)}(t_{ng}^{(k)}(z_n), k)$
. This velocity depends on the latent time $t_{ng}^{(k)}(z_n)$
 (which is a function of $z_n$
 and k) and the state k itself.</p>
<p><strong>The Probabilities:</strong> The probability of being in state k is $P(k_{ng} = k|z_n)$
. As derived previously, this is given by the expected value of the k-th component of $\pi_{ng}$
 under the Dirichlet distribution:</p>

$$P(k_{ng} = k|z_n) = \mathbb{E}_{q_\phi(\pi_{ng}|z_n)}[\pi_{ng,k}] = \frac{\alpha_{q,ng,k}(z_n)}{\sum_{j=1}^K \alpha_{q,ng,j}(z_n)} \tag{3}$$


<p>Putting it together, the inner expectation for velocity becomes:</p>

$$\mathbb{E}_{q_\phi(\pi_{ng}|z_n)} \left[v^{(g)}(t(k_{ng})_{ng}, k_{ng})\right] = \sum_{k=1}^K P(k_{ng} = k|z_n) \cdot v^{(g)}(t_{ng}^{(k)}(z_n), k) \tag{4}$$


<p>This will result in a single scalar value for each gene and cell, given a specific $z_n$
.</p>
<h2 id="numeric-example-for-cell-1-gene-a-2">Numeric Example for Cell 1, Gene A</h2>
<p>Let&rsquo;s use the same example inputs from our previous discussions:</p>
<p><strong>Assumed Sampled</strong> $z_1^*$
: $(0.6558, -0.3779)$
 (This comes from the first step in the &ldquo;posterior predictive velocity distribution&rdquo; process).</p>
<p><strong>Learned Biophysical Parameters (Œ∏) for Gene A:</strong></p>
<ul>
<li>$\alpha_A = 1.2$
 (Transcription rate in Induction)</li>
<li>$\beta_A = 0.2$
 (Splicing rate)</li>
<li>$\gamma_A = 0.1$
 (Degradation rate)</li>
<li>$t_A^s = 8.0$
 (Switching time)</li>
</ul>
<p><strong>Encoder Output for</strong> $\pi_{1A}$
 <strong>(given</strong> $z_1^*$
**):**</p>
<ul>
<li>$\alpha_{q,1A}(z_1^*) = (5.0, 0.1, 0.2, 0.1)$
</li>
<li>Sum = 5.4</li>
</ul>
<p>This gives us the state probabilities $P(k_{1A} = k|z_1^*)$
:</p>
<ul>
<li>$P(k_{1A} = 1|z_1^*) = 5.0/5.4 \approx 0.9259$
</li>
<li>$P(k_{1A} = 2|z_1^*) = 0.1/5.4 \approx 0.0185$
</li>
<li>$P(k_{1A} = 3|z_1^*) = 0.2/5.4 \approx 0.0370$
</li>
<li>$P(k_{1A} = 4|z_1^*) = 0.1/5.4 \approx 0.0185$
</li>
</ul>
<p><strong>Decoder Output for</strong> $\rho_{1A}^{(k)}$
 <strong>(given</strong> $z_1^*$
**) and corresponding latent times** $t_{1A}^{(k)}$
**:**</p>
<ul>
<li>$\rho_{1A}^{(1)} = 0.5 \Rightarrow t_{1A}^{(1)} = 0.5 \times 8.0 = 4.0$
</li>
<li>$\rho_{1A}^{(2)} = 0.3 \Rightarrow t_{1A}^{(2)} = 0.3 \times 8.0 = 2.4$
</li>
<li>$\rho_{1A}^{(3)} = 0.8 \Rightarrow t_{1A}^{(3)} = 0.8 \times 8.0 = 6.4$
</li>
<li>$\rho_{1A}^{(4)} = 0.1 \Rightarrow t_{1A}^{(4)} = 0.1 \times 8.0 = 0.8$
</li>
</ul>
<p><strong>The Velocity Formula:</strong></p>

$$v^{(g)}(t^{(k)}, k) = \beta_g \bar{u}^{(g)}(t^{(k)}, k) - \gamma_g \bar{s}^{(g)}(t^{(k)}, k) \tag{5}$$


<p>We need the expected unspliced ($\bar{u}$
) and spliced ($\bar{s}$
) counts for each state at its corresponding latent time. These are given by the kinetic equations in VeloVI.</p>
<h3 id="calculations-for-each-state-k-and-its-corresponding-hahahugoshortcode13s261hbhb">Calculations for each state k and its corresponding $t_{1A}^{(k)}$
:</h3>
<p><strong>State 1 (Induction, k=1): t=4.0</strong></p>

$$\bar{u}^{(A)}(4.0, 1) = \frac{\alpha_A}{\beta_A}(1 - e^{-\beta_A t}) = \frac{1.2}{0.2}(1 - e^{-0.2 \times 4.0}) = 6(1 - e^{-0.8}) \approx 6(1 - 0.4493) = 3.3042 \tag{6}$$



$$\bar{s}^{(A)}(4.0, 1) = \frac{\alpha_A}{\gamma_A}\left(1 - \frac{\gamma_A}{\gamma_A - \beta_A}e^{-\beta_A t} + \frac{\beta_A}{\gamma_A - \beta_A}e^{-\gamma_A t}\right) \tag{7}$$



$$= \frac{1.2}{0.1}\left(1 - \frac{0.1}{0.1 - 0.2}e^{-0.8} + \frac{0.2}{0.1 - 0.2}e^{-0.4}\right) = 12(1 + e^{-0.8} - 2e^{-0.4}) \tag{8}$$



$$\approx 12(1 + 0.4493 - 2 \times 0.6703) = 12(1.4493 - 1.3406) \approx 1.3044 \tag{9}$$



$$v^{(A)}(4.0, 1) = \beta_A \bar{u}^{(A)}(4.0, 1) - \gamma_A \bar{s}^{(A)}(4.0, 1) = (0.2 \times 3.3042) - (0.1 \times 1.3044) = 0.66084 - 0.13044 = 0.5304 \tag{10}$$


<p><strong>State 2 (Repression, k=2): t=2.4</strong></p>
<p>(For repression, we need the values at $t_A^s = 8.0$
 (end of induction) as initial conditions for decay.)</p>

$$\bar{u}^{(A)}(t_A^s, 1) = 6(1 - e^{-0.2 \times 8.0}) \approx 6(1 - 0.2019) = 4.7886 \tag{11}$$



$$\bar{s}^{(A)}(t_A^s, 1) = 12(1 + e^{-1.6} - 2e^{-0.8}) \approx 12(1 + 0.2019 - 2 \times 0.4493) = 3.6396 \tag{12}$$



$$\bar{u}^{(A)}(2.4, 2) = \bar{u}^{(A)}(t_A^s, 1)e^{-\beta_A t_{1A}^{(2)}} = 4.7886 \times e^{-0.2 \times 2.4} \approx 4.7886 \times 0.6188 = 2.9634 \tag{13}$$



$$\bar{s}^{(A)}(2.4, 2) = \bar{s}^{(A)}(t_A^s, 1)e^{-\gamma_A t_{1A}^{(2)}} + \frac{\beta_A \bar{u}^{(A)}(t_A^s, 1)}{\gamma_A - \beta_A}(e^{-\beta_A t_{1A}^{(2)}} - e^{-\gamma_A t_{1A}^{(2)}}) \tag{14}$$



$$= 3.6396e^{-0.1 \times 2.4} + \frac{0.2 \times 4.7886}{0.1 - 0.2}(e^{-0.2 \times 2.4} - e^{-0.1 \times 2.4}) \tag{15}$$



$$\approx 3.6396 \times 0.7866 + (-9.5772)(0.6188 - 0.7866) \approx 2.8617 + 1.6072 = 4.4689 \tag{16}$$



$$v^{(A)}(2.4, 2) = (0.2 \times 2.9634) - (0.1 \times 4.4689) = 0.59268 - 0.44689 = 0.1458 \tag{17}$$


<p><strong>State 3 (Steady-State Induction, k=3): t=6.4</strong></p>

$$\bar{u}^{(A)}(6.4, 3) = \alpha_A/\beta_A = 1.2/0.2 = 6.0 \tag{18}$$



$$\bar{s}^{(A)}(6.4, 3) = \alpha_A/\gamma_A = 1.2/0.1 = 12.0 \tag{19}$$



$$v^{(A)}(6.4, 3) = \beta_A \bar{u}^{(A)}(6.4, 3) - \gamma_A \bar{s}^{(A)}(6.4, 3) = (0.2 \times 6.0) - (0.1 \times 12.0) = 1.2 - 1.2 = 0.0 \tag{20}$$


<p><strong>State 4 (Steady-State Repression, k=4): t=0.8</strong></p>

$$\bar{u}^{(A)}(0.8, 4) = 0.0 \tag{21}$$



$$\bar{s}^{(A)}(0.8, 4) = 0.0 \tag{22}$$



$$v^{(A)}(0.8, 4) = \beta_A \bar{u}^{(A)}(0.8, 4) - \gamma_A \bar{s}^{(A)}(0.8, 4) = (0.2 \times 0.0) - (0.1 \times 0.0) = 0.0 \tag{23}$$


<h3 id="calculate-the-weighted-sum">Calculate the weighted sum:</h3>
<p>Now, we multiply each calculated velocity by its corresponding state probability $P(k_{1A} = k|z_1^*)$
 and sum them up:</p>

$$\mathbb{E}_{q_\phi(\pi_{1A}|z_1^*)} \left[v^{(A)}(t(k_{1A})_{1A}, k_{1A})\right] = \tag{24}$$



$$(0.9259 \times 0.5304) + (0.0185 \times 0.1458) + (0.0370 \times 0.0) + (0.0185 \times 0.0) \tag{25}$$



$$= 0.4900 + 0.0027 + 0.0 + 0.0 \approx 0.4927 \tag{26}$$


<h2 id="get-posterior-predictive-velocity-distribution">Get Posterior Predictive Velocity Distribution</h2>
<p>The value <strong>0.4927</strong> is the expected velocity for Gene A in Cell 1, given the specific sampled latent variable $z_1^* = (0.6558, -0.3779)$
.</p>
<p>This value is one sample of the posterior predictive velocity. To get the full posterior predictive velocity distribution, you would repeat this entire calculation for many different samples of $z_n^*$
 (drawn from $q_\phi(z_n|u_n, s_n)$
) and then analyze the distribution of these resulting expected velocity values.</p>

      </div>

      <div class="mt-16"></div>

      <div class="container mx-auto prose prose-slate lg:prose-xl dark:prose-invert mt-5">
        
        <div class="max-w-prose print:hidden">
  
  

  


  
<section class="flex flex-row flex-wrap justify-center pt-4 text-xl">
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F11b.velocity_velovi%2Fvelovi_expected_state_n_latent_time%2F&amp;text="
    title="X"
    aria-label="X"
    id="share-link-x"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://www.facebook.com/sharer.php?u=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F11b.velocity_velovi%2Fvelovi_expected_state_n_latent_time%2F&amp;t="
    title="Facebook"
    aria-label="Facebook"
    id="share-link-facebook"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="currentColor" d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12c0 4.84 3.44 8.87 8 9.8V15H8v-3h2V9.5C10 7.57 11.57 6 13.5 6H16v3h-2c-.55 0-1 .45-1 1v2h3v3h-3v6.95c5.05-.5 9-4.76 9-9.95z"/></svg>
  </a>
  

  
  
  
  
  
  
    
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="mailto:?subject=&amp;body=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F11b.velocity_velovi%2Fvelovi_expected_state_n_latent_time%2F"
    title="Email"
    aria-label="Email"
    id="share-link-email"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-width="1.5" d="M16.5 12a4.5 4.5 0 1 1-9 0a4.5 4.5 0 0 1 9 0Zm0 0c0 1.657 1.007 3 2.25 3S21 13.657 21 12a9 9 0 1 0-2.636 6.364M16.5 12V8.25"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://www.linkedin.com/shareArticle?url=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F11b.velocity_velovi%2Fvelovi_expected_state_n_latent_time%2F&amp;title="
    title="LinkedIn"
    aria-label="LinkedIn"
    id="share-link-linkedin"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" height="1em" viewBox="0 0 448 512"><path fill="currentColor" d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="whatsapp://send?text=%20http%3A%2F%2Flocalhost%3A1313%2Fpost%2F11b.velocity_velovi%2Fvelovi_expected_state_n_latent_time%2F"
    title="WhatsApp"
    aria-label="WhatsApp"
    id="share-link-whatsapp"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256" fill="currentColor"><path d="m187.58 144.84-32-16a8 8 0 0 0-8 .5l-14.69 9.8a40.55 40.55 0 0 1-16-16l9.8-14.69a8 8 0 0 0 .5-8l-16-32A8 8 0 0 0 104 64a40 40 0 0 0-40 40 88.1 88.1 0 0 0 88 88 40 40 0 0 0 40-40 8 8 0 0 0-4.42-7.16ZM152 176a72.08 72.08 0 0 1-72-72 24 24 0 0 1 19.29-23.54l11.48 23L101 118a8 8 0 0 0-.73 7.51 56.47 56.47 0 0 0 30.15 30.15A8 8 0 0 0 138 155l14.61-9.74 23 11.48A24 24 0 0 1 152 176ZM128 24a104 104 0 0 0-91.82 152.88l-11.35 34.05a16 16 0 0 0 20.24 20.24l34.05-11.35A104 104 0 1 0 128 24Zm0 192a87.87 87.87 0 0 1-44.06-11.81 8 8 0 0 0-6.54-.67L40 216l12.47-37.4a8 8 0 0 0-.66-6.54A88 88 0 1 1 128 216Z"/></svg>
  </a>
  
</section>


  








  
  



  
  
  
    
  
  
  

<div class="flex pt-12 pb-4">
  
  
  <img
    class="mr-4 h-24 w-24 rounded-full"
    width="96"
    height="96"
    alt="Jiyuan Liu"
  src="/author/jiyuan-liu/avatar_hu6328122812819896191.jpg"
  loading="lazy"
  />
  
  <div class="place-self-center">
    <div class="text-[0.6rem] uppercase leading-3 text-neutral-500 dark:text-neutral-400">
      Authors
    </div>
    <div class="font-semibold leading-6 text-neutral-800 dark:text-neutral-300">
      <a href="http://localhost:1313/" class="no-underline">
      Jiyuan Liu
      </a>
    </div>

    
    <div class="text-sm font-bold text-neutral-700 dark:text-neutral-300">
    Postdoctoral Scientist
    </div>
    


    

    <div class="text-2xl sm:text-lg pt-1">

      
<div class="flex flex-wrap text-neutral-500 dark:text-neutral-300">
  
    
    
    
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="mailto:jliu78@mgh.harvard.edu"
      
      aria-label="At-Symbol"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-width="1.5" d="M16.5 12a4.5 4.5 0 1 1-9 0a4.5 4.5 0 0 1 9 0Zm0 0c0 1.657 1.007 3 2.25 3S21 13.657 21 12a9 9 0 1 0-2.636 6.364M16.5 12V8.25"/></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://github.com/stars/jliu678/lists/owned-contributed-by-me"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Brands/Github"
    ><svg style="height: 1em;" fill="currentColor" viewBox="3 3 18 18"><path d="M12 3C7.0275 3 3 7.12937 3 12.2276C3 16.3109 5.57625 19.7597 9.15374 20.9824C9.60374 21.0631 9.77249 20.7863 9.77249 20.5441C9.77249 20.3249 9.76125 19.5982 9.76125 18.8254C7.5 19.2522 6.915 18.2602 6.735 17.7412C6.63375 17.4759 6.19499 16.6569 5.8125 16.4378C5.4975 16.2647 5.0475 15.838 5.80124 15.8264C6.51 15.8149 7.01625 16.4954 7.18499 16.7723C7.99499 18.1679 9.28875 17.7758 9.80625 17.5335C9.885 16.9337 10.1212 16.53 10.38 16.2993C8.3775 16.0687 6.285 15.2728 6.285 11.7432C6.285 10.7397 6.63375 9.9092 7.20749 9.26326C7.1175 9.03257 6.8025 8.08674 7.2975 6.81794C7.2975 6.81794 8.05125 6.57571 9.77249 7.76377C10.4925 7.55615 11.2575 7.45234 12.0225 7.45234C12.7875 7.45234 13.5525 7.55615 14.2725 7.76377C15.9937 6.56418 16.7475 6.81794 16.7475 6.81794C17.2424 8.08674 16.9275 9.03257 16.8375 9.26326C17.4113 9.9092 17.76 10.7281 17.76 11.7432C17.76 15.2843 15.6563 16.0687 13.6537 16.2993C13.98 16.5877 14.2613 17.1414 14.2613 18.0065C14.2613 19.2407 14.25 20.2326 14.25 20.5441C14.25 20.7863 14.4188 21.0746 14.8688 20.9824C16.6554 20.364 18.2079 19.1866 19.3078 17.6162C20.4077 16.0457 20.9995 14.1611 21 12.2276C21 7.12937 16.9725 3 12 3Z"></path></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://www.linkedin.com/in/jiyuan-liu-7608a42a6/"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Brands/Linkedin"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" height="1em" viewBox="0 0 448 512"><path fill="currentColor" d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://scholar.google.com/citations?view_op=list_works&amp;hl=en&amp;hl=en&amp;user=5c1kGO4AAAAJ&amp;sortby=pubdate"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Academicons/Google-Scholar"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M343.759 106.662V79.43L363.524 64h-213.89L20.476 176.274h85.656a82.339 82.339 0 0 0-.219 6.225c0 20.845 7.22 38.087 21.672 51.861c14.453 13.797 32.252 20.648 53.327 20.648c4.923 0 9.75-.368 14.438-1.024c-2.907 6.5-4.374 12.523-4.374 18.142c0 9.875 4.499 20.43 13.467 31.642c-39.234 2.67-68.061 9.732-86.437 21.163c-10.531 6.5-19 14.704-25.39 24.531c-6.391 9.9-9.578 20.515-9.578 31.962c0 9.648 2.062 18.336 6.219 26.062c4.156 7.726 9.578 14.07 16.312 18.984c6.718 4.968 14.469 9.101 23.219 12.469c8.734 3.344 17.406 5.718 26.061 7.062A167.052 167.052 0 0 0 180.555 448c13.469 0 26.953-1.734 40.547-5.187c13.562-3.485 26.28-8.642 38.171-15.493c11.86-6.805 21.515-16.086 28.922-27.718c7.39-11.68 11.094-24.805 11.094-39.336c0-11.016-2.25-21.039-6.75-30.14c-4.468-9.073-9.938-16.542-16.452-22.345c-6.501-5.813-13-11.155-19.516-15.968c-6.5-4.845-12-9.75-16.468-14.813c-4.485-5.046-6.735-10.054-6.735-14.984c0-4.921 1.734-9.672 5.216-14.265c3.455-4.61 7.674-9.048 12.61-13.306c4.937-4.25 9.875-8.968 14.796-14.133c4.922-5.147 9.141-11.827 12.61-20.008c3.485-8.18 5.203-17.445 5.203-27.757c0-13.453-2.547-24.46-7.547-33.314c-.594-1.022-1.218-1.803-1.875-3.022l56.907-46.672v17.119c-7.393.93-6.624 5.345-6.624 10.635V245.96c0 5.958 4.875 10.834 10.834 10.834h3.989c5.958 0 10.833-4.875 10.833-10.834V117.293c0-5.277.778-9.688-6.561-10.63zm-107.36 222.48c1.14.75 3.704 2.78 7.718 6.038c4.05 3.243 6.797 5.695 8.266 7.414a443.553 443.553 0 0 1 6.376 7.547c2.813 3.375 4.718 6.304 5.718 8.734c1 2.477 2.016 5.461 3.047 8.946a38.27 38.27 0 0 1 1.485 10.562c0 17.048-6.564 29.68-19.656 37.859c-13.125 8.18-28.767 12.274-46.938 12.274c-9.187 0-18.203-1.093-27.063-3.196c-8.843-2.116-17.311-5.336-25.39-9.601c-8.078-4.258-14.577-10.204-19.5-17.797c-4.938-7.64-7.407-16.415-7.407-26.25c0-10.32 2.797-19.29 8.422-26.906c5.594-7.625 12.938-13.391 22.032-17.315c9.063-3.946 18.25-6.742 27.562-8.398a157.865 157.865 0 0 1 28.438-2.555c4.47 0 7.936.25 10.405.696c.455.219 3.032 2.07 7.735 5.563c4.704 3.462 7.625 5.595 8.75 6.384zm-3.359-100.579c-7.406 8.86-17.734 13.288-30.953 13.288c-11.86 0-22.298-4.764-31.266-14.312c-9-9.523-15.422-20.328-19.344-32.43c-3.937-12.109-5.906-23.984-5.906-35.648c0-13.694 3.596-25.352 10.781-34.976c7.187-9.65 17.5-14.485 30.938-14.485c11.875 0 22.374 5.038 31.437 15.157c9.094 10.085 15.61 21.413 19.517 33.968c3.922 12.54 5.873 24.53 5.873 35.984c0 13.446-3.702 24.61-11.076 33.454z"/></svg></a>
  
</div>



    </div>
  </div>
</div>







  
  
    
    
    
      
      
    
<div class="pt-1 no-prose w-full">
  <hr class="border-dotted border-neutral-300 dark:border-neutral-600" />
  <div class="flex flex-col md:flex-row flex-nowrap justify-between gap-5 pt-2">
    <div class="">
      
        <a class="group flex no-underline" href="/post/11b.velocity_velovi/velovi_elbo_term_with_expectation/">
          <span
            class="mt-[-0.3rem] me-2 text-neutral-700 transition-transform group-hover:-translate-x-[2px] group-hover:text-primary-600 dark:text-neutral dark:group-hover:text-primary-400"
          ><span class="ltr:inline rtl:hidden">&larr;</span></span>
          <span class="flex flex-col">
            <span
              class="mt-[0.1rem] leading-6 group-hover:underline group-hover:decoration-primary-500"
            ></span>
            <span class="mt-[0.1rem] text-xs text-neutral-500 dark:text-neutral-400">
              
                Jan 1, 0001
              
            </span>
          </span>
        </a>
      
    </div>
    <div class="">
      
        <a class="group flex text-right no-underline" href="/post/11b.velocity_velovi/velovi_model_n_inference/">
          <span class="flex flex-col">
            <span
              class="mt-[0.1rem] leading-6 group-hover:underline group-hover:decoration-primary-500"
            ></span
            >
            <span class="mt-[0.1rem] text-xs text-neutral-500 dark:text-neutral-400">
              
                Jan 1, 0001
              
            </span>
          </span>
          <span
            class="mt-[-0.3rem] ms-2 text-neutral-700 transition-transform group-hover:-translate-x-[2px] group-hover:text-primary-600 dark:text-neutral dark:group-hover:text-primary-400"
          ><span class="ltr:inline">&rarr;</span></span>
        </a>
      
    </div>
  </div>
</div>



  


  



</div>

      </div>

    </main>
  </article>
</div>

    </div>
    <div class="page-footer">
      <footer class="container mx-auto flex flex-col justify-items-center text-sm leading-6 mt-24 mb-4 text-slate-700 dark:text-slate-200">

  












  
  
  
  
  














  
  
  

  
  
    
  
  
    
  

  

  
  <p class="powered-by text-center">
    ¬© 2025 Me. This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-nd/4.0" rel="noopener noreferrer" target="_blank">CC BY NC ND 4.0</a>
  </p>
  

  <p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-nd/4.0" rel="noopener noreferrer" target="_blank" aria-label="Creative Commons">
      <i class="fab fa-creative-commons fa-2x" aria-hidden="true"></i>
      <i class="fab fa-creative-commons-by fa-2x" aria-hidden="true"></i>
      
        <i class="fab fa-creative-commons-nc fa-2x" aria-hidden="true"></i>
      
      
        <i class="fab fa-creative-commons-nd fa-2x" aria-hidden="true"></i>
      
    </a>
  </p>





  <p class="powered-by text-center">
    
    
    
      
      
      
      
      
      
      Published with <a href="https://hugoblox.com/?utm_campaign=poweredby" target="_blank" rel="noopener">Hugo Blox Builder</a> ‚Äî the free, <a href="https://github.com/HugoBlox/hugo-blox-builder" target="_blank" rel="noopener">open source</a> website builder that empowers creators.
    
  </p>

</footer>

    </div>

    
    











  </body>
</html>
