<!doctype html>
<!-- This site was created with Hugo Blox. https://hugoblox.com -->
<!-- Last Published: May 18, 2025 --><html lang="en-us" dir="ltr"
      data-wc-theme-default="system">
  
<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <meta name="generator" content="Hugo Blox Builder 0.3.1" />

  
  












  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Jiyuan Liu" />

  
  
  
    
  
  <meta name="description" content="Human Endogenous Retroviruses (hERVs) are ancient viral sequences embedded in the human genome. Transgenes are common in transgenic mouse models. To quantify them from sequencing reads, we need-- a) modify fasta and gtf files to include the their sequencing and annotation; b) a feature quantification algorithm to handle multimapping commonly seen for hERV and transgenes. Here I discussed the algorithms for feature quantification, and successfully quantified hERV and transgenes by implementing an EM algorithm." />

  
  <link rel="alternate" hreflang="en-us" href="http://localhost:1313/post/01.bulkrnaseq_allow_herv_quantification/" />

  
  
  
  
    
    <link rel="stylesheet" href="/css/themes/emerald.min.css" />
  

  
  
    
    <link href="/dist/wc.min.css" rel="stylesheet" />
  

  
  
  

  

  <script>
     
    window.hbb = {
       defaultTheme: document.documentElement.dataset.wcThemeDefault,
       setDarkTheme: () => {
        document.documentElement.classList.add("dark");
        document.documentElement.style.colorScheme = "dark";
      },
       setLightTheme: () => {
        document.documentElement.classList.remove("dark");
        document.documentElement.style.colorScheme = "light";
      }
    }

    console.debug(`Default Hugo Blox Builder theme is ${window.hbb.defaultTheme}`);

    if ("wc-color-theme" in localStorage) {
      localStorage.getItem("wc-color-theme") === "dark" ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
    } else {
      window.hbb.defaultTheme === "dark" ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
      if (window.hbb.defaultTheme === "system") {
        window.matchMedia("(prefers-color-scheme: dark)").matches ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
      }
    }
  </script>

  <script>
    
    document.addEventListener('DOMContentLoaded', function () {
      
      let checkboxes = document.querySelectorAll('li input[type=\'checkbox\'][disabled]');
      checkboxes.forEach(e => {
        e.parentElement.parentElement.classList.add('task-list');
      });

      
      const liNodes = document.querySelectorAll('.task-list li');
      liNodes.forEach(nodes => {
        let textNodes = Array.from(nodes.childNodes).filter(node => node.nodeType === 3 && node.textContent.trim().length > 1);
        if (textNodes.length > 0) {
          const span = document.createElement('label');
          textNodes[0].after(span);  
          span.appendChild(nodes.querySelector('input[type=\'checkbox\']'));
          span.appendChild(textNodes[0]);
        }
      });
    });
  </script>

  
  
  




































  
  

  
  <link rel="icon" type="image/png" href="/media/icon_hu_c54cc46c793e1f2b.png" />
  <link rel="apple-touch-icon" type="image/png" href="/media/icon_hu_68ea7b4ed3f6b794.png" />

  <link rel="canonical" href="http://localhost:1313/post/01.bulkrnaseq_allow_herv_quantification/" />

  
  
  
  
  
  
  
  
    
  
  

  
  
    
    
  
  <meta property="twitter:card" content="summary_large_image" />
  
    <meta property="twitter:site" content="@GetResearchDev" />
    <meta property="twitter:creator" content="@GetResearchDev" />
  
  <meta property="og:site_name" content="Jiyuan Liu" />
  <meta property="og:url" content="http://localhost:1313/post/01.bulkrnaseq_allow_herv_quantification/" />
  <meta property="og:title" content="🧬 Quantify HERV and transgenes from Bulk-RNAseq | Jiyuan Liu" />
  <meta property="og:description" content="Human Endogenous Retroviruses (hERVs) are ancient viral sequences embedded in the human genome. Transgenes are common in transgenic mouse models. To quantify them from sequencing reads, we need-- a) modify fasta and gtf files to include the their sequencing and annotation; b) a feature quantification algorithm to handle multimapping commonly seen for hERV and transgenes. Here I discussed the algorithms for feature quantification, and successfully quantified hERV and transgenes by implementing an EM algorithm." /><meta property="og:image" content="http://localhost:1313/post/01.bulkrnaseq_allow_herv_quantification/featured.jpg" />
    <meta property="twitter:image" content="http://localhost:1313/post/01.bulkrnaseq_allow_herv_quantification/featured.jpg" /><meta property="og:locale" content="en-us" />
  
    
      <meta
        property="article:published_time"
        content="2025-03-24T00:00:00&#43;00:00"
      />
    
    <meta property="article:modified_time" content="2025-03-24T00:00:00&#43;00:00">
  

  



  


  <title>🧬 Quantify HERV and transgenes from Bulk-RNAseq | Jiyuan Liu</title>

  
  
  
  
  
    
    
  
  
  <style>
    @font-face {
      font-family: 'Inter var';
      font-style: normal;
      font-weight: 100 900;
      font-display: swap;
      src: url(/dist/font/Inter.var.woff2) format(woff2);
    }
  </style>

  

  
  


  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
    
  
  
  
  
  
  
  
  
  
  
  
  







<link type="text/css" rel="stylesheet" href="/dist/pagefind/pagefind-ui.be766eb419317a14ec769d216e9779bfe8f3737c80e780f4ba0dafb57a41a482.css" integrity="sha256-vnZutBkxehTsdp0hbpd5v&#43;jzc3yA54D0ug2vtXpBpII=" />


<script src="/dist/pagefind/pagefind-ui.87693d7c6f2b3b347ce359d0ede762c033419f0a32b22ce508c335a81d841f1b.js" integrity="sha256-h2k9fG8rOzR841nQ7ediwDNBnwoysizlCMM1qB2EHxs="></script>


<script>window.hbb.pagefind = {"baseUrl":"/"};</script>

<style>
  html.dark {
    --pagefind-ui-primary: #eeeeee;
    --pagefind-ui-text: #eeeeee;
    --pagefind-ui-background: #152028;
    --pagefind-ui-border: #152028;
    --pagefind-ui-tag: #152028;
  }
</style>

<script>
  window.addEventListener('DOMContentLoaded', (event) => {
    new PagefindUI({
      element: "#search",
      showSubResults: true,
      baseUrl: window.hbb.pagefind.baseUrl,
      bundlePath: window.hbb.pagefind.baseUrl + "pagefind/",
    });
  });
  document.addEventListener('DOMContentLoaded', () => {
    let element = document.getElementById('search');
    let trigger = document.getElementById('search_toggle');

    if (trigger) {
      trigger.addEventListener('click', () => {
        element.classList.toggle('hidden');
        element.querySelector("input").value = ""
        element.querySelector("input").focus()

        if (!element.classList.contains('hidden')) {
          let clear_trigger = document.querySelector('.pagefind-ui__search-clear');

          if (clear_trigger && !clear_trigger.hasAttribute('listenerOnClick')) {
            clear_trigger.setAttribute('listenerOnClick', 'true');

            clear_trigger.addEventListener('click', () => {
              element.classList.toggle('hidden');
            });
          }
        }

      });
    }
  });
</script>










  
  
  <link type="text/css" rel="stylesheet" href="/dist/lib/katex/katex.min.505d5f829022bb7b4f24dfee0aa1141cd7bba67afe411d1240335f820960b5c3.css" integrity="sha256-UF1fgpAiu3tPJN/uCqEUHNe7pnr&#43;QR0SQDNfgglgtcM=" />
  
  
  <script defer src="/dist/lib/katex/katex.min.dc84b296ec3e884de093158f760fd9d45b6c7abe58b5381557f4e138f46a58ae.js" integrity="sha256-3ISyluw&#43;iE3gkxWPdg/Z1Ftser5YtTgVV/ThOPRqWK4="></script>
  
  
  
  
  <script defer src="/js/katex-renderer.6579ec9683211cfb952064aedf3a3baea5eeb17a061775b32b70917474637c80.js" integrity="sha256-ZXnsloMhHPuVIGSu3zo7rqXusXoGF3WzK3CRdHRjfIA="></script>
  
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  






  
  
  
  
  
  
  
  <script
    defer
    src="/js/hugo-blox-en.min.js"
    integrity=""
  ></script>

  
  








  
    
      
      <script async defer src="https://buttons.github.io/buttons.js"></script>

      
    
  




</head>

  <body class="dark:bg-hb-dark dark:text-white page-wrapper" id="top">
    <div id="page-bg"></div>
    <div class="page-header sticky top-0 z-30">
      
      
      
        
        
        
          <header id="site-header" class="header">
  <nav class="navbar px-3 flex justify-left">
    <div class="order-0 h-100">
      
      <a class="navbar-brand" href="/" title="Jiyuan Liu">
        Home
      </a>
    </div>
    
    <input id="nav-toggle" type="checkbox" class="hidden" />
    <label
      for="nav-toggle"
      class="order-3 cursor-pointer flex items-center lg:hidden text-dark dark:text-white lg:order-1">
      <svg id="show-button" class="h-6 fill-current block" viewBox="0 0 20 20">
        <title>Open Menu</title>
        <path d="M0 3h20v2H0V3z m0 6h20v2H0V9z m0 6h20v2H0V0z"></path>
      </svg>
      <svg id="hide-button" class="h-6 fill-current hidden" viewBox="0 0 20 20">
        <title>Close Menu</title>
        <polygon
          points="11 9 22 9 22 11 11 11 11 22 9 22 9 11 -2 11 -2 9 9 9 9 -2 11 -2"
          transform="rotate(45 10 10)"></polygon>
      </svg>
    </label>
    

    
    
    <ul
      id="nav-menu"
      class="navbar-nav order-3 hidden lg:flex w-full pb-6 lg:order-1 lg:w-auto lg:space-x-2 lg:pb-0 xl:space-x-8 justify-left
      ">
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/"
        >Bio</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/#papers"
        >Papers</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/#talks"
        >Talks</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/#news"
        >News</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/experience/"
        >Experience</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/projects/"
        >Projects</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/teaching/"
        >Teaching</a
        >
      </li>
      
      
      
    </ul>

    <div class="order-1 ml-auto flex items-center md:order-2 lg:ml-0">

      
      
      
      <button
        aria-label="search"
        class="text-black hover:text-primary  inline-block px-3 text-xl dark:text-white"
        id="search_toggle">
        <svg xmlns="http://www.w3.org/2000/svg" height="16" width="16" viewBox="0 0 512 512" fill="currentColor"><path d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"/></svg>
      </button>
      

      
      
      <div class="px-3 text-black hover:text-primary-700 dark:text-white dark:hover:text-primary-300
            [&.active]:font-bold [&.active]:text-black/90 dark:[&.active]:text-white">
        <button class="theme-toggle mt-1" accesskey="t" title="appearance">
          <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
               fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
               stroke-linejoin="round" class="dark:hidden">
            <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
          </svg>
          <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
               fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
               stroke-linejoin="round" class=" dark:block [&:not(dark)]:hidden">
            <circle cx="12" cy="12" r="5"></circle>
            <line x1="12" y1="1" x2="12" y2="3"></line>
            <line x1="12" y1="21" x2="12" y2="23"></line>
            <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
            <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
            <line x1="1" y1="12" x2="3" y2="12"></line>
            <line x1="21" y1="12" x2="23" y2="12"></line>
            <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
            <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
          </svg>
        </button>
      </div>
      

      
      

      
      
    </div>
  </nav>
</header>


<div id="search" class="hidden p-3"></div>


        
      
    </div>
    <div class="page-body  my-10">
      





<div class="mx-auto flex max-w-screen-xl">
  



<aside class="hb-sidebar-container max-lg:[transform:translate3d(0,-100%,0)] lg:hidden xl:block">
  
  <div class="px-4 pt-4 lg:hidden">
    
    
  </div>
  <div class="hb-scrollbar lg:h-[calc(100vh-var(--navbar-height))]">
    <ul class="flex flex-col gap-1 lg:hidden">
      
      
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/event/"
    
  >Recent &amp; Upcoming Talks
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/event/example/"
    
  >Example Talk
    </a>
              
            </li></ul>
      </div></li>
        <li class="open"><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/"
    
  >Blog
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19.ememwithmapvariationalbayesian/"
    
  >🧬 EM algorithm, EM with max a posterior, full Bayesian inferences (variational Bayesian) which might be used in auto-coder
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/18.what_make_neural_network_powerful_popular/"
    
  >🧬 what makes neural network so powerful and popular
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/17.smooth.tests.of.goodness.of.fit/"
    
  >🧬 Smooth tests of goodness of Fit used in Stijn Hawinkel ’s plos one paper
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/16.how.good.is.negative.binominal/"
    
  >🧬 how good is negative binomial, why AI works for biology
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/15.use.seurat.for.bulk.rnaseq/"
    
  >🧬 Seurat to analysis large sample size of bulk RNAseq. Discuss Wixcox is better for clinical samples
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/14.spatial_genomics_presentation/"
    
  >🧬 Spatial genomics presentation
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/13.yolov8/"
    
  >🧬 Bayesian abundance statistics in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/12.basian_abundance_statistics/"
    
  >🧬 Yolov8
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11.velocity_reasoning/"
    
  >🧬 Velocity reasoning and discussion
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/10.cca_reasoning/"
    
  >🧬 CCA reasoning and integration discussion
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/09.sctransform/"
    
  >🧬 sctransfrom reasoning
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/08.clinical_relevance/"
    
  >🧬 Clinical relevance evaluation in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/07.cord_diagram/"
    
  >🧬 Expressive cord diagram in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/06.noise_or_true_stem_cell/"
    
  >🧬 Determine whether it’s noise or true stem like cells in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/05.xensoft_caki_paper/"
    
  >🧬 quantify xenograft and host gene expression from the same sample
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/04.huangcong_oncogene/"
    
  >🧬 counteract treatment-specific background in differential expressed gene analysis (huangcong paper)
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/03.well-based_scrnaseq_herv_quantification/"
    
  >🧬 quantify transgenes and herv from well-based single-cell RNAseq
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/02.10x_scrnaseq_herv_quantification/"
    
  >🧬 quantify transgenes and herv from 10x single-cell RNAseq
    </a>
              
            </li><li class="flex flex-col open"><a
    class="hb-sidebar-custom-link
      sidebar-active-item bg-primary-100 font-semibold text-primary-800 dark:bg-primary-300 dark:text-primary-900"
    href="/post/01.bulkrnaseq_allow_herv_quantification/"
    
  >🧬 Quantify HERV and transgenes from Bulk-RNAseq
    </a>
  
    <ul class="hb-sidebar-mobile-toc"><li>
              <a
                href="#what-are-hervs-and-transgenes"
                class="hb-docs-link"
              >What are hERVs and transgenes</a>
            </li>
          <li>
              <a
                href="#how-to-quantify-them"
                class="hb-docs-link"
              >How to quantify them</a>
            </li>
          <li>
              <a
                href="#quantificaiton-using-salmon"
                class="hb-docs-link"
              >Quantificaiton using Salmon</a>
            </li>
          </ul>
  
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/00.model_aerosol_depostion/"
    
  >💨 Modeling the deposition of infectious agents suggests that aerosolized drugs offer enhanced specificity
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.get-started/"
    
  >🎉 Easily create your own simple yet highly customizable blog
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.second-brain/"
    
  >🧠 Sharpen your thinking with a second brain
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.data-visualization/"
    
  >📈 Communicate your results effectively with the best data visualizations
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.teach-courses/"
    
  >👩🏼‍🏫 Teach academic courses
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.project-management/"
    
  >✅ Manage your projects
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/projects/"
    
  >Projects
    </a></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/"
    
  >Projects
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/pandas/"
    
  >Pandas
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/pytorch/"
    
  >PyTorch
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/scikit/"
    
  >scikit-learn
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/experience/"
    
  >Experience
    </a></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/"
    
  >Teaching
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/js/"
    
  >Learn JavaScript
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/python/"
    
  >Learn Python
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/"
    
  >Publications
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/preprint/"
    
  >An example preprint / working paper
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/journal-article/"
    
  >An example journal article
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/conference-paper/"
    
  >An example conference paper
    </a>
              
            </li></ul>
      </div></li>
    </ul>

    <div class="max-xl:hidden h-0 w-64 shrink-0"></div></div>

</aside>
  

<nav class="hb-toc order-last hidden w-64 shrink-0 xl:block print:hidden px-4" aria-label="table of contents">
  











  <div class="hb-scrollbar text-sm [hyphens:auto] sticky top-16 overflow-y-auto pr-4 pt-6 max-h-[calc(100vh-var(--navbar-height)-env(safe-area-inset-bottom))] -mr-4 rtl:-ml-4"><p class="mb-4 font-semibold tracking-tight">On this page</p><ul>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#what-are-hervs-and-transgenes">What are hERVs and transgenes</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#how-to-quantify-them">How to quantify them</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#their-fasta-and-gtf">Their fasta and gtf</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#how-tools-deal-with-multimapping">How tools deal with multimapping</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#why-em-algorithm-for-multimapping">Why EM algorithm for multimapping</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#em-is-proved-to-resolve-gmm-well">EM is proved to resolve GMM well</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#multimapping-is-modeled-as-mixtures-of-distributions-that-account-for-mapping-bias">Multimapping is modeled as mixtures of distributions that account for mapping bias</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-12 rtl:pr-12 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#e-step">E-Step</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-12 rtl:pr-12 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#bias-corrections">Bias Corrections</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-12 rtl:pr-12 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#m-step">M-Step</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#quantificaiton-using-salmon">Quantificaiton using Salmon</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#excerpt-scripts">Excerpt scripts</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#build-index">build index</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#qc">QC</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#quantification">quantification</a>
      </li></ul>

  
  
    
    
  



    












  </div>
  </nav>


  <article class="w-full break-words flex min-h-[calc(100vh-var(--navbar-height))] min-w-0 justify-center pb-8 pr-[calc(env(safe-area-inset-right)-1.5rem)]">
    <main class="w-full min-w-0 max-w-6xl px-6 pt-4 md:px-12">

      

      <h1 class="mt-2 text-4xl font-bold tracking-tight text-slate-900 dark:text-slate-100">🧬 Quantify HERV and transgenes from Bulk-RNAseq</h1>

      <div class="mt-4 mb-16">
      <div class="text-gray-500 dark:text-gray-300 text-sm flex items-center flex-wrap gap-y-2"><span class="mr-1">Mar 24, 2025</span><span class="mx-1">·</span>
        
          
          
          <div class="group inline-flex items-center text-current gap-x-1.5 mx-1">
            
            
            <img src="/author/jiyuan-liu/avatar_hu_98ce379ab30758a2.webp" alt="Jiyuan Liu" class="inline-block h-4 w-4 rounded-full border border-current" loading="lazy" />
            
            <div >Jiyuan Liu</div>
          </div>
          
        

        
        <span class="mx-1">·</span>
        <span class="mx-1">
          8 min read
        </span>
        
        </div>

        <div class="mt-3">
          





        </div>
      </div>



      
      
      
      
      

      
      
      
      
      
      
      
      
      
      
      <div class="article-header article-container featured-image-wrapper mt-4 mb-16" style="max-width: 720px; max-height: 720px;">
        <div style="position: relative">
          <img src="/post/01.bulkrnaseq_allow_herv_quantification/featured_hu_b87379d4f1989fc.webp" width="720" height="720" alt="" class="featured-image">
          <span class="article-header-caption">Image credit: <a href="https://unsplash.com" target="_blank" rel="noopener"><strong>Illustration by Round Icons on unsplash</strong></a></span>
        </div>
      </div>
      

      
      

      <div class="prose prose-slate lg:prose-xl dark:prose-invert">
        <h2 id="what-are-hervs-and-transgenes">What are hERVs and transgenes</h2>
<p>Human Endogenous Retroviruses (hERVs) are ancient viral sequences embedded in the human genome. The roles of hERVs in gene regulation, immunity, development and cancer are under intense research. Transgenes including GFP, CRE, Luciferase, rtTA/Tet-On/Tet-Off and epitopically expressed genes are common targets to be quantified in sample from transgenic mouse models.</p>
<h2 id="how-to-quantify-them">How to quantify them</h2>
<p>To quantify them, we of course need modify fasta and gtf files to include the their sequencing and annotation. Also, because their sequencing reads are frequently repetitive or overlaping with host genome, we need a feature quantification algorithm to handle multimapping commonly seen for HERV and transgenes.</p>
<h3 id="their-fasta-and-gtf">Their fasta and gtf</h3>
<p>The transgenes can be included as chromosomes in fasta as below</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">&gt; EGFP
</span></span><span class="line"><span class="cl">ATGGTGAGCAAGGGCGAGGAGCTGTTCACCGGGGTGGTGCCCATCCTGGTCGAGCTGGACGGCG
</span></span><span class="line"><span class="cl">ACGTAAACGGCCACAAGTTCAGCGTGTCCGGCGAGGGCGAGGGCGATGCCACCTACGGCAAGCT
</span></span><span class="line"><span class="cl">GACCCTGAAGTTCATCTGCACCACCGGCAAGCTGCCCGTGCCCTGGCCCACCCTCGTGACCACC
</span></span><span class="line"><span class="cl">CTGACCTACGGCGTGCAGTGCTTCAGCCGCTACCCCGACCACATGAAGCAGCACGACTTCTTCA
</span></span><span class="line"><span class="cl">AGTCCGCCATGCCCGAAGGCTACGTCCAGGAGCGCACCATCTTCTTCAAGGACGACGGCAACTA
</span></span><span class="line"><span class="cl">CAAGACCCGCGCCGAGGTGAAGTTCGAGGGCGACACCCTGGTGAACCGCATCGAGCTGAAGGGC
</span></span><span class="line"><span class="cl">ATCGACTTCAAGGAGGACGGCAACATCCTGGGGCACAAGCTGGAGTACAACTACAACAGCCACA
</span></span><span class="line"><span class="cl">ACGTCTATATCATGGCCGACAAGCAGAAGAACGGCATCAAGGTGAACTTCAAGATCCGCCACAA
</span></span><span class="line"><span class="cl">CATCGAGGACGGCAGCGTGCAGCTCGCCGACCACTACCAGCAGAACACCCCCATCGGCGACGGC
</span></span><span class="line"><span class="cl">CCCGTGCTGCTGCCCGACAACCACTACCTGAGCACCCAGTCCGCCCTGAGCAAAGACCCCAACG
</span></span><span class="line"><span class="cl">AGAAGCGCGATCACATGGTCCTGCTGGAGTTCGTGACCGCCGCCGGGATCACTCTCGGCATGGA
</span></span><span class="line"><span class="cl">CGAGCTGTACAAGTAA
</span></span></code></pre></div><p>The transgenes are formated as below in gtf</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">EGFP    self    exon    <span class="m">1</span>       <span class="m">720</span>     <span class="m">500</span>     +       .       gene_id <span class="s2">&#34;custom_3&#34;</span><span class="p">;</span> gene_name <span class="s2">&#34;EGFP&#34;</span>
</span></span><span class="line"><span class="cl">Cre     self    exon    <span class="m">1</span>       <span class="m">1023</span>    <span class="m">500</span>     +       .       gene_id <span class="s2">&#34;custom_4&#34;</span><span class="p">;</span> gene_name <span class="s2">&#34;Cre&#34;</span>
</span></span><span class="line"><span class="cl">ERT2    self    exon    <span class="m">1</span>       <span class="m">931</span>     <span class="m">500</span>     +       .       gene_id <span class="s2">&#34;custom_5&#34;</span><span class="p">;</span> gene_name <span class="s2">&#34;ERT2&#34;</span>
</span></span></code></pre></div><p>The hERV sequences are already in reference genome, but their gtf need be added as below. We will talk about <strong><a href="willbecomposedsoon">how to generate the hERV gtf matching the latest reference genome in anohter blog</a></strong>.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">chr10   RepeatMasker    exon    <span class="m">3083835</span> <span class="m">3085549</span> 4.5     -       .       gene_id <span class="s2">&#34;RepMasker_dc1fbd39-e43b-4ac4-90ef-1da20b1d6249&#34;</span><span class="p">;</span> gene_name <span class="s2">&#34;chr10:LTR/ERVL-MaLR:ORR1A0:3&#34;</span>
</span></span><span class="line"><span class="cl">chr10   RepeatMasker    exon    <span class="m">3114480</span> <span class="m">3115385</span> <span class="m">6</span>       -       .       gene_id <span class="s2">&#34;RepMasker_c49d8024-0b45-46a9-bcaa-c44f7a305a03&#34;</span><span class="p">;</span> gene_name <span class="s2">&#34;chr10:LTR/ERVK:RLTR13F:8&#34;</span>
</span></span><span class="line"><span class="cl">chr10   RepeatMasker    exon    <span class="m">3122665</span> <span class="m">3123530</span> 13.8    +       .       gene_id <span class="s2">&#34;RepMasker_5154950a-1e53-4528-a5cf-37d121e3e1a4&#34;</span><span class="p">;</span> gene_name <span class="s2">&#34;chr10:LTR/ERVK:RLTR13D6:11&#34;</span>
</span></span><span class="line"><span class="cl">chr11   RepeatMasker    exon    <span class="m">3055489</span> <span class="m">3061982</span> 25.9    +       .       gene_id <span class="s2">&#34;RepMasker_f6a34047-357d-4ed5-9744-eaf4c862be26&#34;</span><span class="p">;</span> gene_name <span class="s2">&#34;chr11:LTR/ERVK:IAPLTR3-int:1&#34;</span>
</span></span><span class="line"><span class="cl">chr11   RepeatMasker    exon    <span class="m">3139464</span> <span class="m">3140302</span> 11.7    +       .       gene_id <span class="s2">&#34;RepMasker_ad8b23ab-de54-4f73-be4d-f8405ad13e66&#34;</span><span class="p">;</span> gene_name <span class="s2">&#34;chr11:LTR/ERVK:RLTR33:3&#34;</span>
</span></span><span class="line"><span class="cl">chr11   RepeatMasker    exon    <span class="m">3410867</span> <span class="m">3413285</span> 16.9    +       .       gene_id <span class="s2">&#34;RepMasker_ec06076e-c8d0-4a94-8cfa-358f352bb1e8&#34;</span><span class="p">;</span> gene_name <span class="s2">&#34;chr11:LTR/ERV1:RLTR14-int:7&#34;</span>
</span></span><span class="line"><span class="cl">chr12   RepeatMasker    exon    <span class="m">3052672</span> <span class="m">3059141</span> 1.6     +       .       gene_id <span class="s2">&#34;RepMasker_8ea0db28-2cd0-4d7e-b151-b5b6480ee320&#34;</span><span class="p">;</span> gene_name <span class="s2">&#34;chr12:LTR/ERVK:IAPEz-int:1&#34;</span>
</span></span><span class="line"><span class="cl">chr12   RepeatMasker    exon    <span class="m">3065049</span> <span class="m">3072337</span> 27.3    -       .       gene_id <span class="s2">&#34;RepMasker_f3379b4a-cdfa-4b24-a4c9-5dacbdaf004c&#34;</span><span class="p">;</span> gene_name <span class="s2">&#34;chr12:LTR/ERV1:MMERGLN-int:2&#34;</span>
</span></span><span class="line"><span class="cl">chr12   RepeatMasker    exon    <span class="m">3084084</span> <span class="m">3084938</span> 5.8     -       .       gene_id <span class="s2">&#34;RepMasker_062a5737-d991-4220-9648-33c43ad467a2&#34;</span><span class="p">;</span> gene_name <span class="s2">&#34;chr12:LTR/ERVK:IAPEz-int:4&#34;</span>
</span></span></code></pre></div><h3 id="how-tools-deal-with-multimapping">How tools deal with multimapping</h3>
<p>Below list common feature count/qunatification tools emphasizing on the core algorithms they wrapped to deal with multimapping. <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-017-4002-1" target="_blank" rel="noopener">These common tools have similar quantification accuracy</a>.</p>














<table class="table-auto w-full">
  
    
    
    <thead>
      <tr>  <th class="border-b dark:border-slate-600 font-medium p-4 pt-0 pb-3 text-slate-400 dark:text-slate-200 text-left">Tool</th>  <th class="border-b dark:border-slate-600 font-medium p-4 pt-0 pb-3 text-slate-400 dark:text-slate-200 text-left">Default(option)</th>  <th class="border-b dark:border-slate-600 font-medium p-4 pt-0 pb-3 text-slate-400 dark:text-slate-200 text-left">Core algorithm</th>  </tr>
    </thead>
  
  <tbody>
  
    <tr>
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">HTSeq-count</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">discard(&ndash;nonunique=all/fraction)</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">fraction</td>
        
      
    </tr>
  
    <tr>
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">featureCounts</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">discard(-M &ndash;fraction)</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">fraction</td>
        
      
    </tr>
  
    <tr>
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">RSEM</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">EM Probabilistic</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">EM</td>
        
      
    </tr>
  
    <tr>
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">Salmon</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">EM Probabilistic</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">(Bayesian) EM</td>
        
      
    </tr>
  
    <tr>
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">kallisto</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">EM Probabilistic</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">EM</td>
        
      
    </tr>
  
    <tr>
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">StringTie</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">EM Probabilistic</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">EM</td>
        
      
    </tr>
  
    <tr>
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">TEtranscripts</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">EM Probabilistic</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">EM</td>
        
      
    </tr>
  
    <tr>
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">Telescope</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">EM Probabilistic</td>
        
      
        
          <td data-table-dtype="text" class="border-b border-slate-100 dark:border-slate-700 p-4 text-slate-500 dark:text-slate-400">EM with MAP</td>
        
      
    </tr>
  
  </tbody>
  
    <caption class="table-caption"><strong>Table 1. Multimapping options and algorithms of tools</strong></caption>
  
</table>

<p>They by default may simply discard multimapping reads, which ignores most of hERVs. With carefully specified arguments, they either distribute read counts equally among certian features (fraction) or estimate using EM algorithm.</p>
<hr>
<h3 id="why-em-algorithm-for-multimapping">Why EM algorithm for multimapping</h3>
<p>&ldquo;Fraction&rdquo; algorithm overlooked how likely the reads can match with reference sequences. The biological prior information, if reasonably exploited, can significantly improve the multimapping quantification. Beatifully, multimapping by nature is similar to the problem of Gaussian Mixtures and EM has been proved to resolve Gaussian Mixture Model (GMM) well.</p>
<h4 id="em-is-proved-to-resolve-gmm-well">EM is proved to resolve GMM well</h4>
<p>GMM describes a set of data points of which each comes from one of several Gaussian distributions, but we don&rsquo;t know which one. This missing information makes direct estimation (like via maximum likelihood) of the Gaussian parameters hard. EM solve this by iterating below E and M steps until converge:</p>
<ul>
<li>
<p>E-step (Expectation): Use initiallized parameter to compute the <strong>posterior probability (sounds like bayesian, but it is not in standard EM)</strong> that each data point was generated by each Gaussian (i.e., the responsibility γ for each component).</p>
</li>
<li>
<p>M-step (Maximization): Update the parameters (π, μ, σ²) by performing <strong>weighted MLE</strong> using the expected values (responsibilities) computed in the E-step.</p>
</li>
</ul>
<hr>
<h4 id="multimapping-is-modeled-as-mixtures-of-distributions-that-account-for-mapping-bias">Multimapping is modeled as mixtures of distributions that account for mapping bias</h4>
<p><a href="https://www.nature.com/articles/nmeth.4197" target="_blank" rel="noopener">Salmon model </a> describes a set of fragments (seqeuncing reads) of which each fragment <em>f</em> originates from one of the several transcripts <em>t</em> at a probability distribution  $P(t \mid f)$ 
, but we don&rsquo;t know which one. Thus EM goes:</p>
<h5 id="e-step"><em>E-Step</em></h5>
<p>For each fragment  $f$ 
 and transcript  $t$ 
:</p>

$$
P(t \mid f) = \frac{\alpha_t \cdot P(f \mid t)}{\sum_{t'} \alpha_{t'} \cdot P(f \mid t')}
$$


<p>Where:</p>
<ul>
<li> $\alpha_t$ 
: Current estimate of the relative abundance of transcript  $t$ 
.</li>
<li> $P(f \mid t)$ 
: Probability of observing fragment  $f$ 
 given transcript  $t$ 
, incorporating bias corrections.</li>
</ul>
<hr>
<h5 id="bias-corrections"><em>Bias Corrections</em></h5>
<p>The fragment-transcript probability is modeled as:</p>

$$
P(f \mid t) = P_{\text{len}}(l_f \mid t) \cdot P_{\text{pos}}(p_f \mid t) \cdot P_{\text{seq}}(s_f \mid t)
$$


<p>Where:</p>
<ul>
<li> $l_f$ 
: Length of fragment  $f$ 
.</li>
<li> $p_f$ 
: Position of fragment  $f$ 
 on transcript  $t$ 
.</li>
<li> $s_f$ 
: Sequence content of fragment  $f$ 
.</li>
<li> $P_{\text{len}}$ 
: Probability of fragment length, modeling the empirical fragment length distribution.</li>
<li> $P_{\text{pos}}$ 
: Positional bias model, capturing biases in fragment start locations.</li>
<li> $P_{\text{seq}}$ 
: Sequence-specific bias model, correcting for nucleotide composition effects.</li>
</ul>
<p>Importantly, the bias models:  $P_{\text{len}}$ 
,  $P_{\text{pos}}$ 
 and  $P_{\text{seq}}$ 
 are not directly updated by the EM algorithm during transcript abundance estimation. Instead, these bias models are learned once, typically in a preprocessing or auxiliary phase before the EM iterations begin. The EM algorithm then uses the fixed values of these bias models to compute the probability  $P(f \mid t)$ 
 in the E-step.</p>
<p><strong>The only parameters updated by EM are the relative abundances  $\alpha_t$ 
 of the transcripts.</strong></p>
<hr>
<h5 id="m-step"><em>M-Step</em></h5>
<p>Update the abundance estimates:</p>

$$
\alpha_t = \sum_f P(t \mid f)
$$


<p>Then normalize:</p>

$$
\sum_t \alpha_t = 1
$$


<hr>
<h2 id="quantificaiton-using-salmon">Quantificaiton using Salmon</h2>
<p>Now we see EM algorithm superior than fraction algorithms in quantifing multimappings since EM</p>
<ul>
<li>is well suitable for estimating parameters in Gaussian Mixture Models, which is similar to the multimapping mathematical model</li>
<li>takes multimapping mathematical models accounting for the mapping bias</li>
</ul>
<p>We implemented Salmon since it allows both EM and full bayesian EM. We will talk about <a href="donesoon">EM (with MLE), EM with MAP and bayesian EM</a> as well as <a href="donesoon">Frequentist vs Bayesian</a> in other blogs.</p>
<p>Please see source codes in <a href="https://github.com/jliu678/herv_project_siyi/tree/main" target="_blank" rel="noopener">github</a> that can be deploied in Cloud Cluster Computational platform and local desktops, and allows for:</p>
<ul>
<li>batch computaion with tunable batch size</li>
<li>automatic environment setting up</li>
<li>file preprocessing</li>
<li>QC</li>
<li>index buidling, alignment and feature counting by either <code>subread</code> or <code>salmon</code></li>
</ul>
<p><a href="https://github.com/jliu678/herv_project_siyi/tree/main" target="_blank" rel="noopener">github</a> also contains usage examples using <code>SLURM</code> or <code>LSF</code> job scheduler and handy utility tools.</p>
<hr>
<h3 id="excerpt-scripts">Excerpt scripts</h3>
<p>Below excerpt scripts call each needed analysis step specified as <code>ANALYSIS_STEP</code> array, including file preprocessing, index buidling by <code>build_index()</code>, QC by <code>qc_all()</code>, alignment by <code>align_all()</code> if using <code>subread</code>, and feature counting by <code>count_all()</code>.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">	<span class="k">for</span> i in <span class="si">${</span><span class="nv">ANALYSIS_STEP</span><span class="p">[@]</span><span class="si">}</span><span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">	  timed_print <span class="s2">&#34;</span><span class="nv">$i</span><span class="s2">-ing...&#34;</span>	
</span></span><span class="line"><span class="cl">		<span class="k">case</span> <span class="s2">&#34;</span><span class="nv">$i</span><span class="s2">&#34;</span> in 
</span></span><span class="line"><span class="cl">			<span class="c1"># for the first 2, do not run if they are in a child/forked process</span>
</span></span><span class="line"><span class="cl">			index<span class="o">)</span> <span class="k">if</span> <span class="o">[</span> <span class="nv">$CHILD</span> <span class="o">=</span> <span class="nb">false</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span> build_index<span class="p">;</span> <span class="k">fi</span> <span class="p">;;</span> 
</span></span><span class="line"><span class="cl">			convert<span class="o">)</span> <span class="k">if</span> <span class="o">[</span> <span class="nv">$CHILD</span> <span class="o">=</span> <span class="nb">false</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span> get_pairs_all<span class="p">;</span> <span class="k">fi</span> <span class="p">;;</span>
</span></span><span class="line"><span class="cl">			qc<span class="o">)</span> qc_all <span class="p">;;</span>
</span></span><span class="line"><span class="cl">			align<span class="o">)</span> align_all <span class="p">;;</span>
</span></span><span class="line"><span class="cl">			count<span class="o">)</span> count_all <span class="p">;;</span>
</span></span><span class="line"><span class="cl">		<span class="k">esac</span> 
</span></span><span class="line"><span class="cl">		timed_print <span class="s2">&#34;finished </span><span class="nv">$i</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">	<span class="k">done</span>
</span></span></code></pre></div><p>Some of the core functions in the anylysis are below:</p>
<h4 id="build-index">build index</h4>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">salmon_build_index<span class="o">(){</span> <span class="c1"># supports multiple transcripts</span>
</span></span><span class="line"><span class="cl">	<span class="c1"># get dir name for the index, which is the transcript names seperated by a hyphen(-)</span>
</span></span><span class="line"><span class="cl">	<span class="c1"># eg. for erv.fa and line.fa, you get: erv-line</span>
</span></span><span class="line"><span class="cl">	<span class="nb">local</span> <span class="nv">index_name</span><span class="o">=</span><span class="s2">&#34;salmon/</span><span class="k">$(</span><span class="nv">IFS</span><span class="o">=</span>-<span class="p">;</span> <span class="nb">echo</span> <span class="s2">&#34;</span><span class="si">${</span><span class="nv">TRANSCRIPTS</span><span class="p">[*]%.*</span><span class="si">}</span><span class="s2">&#34;</span><span class="k">)</span><span class="s2">&#34;</span> 
</span></span><span class="line"><span class="cl">	<span class="k">if</span> <span class="o">[</span> <span class="s2">&#34;</span><span class="nv">$OVER_WRITE</span><span class="s2">&#34;</span> <span class="o">=</span> <span class="s2">&#34;true&#34;</span> <span class="o">]</span> <span class="o">||</span> <span class="o">[</span> ! -d <span class="s2">&#34;</span><span class="si">${</span><span class="nv">index_name</span><span class="si">}</span><span class="s2">_index&#34;</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span> 
</span></span><span class="line"><span class="cl">		timed_print <span class="s2">&#34;building salmon index @: </span><span class="si">${</span><span class="nv">index_name</span><span class="si">}</span><span class="s2">_index&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">		<span class="c1"># salmon requires a decoy, which in this case is the entire reference genome</span>
</span></span><span class="line"><span class="cl">		<span class="k">if</span> <span class="o">[</span> <span class="s2">&#34;</span><span class="nv">$OVER_WRITE</span><span class="s2">&#34;</span> <span class="o">=</span> <span class="s2">&#34;true&#34;</span> <span class="o">]</span> <span class="o">||</span> <span class="o">[</span> ! -f <span class="s2">&#34;</span><span class="si">${</span><span class="nv">index_name</span><span class="si">}</span><span class="s2">_decoys.txt&#34;</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
</span></span><span class="line"><span class="cl">			timed_print <span class="s2">&#34;building decoys...&#34;</span>
</span></span><span class="line"><span class="cl">			grep <span class="s2">&#34;^&gt;&#34;</span> <span class="s2">&#34;</span><span class="si">${</span><span class="nv">REF_GENOME</span><span class="si">}</span><span class="s2">&#34;</span> <span class="p">|</span> cut -d <span class="s2">&#34; &#34;</span> -f <span class="m">1</span> &gt; <span class="s2">&#34;</span><span class="si">${</span><span class="nv">index_name</span><span class="si">}</span><span class="s2">_decoys.txt&#34;</span>
</span></span><span class="line"><span class="cl">			sed -i.bak -e <span class="s1">&#39;s/&gt;//g&#39;</span> <span class="s2">&#34;</span><span class="si">${</span><span class="nv">index_name</span><span class="si">}</span><span class="s2">_decoys.txt&#34;</span>
</span></span><span class="line"><span class="cl">		<span class="k">fi</span> 
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">		<span class="c1"># salmon requires all the transcripts being quantified to be concat into one file</span>
</span></span><span class="line"><span class="cl">		<span class="k">if</span> <span class="o">[</span> <span class="s2">&#34;</span><span class="nv">$OVER_WRITE</span><span class="s2">&#34;</span> <span class="o">=</span> <span class="s2">&#34;true&#34;</span> <span class="o">]</span> <span class="o">||</span> <span class="o">[</span> ! -f <span class="s2">&#34;</span><span class="si">${</span><span class="nv">index_name</span><span class="si">}</span><span class="s2">_gentrome.fa&#34;</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
</span></span><span class="line"><span class="cl">			timed_print <span class="s2">&#34;building gentrome...&#34;</span>
</span></span><span class="line"><span class="cl">			cat <span class="s2">&#34;</span><span class="si">${</span><span class="nv">TRANSCRIPTS</span><span class="p">[@]</span><span class="si">}</span><span class="s2">&#34;</span> <span class="s2">&#34;</span><span class="si">${</span><span class="nv">REF_TRANSCRIPT</span><span class="si">}</span><span class="s2">&#34;</span> <span class="s2">&#34;</span><span class="si">${</span><span class="nv">REF_GENOME</span><span class="si">}</span><span class="s2">&#34;</span> &gt; <span class="s2">&#34;</span><span class="si">${</span><span class="nv">index_name</span><span class="si">}</span><span class="s2">.fa&#34;</span>
</span></span><span class="line"><span class="cl">		<span class="k">fi</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">		<span class="c1"># adding --gencode flag due to salmon doing some preprocessing for it</span>
</span></span><span class="line"><span class="cl">		<span class="c1"># this flag does nothing for none gencode files (I think)</span>
</span></span><span class="line"><span class="cl">		salmon index -t <span class="si">${</span><span class="nv">index_name</span><span class="si">}</span>.fa -d <span class="s2">&#34;</span><span class="si">${</span><span class="nv">index_name</span><span class="si">}</span><span class="s2">_decoys.txt&#34;</span> -i <span class="s2">&#34;</span><span class="si">${</span><span class="nv">index_name</span><span class="si">}</span><span class="s2">_index&#34;</span> --gencode
</span></span><span class="line"><span class="cl">	<span class="k">fi</span>
</span></span><span class="line"><span class="cl"><span class="o">}</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">build_index<span class="o">()</span> <span class="o">{</span>
</span></span><span class="line"><span class="cl">	<span class="k">case</span> <span class="nv">$ALIGN_METHOD</span> in
</span></span><span class="line"><span class="cl">		subread<span class="o">)</span> subread_build_index <span class="p">;;</span>
</span></span><span class="line"><span class="cl">		salmon<span class="o">)</span> salmon_build_index <span class="p">;;</span>
</span></span><span class="line"><span class="cl">	<span class="k">esac</span>
</span></span><span class="line"><span class="cl"><span class="o">}</span>
</span></span></code></pre></div><h4 id="qc">QC</h4>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">fastp_qc<span class="o">(){</span> 
</span></span><span class="line"><span class="cl">	<span class="c1"># results stored in /tmp/{name of the source}/qc/* and all end in *.qc.fq.gz</span>
</span></span><span class="line"><span class="cl">	<span class="k">if</span> <span class="o">[</span> <span class="s2">&#34;</span><span class="nv">$OVER_WRITE</span><span class="s2">&#34;</span> <span class="o">=</span> <span class="s2">&#34;true&#34;</span> <span class="o">]</span> <span class="o">||</span> <span class="o">[</span> ! -f <span class="s2">&#34;tmp/</span><span class="si">${</span><span class="nv">SOURCE</span><span class="si">}</span><span class="s2">/qc/</span><span class="nv">$1</span><span class="s2">.qc.fq.gz&#34;</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
</span></span><span class="line"><span class="cl">		<span class="k">if</span> <span class="o">[[</span> <span class="nv">$#</span> -eq <span class="m">2</span> <span class="o">]]</span><span class="p">;</span> <span class="k">then</span> <span class="c1"># if pair ended</span>
</span></span><span class="line"><span class="cl">			./fastp -i <span class="s2">&#34;tmp/</span><span class="si">${</span><span class="nv">SOURCE</span><span class="si">}</span><span class="s2">/</span><span class="nv">$1</span><span class="s2">.fq.gz&#34;</span> -o <span class="s2">&#34;tmp/</span><span class="si">${</span><span class="nv">SOURCE</span><span class="si">}</span><span class="s2">/qc/</span><span class="nv">$1</span><span class="s2">.qc.fq.gz&#34;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>				-I <span class="s2">&#34;tmp/</span><span class="si">${</span><span class="nv">SOURCE</span><span class="si">}</span><span class="s2">/</span><span class="nv">$2</span><span class="s2">.fq.gz&#34;</span> -O <span class="s2">&#34;tmp/</span><span class="si">${</span><span class="nv">SOURCE</span><span class="si">}</span><span class="s2">/qc/</span><span class="nv">$2</span><span class="s2">.qc.fq.gz&#34;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>				-j <span class="s2">&#34;results/fastp/</span><span class="nv">$1</span><span class="s2">.json&#34;</span> -h <span class="s2">&#34;results/fastp/</span><span class="nv">$1</span><span class="s2">.html&#34;</span>
</span></span><span class="line"><span class="cl">		<span class="k">elif</span> <span class="o">[[</span> <span class="nv">$#</span> -eq <span class="m">1</span> <span class="o">]]</span><span class="p">;</span> <span class="k">then</span> <span class="c1"># if single ended</span>
</span></span><span class="line"><span class="cl">			./fastp -i <span class="s2">&#34;tmp/</span><span class="si">${</span><span class="nv">SOURCE</span><span class="si">}</span><span class="s2">/</span><span class="nv">$1</span><span class="s2">.fq.gz&#34;</span> -o <span class="s2">&#34;tmp/</span><span class="si">${</span><span class="nv">SOURCE</span><span class="si">}</span><span class="s2">/qc/</span><span class="nv">$1</span><span class="s2">.qc.fq.gz&#34;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>				-j <span class="s2">&#34;results/fastp/</span><span class="nv">$1</span><span class="s2">.json&#34;</span> -h <span class="s2">&#34;results/fastp/</span><span class="nv">$1</span><span class="s2">.html&#34;</span>
</span></span><span class="line"><span class="cl">		<span class="k">fi</span>	
</span></span><span class="line"><span class="cl">	<span class="k">fi</span>
</span></span><span class="line"><span class="cl"><span class="o">}</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">qc_all<span class="o">(){</span> <span class="c1"># only supports fastp as of now</span>
</span></span><span class="line"><span class="cl">	<span class="k">if</span> <span class="o">[[</span> ! -d <span class="s2">&#34;tmp/</span><span class="si">${</span><span class="nv">SOURCE</span><span class="si">}</span><span class="s2">/qc&#34;</span> <span class="o">]]</span><span class="p">;</span> <span class="k">then</span> 
</span></span><span class="line"><span class="cl">		mkdir <span class="s2">&#34;tmp/</span><span class="si">${</span><span class="nv">SOURCE</span><span class="si">}</span><span class="s2">/qc&#34;</span>
</span></span><span class="line"><span class="cl">	<span class="k">fi</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">	<span class="k">if</span> <span class="o">[[</span> ! -d <span class="s2">&#34;results/</span><span class="si">${</span><span class="nv">QC_METHOD</span><span class="si">}</span><span class="s2">&#34;</span> <span class="o">]]</span><span class="p">;</span> <span class="k">then</span> 
</span></span><span class="line"><span class="cl">		mkdir <span class="s2">&#34;results/</span><span class="si">${</span><span class="nv">QC_METHOD</span><span class="si">}</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">	<span class="k">fi</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">	timed_print <span class="s2">&#34;qc-ing with </span><span class="nv">$QC_METHOD</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">	<span class="k">while</span> <span class="nv">IFS</span><span class="o">=</span>, <span class="nb">read</span> -r r1 r2<span class="p">;</span> <span class="k">do</span> <span class="c1"># loop over pair file, csv</span>
</span></span><span class="line"><span class="cl">		<span class="k">if</span> <span class="o">[</span> <span class="s2">&#34;</span><span class="nv">$OVER_WRITE</span><span class="s2">&#34;</span> <span class="o">=</span> <span class="s2">&#34;true&#34;</span> <span class="o">]</span> <span class="o">||</span> <span class="o">[</span> ! -d <span class="s2">&#34;results/</span><span class="si">${</span><span class="nv">ALIGN_METHOD</span><span class="si">}</span><span class="s2">/</span><span class="nv">$r1</span><span class="s2">&#34;</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span> 
</span></span><span class="line"><span class="cl">			<span class="c1">#if salmon quant already exists</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">			timed_print <span class="s2">&#34;qc-ing </span><span class="si">${</span><span class="nv">r1</span><span class="si">}</span><span class="s2"> and </span><span class="si">${</span><span class="nv">r2</span><span class="si">}</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">			<span class="k">case</span> <span class="nv">$QC_METHOD</span> in 
</span></span><span class="line"><span class="cl">				fastp<span class="o">)</span> fastp_qc <span class="nv">$r1</span> <span class="nv">$r2</span> <span class="p">;;</span>
</span></span><span class="line"><span class="cl">			<span class="k">esac</span> 
</span></span><span class="line"><span class="cl">		<span class="k">fi</span>
</span></span><span class="line"><span class="cl">	<span class="k">done</span> &lt; <span class="nv">$PAIR_FILE</span>
</span></span><span class="line"><span class="cl"><span class="o">}</span>
</span></span></code></pre></div><h4 id="quantification">quantification</h4>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">salmon_quant<span class="o">()</span> <span class="o">{</span>
</span></span><span class="line"><span class="cl">	<span class="c1"># get the same index_name as specified in &#34;salmon_build_index()&#34;</span>
</span></span><span class="line"><span class="cl">	<span class="nb">local</span> <span class="nv">index_name</span><span class="o">=</span><span class="s2">&#34;salmon/</span><span class="k">$(</span><span class="nv">IFS</span><span class="o">=</span>-<span class="p">;</span> <span class="nb">echo</span> <span class="s2">&#34;</span><span class="si">${</span><span class="nv">TRANSCRIPTS</span><span class="p">[*]%.*</span><span class="si">}</span><span class="s2">&#34;</span><span class="k">)</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">	<span class="c1">#salmon outputs into a dir, so we check for that instead</span>
</span></span><span class="line"><span class="cl">	<span class="c1">#results stored in results/salmon/*</span>
</span></span><span class="line"><span class="cl">	<span class="k">if</span> <span class="o">[</span> <span class="s2">&#34;</span><span class="nv">$OVER_WRITE</span><span class="s2">&#34;</span> <span class="o">=</span> <span class="s2">&#34;true&#34;</span> <span class="o">]</span> <span class="o">||</span> <span class="o">[</span> ! -d <span class="s2">&#34;results/salmon/</span><span class="nv">$1</span><span class="s2">&#34;</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
</span></span><span class="line"><span class="cl">		mkdir results/salmon/<span class="nv">$1</span>
</span></span><span class="line"><span class="cl">		<span class="k">if</span> <span class="o">[[</span> <span class="nv">$#</span> -eq <span class="m">2</span> <span class="o">]]</span><span class="p">;</span> <span class="k">then</span> <span class="c1"># if pair ended</span>
</span></span><span class="line"><span class="cl">			salmon quant -i <span class="s2">&#34;</span><span class="si">${</span><span class="nv">index_name</span><span class="si">}</span><span class="s2">_index&#34;</span> -l A -1 <span class="s2">&#34;tmp/</span><span class="si">${</span><span class="nv">SOURCE</span><span class="si">}</span><span class="s2">/qc/</span><span class="nv">$1</span><span class="s2">.qc.fq.gz&#34;</span> -2 <span class="s2">&#34;tmp/</span><span class="si">${</span><span class="nv">SOURCE</span><span class="si">}</span><span class="s2">/qc/</span><span class="nv">$2</span><span class="s2">.qc.fq.gz&#34;</span> --validateMappings -o <span class="s2">&#34;results/salmon/</span><span class="nv">$1</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">			<span class="c1">#sleep 1</span>
</span></span><span class="line"><span class="cl">		<span class="k">elif</span> <span class="o">[[</span> <span class="nv">$#</span> -eq <span class="m">1</span> <span class="o">]]</span><span class="p">;</span> <span class="k">then</span> <span class="c1"># if single ended</span>
</span></span><span class="line"><span class="cl">			salmon quant -i <span class="s2">&#34;</span><span class="si">${</span><span class="nv">index_name</span><span class="si">}</span><span class="s2">_index&#34;</span> -l A -r <span class="s2">&#34;tmp/</span><span class="si">${</span><span class="nv">SOURCE</span><span class="si">}</span><span class="s2">/qc/</span><span class="nv">$1</span><span class="s2">.qc.fq.gz&#34;</span> --validateMappings -o <span class="s2">&#34;results/salmon/</span><span class="nv">$1</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">			<span class="c1">#sleep 1</span>
</span></span><span class="line"><span class="cl">		<span class="k">fi</span> 
</span></span><span class="line"><span class="cl">	<span class="k">fi</span>
</span></span><span class="line"><span class="cl"><span class="o">}</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">salmon_count<span class="o">()</span> <span class="o">{</span>
</span></span><span class="line"><span class="cl">	<span class="k">while</span> <span class="nv">IFS</span><span class="o">=</span>, <span class="nb">read</span> -r r1 r2<span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">		salmon_quant <span class="nv">$r1</span> <span class="nv">$r2</span>
</span></span><span class="line"><span class="cl">	<span class="k">done</span> &lt; <span class="nv">$PAIR_FILE</span>
</span></span><span class="line"><span class="cl"><span class="o">}</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">count_all<span class="o">(){</span>
</span></span><span class="line"><span class="cl">	<span class="k">if</span> <span class="o">[</span> ! -d <span class="s2">&#34;results/</span><span class="si">${</span><span class="nv">ALIGN_METHOD</span><span class="si">}</span><span class="s2">&#34;</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span> 
</span></span><span class="line"><span class="cl">		mkdir <span class="s2">&#34;results/</span><span class="si">${</span><span class="nv">ALIGN_METHOD</span><span class="si">}</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">	<span class="k">fi</span>
</span></span><span class="line"><span class="cl">	
</span></span><span class="line"><span class="cl">	<span class="k">case</span> <span class="nv">$ALIGN_METHOD</span> in 
</span></span><span class="line"><span class="cl">		subread<span class="o">)</span> subread_count <span class="p">;;</span>
</span></span><span class="line"><span class="cl">		salmon<span class="o">)</span> salmon_count
</span></span><span class="line"><span class="cl">	<span class="k">esac</span>
</span></span><span class="line"><span class="cl"><span class="o">}</span>
</span></span></code></pre></div>
      </div>

      
  <time class="mt-12 mb-8 block text-xs text-gray-500 ltr:text-right rtl:text-left dark:text-gray-400" datetime="2025-03-24T00:00:00.000Z">
    <span>Last updated on</span>
    Mar 24, 2025</time>

      <div class="container mx-auto prose prose-slate lg:prose-xl dark:prose-invert mt-5">
        
        <div class="max-w-prose print:hidden">
  
  

  

<div class="flex justify-center">
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/herv/">HERV</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/transgene/">Transgene</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/feature-quantification/">Feature Quantification</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/algorithm/">Algorithm</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/expectation-maximization-em/">Expectation-Maximization (EM)</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/em-with-map/">EM With MAP</a>
  
</div>


  
<section class="flex flex-row flex-wrap justify-center pt-4 text-xl">
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F01.bulkrnaseq_allow_herv_quantification%2F&amp;text=%F0%9F%A7%AC&#43;Quantify&#43;HERV&#43;and&#43;transgenes&#43;from&#43;Bulk-RNAseq"
    title="X"
    aria-label="X"
    id="share-link-x"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://www.facebook.com/sharer.php?u=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F01.bulkrnaseq_allow_herv_quantification%2F&amp;t=%F0%9F%A7%AC&#43;Quantify&#43;HERV&#43;and&#43;transgenes&#43;from&#43;Bulk-RNAseq"
    title="Facebook"
    aria-label="Facebook"
    id="share-link-facebook"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="currentColor" d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12c0 4.84 3.44 8.87 8 9.8V15H8v-3h2V9.5C10 7.57 11.57 6 13.5 6H16v3h-2c-.55 0-1 .45-1 1v2h3v3h-3v6.95c5.05-.5 9-4.76 9-9.95z"/></svg>
  </a>
  

  
  
  
  
  
  
    
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="mailto:?subject=%F0%9F%A7%AC%20Quantify%20HERV%20and%20transgenes%20from%20Bulk-RNAseq&amp;body=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F01.bulkrnaseq_allow_herv_quantification%2F"
    title="Email"
    aria-label="Email"
    id="share-link-email"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-width="1.5" d="M16.5 12a4.5 4.5 0 1 1-9 0a4.5 4.5 0 0 1 9 0Zm0 0c0 1.657 1.007 3 2.25 3S21 13.657 21 12a9 9 0 1 0-2.636 6.364M16.5 12V8.25"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://www.linkedin.com/shareArticle?url=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F01.bulkrnaseq_allow_herv_quantification%2F&amp;title=%F0%9F%A7%AC&#43;Quantify&#43;HERV&#43;and&#43;transgenes&#43;from&#43;Bulk-RNAseq"
    title="LinkedIn"
    aria-label="LinkedIn"
    id="share-link-linkedin"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" height="1em" viewBox="0 0 448 512"><path fill="currentColor" d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="whatsapp://send?text=%F0%9F%A7%AC&#43;Quantify&#43;HERV&#43;and&#43;transgenes&#43;from&#43;Bulk-RNAseq%20http%3A%2F%2Flocalhost%3A1313%2Fpost%2F01.bulkrnaseq_allow_herv_quantification%2F"
    title="WhatsApp"
    aria-label="WhatsApp"
    id="share-link-whatsapp"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256" fill="currentColor"><path d="m187.58 144.84-32-16a8 8 0 0 0-8 .5l-14.69 9.8a40.55 40.55 0 0 1-16-16l9.8-14.69a8 8 0 0 0 .5-8l-16-32A8 8 0 0 0 104 64a40 40 0 0 0-40 40 88.1 88.1 0 0 0 88 88 40 40 0 0 0 40-40 8 8 0 0 0-4.42-7.16ZM152 176a72.08 72.08 0 0 1-72-72 24 24 0 0 1 19.29-23.54l11.48 23L101 118a8 8 0 0 0-.73 7.51 56.47 56.47 0 0 0 30.15 30.15A8 8 0 0 0 138 155l14.61-9.74 23 11.48A24 24 0 0 1 152 176ZM128 24a104 104 0 0 0-91.82 152.88l-11.35 34.05a16 16 0 0 0 20.24 20.24l34.05-11.35A104 104 0 1 0 128 24Zm0 192a87.87 87.87 0 0 1-44.06-11.81 8 8 0 0 0-6.54-.67L40 216l12.47-37.4a8 8 0 0 0-.66-6.54A88 88 0 1 1 128 216Z"/></svg>
  </a>
  
</section>


  








  
  
    



  
  
  
    
  
  
  

<div class="flex pt-12 pb-4">
  
  
  <img
    class="mr-4 h-24 w-24 rounded-full"
    width="96"
    height="96"
    alt="Jiyuan Liu"
  src="/author/jiyuan-liu/avatar_hu_6297b4a1a274518.jpg"
  loading="lazy"
  />
  
  <div class="place-self-center">
    <div class="text-[0.6rem] uppercase leading-3 text-neutral-500 dark:text-neutral-400">
      Authors
    </div>
    <div class="font-semibold leading-6 text-neutral-800 dark:text-neutral-300">
      <a href="http://localhost:1313/" class="no-underline">
      Jiyuan Liu
      </a>
    </div>

    
    <div class="text-sm font-bold text-neutral-700 dark:text-neutral-300">
    Postdoctoral Scientist
    </div>
    


    

    <div class="text-2xl sm:text-lg pt-1">

      
<div class="flex flex-wrap text-neutral-500 dark:text-neutral-300">
  
    
    
    
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="mailto:jliu78@mgh.harvard.edu"
      
      aria-label="At-Symbol"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-width="1.5" d="M16.5 12a4.5 4.5 0 1 1-9 0a4.5 4.5 0 0 1 9 0Zm0 0c0 1.657 1.007 3 2.25 3S21 13.657 21 12a9 9 0 1 0-2.636 6.364M16.5 12V8.25"/></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://github.com/jliu678"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Brands/Github"
    ><svg style="height: 1em;" fill="currentColor" viewBox="3 3 18 18"><path d="M12 3C7.0275 3 3 7.12937 3 12.2276C3 16.3109 5.57625 19.7597 9.15374 20.9824C9.60374 21.0631 9.77249 20.7863 9.77249 20.5441C9.77249 20.3249 9.76125 19.5982 9.76125 18.8254C7.5 19.2522 6.915 18.2602 6.735 17.7412C6.63375 17.4759 6.19499 16.6569 5.8125 16.4378C5.4975 16.2647 5.0475 15.838 5.80124 15.8264C6.51 15.8149 7.01625 16.4954 7.18499 16.7723C7.99499 18.1679 9.28875 17.7758 9.80625 17.5335C9.885 16.9337 10.1212 16.53 10.38 16.2993C8.3775 16.0687 6.285 15.2728 6.285 11.7432C6.285 10.7397 6.63375 9.9092 7.20749 9.26326C7.1175 9.03257 6.8025 8.08674 7.2975 6.81794C7.2975 6.81794 8.05125 6.57571 9.77249 7.76377C10.4925 7.55615 11.2575 7.45234 12.0225 7.45234C12.7875 7.45234 13.5525 7.55615 14.2725 7.76377C15.9937 6.56418 16.7475 6.81794 16.7475 6.81794C17.2424 8.08674 16.9275 9.03257 16.8375 9.26326C17.4113 9.9092 17.76 10.7281 17.76 11.7432C17.76 15.2843 15.6563 16.0687 13.6537 16.2993C13.98 16.5877 14.2613 17.1414 14.2613 18.0065C14.2613 19.2407 14.25 20.2326 14.25 20.5441C14.25 20.7863 14.4188 21.0746 14.8688 20.9824C16.6554 20.364 18.2079 19.1866 19.3078 17.6162C20.4077 16.0457 20.9995 14.1611 21 12.2276C21 7.12937 16.9725 3 12 3Z"></path></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://www.linkedin.com/in/jiyuan-liu-7608a42a6/"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Brands/Linkedin"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" height="1em" viewBox="0 0 448 512"><path fill="currentColor" d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://scholar.google.com/citations?view_op=list_works&amp;hl=en&amp;hl=en&amp;user=5c1kGO4AAAAJ&amp;sortby=pubdate"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Academicons/Google-Scholar"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M343.759 106.662V79.43L363.524 64h-213.89L20.476 176.274h85.656a82.339 82.339 0 0 0-.219 6.225c0 20.845 7.22 38.087 21.672 51.861c14.453 13.797 32.252 20.648 53.327 20.648c4.923 0 9.75-.368 14.438-1.024c-2.907 6.5-4.374 12.523-4.374 18.142c0 9.875 4.499 20.43 13.467 31.642c-39.234 2.67-68.061 9.732-86.437 21.163c-10.531 6.5-19 14.704-25.39 24.531c-6.391 9.9-9.578 20.515-9.578 31.962c0 9.648 2.062 18.336 6.219 26.062c4.156 7.726 9.578 14.07 16.312 18.984c6.718 4.968 14.469 9.101 23.219 12.469c8.734 3.344 17.406 5.718 26.061 7.062A167.052 167.052 0 0 0 180.555 448c13.469 0 26.953-1.734 40.547-5.187c13.562-3.485 26.28-8.642 38.171-15.493c11.86-6.805 21.515-16.086 28.922-27.718c7.39-11.68 11.094-24.805 11.094-39.336c0-11.016-2.25-21.039-6.75-30.14c-4.468-9.073-9.938-16.542-16.452-22.345c-6.501-5.813-13-11.155-19.516-15.968c-6.5-4.845-12-9.75-16.468-14.813c-4.485-5.046-6.735-10.054-6.735-14.984c0-4.921 1.734-9.672 5.216-14.265c3.455-4.61 7.674-9.048 12.61-13.306c4.937-4.25 9.875-8.968 14.796-14.133c4.922-5.147 9.141-11.827 12.61-20.008c3.485-8.18 5.203-17.445 5.203-27.757c0-13.453-2.547-24.46-7.547-33.314c-.594-1.022-1.218-1.803-1.875-3.022l56.907-46.672v17.119c-7.393.93-6.624 5.345-6.624 10.635V245.96c0 5.958 4.875 10.834 10.834 10.834h3.989c5.958 0 10.833-4.875 10.833-10.834V117.293c0-5.277.778-9.688-6.561-10.63zm-107.36 222.48c1.14.75 3.704 2.78 7.718 6.038c4.05 3.243 6.797 5.695 8.266 7.414a443.553 443.553 0 0 1 6.376 7.547c2.813 3.375 4.718 6.304 5.718 8.734c1 2.477 2.016 5.461 3.047 8.946a38.27 38.27 0 0 1 1.485 10.562c0 17.048-6.564 29.68-19.656 37.859c-13.125 8.18-28.767 12.274-46.938 12.274c-9.187 0-18.203-1.093-27.063-3.196c-8.843-2.116-17.311-5.336-25.39-9.601c-8.078-4.258-14.577-10.204-19.5-17.797c-4.938-7.64-7.407-16.415-7.407-26.25c0-10.32 2.797-19.29 8.422-26.906c5.594-7.625 12.938-13.391 22.032-17.315c9.063-3.946 18.25-6.742 27.562-8.398a157.865 157.865 0 0 1 28.438-2.555c4.47 0 7.936.25 10.405.696c.455.219 3.032 2.07 7.735 5.563c4.704 3.462 7.625 5.595 8.75 6.384zm-3.359-100.579c-7.406 8.86-17.734 13.288-30.953 13.288c-11.86 0-22.298-4.764-31.266-14.312c-9-9.523-15.422-20.328-19.344-32.43c-3.937-12.109-5.906-23.984-5.906-35.648c0-13.694 3.596-25.352 10.781-34.976c7.187-9.65 17.5-14.485 30.938-14.485c11.875 0 22.374 5.038 31.437 15.157c9.094 10.085 15.61 21.413 19.517 33.968c3.922 12.54 5.873 24.53 5.873 35.984c0 13.446-3.702 24.61-11.076 33.454z"/></svg></a>
  
</div>



    </div>
  </div>
</div>



  




  
  
    
    
    
      
      
    
<div class="pt-1 no-prose w-full">
  <hr class="border-dotted border-neutral-300 dark:border-neutral-600" />
  <div class="flex flex-col md:flex-row flex-nowrap justify-between gap-5 pt-2">
    <div class="">
      
        <a class="group flex no-underline" href="/post/02.10x_scrnaseq_herv_quantification/">
          <span
            class="mt-[-0.3rem] me-2 text-neutral-700 transition-transform group-hover:-translate-x-[2px] group-hover:text-primary-600 dark:text-neutral dark:group-hover:text-primary-400"
          ><span class="ltr:inline rtl:hidden">&larr;</span></span>
          <span class="flex flex-col">
            <span
              class="mt-[0.1rem] leading-6 group-hover:underline group-hover:decoration-primary-500"
            >🧬 quantify transgenes and herv from 10x single-cell RNAseq</span>
            <span class="mt-[0.1rem] text-xs text-neutral-500 dark:text-neutral-400">
              
                Mar 25, 2025
              
            </span>
          </span>
        </a>
      
    </div>
    <div class="">
      
        <a class="group flex text-right no-underline" href="/post/00.model_aerosol_depostion/">
          <span class="flex flex-col">
            <span
              class="mt-[0.1rem] leading-6 group-hover:underline group-hover:decoration-primary-500"
            >💨 Modeling the deposition of infectious agents suggests that aerosolized drugs offer enhanced specificity</span
            >
            <span class="mt-[0.1rem] text-xs text-neutral-500 dark:text-neutral-400">
              
                Mar 23, 2025
              
            </span>
          </span>
          <span
            class="mt-[-0.3rem] ms-2 text-neutral-700 transition-transform group-hover:-translate-x-[2px] group-hover:text-primary-600 dark:text-neutral dark:group-hover:text-primary-400"
          ><span class="ltr:inline">&rarr;</span></span>
        </a>
      
    </div>
  </div>
</div>



  


  



</div>

      </div>

    </main>
  </article>
</div>

    </div>
    <div class="page-footer">
      <footer class="container mx-auto flex flex-col justify-items-center text-sm leading-6 mt-24 mb-4 text-slate-700 dark:text-slate-200">

  












  
  
  
  
  














  
  
  

  
  
    
  
  
    
  

  

  
  <p class="powered-by text-center">
    © 2025 Me. This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-nd/4.0" rel="noopener noreferrer" target="_blank">CC BY NC ND 4.0</a>
  </p>
  

  <p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-nd/4.0" rel="noopener noreferrer" target="_blank" aria-label="Creative Commons">
      <i class="fab fa-creative-commons fa-2x" aria-hidden="true"></i>
      <i class="fab fa-creative-commons-by fa-2x" aria-hidden="true"></i>
      
        <i class="fab fa-creative-commons-nc fa-2x" aria-hidden="true"></i>
      
      
        <i class="fab fa-creative-commons-nd fa-2x" aria-hidden="true"></i>
      
    </a>
  </p>





  <p class="powered-by text-center">
    
    
    
      
      
      
      
      
      
      Published with <a href="https://hugoblox.com/?utm_campaign=poweredby" target="_blank" rel="noopener">Hugo Blox Builder</a> — the free, <a href="https://github.com/HugoBlox/hugo-blox-builder" target="_blank" rel="noopener">open source</a> website builder that empowers creators.
    
  </p>

</footer>

    </div>

    
    











  </body>
</html>
