<!doctype html>
<!-- This site was created with Hugo Blox. https://hugoblox.com -->
<!-- Last Published: July 7, 2025 --><html lang="en-us" dir="ltr"
      data-wc-theme-default="system">
  
<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <meta name="generator" content="Hugo Blox Builder 0.3.1" />

  
  












  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Jiyuan Liu" />

  
  
  
    
  
  <meta name="description" content="I developed SeqWins to provide memory-efficient, chunk-wise processing of FASTQ files and offer alignment performance that is competitive with or faster than many Linux-based aligners." />

  
  <link rel="alternate" hreflang="en-us" href="http://localhost:1313/post/00a.seqwins/" />

  
  
  
  
    
    <link rel="stylesheet" href="/css/themes/emerald.min.css" />
  

  
  
    
    <link href="/dist/wc.min.css" rel="stylesheet" />
  

  
  
  

  
    
    <link href="/css/custom.min.9a8c1508a999d285edca71446056519c3773544bdbe9fd11396354c75238a016.css" rel="stylesheet" />
  

  <script>
     
    window.hbb = {
       defaultTheme: document.documentElement.dataset.wcThemeDefault,
       setDarkTheme: () => {
        document.documentElement.classList.add("dark");
        document.documentElement.style.colorScheme = "dark";
      },
       setLightTheme: () => {
        document.documentElement.classList.remove("dark");
        document.documentElement.style.colorScheme = "light";
      }
    }

    console.debug(`Default Hugo Blox Builder theme is ${window.hbb.defaultTheme}`);

    if ("wc-color-theme" in localStorage) {
      localStorage.getItem("wc-color-theme") === "dark" ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
    } else {
      window.hbb.defaultTheme === "dark" ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
      if (window.hbb.defaultTheme === "system") {
        window.matchMedia("(prefers-color-scheme: dark)").matches ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
      }
    }
  </script>

  <script>
    
    document.addEventListener('DOMContentLoaded', function () {
      
      let checkboxes = document.querySelectorAll('li input[type=\'checkbox\'][disabled]');
      checkboxes.forEach(e => {
        e.parentElement.parentElement.classList.add('task-list');
      });

      
      const liNodes = document.querySelectorAll('.task-list li');
      liNodes.forEach(nodes => {
        let textNodes = Array.from(nodes.childNodes).filter(node => node.nodeType === 3 && node.textContent.trim().length > 1);
        if (textNodes.length > 0) {
          const span = document.createElement('label');
          textNodes[0].after(span);  
          span.appendChild(nodes.querySelector('input[type=\'checkbox\']'));
          span.appendChild(textNodes[0]);
        }
      });
    });
  </script>

  
  
  




































  
  

  
  <link rel="icon" type="image/png" href="/media/icon_hu_c54cc46c793e1f2b.png" />
  <link rel="apple-touch-icon" type="image/png" href="/media/icon_hu_68ea7b4ed3f6b794.png" />

  <link rel="canonical" href="http://localhost:1313/post/00a.seqwins/" />

  
  
  
  
  
  
  
  
    
  
  

  
  
    
    
  
  <meta property="twitter:card" content="summary_large_image" />
  
    <meta property="twitter:site" content="@GetResearchDev" />
    <meta property="twitter:creator" content="@GetResearchDev" />
  
  <meta property="og:site_name" content="Jiyuan Liu" />
  <meta property="og:url" content="http://localhost:1313/post/00a.seqwins/" />
  <meta property="og:title" content="‚ö° SeqWins, An R Package for Flexible Base Trimming and Comprehensive FASTQ Analysis on Windows | Jiyuan Liu" />
  <meta property="og:description" content="I developed SeqWins to provide memory-efficient, chunk-wise processing of FASTQ files and offer alignment performance that is competitive with or faster than many Linux-based aligners." /><meta property="og:image" content="http://localhost:1313/post/00a.seqwins/featured.jpg" />
    <meta property="twitter:image" content="http://localhost:1313/post/00a.seqwins/featured.jpg" /><meta property="og:locale" content="en-us" />
  
    
      <meta
        property="article:published_time"
        content="2025-03-23T00:00:00&#43;00:00"
      />
    
    <meta property="article:modified_time" content="2025-03-23T00:00:00&#43;00:00">
  

  



  


  <title>‚ö° SeqWins, An R Package for Flexible Base Trimming and Comprehensive FASTQ Analysis on Windows | Jiyuan Liu</title>

  
  
  
  
  
    
    
  
  
  <style>
    @font-face {
      font-family: 'Inter var';
      font-style: normal;
      font-weight: 100 900;
      font-display: swap;
      src: url(/dist/font/Inter.var.woff2) format(woff2);
    }
  </style>

  

  
  


  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
    
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  







<link type="text/css" rel="stylesheet" href="/dist/pagefind/pagefind-ui.be766eb419317a14ec769d216e9779bfe8f3737c80e780f4ba0dafb57a41a482.css" integrity="sha256-vnZutBkxehTsdp0hbpd5v&#43;jzc3yA54D0ug2vtXpBpII=" />


<script src="/dist/pagefind/pagefind-ui.87693d7c6f2b3b347ce359d0ede762c033419f0a32b22ce508c335a81d841f1b.js" integrity="sha256-h2k9fG8rOzR841nQ7ediwDNBnwoysizlCMM1qB2EHxs="></script>


<script>window.hbb.pagefind = {"baseUrl":"/"};</script>

<style>
  html.dark {
    --pagefind-ui-primary: #eeeeee;
    --pagefind-ui-text: #eeeeee;
    --pagefind-ui-background: #152028;
    --pagefind-ui-border: #152028;
    --pagefind-ui-tag: #152028;
  }
</style>

<script>
  window.addEventListener('DOMContentLoaded', (event) => {
    new PagefindUI({
      element: "#search",
      showSubResults: true,
      baseUrl: window.hbb.pagefind.baseUrl,
      bundlePath: window.hbb.pagefind.baseUrl + "pagefind/",
    });
  });
  document.addEventListener('DOMContentLoaded', () => {
    let element = document.getElementById('search');
    let trigger = document.getElementById('search_toggle');

    if (trigger) {
      trigger.addEventListener('click', () => {
        element.classList.toggle('hidden');
        element.querySelector("input").value = ""
        element.querySelector("input").focus()

        if (!element.classList.contains('hidden')) {
          let clear_trigger = document.querySelector('.pagefind-ui__search-clear');

          if (clear_trigger && !clear_trigger.hasAttribute('listenerOnClick')) {
            clear_trigger.setAttribute('listenerOnClick', 'true');

            clear_trigger.addEventListener('click', () => {
              element.classList.toggle('hidden');
            });
          }
        }

      });
    }
  });
</script>















  
  
  
  
  
  
  
  <script
    defer
    src="/js/hugo-blox-en.min.js"
    integrity=""
  ></script>

  
  








  
    
      
      <script async defer src="https://buttons.github.io/buttons.js"></script>

      
    
  




</head>

  <body class="dark:bg-hb-dark dark:text-white page-wrapper" id="top">
    <div id="page-bg"></div>
    <div class="page-header sticky top-0 z-30">
      
      
      
        
        
        
          <header id="site-header" class="header">
  <nav class="navbar px-3 flex justify-left">
    <div class="order-0 h-100">
      
      <a class="navbar-brand" href="/" title="Jiyuan Liu">
        Home
      </a>
    </div>
    
    <input id="nav-toggle" type="checkbox" class="hidden" />
    <label
      for="nav-toggle"
      class="order-3 cursor-pointer flex items-center lg:hidden text-dark dark:text-white lg:order-1">
      <svg id="show-button" class="h-6 fill-current block" viewBox="0 0 20 20">
        <title>Open Menu</title>
        <path d="M0 3h20v2H0V3z m0 6h20v2H0V9z m0 6h20v2H0V0z"></path>
      </svg>
      <svg id="hide-button" class="h-6 fill-current hidden" viewBox="0 0 20 20">
        <title>Close Menu</title>
        <polygon
          points="11 9 22 9 22 11 11 11 11 22 9 22 9 11 -2 11 -2 9 9 9 9 -2 11 -2"
          transform="rotate(45 10 10)"></polygon>
      </svg>
    </label>
    

    
    
    <ul
      id="nav-menu"
      class="navbar-nav order-3 hidden lg:flex w-full pb-6 lg:order-1 lg:w-auto lg:space-x-2 lg:pb-0 xl:space-x-8 justify-left
      ">
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/"
        >Bio</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/#papers"
        >Papers</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/#talks"
        >Talks</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/post"
        >Blogs</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/experience/"
        >Experience</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/projects/"
        >Projects</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/teaching/"
        >Learning</a
        >
      </li>
      
      
      
    </ul>

    <div class="order-1 ml-auto flex items-center md:order-2 lg:ml-0">

      
      
      
      <button
        aria-label="search"
        class="text-black hover:text-primary  inline-block px-3 text-xl dark:text-white"
        id="search_toggle">
        <svg xmlns="http://www.w3.org/2000/svg" height="16" width="16" viewBox="0 0 512 512" fill="currentColor"><path d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"/></svg>
      </button>
      

      
      
      <div class="px-3 text-black hover:text-primary-700 dark:text-white dark:hover:text-primary-300
            [&.active]:font-bold [&.active]:text-black/90 dark:[&.active]:text-white">
        <button class="theme-toggle mt-1" accesskey="t" title="appearance">
          <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
               fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
               stroke-linejoin="round" class="dark:hidden">
            <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
          </svg>
          <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
               fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
               stroke-linejoin="round" class=" dark:block [&:not(dark)]:hidden">
            <circle cx="12" cy="12" r="5"></circle>
            <line x1="12" y1="1" x2="12" y2="3"></line>
            <line x1="12" y1="21" x2="12" y2="23"></line>
            <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
            <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
            <line x1="1" y1="12" x2="3" y2="12"></line>
            <line x1="21" y1="12" x2="23" y2="12"></line>
            <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
            <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
          </svg>
        </button>
      </div>
      

      
      

      
      
    </div>
  </nav>
</header>


<div id="search" class="hidden p-3"></div>


        
      
    </div>
    <div class="page-body  my-10">
      





<div class="mx-auto flex max-w-screen-xl">
  



<aside class="hb-sidebar-container max-lg:[transform:translate3d(0,-100%,0)] lg:hidden xl:block">
  
  <div class="px-4 pt-4 lg:hidden">
    
    
  </div>
  <div class="hb-scrollbar lg:h-[calc(100vh-var(--navbar-height))]">
    <ul class="flex flex-col gap-1 lg:hidden">
      
      
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/event/"
    
  >Recent &amp; Upcoming Talks
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/event/example/"
    
  >Example Talk
    </a>
              
            </li></ul>
      </div></li>
        <li class="open"><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/"
    
  >Computational Blogs
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11d.velocity_dynamic_model_derivation/"
    
  >üß¨ Dynamic RNA velocity model-- (1) math solutions
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11e.velocity_dynamic_model_inference/"
    
  >üß¨ Dynamic RNA velocity model-- (2) parameter inference
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11f.velocity_dynamic_model_posthoc_calculation/scvelo_post-fitting_calculation/"
    
  >üß¨ Dynamic RNA velocity model-- (3) post hoc computation
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11c.velocity_steady_state/"
    
  >üß¨ Math derivation for steady-state RNA velocity model
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19c.bayes_variational_infer_elbo/"
    
  >üéì Math Derivation for ELBO/KL in Bayesian Inference and VAEs
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19b.scrnaseq_impute_saver/"
    
  >üßÆ Math Derivation for scRNAseq Imputation (SAVER)
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19a.conjugate_prior_bayes/"
    
  >üìä Math Derivation in Bayesian Inference-- Bypassing Marginals Using Conjugate Priors
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19.ememwithmapvariationalbayesian/"
    
  >üß¨ EM algorithm, EM with max a posterior, full Bayesian inferences (variational Bayesian) which might be used in auto-coder
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/18.what_make_neural_network_powerful_popular/"
    
  >üß¨ what makes neural network so powerful and popular
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/17.smooth.tests.of.goodness.of.fit/"
    
  >üß¨ Smooth tests of goodness of Fit used in Stijn Hawinkel ‚Äôs plos one paper
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/16.how.good.is.negative.binominal/"
    
  >üß¨ how good is negative binomial, why AI works for biology
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/15.use.seurat.for.bulk.rnaseq/"
    
  >üß¨ Seurat to analysis large sample size of bulk RNAseq. Discuss Wixcox is better for clinical samples
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/13z.scvi_integrate_scrna_spatial/"
    
  >üß¨ Deep Neural Network to integrate scRNAseq and spatial genomics
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/14.spatial_genomics_presentation/"
    
  >üß¨ Spatial genomics presentation
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/12.basian_abundance_statistics/"
    
  >üß¨ Bayesian abundance statistics in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/13.yolov8/"
    
  >üß¨ Yolov8
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11a.velocity_unraveled/"
    
  >üß¨ Math Derivation of CME-defined Stochastic Model of RNA Velocity
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11.velocity_reasoning/"
    
  >üß¨ Velocity reasoning and discussion
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/10.cca_reasoning/"
    
  >üß¨ CCA reasoning and integration discussion
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/09.sctransform/"
    
  >üß¨ sctransfrom reasoning
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/08.clinical_relevance/"
    
  >üß¨ Clinical relevance evaluation in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/07.chord_diagram/"
    
  >üß¨ Expressive cord diagram in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/06.noise_or_true_stem_cell/"
    
  >üß¨ Determine whether it‚Äôs noise or true stem like cells in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/05.xensoft_caki_paper/"
    
  >üß¨ quantify xenograft and host gene expression from the same sample
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/04.huangcong_oncogene/"
    
  >üß¨ counteract treatment-specific background in differential expressed gene analysis (huangcong paper)
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/03.well-based_scrnaseq_herv_quantification/"
    
  >üß¨ A snakemake pipline-- quantify hERV, transgenes and genome from well-based single-cell RNAseq
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/02.snakemake_quantify-herv_10xscrnaseq/"
    
  >üß¨ A snakemake pipline-- quantify hERV, transgenes and genome from 10x scRNAseq
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/01a.repeatmasker/"
    
  >üõ†Ô∏è Generate and validate ERV gtf that matches the latest reference genome
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/01.bulkrnaseq_allow_herv_quantification/"
    
  >üß¨ Quantify hERV and transgenes from Bulk-RNAseq
    </a>
              
            </li><li class="flex flex-col open"><a
    class="hb-sidebar-custom-link
      sidebar-active-item bg-primary-100 font-semibold text-primary-800 dark:bg-primary-300 dark:text-primary-900"
    href="/post/00a.seqwins/"
    
  >‚ö° SeqWins, An R Package for Flexible Base Trimming and Comprehensive FASTQ Analysis on Windows
    </a>
  
    <ul class="hb-sidebar-mobile-toc"><li>
              <a
                href="#github-source-code"
                class="hb-docs-link"
              >Github source code</a>
            </li>
          <li>
              <a
                href="#introduction"
                class="hb-docs-link"
              >Introduction</a>
            </li>
          <li>
              <a
                href="#installation"
                class="hb-docs-link"
              >Installation</a>
            </li>
          <li>
              <a
                href="#input-file-wrangling"
                class="hb-docs-link"
              >Input file wrangling</a>
            </li>
          <li>
              <a
                href="#modules"
                class="hb-docs-link"
              >Modules</a>
            </li>
          </ul>
  
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/00.model_aerosol_depostion/"
    
  >üí® Modeling the deposition of infectious agents suggests that aerosolized drugs offer enhanced specificity
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.get-started/"
    
  >üéâ Easily create your own simple yet highly customizable blog
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.second-brain/"
    
  >üß† Sharpen your thinking with a second brain
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.data-visualization/"
    
  >üìà Communicate your results effectively with the best data visualizations
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.teach-courses/"
    
  >üë©üèº‚Äçüè´ Teach academic courses
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.project-management/"
    
  >‚úÖ Manage your projects
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_derive_time-varying_alpha_model/"
    
  >velovi_derive_time-varying_alpha_model
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_elbo_kl_breakdown/"
    
  >velovi_elbo_kl_breakdown
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_elbo_term_with_expectation/"
    
  >velovi_elbo_term_with_expectation
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_expected_state_n_latent_time/"
    
  >velovi_expected_state_n_latent_time
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_model_n_inference/"
    
  >velovi_model_n_inference
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_parameter_initialize_n_learn/"
    
  >velovi_parameter_initialize_n_learn
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_posterior_predictive_distribution/"
    
  >velovi_posterior_predictive_distribution
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_reparameterization_trick/"
    
  >velovi_reparameterization_trick
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_uncertainty_calculation/"
    
  >velovi_uncertainty_calculation
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/"
    
  >learning
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/visualize_distribution/"
    
  >Dirichlet Distribution
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/stanford_ml_em/"
    
  >Lecture 14 - Expectation-Maximization Algorithms | Stanford CS229 Machine Learning
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/linear_algebra/"
    
  >linear algebra
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/other/home_improvement/"
    
  >linear algebra
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/method_of_charateristics/"
    
  >Method of characteristics
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/method_of_seperation_of_variables/"
    
  >Separation of Variables
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/js/"
    
  >Learn JavaScript
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/python/"
    
  >Learn Python
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/projects/"
    
  >Projects
    </a></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/"
    
  >Projects
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/pandas/"
    
  >Pandas
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/pytorch/"
    
  >PyTorch
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/scikit/"
    
  >scikit-learn
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/experience/"
    
  >Experience
    </a></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/"
    
  >Publications
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/preprint/"
    
  >An example preprint / working paper
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/journal-article/"
    
  >An example journal article
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/conference-paper/"
    
  >An example conference paper
    </a>
              
            </li></ul>
      </div></li>
    </ul>

    <div class="max-xl:hidden h-0 w-64 shrink-0"></div></div>

</aside>
  

<nav class="hb-toc order-last hidden w-64 shrink-0 xl:block print:hidden px-4" aria-label="table of contents">
  











  <div class="hb-scrollbar text-sm [hyphens:auto] sticky top-16 overflow-y-auto pr-4 pt-6 max-h-[calc(100vh-var(--navbar-height)-env(safe-area-inset-bottom))] -mr-4 rtl:-ml-4"><p class="mb-4 font-semibold tracking-tight">On this page</p><ul>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#github-source-code">Github source code</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#introduction">Introduction</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#installation">Installation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#input-file-wrangling">Input file wrangling</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#modules">Modules</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#a-qc-report">A. QC report</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#b-trim-basesfilter-readsalign-and-count-feature">B. Trim bases,filter reads,align and count feature</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#specify-file-path">specify file path</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#specify-trim-and-filter">specify trim and filter</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#summary-of-trimming-and-filtration">summary of trimming and filtration</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#set-aligntype">set alignType</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#set-featurecount">set featureCount</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#even-more-flexibility-is-faciliated">Even more flexibility is faciliated!</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#all-are-exported-for-your-own-pipeline">All are exported for your own pipeline</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#run-from-trimmed-fastqgz-or-aligned-bam">Run from trimmed fastq.gz or aligned bam</a>
      </li></ul>

  
  
    
    
  



    












  </div>
  </nav>


  <article class="w-full break-words flex min-h-[calc(100vh-var(--navbar-height))] min-w-0 justify-center pb-8 pr-[calc(env(safe-area-inset-right)-1.5rem)]">
    <main class="w-full min-w-0 max-w-6xl px-6 pt-4 md:px-12">

      

      <h1 class="mt-2 text-4xl font-bold tracking-tight text-slate-900 dark:text-slate-100">‚ö° SeqWins, An R Package for Flexible Base Trimming and Comprehensive FASTQ Analysis on Windows</h1>

      <div class="mt-4 mb-16">
      <div class="text-gray-500 dark:text-gray-300 text-sm flex items-center flex-wrap gap-y-2"><span class="mr-1">Mar 23, 2025</span><span class="mx-1">¬∑</span>
        
          
          
          <div class="group inline-flex items-center text-current gap-x-1.5 mx-1">
            
            
            <img src="/author/jiyuan-liu/avatar_hu_98ce379ab30758a2.webp" alt="Jiyuan Liu" class="inline-block h-4 w-4 rounded-full border border-current" loading="lazy" />
            
            <div >Jiyuan Liu</div>
          </div>
          
        

        
        <span class="mx-1">¬∑</span>
        <span class="mx-1">
          7 min read
        </span>
        
        </div>

        <div class="mt-3">
          





        </div>
      </div>



      
      
      
      
      

      
      
      
      
      
      
      
      
      
      
      <div class="article-header article-container featured-image-wrapper mt-4 mb-16" style="max-width: 720px; max-height: 720px;">
        <div style="position: relative">
          <img src="/post/00a.seqwins/featured_hu_315fd3607903e916.webp" width="720" height="720" alt="" class="featured-image">
          <span class="article-header-caption">Image credit: <a href="https://unsplash.com" target="_blank" rel="noopener"><strong>Illustration by Art Attack on unsplash</strong></a></span>
        </div>
      </div>
      

      
      

      <div class="prose prose-slate lg:prose-xl dark:prose-invert">
        <h2 id="github-source-code">Github source code</h2>
<p>Please see source codes and Vignette in <a href="https://github.com/jliu678/SeqWins" target="_blank" rel="noopener">github</a>.</p>
<h2 id="introduction">Introduction</h2>
<p>Words prevail that Fastq data cannot be elegantly processed in Windows. However, the fundamental low-level R package ShortRead and Rsubread have been available to provide memory-efficient, chunk-wise processing of FASTQ files and offer alignment performance that is competitive with or faster than many Linux-based aligners. Building on these strengths, I developed SeqWins (fastq <strong>Seq</strong>uence analysis on <strong>Win</strong>dows system), an R package for flexible base trimming and comprehensive FASTQ analysis on Windows. It achieves on <strong>pure Windows system</strong>:</p>
<ul>
<li>memory-efficient, chunk-wise processing of FASTQ files</li>
<li>alignment performance competitive with or faster than many Linux-based aligners</li>
<li>flexible base-level (ATCG) quality control and trimming</li>
<li>convenient high-level whole-process analysis of fastq data&ndash; spanning quality control report, trimming bases and reads accordingly, alignment and feature count</li>
</ul>
<h2 id="installation">Installation</h2>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="c1"># Install dependencies</span>
</span></span><span class="line"><span class="cl"><span class="kr">if</span> <span class="p">(</span><span class="o">!</span><span class="nf">requireNamespace</span><span class="p">(</span><span class="s">&#34;BiocManager&#34;</span><span class="p">,</span> <span class="n">quietly</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">	<span class="nf">install.packages</span><span class="p">(</span><span class="s">&#34;BiocManager&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&#34;ShortRead&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&#34;Rsubread&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Install and load SeqWins</span>
</span></span><span class="line"><span class="cl"><span class="n">devtools</span><span class="o">::</span><span class="nf">install_github</span><span class="p">(</span><span class="s">&#34;jliu678/SeqWins&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">SeqWins</span><span class="p">)</span>
</span></span></code></pre></div><h2 id="input-file-wrangling">Input file wrangling</h2>
<p>These gemonic data of the same species as with your fastq data are required to build index when aligning reads and get feature counts, and can be download from NCBI assembly website, for example <a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.26/" target="_blank" rel="noopener">the hg38 files</a> by clicking the top right &ldquo;<strong>Download Assembly</strong>&rdquo; button and selecting &ldquo;<strong>Genomic FASTA (.fna)</strong>&rdquo; and &ldquo;<strong>Genomic GTF (.gtf)</strong>&rdquo; sequentially on the resultant drop-down options.</p>
<p>If necessary untar the downloaded files as blow</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="nf">untar</span><span class="p">(</span><span class="s">&#34;full/path/genome_assemblies_genome_fasta.tar&#34;</span><span class="p">,</span><span class="n">exdir</span> <span class="o">=</span> <span class="s">&#34;.&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">untar</span><span class="p">(</span><span class="s">&#34;full/path/genome_assemblies_genome_gtf.tar&#34;</span><span class="p">,</span><span class="n">exdir</span> <span class="o">=</span> <span class="s">&#34;.&#34;</span><span class="p">)</span>
</span></span></code></pre></div><h2 id="modules">Modules</h2>
<h3 id="a-qc-report">A. QC report</h3>
<p>Set working directory, it will be where the QC result folder (ShortRead 1.46.0 name it as &ldquo;ShortRead Quality Assessment_files&rdquo;),index files and the folder named &ldquo;bam&rdquo; storing aligned data are located</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="nf">setwd</span><span class="p">(</span><span class="s">&#34;/your/working/directory&#34;</span><span class="p">)</span>
</span></span></code></pre></div><p>and get QC report</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="c1"># call ShortRead::qa to generate QC reports of all fastq.gz files</span>
</span></span><span class="line"><span class="cl"><span class="n">qa</span> <span class="o">&lt;-</span> <span class="nf">qa</span><span class="p">(</span><span class="s">&#34;full/path/FastqFolder&#34;</span><span class="p">,</span> <span class="s">&#34;fastq.gz&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">browseURL</span><span class="p">(</span><span class="nf">report</span><span class="p">(</span><span class="n">qa</span><span class="p">))</span>
</span></span></code></pre></div><p>the above will generate report of the fastq files retrieved by</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="nf">list.files</span><span class="p">(</span><span class="n">path</span> <span class="o">=</span> <span class="s">&#34;full/path/FastqFolder&#34;</span><span class="p">,</span><span class="n">pattern</span> <span class="o">=</span> <span class="s">&#34;fastq.gz&#34;</span><span class="p">)</span>
</span></span></code></pre></div><h3 id="b-trim-basesfilter-readsalign-and-count-feature">B. Trim bases,filter reads,align and count feature</h3>
<p>According to the QC report, customize the parameters used for trimming bases, filtering reads. Although most can be left default,please specify file paths totally decided by yourself; the <code>subReadThreads</code>, <code>shortreadRAM</code> decided by your computer; and the sequence tech type decided by your project.</p>
<h4 id="specify-file-path">specify file path</h4>
<p>If your fastq were generated from single-end sequencing (old-fashioned you are!), only specify <code>fileList1</code> as blow</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="n">fl</span><span class="o">&lt;-</span><span class="nf">list.files</span><span class="p">(</span><span class="n">path</span> <span class="o">=</span> <span class="s">&#34;full/path/FastqFolder&#34;</span><span class="p">,</span><span class="n">pattern</span> <span class="o">=</span> <span class="s">&#34;fastq.gz&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">seqW</span><span class="p">(</span><span class="n">fileList1</span><span class="o">=</span><span class="n">fl</span><span class="p">,</span><span class="n">genomeRefFile</span><span class="o">=</span><span class="s">&#34;./GCF_000001405.26_GRCh38_genomic.fna.gz&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                <span class="n">genomeAnnotFile</span><span class="o">=</span><span class="s">&#34;./GCF_000001405.39_GRCh38.p13_genomic.gtf.gz&#34;</span><span class="p">)</span><span class="c1">#RNAseq</span>
</span></span></code></pre></div><p>If paired-end,specify both <code>fileList1</code> i.e. the &ldquo;read 1&rdquo; files and <code>fileList2</code> i.e. &ldquo;read 2&rdquo; files</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="n">fl_1</span><span class="o">&lt;-</span><span class="nf">list.files</span><span class="p">(</span><span class="n">path</span> <span class="o">=</span> <span class="s">&#34;full/path/FastqFolder&#34;</span><span class="p">,</span><span class="n">pattern</span> <span class="o">=</span> <span class="s">&#34;^.*_1\\.fastq\\.gz$&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">fl_2</span><span class="o">&lt;-</span><span class="nf">list.files</span><span class="p">(</span><span class="n">path</span> <span class="o">=</span> <span class="s">&#34;full/path/FastqFolder&#34;</span><span class="p">,</span><span class="n">pattern</span> <span class="o">=</span> <span class="s">&#34;^.*_2\\.fastq\\.gz$&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nf">seqW</span><span class="p">(</span><span class="n">fileList1</span><span class="o">=</span><span class="n">fl_1</span><span class="p">,</span><span class="n">fileList1</span><span class="o">=</span><span class="n">fl_2</span><span class="p">,</span><span class="n">genomeRefFile</span><span class="o">=</span><span class="s">&#34;./GCF_000001405.26_GRCh38_genomic.fna.gz&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                <span class="n">genomeAnnotFile</span><span class="o">=</span><span class="s">&#34;./GCF_000001405.39_GRCh38.p13_genomic.gtf.gz&#34;</span><span class="p">)</span><span class="c1">#RNAseq</span>
</span></span></code></pre></div><p>Please don&rsquo;t forget to specify <code>genomeRefFile</code> and <code>genomeAnnotFile</code> i.e. the path of &ldquo;<strong>Genomic FASTA (.fna)</strong>&rdquo; and &ldquo;<strong>Genomic GTF (.gtf)</strong>&rdquo; as any of the above examples for the first time when you run <code>seqW</code>.</p>
<p>If you already have index files located in the working dir, you can speed it up by setting <code>indexBasename=&quot;my_index&quot;</code> to avoid regenerating index files like below. And this will make <code>seqW</code> function ignore whatever is assgned to <code>genomeRefFile</code>.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="nf">seqW</span><span class="p">(</span><span class="n">fileList1</span><span class="o">=</span><span class="n">fl_1</span><span class="p">,</span><span class="n">indexBasename</span><span class="o">=</span><span class="s">&#34;my_index&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                <span class="n">genomeAnnotFile</span><span class="o">=</span><span class="s">&#34;./GCF_000001405.39_GRCh38.p13_genomic.gtf.gz&#34;</span><span class="p">)</span><span class="c1">#RNAseq</span>
</span></span></code></pre></div><p><code>alignPairedOutput</code> only works for paired-end input,default is <code>gsub(basename(fileList1),pattern =&quot;_1.*\\.fastq\\.gz&quot;,replacement = &quot;\\.bam&quot;)</code>, optimize it please if it happens to cause overwriting of output files. For example for fastq files named as &ldquo;a_1_sample1.fastq.gz&rdquo;,  &ldquo;a_2_sample1.fastq.gz&rdquo;, &ldquo;a_1_sample2.fastq.gz&rdquo;, as &ldquo;a_2_sample2.fastq.gz&rdquo;, sample1 and sample2 files will both result in &ldquo;a.bam&rdquo; and either overwriting or error will occur.</p>
<h4 id="specify-trim-and-filter">specify trim and filter</h4>
<p>Below shows a complete list of trim and filter parameters. Probably most can be left as default except the adaptor sequence specific to your sequence platform</p>
<ol>
<li>
<p><code>endTrimThrs</code></p>
<p>: Phred score threshold of the end base below which the end base will be trimmed,default &ldquo;?&rdquo;</p>
</li>
<li>
<p><code>endTrimThrsend</code></p>
<p>: mean Phred score threshold of five bases in ends,below which the five bases will be trimmed,default &ldquo;4&rdquo;</p>
</li>
<li>
<p><code>adpter1Seq</code>,<code>adpter2Seq</code></p>
<p>: adapter sequence to be trimmed from end and also inner part of the read</p>
<ul>
<li>
<p>default <code>adpter1Seq</code> &ldquo;AGATCGGAAGAGCACACGTCTGAACTCCAGTCA&rdquo; i.e. illumina HT4000 adapter for read 1</p>
</li>
<li>
<p>default <code>adpter2Seq</code> &ldquo;AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT&rdquo; i.e. illumina HT4000 adapter for read 2</p>
</li>
</ul>
</li>
<li>
<p><code>with.Lindels</code>,<code>with.Rindels</code></p>
<p>: True if your reads contain indels on left end (5&rsquo;,corresponding to <code>with.Lindels</code>) or right end (3&rsquo;,corresponding to <code>with.Rindels</code>), default False</p>
</li>
<li>
<p><code>widthThrs</code></p>
<p>: width threshold of reads to be filtered out,default 14L</p>
</li>
<li>
<p><code>cmplxThrs</code></p>
<p>: complexity threshold of reads to be filtered out,default 0.5 i.e. half of mean complexity of human genome</p>
</li>
<li>
<p><code>innerNThrs</code></p>
<p>: number of N inside the read below which the read will be removed,default 2L</p>
<ul>
<li>
<p><code>innerS</code></p>
<p>: which base should be considered the start of inside part of the read after the above trimming,default 4L</p>
</li>
<li>
<p><code>innerE</code></p>
<p>: which base should be considered the end of inside part of the read after the above trimming. Negative integer X for width(read)-abs(X) default -4L; positive integer X for Xth base</p>
</li>
</ul>
</li>
</ol>
<h4 id="summary-of-trimming-and-filtration">summary of trimming and filtration</h4>
<p>A list containing dataframes of which each reports the result of trimming and filtration will be returned by the <code>seqw</code> function. You can assign the list to an object for further observation.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="n">reportList</span><span class="o">&lt;-</span><span class="nf">seqW</span><span class="p">(</span><span class="n">fileList1</span><span class="o">=</span><span class="n">fl_1</span><span class="p">,</span><span class="n">fileList1</span><span class="o">=</span><span class="n">fl_2</span><span class="p">,</span><span class="n">genomeRefFile</span><span class="o">=</span><span class="s">&#34;./GCF_000001405.26_GRCh38_genomic.fna.gz&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                <span class="n">genomeAnnotFile</span><span class="o">=</span><span class="s">&#34;./GCF_000001405.39_GRCh38.p13_genomic.gtf.gz&#34;</span><span class="p">)</span><span class="c1">#RNAseq</span>
</span></span></code></pre></div><h4 id="set-aligntype">set alignType</h4>
<p>Specifically for RNAseq leave <code>alignType</code> as default (simply don&rsquo;t mention it). By the way, unit of &ldquo;shortreadRAM&rdquo; is byte,don&rsquo;t panic.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="nf">seqW</span><span class="p">(</span><span class="n">fileList1</span><span class="o">=</span><span class="n">fl</span><span class="p">,</span><span class="n">subReadThreads</span><span class="o">=</span><span class="m">3L</span><span class="p">,</span><span class="n">shortreadRAM</span><span class="o">=</span><span class="m">1e8</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                <span class="n">genomeRefFile</span><span class="o">=</span><span class="s">&#34;./GCF_000001405.26_GRCh38_genomic.fna.gz&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                <span class="n">genomeAnnotFile</span><span class="o">=</span><span class="s">&#34;./GCF_000001405.39_GRCh38.p13_genomic.gtf.gz&#34;</span><span class="p">)</span>
</span></span></code></pre></div><p>For DNAseq, longDNAseq, RNAseq aim at exon-exon junction and microRNAseq, set <code>alignType</code> as &lsquo;<code>dna</code>&rsquo;,&rsquo;<code>dnaLong</code>&rsquo;,&rsquo;<code>rnaExon_Exon</code>&lsquo;and &lsquo;<code>microRNA</code>&rsquo; respectively&ndash;</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="nf">seqW</span><span class="p">(</span><span class="n">fileList1</span><span class="o">=</span><span class="n">fl</span><span class="p">,</span><span class="n">alignType</span> <span class="o">=</span><span class="s">&#39;dna&#39;</span><span class="p">)</span><span class="c1">#DNAseq</span>
</span></span><span class="line"><span class="cl"><span class="nf">seqW</span><span class="p">(</span><span class="n">fileList1</span><span class="o">=</span><span class="n">fl</span><span class="p">,</span><span class="n">alignType</span> <span class="o">=</span><span class="s">&#39;dnaLong&#39;</span><span class="p">)</span><span class="c1">#longDNAseq</span>
</span></span><span class="line"><span class="cl"><span class="nf">seqW</span><span class="p">(</span><span class="n">fileList1</span><span class="o">=</span><span class="n">fl</span><span class="p">,</span><span class="n">alignType</span> <span class="o">=</span><span class="s">&#39;rnaExon_Exon&#39;</span><span class="p">)</span><span class="c1">#RNAseq aim at exon-exon junction</span>
</span></span><span class="line"><span class="cl"><span class="nf">seqW</span><span class="p">(</span><span class="n">fileList1</span><span class="o">=</span><span class="n">fl</span><span class="p">,</span><span class="n">alignType</span> <span class="o">=</span><span class="s">&#39;microRNA&#39;</span><span class="p">)</span><span class="c1">#microRNAseq</span>
</span></span></code></pre></div><h4 id="set-featurecount">set featureCount</h4>
<p>If only gene counts are needed to be output, leave <code>useMetaFeatures</code> as default (simply don&rsquo;t mention it). Please note each row in the input &ldquo;<strong>Genomic GTF (.gtf)</strong>&rdquo; is a feature, and many features can belong to one gene while some features have no corresponding gene ID. You can see all fields that define a feature <a href="https://mblab.wustl.edu/GTF22.html" target="_blank" rel="noopener">here</a>. Thus, if feature counts containing information of all the fields rather than gene counts are needed, specify <code>useMetaFeatures=F</code> as below, so that the output will give information like the number of reads mapped to &ldquo;+&rdquo; strand of chromosome &ldquo;II&rdquo; from &ldquo;380&rdquo; to &ldquo;401&rdquo;.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="nf">seqW</span><span class="p">(</span><span class="n">fileList1</span><span class="o">=</span><span class="n">fl</span><span class="p">,</span><span class="n">subReadThreads</span><span class="o">=</span><span class="m">3L</span><span class="p">,</span><span class="n">shortreadRAM</span><span class="o">=</span><span class="m">1e8</span><span class="p">,</span><span class="n">useMetaFeatures</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                <span class="n">genomeRefFile</span><span class="o">=</span><span class="s">&#34;./GCF_000001405.26_GRCh38_genomic.fna.gz&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                <span class="n">genomeAnnotFile</span><span class="o">=</span><span class="s">&#34;./GCF_000001405.39_GRCh38.p13_genomic.gtf.gz&#34;</span><span class="p">)</span>
</span></span></code></pre></div><h3 id="even-more-flexibility-is-faciliated">Even more flexibility is faciliated!</h3>
<p>Hope the package can realize easy but deep appreciation of your fastq data, and you probably have found every parameters you wanna touch through above introduction. But there is even more! <code>...</code> arguments enable <code>seqW</code> function to access all the arguments in the classic functions wrapped inside it, including</p>
<ol>
<li><code>Rsubread::align</code>, called when <code>alignType=&quot;dna&quot;</code> ,<code>&quot;rna&quot;</code> ,<code>&quot;microRNA&quot;</code></li>
<li><code>Rsubread::sublong</code>, called when <code>alignType=&quot;longDNA&quot;</code></li>
<li><code>Rsubread::subjunc</code>, called when <code>alignType=&quot;rnaExon_Exon&quot;</code></li>
<li><code>Rsubread::featureCounts</code>, called every time <code>seqW</code> runs
(
thus simply add their parameters (see what they have by <code>?Rsubread::featureCounts</code> or <a href="https://bioconductor.org/packages/release/bioc/vignettes/Rsubread/inst/doc/SubreadUsersGuide.pdf" target="_blank" rel="noopener">Rsubread vignettes</a>) when calling <code>seqW</code>, e.g.</li>
</ol>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="c1"># minFragLength by default is 50L in Rsubread, but you can modify it easily as blow</span>
</span></span><span class="line"><span class="cl"><span class="nf">seqW</span><span class="p">(</span><span class="n">minFragLength</span><span class="o">=</span><span class="m">40L</span><span class="p">,</span><span class="n">fileList1</span><span class="o">=</span><span class="n">fl</span><span class="p">,</span><span class="n">genomeRefFile</span><span class="o">=</span><span class="s">&#34;./GCF_000001405.26_GRCh38_genomic.fna.gz&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                <span class="n">genomeAnnotFile</span><span class="o">=</span><span class="s">&#34;./GCF_000001405.39_GRCh38.p13_genomic.gtf.gz&#34;</span><span class="p">)</span><span class="c1">#RNAseq</span>
</span></span></code></pre></div><h3 id="all-are-exported-for-your-own-pipeline">All are exported for your own pipeline</h3>
<p>You might even wanna add/alter the trimming and filtration pipeline; No problem! I have exported every units wrapped under the top-level function, including</p>
<ol>
<li><code>trimEnds</code></li>
<li><code>trimTailw</code></li>
<li><code>trimAdapter</code></li>
<li><code>filterWidth</code></li>
<li><code>filterLowQuality</code></li>
<li><code>ShortRead::nFilter</code></li>
<li><code>filterLowComplexity</code></li>
<li><code>filterInnerN</code></li>
</ol>
<p>simply <code>?trimEnds</code> for example, to see their function and how to use it as bricks to make your own trimming and filter flow. And then use aforementioned <code>Rsubread</code> functions to complete alignment and feature count.</p>
<h3 id="run-from-trimmed-fastqgz-or-aligned-bam">Run from trimmed fastq.gz or aligned bam</h3>
<p><code>seqW</code> will sequentially generate</p>
<ol>
<li>index files in working dir, whose names by default start with &ldquo;my_index&rdquo;;</li>
<li>trimmed &ldquo;.fastq.gz&rdquo; files whose names contain &ldquo;trimed&rdquo; in the same folder as for you initially input &ldquo;.fastq.gz&rdquo; files;</li>
<li>aligned &ldquo;.bam&rdquo; files in the folder named &ldquo;bam&rdquo; located in working dir;</li>
<li>feature counts in &ldquo;.rds&rdquo; format in working dir.</li>
</ol>
<p>The above steps are carried out by <code>Rsubread::buildindex</code>, <code>TrimAndFilter</code>, <code>Rsubread::align</code> or its variant and <code>Rsubread::featureCounts</code>, respectively. We have earlier discussed how to use <code>seqW</code> funciton avoid repeating generation of index files when there are already index files. Besides, if satisfactory trimmed fastq exist, simply sequentially call necessary functions listed above to achieve your goal, e.g. building index by <code>Rsubread::buildindex</code> then aligning and count features by  <code>Rsubread::align</code> and <code>Rsubread::featureCounts</code>. If you even have aligned bam, just run <code>Rsubread::buildindex</code> and <code>Rsubread::featureCounts</code>. If you have got feature counts at hand, well, thank you a lot for using or visiting this package.</p>

      </div>

      
  <time class="mt-12 mb-8 block text-xs text-gray-500 ltr:text-right rtl:text-left dark:text-gray-400" datetime="2025-03-23T00:00:00.000Z">
    <span>Last updated on</span>
    Mar 23, 2025</time>

      <div class="container mx-auto prose prose-slate lg:prose-xl dark:prose-invert mt-5">
        
        <div class="max-w-prose print:hidden">
  
  

  

<div class="flex justify-center">
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/seqwins/">SeqWins</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/fastq/">Fastq</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/windows/">Windows</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/base-trimming/">Base Trimming</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/read-filtration/">Read Filtration</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/rnaseq/">RNAseq</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/quality-control/">Quality Control</a>
  
</div>


  
<section class="flex flex-row flex-wrap justify-center pt-4 text-xl">
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F00a.seqwins%2F&amp;text=%E2%9A%A1&#43;SeqWins%2C&#43;An&#43;R&#43;Package&#43;for&#43;Flexible&#43;Base&#43;Trimming&#43;and&#43;Comprehensive&#43;FASTQ&#43;Analysis&#43;on&#43;Windows"
    title="X"
    aria-label="X"
    id="share-link-x"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://www.facebook.com/sharer.php?u=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F00a.seqwins%2F&amp;t=%E2%9A%A1&#43;SeqWins%2C&#43;An&#43;R&#43;Package&#43;for&#43;Flexible&#43;Base&#43;Trimming&#43;and&#43;Comprehensive&#43;FASTQ&#43;Analysis&#43;on&#43;Windows"
    title="Facebook"
    aria-label="Facebook"
    id="share-link-facebook"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="currentColor" d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12c0 4.84 3.44 8.87 8 9.8V15H8v-3h2V9.5C10 7.57 11.57 6 13.5 6H16v3h-2c-.55 0-1 .45-1 1v2h3v3h-3v6.95c5.05-.5 9-4.76 9-9.95z"/></svg>
  </a>
  

  
  
  
  
  
  
    
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="mailto:?subject=%E2%9A%A1%20SeqWins%2C%20An%20R%20Package%20for%20Flexible%20Base%20Trimming%20and%20Comprehensive%20FASTQ%20Analysis%20on%20Windows&amp;body=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F00a.seqwins%2F"
    title="Email"
    aria-label="Email"
    id="share-link-email"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-width="1.5" d="M16.5 12a4.5 4.5 0 1 1-9 0a4.5 4.5 0 0 1 9 0Zm0 0c0 1.657 1.007 3 2.25 3S21 13.657 21 12a9 9 0 1 0-2.636 6.364M16.5 12V8.25"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://www.linkedin.com/shareArticle?url=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F00a.seqwins%2F&amp;title=%E2%9A%A1&#43;SeqWins%2C&#43;An&#43;R&#43;Package&#43;for&#43;Flexible&#43;Base&#43;Trimming&#43;and&#43;Comprehensive&#43;FASTQ&#43;Analysis&#43;on&#43;Windows"
    title="LinkedIn"
    aria-label="LinkedIn"
    id="share-link-linkedin"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" height="1em" viewBox="0 0 448 512"><path fill="currentColor" d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="whatsapp://send?text=%E2%9A%A1&#43;SeqWins%2C&#43;An&#43;R&#43;Package&#43;for&#43;Flexible&#43;Base&#43;Trimming&#43;and&#43;Comprehensive&#43;FASTQ&#43;Analysis&#43;on&#43;Windows%20http%3A%2F%2Flocalhost%3A1313%2Fpost%2F00a.seqwins%2F"
    title="WhatsApp"
    aria-label="WhatsApp"
    id="share-link-whatsapp"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256" fill="currentColor"><path d="m187.58 144.84-32-16a8 8 0 0 0-8 .5l-14.69 9.8a40.55 40.55 0 0 1-16-16l9.8-14.69a8 8 0 0 0 .5-8l-16-32A8 8 0 0 0 104 64a40 40 0 0 0-40 40 88.1 88.1 0 0 0 88 88 40 40 0 0 0 40-40 8 8 0 0 0-4.42-7.16ZM152 176a72.08 72.08 0 0 1-72-72 24 24 0 0 1 19.29-23.54l11.48 23L101 118a8 8 0 0 0-.73 7.51 56.47 56.47 0 0 0 30.15 30.15A8 8 0 0 0 138 155l14.61-9.74 23 11.48A24 24 0 0 1 152 176ZM128 24a104 104 0 0 0-91.82 152.88l-11.35 34.05a16 16 0 0 0 20.24 20.24l34.05-11.35A104 104 0 1 0 128 24Zm0 192a87.87 87.87 0 0 1-44.06-11.81 8 8 0 0 0-6.54-.67L40 216l12.47-37.4a8 8 0 0 0-.66-6.54A88 88 0 1 1 128 216Z"/></svg>
  </a>
  
</section>


  








  
  
    



  
  
  
    
  
  
  

<div class="flex pt-12 pb-4">
  
  
  <img
    class="mr-4 h-24 w-24 rounded-full"
    width="96"
    height="96"
    alt="Jiyuan Liu"
  src="/author/jiyuan-liu/avatar_hu_6297b4a1a274518.jpg"
  loading="lazy"
  />
  
  <div class="place-self-center">
    <div class="text-[0.6rem] uppercase leading-3 text-neutral-500 dark:text-neutral-400">
      Authors
    </div>
    <div class="font-semibold leading-6 text-neutral-800 dark:text-neutral-300">
      <a href="http://localhost:1313/" class="no-underline">
      Jiyuan Liu
      </a>
    </div>

    
    <div class="text-sm font-bold text-neutral-700 dark:text-neutral-300">
    Postdoctoral Scientist
    </div>
    


    

    <div class="text-2xl sm:text-lg pt-1">

      
<div class="flex flex-wrap text-neutral-500 dark:text-neutral-300">
  
    
    
    
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="mailto:jliu78@mgh.harvard.edu"
      
      aria-label="At-Symbol"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-width="1.5" d="M16.5 12a4.5 4.5 0 1 1-9 0a4.5 4.5 0 0 1 9 0Zm0 0c0 1.657 1.007 3 2.25 3S21 13.657 21 12a9 9 0 1 0-2.636 6.364M16.5 12V8.25"/></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://github.com/stars/jliu678/lists/owned-contributed-by-me"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Brands/Github"
    ><svg style="height: 1em;" fill="currentColor" viewBox="3 3 18 18"><path d="M12 3C7.0275 3 3 7.12937 3 12.2276C3 16.3109 5.57625 19.7597 9.15374 20.9824C9.60374 21.0631 9.77249 20.7863 9.77249 20.5441C9.77249 20.3249 9.76125 19.5982 9.76125 18.8254C7.5 19.2522 6.915 18.2602 6.735 17.7412C6.63375 17.4759 6.19499 16.6569 5.8125 16.4378C5.4975 16.2647 5.0475 15.838 5.80124 15.8264C6.51 15.8149 7.01625 16.4954 7.18499 16.7723C7.99499 18.1679 9.28875 17.7758 9.80625 17.5335C9.885 16.9337 10.1212 16.53 10.38 16.2993C8.3775 16.0687 6.285 15.2728 6.285 11.7432C6.285 10.7397 6.63375 9.9092 7.20749 9.26326C7.1175 9.03257 6.8025 8.08674 7.2975 6.81794C7.2975 6.81794 8.05125 6.57571 9.77249 7.76377C10.4925 7.55615 11.2575 7.45234 12.0225 7.45234C12.7875 7.45234 13.5525 7.55615 14.2725 7.76377C15.9937 6.56418 16.7475 6.81794 16.7475 6.81794C17.2424 8.08674 16.9275 9.03257 16.8375 9.26326C17.4113 9.9092 17.76 10.7281 17.76 11.7432C17.76 15.2843 15.6563 16.0687 13.6537 16.2993C13.98 16.5877 14.2613 17.1414 14.2613 18.0065C14.2613 19.2407 14.25 20.2326 14.25 20.5441C14.25 20.7863 14.4188 21.0746 14.8688 20.9824C16.6554 20.364 18.2079 19.1866 19.3078 17.6162C20.4077 16.0457 20.9995 14.1611 21 12.2276C21 7.12937 16.9725 3 12 3Z"></path></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://www.linkedin.com/in/jiyuan-liu-7608a42a6/"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Brands/Linkedin"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" height="1em" viewBox="0 0 448 512"><path fill="currentColor" d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://scholar.google.com/citations?view_op=list_works&amp;hl=en&amp;hl=en&amp;user=5c1kGO4AAAAJ&amp;sortby=pubdate"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Academicons/Google-Scholar"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M343.759 106.662V79.43L363.524 64h-213.89L20.476 176.274h85.656a82.339 82.339 0 0 0-.219 6.225c0 20.845 7.22 38.087 21.672 51.861c14.453 13.797 32.252 20.648 53.327 20.648c4.923 0 9.75-.368 14.438-1.024c-2.907 6.5-4.374 12.523-4.374 18.142c0 9.875 4.499 20.43 13.467 31.642c-39.234 2.67-68.061 9.732-86.437 21.163c-10.531 6.5-19 14.704-25.39 24.531c-6.391 9.9-9.578 20.515-9.578 31.962c0 9.648 2.062 18.336 6.219 26.062c4.156 7.726 9.578 14.07 16.312 18.984c6.718 4.968 14.469 9.101 23.219 12.469c8.734 3.344 17.406 5.718 26.061 7.062A167.052 167.052 0 0 0 180.555 448c13.469 0 26.953-1.734 40.547-5.187c13.562-3.485 26.28-8.642 38.171-15.493c11.86-6.805 21.515-16.086 28.922-27.718c7.39-11.68 11.094-24.805 11.094-39.336c0-11.016-2.25-21.039-6.75-30.14c-4.468-9.073-9.938-16.542-16.452-22.345c-6.501-5.813-13-11.155-19.516-15.968c-6.5-4.845-12-9.75-16.468-14.813c-4.485-5.046-6.735-10.054-6.735-14.984c0-4.921 1.734-9.672 5.216-14.265c3.455-4.61 7.674-9.048 12.61-13.306c4.937-4.25 9.875-8.968 14.796-14.133c4.922-5.147 9.141-11.827 12.61-20.008c3.485-8.18 5.203-17.445 5.203-27.757c0-13.453-2.547-24.46-7.547-33.314c-.594-1.022-1.218-1.803-1.875-3.022l56.907-46.672v17.119c-7.393.93-6.624 5.345-6.624 10.635V245.96c0 5.958 4.875 10.834 10.834 10.834h3.989c5.958 0 10.833-4.875 10.833-10.834V117.293c0-5.277.778-9.688-6.561-10.63zm-107.36 222.48c1.14.75 3.704 2.78 7.718 6.038c4.05 3.243 6.797 5.695 8.266 7.414a443.553 443.553 0 0 1 6.376 7.547c2.813 3.375 4.718 6.304 5.718 8.734c1 2.477 2.016 5.461 3.047 8.946a38.27 38.27 0 0 1 1.485 10.562c0 17.048-6.564 29.68-19.656 37.859c-13.125 8.18-28.767 12.274-46.938 12.274c-9.187 0-18.203-1.093-27.063-3.196c-8.843-2.116-17.311-5.336-25.39-9.601c-8.078-4.258-14.577-10.204-19.5-17.797c-4.938-7.64-7.407-16.415-7.407-26.25c0-10.32 2.797-19.29 8.422-26.906c5.594-7.625 12.938-13.391 22.032-17.315c9.063-3.946 18.25-6.742 27.562-8.398a157.865 157.865 0 0 1 28.438-2.555c4.47 0 7.936.25 10.405.696c.455.219 3.032 2.07 7.735 5.563c4.704 3.462 7.625 5.595 8.75 6.384zm-3.359-100.579c-7.406 8.86-17.734 13.288-30.953 13.288c-11.86 0-22.298-4.764-31.266-14.312c-9-9.523-15.422-20.328-19.344-32.43c-3.937-12.109-5.906-23.984-5.906-35.648c0-13.694 3.596-25.352 10.781-34.976c7.187-9.65 17.5-14.485 30.938-14.485c11.875 0 22.374 5.038 31.437 15.157c9.094 10.085 15.61 21.413 19.517 33.968c3.922 12.54 5.873 24.53 5.873 35.984c0 13.446-3.702 24.61-11.076 33.454z"/></svg></a>
  
</div>



    </div>
  </div>
</div>



  




  
  
    
    
    
      
      
    
<div class="pt-1 no-prose w-full">
  <hr class="border-dotted border-neutral-300 dark:border-neutral-600" />
  <div class="flex flex-col md:flex-row flex-nowrap justify-between gap-5 pt-2">
    <div class="">
      
        <a class="group flex no-underline" href="/post/01.bulkrnaseq_allow_herv_quantification/">
          <span
            class="mt-[-0.3rem] me-2 text-neutral-700 transition-transform group-hover:-translate-x-[2px] group-hover:text-primary-600 dark:text-neutral dark:group-hover:text-primary-400"
          ><span class="ltr:inline rtl:hidden">&larr;</span></span>
          <span class="flex flex-col">
            <span
              class="mt-[0.1rem] leading-6 group-hover:underline group-hover:decoration-primary-500"
            >üß¨ Quantify hERV and transgenes from Bulk-RNAseq</span>
            <span class="mt-[0.1rem] text-xs text-neutral-500 dark:text-neutral-400">
              
                Mar 24, 2025
              
            </span>
          </span>
        </a>
      
    </div>
    <div class="">
      
        <a class="group flex text-right no-underline" href="/post/00.model_aerosol_depostion/">
          <span class="flex flex-col">
            <span
              class="mt-[0.1rem] leading-6 group-hover:underline group-hover:decoration-primary-500"
            >üí® Modeling the deposition of infectious agents suggests that aerosolized drugs offer enhanced specificity</span
            >
            <span class="mt-[0.1rem] text-xs text-neutral-500 dark:text-neutral-400">
              
                Mar 23, 2025
              
            </span>
          </span>
          <span
            class="mt-[-0.3rem] ms-2 text-neutral-700 transition-transform group-hover:-translate-x-[2px] group-hover:text-primary-600 dark:text-neutral dark:group-hover:text-primary-400"
          ><span class="ltr:inline">&rarr;</span></span>
        </a>
      
    </div>
  </div>
</div>



  


  



</div>

      </div>

    </main>
  </article>
</div>

    </div>
    <div class="page-footer">
      <footer class="container mx-auto flex flex-col justify-items-center text-sm leading-6 mt-24 mb-4 text-slate-700 dark:text-slate-200">

  












  
  
  
  
  














  
  
  

  
  
    
  
  
    
  

  

  
  <p class="powered-by text-center">
    ¬© 2025 Me. This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-nd/4.0" rel="noopener noreferrer" target="_blank">CC BY NC ND 4.0</a>
  </p>
  

  <p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-nd/4.0" rel="noopener noreferrer" target="_blank" aria-label="Creative Commons">
      <i class="fab fa-creative-commons fa-2x" aria-hidden="true"></i>
      <i class="fab fa-creative-commons-by fa-2x" aria-hidden="true"></i>
      
        <i class="fab fa-creative-commons-nc fa-2x" aria-hidden="true"></i>
      
      
        <i class="fab fa-creative-commons-nd fa-2x" aria-hidden="true"></i>
      
    </a>
  </p>





  <p class="powered-by text-center">
    
    
    
      
      
      
      
      
      
      Published with <a href="https://hugoblox.com/?utm_campaign=poweredby" target="_blank" rel="noopener">Hugo Blox Builder</a> ‚Äî the free, <a href="https://github.com/HugoBlox/hugo-blox-builder" target="_blank" rel="noopener">open source</a> website builder that empowers creators.
    
  </p>

</footer>

    </div>

    
    











  </body>
</html>
