<!doctype html>
<!-- This site was created with Hugo Blox. https://hugoblox.com -->
<!-- Last Published: May 26, 2025 --><html lang="en-us" dir="ltr"
      data-wc-theme-default="system">
  
<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <meta name="generator" content="Hugo Blox Builder 0.3.1" />

  
  












  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Jiyuan Liu" />

  
  
  
    
  
  <meta name="description" content="ERV gtf annotation is needed to quantify ERV from sequencing data. But the ERV gtf compatible with the latest mouse genome GRCm39 is not publicly available. Here I generated and validated the ERV gtf based on GRCm39 using RepeatMasker and customized scripts." />

  
  <link rel="alternate" hreflang="en-us" href="http://localhost:1313/post/01a.repeatmasker/" />

  
  
  
  
    
    <link rel="stylesheet" href="/css/themes/emerald.min.css" />
  

  
  
    
    <link href="/dist/wc.min.css" rel="stylesheet" />
  

  
  
  

  

  <script>
     
    window.hbb = {
       defaultTheme: document.documentElement.dataset.wcThemeDefault,
       setDarkTheme: () => {
        document.documentElement.classList.add("dark");
        document.documentElement.style.colorScheme = "dark";
      },
       setLightTheme: () => {
        document.documentElement.classList.remove("dark");
        document.documentElement.style.colorScheme = "light";
      }
    }

    console.debug(`Default Hugo Blox Builder theme is ${window.hbb.defaultTheme}`);

    if ("wc-color-theme" in localStorage) {
      localStorage.getItem("wc-color-theme") === "dark" ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
    } else {
      window.hbb.defaultTheme === "dark" ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
      if (window.hbb.defaultTheme === "system") {
        window.matchMedia("(prefers-color-scheme: dark)").matches ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
      }
    }
  </script>

  <script>
    
    document.addEventListener('DOMContentLoaded', function () {
      
      let checkboxes = document.querySelectorAll('li input[type=\'checkbox\'][disabled]');
      checkboxes.forEach(e => {
        e.parentElement.parentElement.classList.add('task-list');
      });

      
      const liNodes = document.querySelectorAll('.task-list li');
      liNodes.forEach(nodes => {
        let textNodes = Array.from(nodes.childNodes).filter(node => node.nodeType === 3 && node.textContent.trim().length > 1);
        if (textNodes.length > 0) {
          const span = document.createElement('label');
          textNodes[0].after(span);  
          span.appendChild(nodes.querySelector('input[type=\'checkbox\']'));
          span.appendChild(textNodes[0]);
        }
      });
    });
  </script>

  
  
  




































  
  

  
  <link rel="icon" type="image/png" href="/media/icon_hu_c54cc46c793e1f2b.png" />
  <link rel="apple-touch-icon" type="image/png" href="/media/icon_hu_68ea7b4ed3f6b794.png" />

  <link rel="canonical" href="http://localhost:1313/post/01a.repeatmasker/" />

  
  
  
  
  
  
  
  
    
  
  

  
  
    
    
  
  <meta property="twitter:card" content="summary_large_image" />
  
    <meta property="twitter:site" content="@GetResearchDev" />
    <meta property="twitter:creator" content="@GetResearchDev" />
  
  <meta property="og:site_name" content="Jiyuan Liu" />
  <meta property="og:url" content="http://localhost:1313/post/01a.repeatmasker/" />
  <meta property="og:title" content="üõ†Ô∏è Generate and validate ERV gtf that matches the latest reference genome | Jiyuan Liu" />
  <meta property="og:description" content="ERV gtf annotation is needed to quantify ERV from sequencing data. But the ERV gtf compatible with the latest mouse genome GRCm39 is not publicly available. Here I generated and validated the ERV gtf based on GRCm39 using RepeatMasker and customized scripts." /><meta property="og:image" content="http://localhost:1313/post/01a.repeatmasker/featured.jpg" />
    <meta property="twitter:image" content="http://localhost:1313/post/01a.repeatmasker/featured.jpg" /><meta property="og:locale" content="en-us" />
  
    
      <meta
        property="article:published_time"
        content="2025-03-24T00:00:00&#43;00:00"
      />
    
    <meta property="article:modified_time" content="2025-03-24T00:00:00&#43;00:00">
  

  



  


  <title>üõ†Ô∏è Generate and validate ERV gtf that matches the latest reference genome | Jiyuan Liu</title>

  
  
  
  
  
    
    
  
  
  <style>
    @font-face {
      font-family: 'Inter var';
      font-style: normal;
      font-weight: 100 900;
      font-display: swap;
      src: url(/dist/font/Inter.var.woff2) format(woff2);
    }
  </style>

  

  
  


  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
    
  
  
  
  
  
  
  
  
  
  
  
  







<link type="text/css" rel="stylesheet" href="/dist/pagefind/pagefind-ui.be766eb419317a14ec769d216e9779bfe8f3737c80e780f4ba0dafb57a41a482.css" integrity="sha256-vnZutBkxehTsdp0hbpd5v&#43;jzc3yA54D0ug2vtXpBpII=" />


<script src="/dist/pagefind/pagefind-ui.87693d7c6f2b3b347ce359d0ede762c033419f0a32b22ce508c335a81d841f1b.js" integrity="sha256-h2k9fG8rOzR841nQ7ediwDNBnwoysizlCMM1qB2EHxs="></script>


<script>window.hbb.pagefind = {"baseUrl":"/"};</script>

<style>
  html.dark {
    --pagefind-ui-primary: #eeeeee;
    --pagefind-ui-text: #eeeeee;
    --pagefind-ui-background: #152028;
    --pagefind-ui-border: #152028;
    --pagefind-ui-tag: #152028;
  }
</style>

<script>
  window.addEventListener('DOMContentLoaded', (event) => {
    new PagefindUI({
      element: "#search",
      showSubResults: true,
      baseUrl: window.hbb.pagefind.baseUrl,
      bundlePath: window.hbb.pagefind.baseUrl + "pagefind/",
    });
  });
  document.addEventListener('DOMContentLoaded', () => {
    let element = document.getElementById('search');
    let trigger = document.getElementById('search_toggle');

    if (trigger) {
      trigger.addEventListener('click', () => {
        element.classList.toggle('hidden');
        element.querySelector("input").value = ""
        element.querySelector("input").focus()

        if (!element.classList.contains('hidden')) {
          let clear_trigger = document.querySelector('.pagefind-ui__search-clear');

          if (clear_trigger && !clear_trigger.hasAttribute('listenerOnClick')) {
            clear_trigger.setAttribute('listenerOnClick', 'true');

            clear_trigger.addEventListener('click', () => {
              element.classList.toggle('hidden');
            });
          }
        }

      });
    }
  });
</script>















  
  
  
  
  
  
  
  <script
    defer
    src="/js/hugo-blox-en.min.js"
    integrity=""
  ></script>

  
  








  
    
      
      <script async defer src="https://buttons.github.io/buttons.js"></script>

      
    
  




</head>

  <body class="dark:bg-hb-dark dark:text-white page-wrapper" id="top">
    <div id="page-bg"></div>
    <div class="page-header sticky top-0 z-30">
      
      
      
        
        
        
          <header id="site-header" class="header">
  <nav class="navbar px-3 flex justify-left">
    <div class="order-0 h-100">
      
      <a class="navbar-brand" href="/" title="Jiyuan Liu">
        Home
      </a>
    </div>
    
    <input id="nav-toggle" type="checkbox" class="hidden" />
    <label
      for="nav-toggle"
      class="order-3 cursor-pointer flex items-center lg:hidden text-dark dark:text-white lg:order-1">
      <svg id="show-button" class="h-6 fill-current block" viewBox="0 0 20 20">
        <title>Open Menu</title>
        <path d="M0 3h20v2H0V3z m0 6h20v2H0V9z m0 6h20v2H0V0z"></path>
      </svg>
      <svg id="hide-button" class="h-6 fill-current hidden" viewBox="0 0 20 20">
        <title>Close Menu</title>
        <polygon
          points="11 9 22 9 22 11 11 11 11 22 9 22 9 11 -2 11 -2 9 9 9 9 -2 11 -2"
          transform="rotate(45 10 10)"></polygon>
      </svg>
    </label>
    

    
    
    <ul
      id="nav-menu"
      class="navbar-nav order-3 hidden lg:flex w-full pb-6 lg:order-1 lg:w-auto lg:space-x-2 lg:pb-0 xl:space-x-8 justify-left
      ">
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/"
        >Bio</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/#papers"
        >Papers</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/#talks"
        >Talks</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/post"
        >Blogs</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/experience/"
        >Experience</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/projects/"
        >Projects</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/teaching/"
        >Learning</a
        >
      </li>
      
      
      
    </ul>

    <div class="order-1 ml-auto flex items-center md:order-2 lg:ml-0">

      
      
      
      <button
        aria-label="search"
        class="text-black hover:text-primary  inline-block px-3 text-xl dark:text-white"
        id="search_toggle">
        <svg xmlns="http://www.w3.org/2000/svg" height="16" width="16" viewBox="0 0 512 512" fill="currentColor"><path d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"/></svg>
      </button>
      

      
      
      <div class="px-3 text-black hover:text-primary-700 dark:text-white dark:hover:text-primary-300
            [&.active]:font-bold [&.active]:text-black/90 dark:[&.active]:text-white">
        <button class="theme-toggle mt-1" accesskey="t" title="appearance">
          <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
               fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
               stroke-linejoin="round" class="dark:hidden">
            <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
          </svg>
          <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
               fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
               stroke-linejoin="round" class=" dark:block [&:not(dark)]:hidden">
            <circle cx="12" cy="12" r="5"></circle>
            <line x1="12" y1="1" x2="12" y2="3"></line>
            <line x1="12" y1="21" x2="12" y2="23"></line>
            <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
            <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
            <line x1="1" y1="12" x2="3" y2="12"></line>
            <line x1="21" y1="12" x2="23" y2="12"></line>
            <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
            <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
          </svg>
        </button>
      </div>
      

      
      

      
      
    </div>
  </nav>
</header>


<div id="search" class="hidden p-3"></div>


        
      
    </div>
    <div class="page-body  my-10">
      





<div class="mx-auto flex max-w-screen-xl">
  



<aside class="hb-sidebar-container max-lg:[transform:translate3d(0,-100%,0)] lg:hidden xl:block">
  
  <div class="px-4 pt-4 lg:hidden">
    
    
  </div>
  <div class="hb-scrollbar lg:h-[calc(100vh-var(--navbar-height))]">
    <ul class="flex flex-col gap-1 lg:hidden">
      
      
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/event/"
    
  >Recent &amp; Upcoming Talks
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/event/example/"
    
  >Example Talk
    </a>
              
            </li></ul>
      </div></li>
        <li class="open"><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/"
    
  >computational blog
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19.ememwithmapvariationalbayesian/"
    
  >üß¨ EM algorithm, EM with max a posterior, full Bayesian inferences (variational Bayesian) which might be used in auto-coder
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/18.what_make_neural_network_powerful_popular/"
    
  >üß¨ what makes neural network so powerful and popular
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/17.smooth.tests.of.goodness.of.fit/"
    
  >üß¨ Smooth tests of goodness of Fit used in Stijn Hawinkel ‚Äôs plos one paper
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/16.how.good.is.negative.binominal/"
    
  >üß¨ how good is negative binomial, why AI works for biology
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/15.use.seurat.for.bulk.rnaseq/"
    
  >üß¨ Seurat to analysis large sample size of bulk RNAseq. Discuss Wixcox is better for clinical samples
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/14.spatial_genomics_presentation/"
    
  >üß¨ Spatial genomics presentation
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/13.yolov8/"
    
  >üß¨ Bayesian abundance statistics in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/12.basian_abundance_statistics/"
    
  >üß¨ Yolov8
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11.velocity_reasoning/"
    
  >üß¨ Velocity reasoning and discussion
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/10.cca_reasoning/"
    
  >üß¨ CCA reasoning and integration discussion
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/09.sctransform/"
    
  >üß¨ sctransfrom reasoning
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/08.clinical_relevance/"
    
  >üß¨ Clinical relevance evaluation in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/07.cord_diagram/"
    
  >üß¨ Expressive cord diagram in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/06.noise_or_true_stem_cell/"
    
  >üß¨ Determine whether it‚Äôs noise or true stem like cells in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/05.xensoft_caki_paper/"
    
  >üß¨ quantify xenograft and host gene expression from the same sample
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/04.huangcong_oncogene/"
    
  >üß¨ counteract treatment-specific background in differential expressed gene analysis (huangcong paper)
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/03.well-based_scrnaseq_herv_quantification/"
    
  >üß¨ A snakemake pipline-- quantify hERV, transgenes and genome from well-based single-cell RNAseq
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/02.snakemake_quantify-herv_10xscrnaseq/"
    
  >üß¨ A snakemake pipline-- quantify hERV, transgenes and genome from 10x scRNAseq
    </a>
              
            </li><li class="flex flex-col open"><a
    class="hb-sidebar-custom-link
      sidebar-active-item bg-primary-100 font-semibold text-primary-800 dark:bg-primary-300 dark:text-primary-900"
    href="/post/01a.repeatmasker/"
    
  >üõ†Ô∏è Generate and validate ERV gtf that matches the latest reference genome
    </a>
  
    <ul class="hb-sidebar-mobile-toc"><li>
              <a
                href="#github-source-code"
                class="hb-docs-link"
              >Github source code</a>
            </li>
          <li>
              <a
                href="#what-are-erv"
                class="hb-docs-link"
              >What are ERV</a>
            </li>
          <li>
              <a
                href="#why-is-erv-gtf-needed"
                class="hb-docs-link"
              >Why is ERV gtf needed</a>
            </li>
          <li>
              <a
                href="#why-is-erv-gtf-compatible-with-grcm39-needed"
                class="hb-docs-link"
              >Why is ERV gtf compatible with GRCm39 needed</a>
            </li>
          <li>
              <a
                href="#how-to-generate-and-validate-the-erv-gtf-compatible-with-grcm39"
                class="hb-docs-link"
              >How to generate and validate the ERV gtf compatible with GRCm39</a>
            </li>
          <li>
              <a
                href="#generate-grcm39-erv"
                class="hb-docs-link"
              >Generate GRCm39 ERV</a>
            </li>
          <li>
              <a
                href="#vadlidate-our-grcm39-erv"
                class="hb-docs-link"
              >Vadlidate our GRCm39 ERV</a>
            </li>
          <li>
              <a
                href="#conclusion"
                class="hb-docs-link"
              >Conclusion</a>
            </li>
          </ul>
  
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/01.bulkrnaseq_allow_herv_quantification/"
    
  >üß¨ Quantify hERV and transgenes from Bulk-RNAseq
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/00a.seqwins/"
    
  >‚ö° SeqWins, An R Package for Flexible Base Trimming and Comprehensive FASTQ Analysis on Windows
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/00.model_aerosol_depostion/"
    
  >üí® Modeling the deposition of infectious agents suggests that aerosolized drugs offer enhanced specificity
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.get-started/"
    
  >üéâ Easily create your own simple yet highly customizable blog
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.second-brain/"
    
  >üß† Sharpen your thinking with a second brain
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.data-visualization/"
    
  >üìà Communicate your results effectively with the best data visualizations
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.teach-courses/"
    
  >üë©üèº‚Äçüè´ Teach academic courses
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.project-management/"
    
  >‚úÖ Manage your projects
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/projects/"
    
  >Projects
    </a></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/"
    
  >Projects
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/pandas/"
    
  >Pandas
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/pytorch/"
    
  >PyTorch
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/scikit/"
    
  >scikit-learn
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/experience/"
    
  >Experience
    </a></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/"
    
  >Teaching
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/js/"
    
  >Learn JavaScript
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/python/"
    
  >Learn Python
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/"
    
  >Publications
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/preprint/"
    
  >An example preprint / working paper
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/journal-article/"
    
  >An example journal article
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/conference-paper/"
    
  >An example conference paper
    </a>
              
            </li></ul>
      </div></li>
    </ul>

    <div class="max-xl:hidden h-0 w-64 shrink-0"></div></div>

</aside>
  

<nav class="hb-toc order-last hidden w-64 shrink-0 xl:block print:hidden px-4" aria-label="table of contents">
  











  <div class="hb-scrollbar text-sm [hyphens:auto] sticky top-16 overflow-y-auto pr-4 pt-6 max-h-[calc(100vh-var(--navbar-height)-env(safe-area-inset-bottom))] -mr-4 rtl:-ml-4"><p class="mb-4 font-semibold tracking-tight">On this page</p><ul>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#github-source-code">Github source code</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#what-are-erv">What are ERV</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#why-is-erv-gtf-needed">Why is ERV gtf needed</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#why-is-erv-gtf-compatible-with-grcm39-needed">Why is ERV gtf compatible with GRCm39 needed</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#how-to-generate-and-validate-the-erv-gtf-compatible-with-grcm39">How to generate and validate the ERV gtf compatible with GRCm39</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#generate-grcm39-erv">Generate GRCm39 ERV</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#1-install-repeatmasker">1. Install RepeatMasker</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#2-file-wrangling-and-run-repeatmasker">2. File wrangling and run RepeatMasker</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#3-generate-gtf-file-from-repeatmasker-outputs">3. Generate GTF file from RepeatMasker outputs</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#4-modify-gtf-for-starsolo">4. Modify GTF for STARsolo</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#5-generate-fasta-from-gtf-for-validation">5. Generate FASTA from GTF for validation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#vadlidate-our-grcm39-erv">Vadlidate our GRCm39 ERV</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#1-wrange-files-of-grcm38-erv">1. Wrange files of GRCm38 ERV</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#2-blastn-geve-as-reference-our-grcm39-erv-as-query">2. BLASTn: gEVE as reference, our GRCm39 ERV as query</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#parrellel-computation">parrellel computation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#3-blastn-our-grcm39-erv-as-reference-geve-as-query">3. BLASTn: our GRCm39 ERV as reference, gEVE as query</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#conclusion">Conclusion</a>
      </li></ul>

  
  
    
    
  



    












  </div>
  </nav>


  <article class="w-full break-words flex min-h-[calc(100vh-var(--navbar-height))] min-w-0 justify-center pb-8 pr-[calc(env(safe-area-inset-right)-1.5rem)]">
    <main class="w-full min-w-0 max-w-6xl px-6 pt-4 md:px-12">

      

      <h1 class="mt-2 text-4xl font-bold tracking-tight text-slate-900 dark:text-slate-100">üõ†Ô∏è Generate and validate ERV gtf that matches the latest reference genome</h1>

      <div class="mt-4 mb-16">
      <div class="text-gray-500 dark:text-gray-300 text-sm flex items-center flex-wrap gap-y-2"><span class="mr-1">Mar 24, 2025</span><span class="mx-1">¬∑</span>
        
          
          
          <div class="group inline-flex items-center text-current gap-x-1.5 mx-1">
            
            
            <img src="/author/jiyuan-liu/avatar_hu_98ce379ab30758a2.webp" alt="Jiyuan Liu" class="inline-block h-4 w-4 rounded-full border border-current" loading="lazy" />
            
            <div >Jiyuan Liu</div>
          </div>
          
        

        
        <span class="mx-1">¬∑</span>
        <span class="mx-1">
          11 min read
        </span>
        
        </div>

        <div class="mt-3">
          





        </div>
      </div>



      
      
      
      
      

      
      
      
      
      
      
      
      
      
      
      <div class="article-header article-container featured-image-wrapper mt-4 mb-16" style="max-width: 720px; max-height: 720px;">
        <div style="position: relative">
          <img src="/post/01a.repeatmasker/featured_hu_4fe1142597e1d5c2.webp" width="720" height="720" alt="" class="featured-image">
          <span class="article-header-caption">Image credit: <a href="https://unsplash.com" target="_blank" rel="noopener"><strong>Heather Green on Unsplash</strong></a></span>
        </div>
      </div>
      

      
      

      <div class="prose prose-slate lg:prose-xl dark:prose-invert">
        <h2 id="github-source-code">Github source code</h2>
<p>Please see my <a href="https://github.com/jliu678/Generate_validate_GRCm39_ERV" target="_blank" rel="noopener">github repo</a>.</p>
<h2 id="what-are-erv">What are ERV</h2>
<p>Endogenous Retroviruses (ERV) are ancient viral sequences embedded in the mammalian genome. The roles of hERVs in gene regulation, immunity, development and cancer are under intense research.</p>
<h2 id="why-is-erv-gtf-needed">Why is ERV gtf needed</h2>
<p>ERV gtf is the annotation file describes the start and end position of each of ERV in the genome. And itself or its corresponding fasta file (containing ERV sequences) is required to quantify the ERV from the RNAseq reads.</p>
<h2 id="why-is-erv-gtf-compatible-with-grcm39-needed">Why is ERV gtf compatible with GRCm39 needed</h2>
<p>GRCm39 is preferred to have most accurate and up-to-date quantification of genes or other features from RNAseq reads, because of the key improvements made in GRCm39 compared with GRCm38:</p>
<ul>
<li>Improved accuracy in assembly with updated sequencing data</li>
<li>More corrected sequencing gaps and mis-assemblies</li>
<li>Better representation of segmental duplications and structural variations</li>
<li>Enhanced annotation coverage</li>
</ul>
<p>Moreover, ERV and genes have to be quantify simultaneously for below reasons. Therefore, using the GRCm38 and its ERV annotation instead of the GRCm39 pair results in less accurate quantification of both genes and ERVs</p>
<ul>
<li>properly handling multi-mapping: ERV sequences can be similar or overlap with endogenous genes. If ERVs and genes aren‚Äôt quantified together, reads may be incorrectly assigned to one while the other is completely ignored.</li>
<li>accurately read assignment: regardless multimappings, quantifying ERV with genes ensures a good control of false positives by including a competitive pool of unmatched features.</li>
</ul>
<h2 id="how-to-generate-and-validate-the-erv-gtf-compatible-with-grcm39">How to generate and validate the ERV gtf compatible with GRCm39</h2>
<p><a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC4885607/" target="_blank" rel="noopener">gEVE</a> database was generates using RepeatMasker, thus we were encouraged to generate the latest ERV gtf in the same way. We hypothesized that the ERV gtf generated by us should show a good degree of conservation when compared with GRCm38 ERV gtf in <a href="http://geve.med.u-tokai.ac.jp/" target="_blank" rel="noopener">gEVE database</a>, and validated it correspondingly.</p>
<h2 id="generate-grcm39-erv">Generate GRCm39 ERV</h2>
<h3 id="1-install-repeatmasker">1. Install RepeatMasker</h3>
<p>Please refer to its <a href="https://www.repeatmasker.org/RepeatMasker/" target="_blank" rel="noopener">official manuel</a></p>
<h3 id="2-file-wrangling-and-run-repeatmasker">2. File wrangling and run RepeatMasker</h3>
<p>Download fasta of GRCm39 (i.e. mm39), and split fasta by chromosome names by <code>faidx</code> provided by <a href="http://www.htslib.org/" target="_blank" rel="noopener">samtools</a> or <a href="https://pythonhosted.org/pyfaidx/" target="_blank" rel="noopener">pyfaidx</a>.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash 
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="nv">CHRS</span><span class="o">=(</span><span class="k">$(</span>seq <span class="m">1</span> 19<span class="k">)</span> <span class="s2">&#34;X&#34;</span> <span class="s2">&#34;Y&#34;</span><span class="o">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">mkdir mus39_split
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">for</span> chr in <span class="s2">&#34;</span><span class="si">${</span><span class="nv">CHRS</span><span class="p">[@]</span><span class="si">}</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="s2">&#34;extracting chr</span><span class="si">${</span><span class="nv">chr</span><span class="si">}</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">	faidx --regex <span class="s2">&#34;.*chr</span><span class="si">${</span><span class="nv">chr</span><span class="si">}</span><span class="s2">.*&#34;</span> Mmus39.fa -o <span class="s2">&#34;mus39_split/chr</span><span class="si">${</span><span class="nv">chr</span><span class="si">}</span><span class="s2">.fa&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span></code></pre></div><p>Then call <code>run_repeatmasker.sh</code> as below to run repeatmasker on each mouse chromosome.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nv">chroms</span><span class="o">=(</span> <span class="k">$(for</span> i in <span class="k">$(</span>seq <span class="m">1</span> 19<span class="k">)</span> X Y<span class="p">;</span> <span class="k">do</span> <span class="nb">echo</span> <span class="s2">&#34;chr</span><span class="nv">$i</span><span class="s2">&#34;</span><span class="p">;</span> <span class="k">done)</span> <span class="o">)</span>
</span></span><span class="line"><span class="cl">bash run_repeatmasker.sh <span class="s2">&#34;</span><span class="si">${</span><span class="nv">chroms</span><span class="p">[@]</span><span class="si">}</span><span class="s2">&#34;</span>
</span></span></code></pre></div><p>The <code>run_repeatmasker.sh</code> is below</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl"><span class="c1"># save this file as run_repeatmasker.sh</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># for every chr in listed in input</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> chr in <span class="nv">$@</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="c1"># ./repeatmasker/${chr} is output dir</span>
</span></span><span class="line"><span class="cl">    mkdir ./repeatmasker/<span class="si">${</span><span class="nv">chr</span><span class="si">}</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">    <span class="c1"># hmmer use 2 cores for each instance, 24 cores means 12 instances of hmmer</span>
</span></span><span class="line"><span class="cl">    RepeatMasker -engine hmmer -parallel <span class="m">12</span> -species <span class="s2">&#34;Mus musculus&#34;</span><span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -gccalc -dir ./repeatmasker/<span class="si">${</span><span class="nv">chr</span><span class="si">}</span> -gff ./chrs/<span class="si">${</span><span class="nv">chr</span><span class="si">}</span>.fa
</span></span><span class="line"><span class="cl"><span class="k">done</span> 
</span></span></code></pre></div><h3 id="3-generate-gtf-file-from-repeatmasker-outputs">3. Generate GTF file from <code>RepeatMasker</code> outputs</h3>
<p>Below is R function <code>repmask2gtf()</code> to generate GTF from <code>RepeatMasker</code> output files. We will dissect it later.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">repmask2gtf</span> <span class="o">&lt;-</span> <span class="kr">function</span> <span class="p">(</span><span class="n">name</span><span class="p">)</span> <span class="p">{</span>
</span></span><span class="line"><span class="cl">  <span class="n">out_name</span> <span class="o">&lt;-</span> <span class="nf">sprintf</span><span class="p">(</span><span class="s">&#34;chr%1$s/chr%1$s.fa.out&#34;</span><span class="p">,</span> <span class="n">name</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">gff_name</span> <span class="o">&lt;-</span> <span class="nf">sprintf</span><span class="p">(</span><span class="s">&#34;chr%1$s/chr%1$s.fa.out.gff&#34;</span><span class="p">,</span> <span class="n">name</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf_name</span> <span class="o">&lt;-</span> <span class="nf">sprintf</span><span class="p">(</span><span class="s">&#34;gtfs/chr%1$s.gtf&#34;</span><span class="p">,</span> <span class="n">name</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">header</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;bit score&#34;</span><span class="p">,</span> <span class="s">&#34;perc div.&#34;</span><span class="p">,</span> <span class="s">&#34;perc del.&#34;</span><span class="p">,</span> <span class="s">&#34;perc ins.&#34;</span><span class="p">,</span> <span class="s">&#34;chr&#34;</span><span class="p">,</span> <span class="s">&#34;begin&#34;</span><span class="p">,</span> <span class="s">&#34;end&#34;</span><span class="p">,</span> <span class="s">&#34;(left)&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">              <span class="s">&#34;side&#34;</span><span class="p">,</span> <span class="s">&#34;matching repeat&#34;</span><span class="p">,</span> <span class="s">&#34;repeat class/family&#34;</span><span class="p">,</span> <span class="s">&#34;rep begin&#34;</span><span class="p">,</span> <span class="s">&#34;rep end&#34;</span><span class="p">,</span> <span class="s">&#34;rep (left)&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">              <span class="s">&#34;ID&#34;</span><span class="p">,</span> <span class="s">&#34;inc higer score&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">data</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span><span class="n">out_name</span><span class="p">,</span> <span class="n">skip</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">fill</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">col.names</span> <span class="o">=</span> <span class="nf">paste0</span><span class="p">(</span><span class="s">&#34;V&#34;</span><span class="p">,</span><span class="nf">seq_len</span><span class="p">(</span><span class="m">16</span><span class="p">)))</span>
</span></span><span class="line"><span class="cl">  <span class="nf">names</span><span class="p">(</span><span class="n">data</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">header</span>
</span></span><span class="line"><span class="cl">  <span class="n">gff</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span><span class="n">gff_name</span><span class="p">,</span> <span class="n">skip</span><span class="o">=</span><span class="m">2</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">gff</span> <span class="o">&lt;-</span> <span class="n">gff[</span><span class="o">!</span><span class="nf">grepl</span><span class="p">(</span><span class="s">&#34;Simple|Unknown&#34;</span><span class="p">,</span> <span class="n">data</span><span class="o">$</span><span class="n">`repeat class/family`</span><span class="p">),</span><span class="n">]</span>
</span></span><span class="line"><span class="cl">  <span class="n">data</span> <span class="o">&lt;-</span> <span class="n">data[</span><span class="o">!</span><span class="nf">grepl</span><span class="p">(</span><span class="s">&#34;Simple|Unknown&#34;</span><span class="p">,</span> <span class="n">data</span><span class="o">$</span><span class="n">`repeat class/family`</span><span class="p">),</span><span class="n">]</span>
</span></span><span class="line"><span class="cl">  <span class="n">threshold</span> <span class="o">&lt;-</span> <span class="m">750</span>
</span></span><span class="line"><span class="cl">  <span class="n">gff</span> <span class="o">&lt;-</span> <span class="n">gff</span><span class="nf">[</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">`bit score`</span> <span class="o">&gt;</span> <span class="n">threshold</span><span class="p">),</span><span class="n">]</span>
</span></span><span class="line"><span class="cl">  <span class="n">data</span> <span class="o">&lt;-</span> <span class="n">data</span><span class="nf">[</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">`bit score`</span> <span class="o">&gt;</span> <span class="n">threshold</span><span class="p">),</span><span class="n">]</span>
</span></span><span class="line"><span class="cl">  <span class="kr">if</span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">gff</span><span class="p">)</span> <span class="o">==</span> <span class="m">0</span> <span class="o">||</span> <span class="nf">nrow</span><span class="p">(</span><span class="n">data</span><span class="p">)</span> <span class="o">==</span> <span class="m">0</span><span class="p">)</span> <span class="p">{</span>
</span></span><span class="line"><span class="cl">    <span class="nf">cat</span><span class="p">(</span><span class="s">&#39;No Repeatmasker annotation after filtering for&#39;</span><span class="p">,</span><span class="n">name</span><span class="p">,</span><span class="s">&#39;\n&#39;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">    <span class="kr">return</span><span class="p">()</span>
</span></span><span class="line"><span class="cl">  <span class="p">}</span>
</span></span><span class="line"><span class="cl">  <span class="n">seqname</span> <span class="o">&lt;-</span> <span class="n">data</span><span class="o">$</span><span class="n">chr</span>
</span></span><span class="line"><span class="cl">  <span class="n">source</span> <span class="o">&lt;-</span> <span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&#34;RepeatMasker&#34;</span><span class="p">),</span> <span class="n">each</span><span class="o">=</span><span class="nf">length</span><span class="p">(</span><span class="n">seqname</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">  <span class="n">feature</span> <span class="o">&lt;-</span> <span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&#34;mRNA&#34;</span><span class="p">),</span> <span class="n">each</span><span class="o">=</span><span class="nf">length</span><span class="p">(</span><span class="n">seqname</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">  <span class="n">start</span> <span class="o">&lt;-</span> <span class="n">data</span><span class="o">$</span><span class="n">begin</span>
</span></span><span class="line"><span class="cl">  <span class="n">end</span> <span class="o">&lt;-</span> <span class="n">data</span><span class="o">$</span><span class="n">end</span>
</span></span><span class="line"><span class="cl">  <span class="n">score</span> <span class="o">&lt;-</span> <span class="n">data</span><span class="o">$</span><span class="n">`perc div.`</span>
</span></span><span class="line"><span class="cl">  <span class="n">strand</span> <span class="o">&lt;-</span> <span class="n">gff</span><span class="o">$</span><span class="n">V7</span>
</span></span><span class="line"><span class="cl">  <span class="n">frame</span> <span class="o">&lt;-</span> <span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&#34;.&#34;</span><span class="p">),</span> <span class="n">each</span><span class="o">=</span><span class="nf">length</span><span class="p">(</span><span class="n">seqname</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">  <span class="n">gene_id</span> <span class="o">&lt;-</span> <span class="nf">sapply</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">seqname</span><span class="p">),</span> <span class="kr">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="nf">paste0</span><span class="p">(</span><span class="s">&#34;gene_id \&#34;RepMasker&#34;</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="s">&#34;\&#34;&#34;</span><span class="p">,</span> <span class="n">collapse</span><span class="o">=</span><span class="s">&#39;&#39;</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">  <span class="n">gene_name</span> <span class="o">&lt;-</span> <span class="nf">mapply</span><span class="p">(</span>
</span></span><span class="line"><span class="cl">    <span class="kr">function</span><span class="p">(</span><span class="n">i</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="n">z</span><span class="p">)</span> <span class="nf">paste0</span><span class="p">(</span><span class="s">&#34;gene_name \&#34;&#34;</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="s">&#34;:&#34;</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="s">&#34;:&#34;</span><span class="p">,</span> <span class="n">z</span><span class="p">,</span> <span class="s">&#34;:&#34;</span><span class="p">,</span> <span class="n">i</span><span class="p">,</span> <span class="s">&#34;\&#34;&#34;</span><span class="p">,</span> <span class="n">collapse</span><span class="o">=</span><span class="s">&#39;&#39;</span><span class="p">),</span>
</span></span><span class="line"><span class="cl">    <span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">seqname</span><span class="p">),</span> 
</span></span><span class="line"><span class="cl">    <span class="n">data</span><span class="o">$</span><span class="n">chr</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">    <span class="n">data</span><span class="o">$</span><span class="n">`repeat class/family`</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">    <span class="n">data</span><span class="o">$</span><span class="n">`matching repeat`</span>
</span></span><span class="line"><span class="cl">    <span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">ID</span> <span class="o">&lt;-</span> <span class="nf">sapply</span><span class="p">(</span>
</span></span><span class="line"><span class="cl">    <span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">seqname</span><span class="p">),</span> 
</span></span><span class="line"><span class="cl">    <span class="kr">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="nf">paste0</span><span class="p">(</span><span class="s">&#34;ID \&#34;RepMasker&#34;</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="s">&#34;\&#34;&#34;</span><span class="p">,</span> <span class="n">collapse</span><span class="o">=</span><span class="s">&#39;&#39;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">    <span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">attr</span> <span class="o">&lt;-</span> <span class="nf">mapply</span><span class="p">(</span>
</span></span><span class="line"><span class="cl">    <span class="kr">function</span><span class="p">(</span><span class="n">id</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">)</span> <span class="nf">paste0</span><span class="p">(</span><span class="n">id</span><span class="p">,</span> <span class="s">&#34;; &#34;</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="s">&#34;; &#34;</span><span class="p">,</span> <span class="n">y</span><span class="p">),</span>
</span></span><span class="line"><span class="cl">    <span class="n">ID</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">    <span class="n">gene_id</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">    <span class="n">gene_name</span>
</span></span><span class="line"><span class="cl">    <span class="p">)</span>
</span></span><span class="line"><span class="cl">    <span class="n">gtf</span> <span class="o">&lt;-</span> <span class="nf">data.table</span><span class="p">(</span><span class="n">seqname</span><span class="o">=</span><span class="n">seqname</span><span class="p">,</span><span class="n">source</span><span class="o">=</span><span class="n">source</span><span class="p">,</span><span class="n">feature</span><span class="o">=</span><span class="n">feature</span><span class="p">,</span><span class="n">start</span><span class="o">=</span><span class="n">start</span><span class="p">,</span><span class="n">end</span><span class="o">=</span><span class="n">end</span><span class="p">,</span><span class="n">score</span><span class="o">=</span><span class="n">score</span><span class="p">,</span><span class="n">strand</span><span class="o">=</span><span class="n">strand</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">    <span class="n">frame</span><span class="o">=</span><span class="n">frame</span><span class="p">,</span><span class="n">attribute</span><span class="o">=</span><span class="n">attr</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="nf">write.table</span><span class="p">(</span><span class="n">gtf</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="n">gtf_name</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&#39;\t&#39;</span><span class="p">,</span> <span class="n">col.names</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span> <span class="n">row.names</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="p">}</span>
</span></span></code></pre></div><p>The below lines get the names of <code>.out</code> and <code>.out.gff</code> files output by <code>repeatmasker</code>, and define the name of the GTF to return.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl">  <span class="n">out_name</span> <span class="o">&lt;-</span> <span class="nf">sprintf</span><span class="p">(</span><span class="s">&#34;chr%1$s/chr%1$s.fa.out&#34;</span><span class="p">,</span> <span class="n">name</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">gff_name</span> <span class="o">&lt;-</span> <span class="nf">sprintf</span><span class="p">(</span><span class="s">&#34;chr%1$s/chr%1$s.fa.out.gff&#34;</span><span class="p">,</span> <span class="n">name</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf_name</span> <span class="o">&lt;-</span> <span class="nf">sprintf</span><span class="p">(</span><span class="s">&#34;gtfs/chr%1$s.gtf&#34;</span><span class="p">,</span> <span class="n">name</span><span class="p">)</span>
</span></span></code></pre></div><p>The next few lines simply reads the files that are generated by <code>RepeatMasker</code>. Note that we are using the <code>-gff</code> setting in RepeatMasker so that it generates a <code>.out.gff</code> file, which besides <code>.out</code> file contains information required for GTF. We&rsquo;re skipping the first two lines of the .out and .gff files because they are just headers. I define a header for the data read from <code>.out.gff</code> file.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl">  <span class="c1"># make sure we&#39;re reading 16 columns, because the last column is an optional one </span>
</span></span><span class="line"><span class="cl">  <span class="c1"># and some rows many not have them </span>
</span></span><span class="line"><span class="cl">  <span class="n">data</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span><span class="n">out_name</span><span class="p">,</span> <span class="n">skip</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">fill</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                     <span class="n">col.names</span> <span class="o">=</span> <span class="nf">paste0</span><span class="p">(</span><span class="s">&#34;V&#34;</span><span class="p">,</span><span class="nf">seq_len</span><span class="p">(</span><span class="m">16</span><span class="p">)))</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  <span class="nf">names</span><span class="p">(</span><span class="n">data</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;bit score&#34;</span><span class="p">,</span> <span class="s">&#34;perc div.&#34;</span><span class="p">,</span> <span class="s">&#34;perc del.&#34;</span><span class="p">,</span> <span class="s">&#34;perc ins.&#34;</span><span class="p">,</span> <span class="s">&#34;chr&#34;</span><span class="p">,</span> <span class="s">&#34;begin&#34;</span><span class="p">,</span> <span class="s">&#34;end&#34;</span><span class="p">,</span> <span class="s">&#34;(left)&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">              <span class="s">&#34;side&#34;</span><span class="p">,</span> <span class="s">&#34;matching repeat&#34;</span><span class="p">,</span> <span class="s">&#34;repeat class/family&#34;</span><span class="p">,</span> <span class="s">&#34;rep begin&#34;</span><span class="p">,</span> <span class="s">&#34;rep end&#34;</span><span class="p">,</span> <span class="s">&#34;rep (left)&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">              <span class="s">&#34;ID&#34;</span><span class="p">,</span> <span class="s">&#34;inc higer score&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">  <span class="n">gff</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span><span class="n">gff_name</span><span class="p">,</span> <span class="n">skip</span><span class="o">=</span><span class="m">2</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>
</span></span></code></pre></div><p>Next, we&rsquo;re filtering the ERV output by RepeatMasker to only keep the sequences more likely to be genetically significant</p>
<ul>
<li>many of the identified sequences overlap and often the classification that best represents the true origin of the sequence has the highest score. So we are filtering for that by having a threshold for the <code>bit score</code>.</li>
<li>the <code>.out.gff</code> file is ordered the same way as the <code>.out</code> file, so can also filter based on the <code>Simple/Unknown</code> classification in <code>.out</code> file. Simple repeats tend to score higher than ERV or LINE.</li>
</ul>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl">  <span class="n">gff</span> <span class="o">&lt;-</span> <span class="n">gff[</span><span class="o">!</span><span class="nf">grepl</span><span class="p">(</span><span class="s">&#34;Simple|Unknown&#34;</span><span class="p">,</span> <span class="n">data</span><span class="o">$</span><span class="n">`repeat class/family`</span><span class="p">),</span><span class="n">]</span>
</span></span><span class="line"><span class="cl">  <span class="n">data</span> <span class="o">&lt;-</span> <span class="n">data[</span><span class="o">!</span><span class="nf">grepl</span><span class="p">(</span><span class="s">&#34;Simple|Unknown&#34;</span><span class="p">,</span> <span class="n">data</span><span class="o">$</span><span class="n">`repeat class/family`</span><span class="p">),</span><span class="n">]</span>
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">  <span class="kr">if</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">gff</span><span class="p">)</span> <span class="o">==</span> <span class="m">0</span> <span class="o">||</span> <span class="nf">length</span><span class="p">(</span><span class="n">data</span><span class="p">)</span> <span class="o">==</span> <span class="m">0</span><span class="p">)</span> <span class="p">{</span><span class="kr">return</span><span class="p">()}</span>
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">  <span class="n">threshold</span> <span class="o">&lt;-</span> <span class="m">750</span>
</span></span><span class="line"><span class="cl">  <span class="n">gff</span> <span class="o">&lt;-</span> <span class="n">gff</span><span class="nf">[</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">`bit score`</span> <span class="o">&gt;</span> <span class="n">threshold</span><span class="p">),</span><span class="n">]</span>
</span></span><span class="line"><span class="cl">  <span class="n">data</span> <span class="o">&lt;-</span> <span class="n">data</span><span class="nf">[</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">`bit score`</span> <span class="o">&gt;</span> <span class="n">threshold</span><span class="p">),</span><span class="n">]</span>
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">  <span class="kr">if</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">gff</span><span class="p">)</span> <span class="o">==</span> <span class="m">0</span> <span class="o">||</span> <span class="nf">length</span><span class="p">(</span><span class="n">data</span><span class="p">)</span> <span class="o">==</span> <span class="m">0</span><span class="p">)</span> <span class="p">{</span><span class="kr">return</span><span class="p">()}</span>
</span></span></code></pre></div><p>Then, we generate each column for the GTF. A major issue is that <code>AGAT</code>, the tool we use to turn GTF files into FASTA files for validating our ERV GTF, requires that each sequence has an ID and be marked as <strong>mRNA</strong>.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl">  <span class="n">seqname</span> <span class="o">&lt;-</span> <span class="n">data</span><span class="o">$</span><span class="n">chr</span>
</span></span><span class="line"><span class="cl">  <span class="n">source</span> <span class="o">&lt;-</span> <span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&#34;RepeatMasker&#34;</span><span class="p">),</span> <span class="n">each</span><span class="o">=</span><span class="nf">length</span><span class="p">(</span><span class="n">seqname</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">  <span class="n">feature</span> <span class="o">&lt;-</span> <span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&#34;mRNA&#34;</span><span class="p">),</span> <span class="n">each</span><span class="o">=</span><span class="nf">length</span><span class="p">(</span><span class="n">seqname</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">  <span class="n">start</span> <span class="o">&lt;-</span> <span class="n">data</span><span class="o">$</span><span class="n">begin</span>
</span></span><span class="line"><span class="cl">  <span class="n">end</span> <span class="o">&lt;-</span> <span class="n">data</span><span class="o">$</span><span class="n">end</span>
</span></span><span class="line"><span class="cl">  <span class="n">score</span> <span class="o">&lt;-</span> <span class="n">data</span><span class="o">$</span><span class="n">`perc div.`</span>
</span></span><span class="line"><span class="cl">  <span class="n">strand</span> <span class="o">&lt;-</span> <span class="n">gff</span><span class="o">$</span><span class="n">V7</span>
</span></span><span class="line"><span class="cl">  <span class="n">frame</span> <span class="o">&lt;-</span> <span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&#34;.&#34;</span><span class="p">),</span> <span class="n">each</span><span class="o">=</span><span class="nf">length</span><span class="p">(</span><span class="n">seqname</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">  <span class="n">gene_id</span> <span class="o">&lt;-</span> <span class="nf">sapply</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">seqname</span><span class="p">),</span> 
</span></span><span class="line"><span class="cl">                    <span class="kr">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
</span></span><span class="line"><span class="cl">                      <span class="nf">paste0</span><span class="p">(</span><span class="s">&#34;gene_id \&#34;RepMasker&#34;</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">                             <span class="n">x</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">                             <span class="s">&#34;\&#34;&#34;</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">                             <span class="n">collapse</span><span class="o">=</span><span class="s">&#39;&#39;</span><span class="p">)}</span>
</span></span><span class="line"><span class="cl">                    <span class="p">)</span>
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">  <span class="n">gene_name</span> <span class="o">&lt;-</span> <span class="nf">mapply</span><span class="p">(</span>
</span></span><span class="line"><span class="cl">    <span class="kr">function</span><span class="p">(</span><span class="n">i</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="n">z</span><span class="p">){</span>
</span></span><span class="line"><span class="cl">      <span class="nf">paste0</span><span class="p">(</span><span class="s">&#34;gene_name \&#34;&#34;</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">             <span class="n">x</span><span class="p">,</span> <span class="s">&#34;:&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">             <span class="n">y</span><span class="p">,</span> <span class="s">&#34;:&#34;</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">             <span class="n">z</span><span class="p">,</span> <span class="s">&#34;:&#34;</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">             <span class="n">i</span><span class="p">,</span> <span class="s">&#34;\&#34;&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">             <span class="n">collapse</span><span class="o">=</span><span class="s">&#39;&#39;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">    <span class="p">},</span>
</span></span><span class="line"><span class="cl">    <span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">seqname</span><span class="p">),</span> 
</span></span><span class="line"><span class="cl">    <span class="n">data</span><span class="o">$</span><span class="n">chr</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">    <span class="n">data</span><span class="o">$</span><span class="n">`repeat class/family`</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">    <span class="n">data</span><span class="o">$</span><span class="n">`matching repeat`</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">  <span class="n">ID</span> <span class="o">&lt;-</span> <span class="nf">sapply</span><span class="p">(</span>
</span></span><span class="line"><span class="cl">    <span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">seqname</span><span class="p">),</span> 
</span></span><span class="line"><span class="cl">    <span class="kr">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
</span></span><span class="line"><span class="cl">      <span class="nf">paste0</span><span class="p">(</span><span class="s">&#34;ID \&#34;RepMasker&#34;</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="s">&#34;\&#34;&#34;</span><span class="p">,</span> <span class="n">collapse</span><span class="o">=</span><span class="s">&#39;&#39;</span><span class="p">)}</span>
</span></span><span class="line"><span class="cl">    <span class="p">)</span>
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">  <span class="n">attr</span> <span class="o">&lt;-</span> <span class="nf">mapply</span><span class="p">(</span>
</span></span><span class="line"><span class="cl">    <span class="kr">function</span><span class="p">(</span><span class="n">id</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">)</span> <span class="nf">paste0</span><span class="p">(</span><span class="n">id</span><span class="p">,</span> <span class="s">&#34;; &#34;</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="s">&#34;; &#34;</span><span class="p">,</span> <span class="n">y</span><span class="p">),</span> 
</span></span><span class="line"><span class="cl">    <span class="n">ID</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">    <span class="n">gene_id</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">    <span class="n">gene_name</span><span class="p">)</span>
</span></span></code></pre></div><p>Last, output the GTF</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl">  <span class="n">gtf</span> <span class="o">&lt;-</span> <span class="nf">data.table</span><span class="p">(</span>
</span></span><span class="line"><span class="cl">    <span class="n">seqname</span><span class="o">=</span><span class="n">seqname</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">    <span class="n">source</span><span class="o">=</span><span class="n">source</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">    <span class="n">feature</span><span class="o">=</span><span class="n">feature</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">    <span class="n">start</span><span class="o">=</span><span class="n">start</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">    <span class="n">end</span><span class="o">=</span><span class="n">end</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">    <span class="n">score</span><span class="o">=</span><span class="n">score</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">    <span class="n">strand</span><span class="o">=</span><span class="n">strand</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">    <span class="n">frame</span><span class="o">=</span><span class="n">frame</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">    <span class="n">attribute</span><span class="o">=</span><span class="n">attr</span>
</span></span><span class="line"><span class="cl">  <span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="nf">write.table</span><span class="p">(</span><span class="n">gtf</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="n">gtf_name</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&#39;\t&#39;</span><span class="p">,</span> <span class="n">col.names</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span> <span class="n">row.names</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>
</span></span></code></pre></div><p>A loop below to call the above <code>repmask2gtf()</code> to generate the GTF for all the files that are in the current dir.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="kr">for</span> <span class="p">(</span><span class="n">chr</span> <span class="kr">in</span> <span class="nf">list.dirs</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="s">&#34;.&#34;</span><span class="p">,</span> <span class="n">full.names</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> <span class="n">recursive</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)){</span>
</span></span><span class="line"><span class="cl">  <span class="kr">if</span> <span class="p">(</span><span class="nf">grepl</span><span class="p">(</span><span class="s">&#34;chr&#34;</span><span class="p">,</span> <span class="n">chr</span><span class="p">,</span> <span class="n">fixed</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)){</span>
</span></span><span class="line"><span class="cl">    <span class="nf">print</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#34;generating GTF for &#34;</span><span class="p">,</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;./chr&#34;</span><span class="p">,</span> <span class="s">&#34;&#34;</span><span class="p">,</span> <span class="n">chr</span><span class="p">)))</span>
</span></span><span class="line"><span class="cl">    <span class="kr">if</span> <span class="p">(</span><span class="o">!</span><span class="nf">file.exists</span><span class="p">(</span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&#34;gtfs/chr%1$s.gtf&#34;</span><span class="p">,</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;./chr&#34;</span><span class="p">,</span> <span class="s">&#34;&#34;</span><span class="p">,</span> <span class="n">chr</span><span class="p">)))){</span>
</span></span><span class="line"><span class="cl">      <span class="nf">repmask2gtf</span><span class="p">(</span><span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;./chr&#34;</span><span class="p">,</span> <span class="s">&#34;&#34;</span><span class="p">,</span> <span class="n">chr</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">    <span class="p">}</span><span class="kr">else</span><span class="p">{</span>
</span></span><span class="line"><span class="cl">      <span class="nf">cat</span><span class="p">(</span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&#34;gtfs/chr%1$s.gtf&#34;</span><span class="p">,</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;./chr&#34;</span><span class="p">,</span> <span class="s">&#34;&#34;</span><span class="p">,</span> <span class="n">chr</span><span class="p">)),</span> <span class="s">&#39;already exist\n&#39;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">    <span class="p">}</span>
</span></span><span class="line"><span class="cl">  <span class="p">}</span>
</span></span><span class="line"><span class="cl"><span class="p">}</span>
</span></span></code></pre></div><h3 id="4-modify-gtf-for-starsolo">4. Modify GTF for STARsolo</h3>
<p>Modify GTF for a couple of reasons:</p>
<ul>
<li>STAR does not accept that we&rsquo;re using <em>mRNA</em> instead of <em>exons</em></li>
<li>The <em>gene id</em>, which are <em>RepeatMasker</em> + <em>a number</em>, may not work when we concat all chromosomes together due to  duplicated IDs.</li>
<li>STAR also does not need a <em>ID</em> besides <em>gene id</em>, so we&rsquo;ll remove that as well.</li>
</ul>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="n">change_feature_gtf</span> <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span><span class="n">gtf_tab</span><span class="p">)</span> <span class="p">{</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf_tab</span><span class="o">$</span><span class="n">V3</span> <span class="o">&lt;-</span> <span class="nf">rep</span><span class="p">(</span><span class="s">&#34;exon&#34;</span><span class="p">,</span> <span class="nf">nrow</span><span class="p">(</span><span class="n">gtf_tab</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf_tab</span>
</span></span><span class="line"><span class="cl"><span class="p">}</span>
</span></span><span class="line"><span class="cl"><span class="n">rm_id_gtf</span> <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span><span class="n">gtf_tab</span><span class="p">)</span> <span class="p">{</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf_tab</span><span class="o">$</span><span class="n">V9</span> <span class="o">&lt;-</span> <span class="nf">lapply</span><span class="p">(</span>
</span></span><span class="line"><span class="cl">    <span class="n">gtf_tab</span><span class="o">$</span><span class="n">V9</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">    <span class="kr">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;(ID \&#34;[A-Za-z0-9]*\&#34;; )&#34;</span><span class="p">,</span> <span class="s">&#34;&#34;</span><span class="p">,</span> <span class="n">x</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf_tab</span>
</span></span><span class="line"><span class="cl"><span class="p">}</span>
</span></span><span class="line"><span class="cl"><span class="n">rename_gtf</span> <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span><span class="n">gtf_tab</span><span class="p">)</span> <span class="p">{</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf_tab</span><span class="o">$</span><span class="n">V9</span> <span class="o">&lt;-</span> <span class="nf">lapply</span><span class="p">(</span><span class="n">gtf_tab</span><span class="o">$</span><span class="n">V9</span><span class="p">,</span> <span class="kr">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="p">{</span>
</span></span><span class="line"><span class="cl">    <span class="n">new_id</span> <span class="o">&lt;-</span> <span class="nf">sprintf</span><span class="p">(</span><span class="s">&#34;gene_id \&#34;RepMasker_%s\&#34;; &#34;</span><span class="p">,</span> <span class="n">uuid</span><span class="o">::</span><span class="nf">UUIDgenerate</span><span class="p">())</span>
</span></span><span class="line"><span class="cl">    <span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;(gene_id \&#34;[A-Za-z0-9]*\&#34;; )&#34;</span><span class="p">,</span> <span class="n">new_id</span><span class="p">,</span> <span class="n">x</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="p">})</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf_tab</span>
</span></span><span class="line"><span class="cl"><span class="p">}</span>
</span></span><span class="line"><span class="cl"><span class="n">mod_gtf</span> <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span><span class="n">chr_name</span><span class="p">)</span> <span class="p">{</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span><span class="n">chr_name</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#34;\t&#34;</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="s">&#34;&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf</span> <span class="o">&lt;-</span> <span class="nf">change_feature_gtf</span><span class="p">(</span><span class="n">gtf</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf</span> <span class="o">&lt;-</span> <span class="nf">rm_id_gtf</span><span class="p">(</span><span class="n">gtf</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf</span> <span class="o">&lt;-</span> <span class="nf">rename_gtf</span><span class="p">(</span><span class="n">gtf</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf</span> <span class="o">&lt;-</span> <span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">gtf</span><span class="p">,</span><span class="n">unlist</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">  <span class="nf">write.table</span><span class="p">(</span><span class="n">gtf</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;./gtfs/&#34;</span><span class="p">,</span> <span class="s">&#34;./gtfs2/&#34;</span><span class="p">,</span> <span class="n">chr_name</span><span class="p">),</span> <span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&#39;\t&#39;</span><span class="p">,</span> <span class="n">col.names</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span> <span class="n">row.names</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="p">}</span>
</span></span></code></pre></div><p>A <code>for</code> loop below to apply <code>mod_gtf()</code> to every GTF file</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="kr">for</span> <span class="p">(</span><span class="n">chr</span> <span class="kr">in</span> <span class="nf">list.files</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="s">&#34;./gtfs&#34;</span><span class="p">,</span> <span class="n">full.names</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span> <span class="n">recursive</span><span class="o">=</span><span class="bp">F</span><span class="p">))</span> <span class="p">{</span>
</span></span><span class="line"><span class="cl">  <span class="kr">if</span> <span class="p">(</span><span class="o">!</span><span class="nf">file.exists</span><span class="p">(</span><span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;./gtfs/&#34;</span><span class="p">,</span> <span class="s">&#34;./gtfs2/&#34;</span><span class="p">,</span> <span class="n">chr</span><span class="p">))){</span>
</span></span><span class="line"><span class="cl">    <span class="nf">print</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#34;generating new gtfs for: &#34;</span><span class="p">,</span> <span class="n">chr</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">    <span class="nf">mod_gtf</span><span class="p">(</span><span class="n">chr</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="p">}</span>
</span></span><span class="line"><span class="cl"><span class="p">}</span>
</span></span></code></pre></div><h3 id="5-generate-fasta-from-gtf-for-validation">5. Generate FASTA from GTF for validation</h3>
<p>We first modify the GTF generated for STAR</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="n">mod_gtf_for_AGAT</span> <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span><span class="n">chr_name</span><span class="p">,</span><span class="n">input_dir</span><span class="o">=</span><span class="s">&#34;./gtfs/&#34;</span><span class="p">,</span><span class="n">output_dir</span><span class="o">=</span><span class="s">&#34;./gtfs2/&#34;</span><span class="p">)</span> <span class="p">{</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span><span class="n">chr_name</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#34;\t&#34;</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="s">&#34;&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">t_uuid</span> <span class="o">&lt;-</span> <span class="n">stringr</span><span class="o">::</span><span class="nf">str_extract</span><span class="p">(</span><span class="n">string</span> <span class="o">=</span> <span class="n">gtf</span><span class="o">$</span><span class="n">V9</span><span class="p">,</span><span class="n">pattern</span> <span class="o">=</span> <span class="s">&#39;gene_id( &#34;RepMasker_.*?&#34;; )&#39;</span><span class="p">,</span><span class="n">group</span> <span class="o">=</span> <span class="m">1</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf</span><span class="o">$</span><span class="n">V9</span> <span class="o">&lt;-</span> <span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;ID&#39;</span><span class="p">,</span><span class="n">t_uuid</span><span class="p">,</span><span class="n">gtf</span><span class="o">$</span><span class="n">V9</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf</span><span class="o">$</span><span class="n">V3</span> <span class="o">&lt;-</span> <span class="nf">rep</span><span class="p">(</span><span class="s">&#34;mRNA&#34;</span><span class="p">,</span> <span class="nf">nrow</span><span class="p">(</span><span class="n">gtf</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">  <span class="n">gtf</span> <span class="o">&lt;-</span> <span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">gtf</span><span class="p">,</span><span class="n">unlist</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">  <span class="nf">write.table</span><span class="p">(</span><span class="n">gtf</span><span class="p">,</span><span class="n">file</span><span class="o">=</span><span class="nf">gsub</span><span class="p">(</span><span class="n">input_dir</span><span class="p">,</span> <span class="n">output_dir</span><span class="p">,</span> <span class="n">chr_name</span><span class="p">),</span><span class="n">quote</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&#39;\t&#39;</span><span class="p">,</span><span class="n">col.names</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="n">row.names</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="p">}</span>
</span></span></code></pre></div><p>Then convert GTF into fasta for comparing our GRCm39 ERV GTF with from GRCm38 gEVE database using <a href="https://agat.readthedocs.io/en/latest/" target="_blank" rel="noopener"><code>AGAT</code></a>installed by mamba/conda.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1"># see also in &#39;gtf2fa.sh&#39;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># CHRS=($(seq 1 19) &#34;X&#34; &#34;Y&#34;)</span>
</span></span><span class="line"><span class="cl"><span class="nv">CHRS</span><span class="o">=(</span><span class="s2">&#34;Y&#34;</span><span class="o">)</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> chr in <span class="s2">&#34;</span><span class="si">${</span><span class="nv">CHRS</span><span class="p">[@]</span><span class="si">}</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="s2">&#34;extracting .fa from chr</span><span class="si">${</span><span class="nv">chr</span><span class="si">}</span><span class="s2">.gtf&#34;</span>
</span></span><span class="line"><span class="cl">	agat_sp_extract_sequences.pl -g ./gtfs2/chr<span class="si">${</span><span class="nv">chr</span><span class="si">}</span>.gtf -t mRNA --merge 0<span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -f ../single_cell/mouse/GRCm39.primary_assembly.genome.fa -o ./fa/chr<span class="si">${</span><span class="nv">chr</span><span class="si">}</span>.fa
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span></code></pre></div><h2 id="vadlidate-our-grcm39-erv">Vadlidate our GRCm39 ERV</h2>
<p>We compared ERV using RepeatMasker and its built-in Dfam 4.1 on GRCm39 and GRCm38 ERV in <a href="https://doi.org/10.1093%2Fdatabase%2Fbaw087" target="_blank" rel="noopener">gEVE</a>.</p>
<h3 id="1-wrange-files-of-grcm38-erv">1. Wrange files of GRCm38 ERV</h3>
<p>ERV sequences for the mouse genome assembly GRCm38 are downloaded from the <a href="http://geve.med.u-tokai.ac.jp/download/" target="_blank" rel="noopener">gEVE website</a>. They are split by chromosome names using <a href="https://pythonhosted.org/pyfaidx/" target="_blank" rel="noopener">pyfaidx</a> or <code>faidx</code> provided by <a href="http://www.htslib.org/" target="_blank" rel="noopener">samtools</a></p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># download gEVE mouse annots</span>
</span></span><span class="line"><span class="cl">wget http://geve.med.u-tokai.ac.jp/download_data/nt_fasta/Mmus38.geve.nt_v1.fa.bz2
</span></span><span class="line"><span class="cl">bunzip Mmus38.geve.nt_v1.fa.bz2
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># extract just chromosome Y (we&#39;re going to compare this one)</span>
</span></span><span class="line"><span class="cl">mkdir mus38_split
</span></span><span class="line"><span class="cl">faidx --regex <span class="s2">&#34;.*chrY.*&#34;</span> Mmus38.geve.nt_v1.fa -o <span class="s2">&#34;mus38_split/chrY.fa&#34;</span>
</span></span></code></pre></div><h3 id="2-blastn-geve-as-reference-our-grcm39-erv-as-query">2. BLASTn: gEVE as reference, our GRCm39 ERV as query</h3>
<p>Then, a BLASTn database is created based on the extracted chromosome Y sequences from GRCm38 ERV. Make sure <a href="https://blast.ncbi.nlm.nih.gov/doc/blast-help/downloadblastdata.html" target="_blank" rel="noopener">BLAST</a> is installed on your system and is in system <strong>PATHS</strong> by adding the below to your user bashrc or else <code>makeblastdb</code> and <code>rBLAST</code> would not work. It won&rsquo;t be achieved by running <code>module load blast</code> in eristwo command line interface or rsutdio terminal</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">export</span> <span class="nv">PATH</span><span class="o">=</span><span class="nv">$PATH</span>:/apps/lib/blast/ncbi-blast-2.13.0+/bin/
</span></span><span class="line"><span class="cl">mkdir ./blastdb/mus38_chrY
</span></span><span class="line"><span class="cl">makeblastdb -in mus38_split/chrY.fa -title mus38_chrY <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -dbtype nucl -parse_seqids -out ./blastdb/mus38_chrY/mus38_chrY
</span></span><span class="line"><span class="cl"><span class="c1"># the first mus38_chrY of &#34;./blastdb/mus38_chrY/mus38_chrY &#34; specify dir, the second specify database name </span>
</span></span></code></pre></div><p>In order to compare the ERV sequence generated, we first use the BLASTn to build reference using gEVE sequences and use RepeatMasker sequences are query. We will then do the reverse and use RepeatMasker sequences as reference and gEVE sequences as query. This step simply reads the RepeatMasker sequences and the BLAST database.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">rBLAST</span><span class="p">)</span> <span class="c1"># make sure your default PATH includes BLAST, or else it wouldn&#39;t work</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">Biostrings</span><span class="p">)</span> <span class="c1"># read .fa files</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># allow us to parallelize the rBLAST operations</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">foreach</span><span class="p">)</span> 
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">doParallel</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">reptmask_chrY</span> <span class="o">&lt;-</span> <span class="nf">readDNAStringSet</span><span class="p">(</span><span class="s">&#39;repeatmasker/chrY.fa&#39;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">bla</span> <span class="o">&lt;-</span> <span class="nf">blast</span><span class="p">(</span><span class="n">db</span><span class="o">=</span><span class="s">&#39;./blastdb/mus38_chrY/mus38_chrY&#39;</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="s">&#34;blastn&#34;</span><span class="p">)</span><span class="c1">#read makeblastdb-generated database into R enviorment</span>
</span></span></code></pre></div><p>Below test if the sequences can be read and if rBLAST is working.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="n">reptmask_chrY[1</span><span class="p">,</span><span class="n">]</span>
</span></span><span class="line"><span class="cl"><span class="nf">predict</span><span class="p">(</span><span class="n">bla</span><span class="p">,</span> <span class="n">reptmask_chrY[1</span><span class="p">,</span><span class="n">]</span><span class="p">,</span> <span class="n">BLAST_args</span> <span class="o">=</span> <span class="s">&#34;-perc_identity 99&#34;</span><span class="p">)</span>
</span></span></code></pre></div><p>If it looks good, we can speed it up via multi-threading parrellel computation</p>
<h4 id="parrellel-computation"><strong>parrellel computation</strong></h4>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="c1"># run in cluster, this may take 15~30 minutes with a good (48 core) computer</span>
</span></span><span class="line"><span class="cl"><span class="n">reptmask_matches</span> <span class="o">&lt;-</span> <span class="nf">foreach</span><span class="p">(</span><span class="n">i</span><span class="o">=</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">reptmask_chrY</span><span class="p">))</span> <span class="o">%dopar%</span> <span class="p">{</span>
</span></span><span class="line"><span class="cl">  <span class="nf">library</span><span class="p">(</span><span class="s">&#39;rBLAST&#39;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="nf">library</span><span class="p">(</span><span class="s">&#39;data.table&#39;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">  <span class="n">match_i</span> <span class="o">=</span> <span class="nf">data.table</span><span class="p">(</span><span class="nf">predict</span><span class="p">(</span><span class="n">bla</span><span class="p">,</span> <span class="n">reptmask_chrY[i</span><span class="p">,</span><span class="n">]</span><span class="p">,</span> <span class="n">BLAST_args</span> <span class="o">=</span> <span class="s">&#34;-perc_identity 99&#34;</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">  <span class="n">match_i</span> 
</span></span><span class="line"><span class="cl"><span class="p">}</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># change the name to something more intuitive</span>
</span></span><span class="line"><span class="cl"><span class="nf">names</span><span class="p">(</span><span class="n">reptmask_matches</span><span class="p">)</span><span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="n">pattern</span><span class="o">=</span><span class="s">&#39; .*$&#39;</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                               <span class="n">replacement</span><span class="o">=</span><span class="s">&#39;&#39;</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                               <span class="n">reptmask_chrY</span><span class="o">@</span><span class="n">ranges</span><span class="o">@</span><span class="n">NAMES</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">saveRDS</span><span class="p">(</span><span class="n">reptmask_matches</span><span class="p">,</span><span class="s">&#39;repeatmasker_mus38_full_named_matches.rds&#39;</span><span class="p">)</span>
</span></span></code></pre></div><p>To analyse the matches, we first find number of sequences with matches, and also extract unique matches from the reference.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="nf">length</span><span class="p">(</span><span class="n">reptmask_matches</span><span class="nf">[lapply</span><span class="p">(</span><span class="n">reptmask_matches</span><span class="p">,</span> <span class="n">nrow</span><span class="p">)</span> <span class="o">&gt;</span> <span class="m">0</span><span class="n">]</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="nf">do.call</span><span class="p">(</span><span class="n">c</span><span class="p">,</span> <span class="nf">lapply</span><span class="p">(</span><span class="n">reptmask_matches</span><span class="p">,</span> <span class="kr">function</span> <span class="p">(</span><span class="n">tbl</span><span class="p">)</span> <span class="n">tbl</span><span class="o">$</span><span class="s">&#39;SubjectID&#39;</span><span class="p">))))</span>
</span></span></code></pre></div><p>We can also specify families and see how many matches exist between gEVE and our sequences.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="n">gtflist</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span><span class="s">&#34;repeatmasker/chrY.gtf&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">erv_matches</span> <span class="o">&lt;-</span> <span class="n">full_matches</span><span class="nf">[grepl</span><span class="p">(</span><span class="s">&#34;ERV&#34;</span><span class="p">,</span> <span class="n">gtflist</span><span class="o">$</span><span class="n">V16</span><span class="p">)</span><span class="n">]</span>
</span></span><span class="line"><span class="cl"><span class="nf">length</span><span class="p">(</span><span class="n">erv_matches</span><span class="nf">[lapply</span><span class="p">(</span><span class="n">erv_matches</span><span class="p">,</span> <span class="n">nrow</span><span class="p">)</span> <span class="o">&gt;</span> <span class="m">0</span><span class="n">]</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">erv_matches_list</span> <span class="o">&lt;-</span> <span class="nf">unique</span><span class="p">(</span><span class="nf">do.call</span><span class="p">(</span><span class="n">c</span><span class="p">,</span> <span class="nf">lapply</span><span class="p">(</span><span class="n">erv_matches</span><span class="p">,</span> <span class="kr">function</span> <span class="p">(</span><span class="n">tbl</span><span class="p">)</span> <span class="n">tbl</span><span class="o">$</span><span class="s">&#39;SubjectID&#39;</span><span class="p">)))</span>
</span></span><span class="line"><span class="cl"><span class="nf">length</span><span class="p">(</span><span class="n">erv_matches_list</span><span class="p">)</span>
</span></span></code></pre></div><h3 id="3-blastn-our-grcm39-erv-as-reference-geve-as-query">3. BLASTn: our GRCm39 ERV as reference, gEVE as query</h3>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="n">mus38_chrY</span> <span class="o">&lt;-</span> <span class="nf">readDNAStringSet</span><span class="p">(</span><span class="s">&#39;mus38_split/chrY.fa&#39;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">bla</span> <span class="o">&lt;-</span> <span class="nf">blast</span><span class="p">(</span><span class="n">db</span><span class="o">=</span><span class="s">&#39;./blastdb/rm_mus39_chrY/rm_mus39_chrY&#39;</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="s">&#34;blastn&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">rev_reptmask_matches</span> <span class="o">&lt;-</span> <span class="nf">foreach</span><span class="p">(</span><span class="n">i</span><span class="o">=</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">mus38_chrY</span><span class="p">))</span> <span class="o">%dopar%</span> <span class="p">{</span>
</span></span><span class="line"><span class="cl">  <span class="nf">library</span><span class="p">(</span><span class="s">&#39;rBLAST&#39;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  <span class="nf">library</span><span class="p">(</span><span class="s">&#39;data.table&#39;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">  <span class="n">match_i</span> <span class="o">=</span> <span class="nf">data.table</span><span class="p">(</span><span class="nf">predict</span><span class="p">(</span><span class="n">bla</span><span class="p">,</span> <span class="n">mus38_chrY[i</span><span class="p">,</span><span class="n">]</span><span class="p">,</span> <span class="n">BLAST_args</span> <span class="o">=</span> <span class="s">&#34;-perc_identity 99&#34;</span><span class="p">))</span> 
</span></span><span class="line"><span class="cl">  <span class="n">match_i</span> <span class="c1">#Equivalent to finalMatrix = cbind(finalMatrix, tempMatrix)</span>
</span></span><span class="line"><span class="cl"><span class="p">}</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#comparison</span>
</span></span><span class="line"><span class="cl"><span class="nf">length</span><span class="p">(</span><span class="n">full_matches</span><span class="nf">[lapply</span><span class="p">(</span><span class="n">rev_reptmask_matches</span><span class="p">,</span> <span class="n">nrow</span><span class="p">)</span> <span class="o">&gt;</span> <span class="m">0</span><span class="n">]</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="nf">do.call</span><span class="p">(</span><span class="n">c</span><span class="p">,</span> <span class="nf">lapply</span><span class="p">(</span><span class="n">rev_reptmask_matches</span><span class="p">,</span> <span class="kr">function</span> <span class="p">(</span><span class="n">tbl</span><span class="p">)</span> <span class="n">tbl</span><span class="o">$</span><span class="s">&#39;SubjectID&#39;</span><span class="p">))))</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#family-specific comparison</span>
</span></span><span class="line"><span class="cl"><span class="n">erv_matches</span> <span class="o">&lt;-</span> <span class="n">rev_reptmask_matches</span><span class="nf">[grepl</span><span class="p">(</span><span class="s">&#34;ERV&#34;</span><span class="p">,</span> <span class="n">gtflist</span><span class="o">$</span><span class="n">V16</span><span class="p">)</span><span class="n">]</span>
</span></span><span class="line"><span class="cl"><span class="nf">length</span><span class="p">(</span><span class="n">erv_matches</span><span class="nf">[lapply</span><span class="p">(</span><span class="n">erv_matches</span><span class="p">,</span> <span class="n">nrow</span><span class="p">)</span> <span class="o">&gt;</span> <span class="m">0</span><span class="n">]</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">erv_matches_list</span> <span class="o">&lt;-</span> <span class="nf">unique</span><span class="p">(</span><span class="nf">do.call</span><span class="p">(</span><span class="n">c</span><span class="p">,</span> <span class="nf">lapply</span><span class="p">(</span><span class="n">erv_matches</span><span class="p">,</span> <span class="kr">function</span> <span class="p">(</span><span class="n">tbl</span><span class="p">)</span> <span class="n">tbl</span><span class="o">$</span><span class="s">&#39;SubjectID&#39;</span><span class="p">)))</span>
</span></span><span class="line"><span class="cl"><span class="nf">length</span><span class="p">(</span><span class="n">erv_matches_list</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nf">stopCluster</span><span class="p">(</span><span class="n">cl</span><span class="p">)</span>
</span></span></code></pre></div><h2 id="conclusion">Conclusion</h2>
<p><strong>For chrY:</strong></p>
<ol>
<li>gEVE GRCm38 ERV has 4633 unique sequences; our GRCm39 ERV have 8841 unique sequences including 3701 unique ERV sequences</li>
<li>gEVE sequences as BLASTn query and our GRCm39 ERV as reference:
<ul>
<li><strong>Out of 9043 total sequences in gEVE, 8768 has a BLASTn &gt;99% matched sequence in our GRCm39 ERV, corresponding to 4321 unique GRCm39 sequences.</strong></li>
</ul>
</li>
</ol>
<p>Thus, our GRCm39 ERV capture nearly all information in gEVE GRCm38 ERV, and add ~4000 more unique ERV sequences. We attribute this improvement to the up-to-date GRCm39 and the sensitivity settings available in the newer version of RepeatMasker. Below observation further revealed that the newly identified sequences in GRCm39 ERV predominantly belong to ERV and LINE, both of significant interest in biomedical research.</p>
<ol start="3">
<li>our GRCm39 ERV as BLASTn query and gEVE as reference: <strong>only 55% of GRCm39 ERV segments and 27% of GRCm39 LINE segments have matched sequences in gEVE, specifically</strong></li>
</ol>
<ul>
<li>2794 (out of 5077) of our GRCm39 ERV segments have BLASTn matched sequence in gEVE (corresponding to 6134 sequences);</li>
<li>1803 (out of 6670) of our GRCm39 LINE segments have BLASTn matched sequence in gEVE (corresponding to 1454 sequences)</li>
</ul>

      </div>

      
  <time class="mt-12 mb-8 block text-xs text-gray-500 ltr:text-right rtl:text-left dark:text-gray-400" datetime="2025-03-24T00:00:00.000Z">
    <span>Last updated on</span>
    Mar 24, 2025</time>

      <div class="container mx-auto prose prose-slate lg:prose-xl dark:prose-invert mt-5">
        
        <div class="max-w-prose print:hidden">
  
  

  

<div class="flex justify-center">
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/erv/">ERV</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/gtf/">Gtf</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/grcm39/">GRCm39</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/repeatmasker/">RepeatMasker</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/parallel-computing/">Parallel Computing</a>
  
</div>


  
<section class="flex flex-row flex-wrap justify-center pt-4 text-xl">
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F01a.repeatmasker%2F&amp;text=%F0%9F%9B%A0%EF%B8%8F&#43;Generate&#43;and&#43;validate&#43;ERV&#43;gtf&#43;that&#43;matches&#43;the&#43;latest&#43;reference&#43;genome"
    title="X"
    aria-label="X"
    id="share-link-x"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://www.facebook.com/sharer.php?u=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F01a.repeatmasker%2F&amp;t=%F0%9F%9B%A0%EF%B8%8F&#43;Generate&#43;and&#43;validate&#43;ERV&#43;gtf&#43;that&#43;matches&#43;the&#43;latest&#43;reference&#43;genome"
    title="Facebook"
    aria-label="Facebook"
    id="share-link-facebook"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="currentColor" d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12c0 4.84 3.44 8.87 8 9.8V15H8v-3h2V9.5C10 7.57 11.57 6 13.5 6H16v3h-2c-.55 0-1 .45-1 1v2h3v3h-3v6.95c5.05-.5 9-4.76 9-9.95z"/></svg>
  </a>
  

  
  
  
  
  
  
    
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="mailto:?subject=%F0%9F%9B%A0%EF%B8%8F%20Generate%20and%20validate%20ERV%20gtf%20that%20matches%20the%20latest%20reference%20genome&amp;body=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F01a.repeatmasker%2F"
    title="Email"
    aria-label="Email"
    id="share-link-email"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-width="1.5" d="M16.5 12a4.5 4.5 0 1 1-9 0a4.5 4.5 0 0 1 9 0Zm0 0c0 1.657 1.007 3 2.25 3S21 13.657 21 12a9 9 0 1 0-2.636 6.364M16.5 12V8.25"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://www.linkedin.com/shareArticle?url=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F01a.repeatmasker%2F&amp;title=%F0%9F%9B%A0%EF%B8%8F&#43;Generate&#43;and&#43;validate&#43;ERV&#43;gtf&#43;that&#43;matches&#43;the&#43;latest&#43;reference&#43;genome"
    title="LinkedIn"
    aria-label="LinkedIn"
    id="share-link-linkedin"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" height="1em" viewBox="0 0 448 512"><path fill="currentColor" d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="whatsapp://send?text=%F0%9F%9B%A0%EF%B8%8F&#43;Generate&#43;and&#43;validate&#43;ERV&#43;gtf&#43;that&#43;matches&#43;the&#43;latest&#43;reference&#43;genome%20http%3A%2F%2Flocalhost%3A1313%2Fpost%2F01a.repeatmasker%2F"
    title="WhatsApp"
    aria-label="WhatsApp"
    id="share-link-whatsapp"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256" fill="currentColor"><path d="m187.58 144.84-32-16a8 8 0 0 0-8 .5l-14.69 9.8a40.55 40.55 0 0 1-16-16l9.8-14.69a8 8 0 0 0 .5-8l-16-32A8 8 0 0 0 104 64a40 40 0 0 0-40 40 88.1 88.1 0 0 0 88 88 40 40 0 0 0 40-40 8 8 0 0 0-4.42-7.16ZM152 176a72.08 72.08 0 0 1-72-72 24 24 0 0 1 19.29-23.54l11.48 23L101 118a8 8 0 0 0-.73 7.51 56.47 56.47 0 0 0 30.15 30.15A8 8 0 0 0 138 155l14.61-9.74 23 11.48A24 24 0 0 1 152 176ZM128 24a104 104 0 0 0-91.82 152.88l-11.35 34.05a16 16 0 0 0 20.24 20.24l34.05-11.35A104 104 0 1 0 128 24Zm0 192a87.87 87.87 0 0 1-44.06-11.81 8 8 0 0 0-6.54-.67L40 216l12.47-37.4a8 8 0 0 0-.66-6.54A88 88 0 1 1 128 216Z"/></svg>
  </a>
  
</section>


  








  
  
    



  
  
  
    
  
  
  

<div class="flex pt-12 pb-4">
  
  
  <img
    class="mr-4 h-24 w-24 rounded-full"
    width="96"
    height="96"
    alt="Jiyuan Liu"
  src="/author/jiyuan-liu/avatar_hu_6297b4a1a274518.jpg"
  loading="lazy"
  />
  
  <div class="place-self-center">
    <div class="text-[0.6rem] uppercase leading-3 text-neutral-500 dark:text-neutral-400">
      Authors
    </div>
    <div class="font-semibold leading-6 text-neutral-800 dark:text-neutral-300">
      <a href="http://localhost:1313/" class="no-underline">
      Jiyuan Liu
      </a>
    </div>

    
    <div class="text-sm font-bold text-neutral-700 dark:text-neutral-300">
    Postdoctoral Scientist
    </div>
    


    

    <div class="text-2xl sm:text-lg pt-1">

      
<div class="flex flex-wrap text-neutral-500 dark:text-neutral-300">
  
    
    
    
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="mailto:jliu78@mgh.harvard.edu"
      
      aria-label="At-Symbol"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-width="1.5" d="M16.5 12a4.5 4.5 0 1 1-9 0a4.5 4.5 0 0 1 9 0Zm0 0c0 1.657 1.007 3 2.25 3S21 13.657 21 12a9 9 0 1 0-2.636 6.364M16.5 12V8.25"/></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://github.com/jliu678"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Brands/Github"
    ><svg style="height: 1em;" fill="currentColor" viewBox="3 3 18 18"><path d="M12 3C7.0275 3 3 7.12937 3 12.2276C3 16.3109 5.57625 19.7597 9.15374 20.9824C9.60374 21.0631 9.77249 20.7863 9.77249 20.5441C9.77249 20.3249 9.76125 19.5982 9.76125 18.8254C7.5 19.2522 6.915 18.2602 6.735 17.7412C6.63375 17.4759 6.19499 16.6569 5.8125 16.4378C5.4975 16.2647 5.0475 15.838 5.80124 15.8264C6.51 15.8149 7.01625 16.4954 7.18499 16.7723C7.99499 18.1679 9.28875 17.7758 9.80625 17.5335C9.885 16.9337 10.1212 16.53 10.38 16.2993C8.3775 16.0687 6.285 15.2728 6.285 11.7432C6.285 10.7397 6.63375 9.9092 7.20749 9.26326C7.1175 9.03257 6.8025 8.08674 7.2975 6.81794C7.2975 6.81794 8.05125 6.57571 9.77249 7.76377C10.4925 7.55615 11.2575 7.45234 12.0225 7.45234C12.7875 7.45234 13.5525 7.55615 14.2725 7.76377C15.9937 6.56418 16.7475 6.81794 16.7475 6.81794C17.2424 8.08674 16.9275 9.03257 16.8375 9.26326C17.4113 9.9092 17.76 10.7281 17.76 11.7432C17.76 15.2843 15.6563 16.0687 13.6537 16.2993C13.98 16.5877 14.2613 17.1414 14.2613 18.0065C14.2613 19.2407 14.25 20.2326 14.25 20.5441C14.25 20.7863 14.4188 21.0746 14.8688 20.9824C16.6554 20.364 18.2079 19.1866 19.3078 17.6162C20.4077 16.0457 20.9995 14.1611 21 12.2276C21 7.12937 16.9725 3 12 3Z"></path></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://www.linkedin.com/in/jiyuan-liu-7608a42a6/"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Brands/Linkedin"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" height="1em" viewBox="0 0 448 512"><path fill="currentColor" d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://scholar.google.com/citations?view_op=list_works&amp;hl=en&amp;hl=en&amp;user=5c1kGO4AAAAJ&amp;sortby=pubdate"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Academicons/Google-Scholar"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M343.759 106.662V79.43L363.524 64h-213.89L20.476 176.274h85.656a82.339 82.339 0 0 0-.219 6.225c0 20.845 7.22 38.087 21.672 51.861c14.453 13.797 32.252 20.648 53.327 20.648c4.923 0 9.75-.368 14.438-1.024c-2.907 6.5-4.374 12.523-4.374 18.142c0 9.875 4.499 20.43 13.467 31.642c-39.234 2.67-68.061 9.732-86.437 21.163c-10.531 6.5-19 14.704-25.39 24.531c-6.391 9.9-9.578 20.515-9.578 31.962c0 9.648 2.062 18.336 6.219 26.062c4.156 7.726 9.578 14.07 16.312 18.984c6.718 4.968 14.469 9.101 23.219 12.469c8.734 3.344 17.406 5.718 26.061 7.062A167.052 167.052 0 0 0 180.555 448c13.469 0 26.953-1.734 40.547-5.187c13.562-3.485 26.28-8.642 38.171-15.493c11.86-6.805 21.515-16.086 28.922-27.718c7.39-11.68 11.094-24.805 11.094-39.336c0-11.016-2.25-21.039-6.75-30.14c-4.468-9.073-9.938-16.542-16.452-22.345c-6.501-5.813-13-11.155-19.516-15.968c-6.5-4.845-12-9.75-16.468-14.813c-4.485-5.046-6.735-10.054-6.735-14.984c0-4.921 1.734-9.672 5.216-14.265c3.455-4.61 7.674-9.048 12.61-13.306c4.937-4.25 9.875-8.968 14.796-14.133c4.922-5.147 9.141-11.827 12.61-20.008c3.485-8.18 5.203-17.445 5.203-27.757c0-13.453-2.547-24.46-7.547-33.314c-.594-1.022-1.218-1.803-1.875-3.022l56.907-46.672v17.119c-7.393.93-6.624 5.345-6.624 10.635V245.96c0 5.958 4.875 10.834 10.834 10.834h3.989c5.958 0 10.833-4.875 10.833-10.834V117.293c0-5.277.778-9.688-6.561-10.63zm-107.36 222.48c1.14.75 3.704 2.78 7.718 6.038c4.05 3.243 6.797 5.695 8.266 7.414a443.553 443.553 0 0 1 6.376 7.547c2.813 3.375 4.718 6.304 5.718 8.734c1 2.477 2.016 5.461 3.047 8.946a38.27 38.27 0 0 1 1.485 10.562c0 17.048-6.564 29.68-19.656 37.859c-13.125 8.18-28.767 12.274-46.938 12.274c-9.187 0-18.203-1.093-27.063-3.196c-8.843-2.116-17.311-5.336-25.39-9.601c-8.078-4.258-14.577-10.204-19.5-17.797c-4.938-7.64-7.407-16.415-7.407-26.25c0-10.32 2.797-19.29 8.422-26.906c5.594-7.625 12.938-13.391 22.032-17.315c9.063-3.946 18.25-6.742 27.562-8.398a157.865 157.865 0 0 1 28.438-2.555c4.47 0 7.936.25 10.405.696c.455.219 3.032 2.07 7.735 5.563c4.704 3.462 7.625 5.595 8.75 6.384zm-3.359-100.579c-7.406 8.86-17.734 13.288-30.953 13.288c-11.86 0-22.298-4.764-31.266-14.312c-9-9.523-15.422-20.328-19.344-32.43c-3.937-12.109-5.906-23.984-5.906-35.648c0-13.694 3.596-25.352 10.781-34.976c7.187-9.65 17.5-14.485 30.938-14.485c11.875 0 22.374 5.038 31.437 15.157c9.094 10.085 15.61 21.413 19.517 33.968c3.922 12.54 5.873 24.53 5.873 35.984c0 13.446-3.702 24.61-11.076 33.454z"/></svg></a>
  
</div>



    </div>
  </div>
</div>



  




  
  
    
    
    
      
      
    
<div class="pt-1 no-prose w-full">
  <hr class="border-dotted border-neutral-300 dark:border-neutral-600" />
  <div class="flex flex-col md:flex-row flex-nowrap justify-between gap-5 pt-2">
    <div class="">
      
        <a class="group flex no-underline" href="/post/02.snakemake_quantify-herv_10xscrnaseq/">
          <span
            class="mt-[-0.3rem] me-2 text-neutral-700 transition-transform group-hover:-translate-x-[2px] group-hover:text-primary-600 dark:text-neutral dark:group-hover:text-primary-400"
          ><span class="ltr:inline rtl:hidden">&larr;</span></span>
          <span class="flex flex-col">
            <span
              class="mt-[0.1rem] leading-6 group-hover:underline group-hover:decoration-primary-500"
            >üß¨ A snakemake pipline-- quantify hERV, transgenes and genome from 10x scRNAseq</span>
            <span class="mt-[0.1rem] text-xs text-neutral-500 dark:text-neutral-400">
              
                Mar 25, 2025
              
            </span>
          </span>
        </a>
      
    </div>
    <div class="">
      
        <a class="group flex text-right no-underline" href="/post/01.bulkrnaseq_allow_herv_quantification/">
          <span class="flex flex-col">
            <span
              class="mt-[0.1rem] leading-6 group-hover:underline group-hover:decoration-primary-500"
            >üß¨ Quantify hERV and transgenes from Bulk-RNAseq</span
            >
            <span class="mt-[0.1rem] text-xs text-neutral-500 dark:text-neutral-400">
              
                Mar 24, 2025
              
            </span>
          </span>
          <span
            class="mt-[-0.3rem] ms-2 text-neutral-700 transition-transform group-hover:-translate-x-[2px] group-hover:text-primary-600 dark:text-neutral dark:group-hover:text-primary-400"
          ><span class="ltr:inline">&rarr;</span></span>
        </a>
      
    </div>
  </div>
</div>



  


  



</div>

      </div>

    </main>
  </article>
</div>

    </div>
    <div class="page-footer">
      <footer class="container mx-auto flex flex-col justify-items-center text-sm leading-6 mt-24 mb-4 text-slate-700 dark:text-slate-200">

  












  
  
  
  
  














  
  
  

  
  
    
  
  
    
  

  

  
  <p class="powered-by text-center">
    ¬© 2025 Me. This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-nd/4.0" rel="noopener noreferrer" target="_blank">CC BY NC ND 4.0</a>
  </p>
  

  <p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-nd/4.0" rel="noopener noreferrer" target="_blank" aria-label="Creative Commons">
      <i class="fab fa-creative-commons fa-2x" aria-hidden="true"></i>
      <i class="fab fa-creative-commons-by fa-2x" aria-hidden="true"></i>
      
        <i class="fab fa-creative-commons-nc fa-2x" aria-hidden="true"></i>
      
      
        <i class="fab fa-creative-commons-nd fa-2x" aria-hidden="true"></i>
      
    </a>
  </p>





  <p class="powered-by text-center">
    
    
    
      
      
      
      
      
      
      Published with <a href="https://hugoblox.com/?utm_campaign=poweredby" target="_blank" rel="noopener">Hugo Blox Builder</a> ‚Äî the free, <a href="https://github.com/HugoBlox/hugo-blox-builder" target="_blank" rel="noopener">open source</a> website builder that empowers creators.
    
  </p>

</footer>

    </div>

    
    











  </body>
</html>
