<!doctype html>
<!-- This site was created with Hugo Blox. https://hugoblox.com -->
<!-- Last Published: July 7, 2025 --><html lang="en-us" dir="ltr"
      data-wc-theme-default="system">
  
<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <meta name="generator" content="Hugo Blox Builder 0.3.1" />

  
  












  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Jiyuan Liu" />

  
  
  
    
  
  <meta name="description" content="Here derives the mathematics underpinning the parameter inference of dynamic RNA velocity model, which is the second installment of our blog series to effectively apply the dynamic model in revealing the RNA velocity of single-cell RNAseq." />

  
  <link rel="alternate" hreflang="en-us" href="http://localhost:1313/post/11f.velocity_dynamic_model_posthoc_calculation/scvelo_post-fitting_calculation/" />

  
  
  
  
    
    <link rel="stylesheet" href="/css/themes/emerald.min.css" />
  

  
  
    
    <link href="/dist/wc.min.css" rel="stylesheet" />
  

  
  
  

  
    
    <link href="/css/custom.min.9a8c1508a999d285edca71446056519c3773544bdbe9fd11396354c75238a016.css" rel="stylesheet" />
  

  <script>
     
    window.hbb = {
       defaultTheme: document.documentElement.dataset.wcThemeDefault,
       setDarkTheme: () => {
        document.documentElement.classList.add("dark");
        document.documentElement.style.colorScheme = "dark";
      },
       setLightTheme: () => {
        document.documentElement.classList.remove("dark");
        document.documentElement.style.colorScheme = "light";
      }
    }

    console.debug(`Default Hugo Blox Builder theme is ${window.hbb.defaultTheme}`);

    if ("wc-color-theme" in localStorage) {
      localStorage.getItem("wc-color-theme") === "dark" ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
    } else {
      window.hbb.defaultTheme === "dark" ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
      if (window.hbb.defaultTheme === "system") {
        window.matchMedia("(prefers-color-scheme: dark)").matches ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
      }
    }
  </script>

  <script>
    
    document.addEventListener('DOMContentLoaded', function () {
      
      let checkboxes = document.querySelectorAll('li input[type=\'checkbox\'][disabled]');
      checkboxes.forEach(e => {
        e.parentElement.parentElement.classList.add('task-list');
      });

      
      const liNodes = document.querySelectorAll('.task-list li');
      liNodes.forEach(nodes => {
        let textNodes = Array.from(nodes.childNodes).filter(node => node.nodeType === 3 && node.textContent.trim().length > 1);
        if (textNodes.length > 0) {
          const span = document.createElement('label');
          textNodes[0].after(span);  
          span.appendChild(nodes.querySelector('input[type=\'checkbox\']'));
          span.appendChild(textNodes[0]);
        }
      });
    });
  </script>

  
  
  




































  
  

  
  <link rel="icon" type="image/png" href="/media/icon_hu_c54cc46c793e1f2b.png" />
  <link rel="apple-touch-icon" type="image/png" href="/media/icon_hu_68ea7b4ed3f6b794.png" />

  <link rel="canonical" href="http://localhost:1313/post/11f.velocity_dynamic_model_posthoc_calculation/scvelo_post-fitting_calculation/" />

  
  
  
  
  
  
  
  
    
    
  
  

  
  
    
    
  
  <meta property="twitter:card" content="summary" />
  
    <meta property="twitter:site" content="@GetResearchDev" />
    <meta property="twitter:creator" content="@GetResearchDev" />
  
  <meta property="og:site_name" content="Jiyuan Liu" />
  <meta property="og:url" content="http://localhost:1313/post/11f.velocity_dynamic_model_posthoc_calculation/scvelo_post-fitting_calculation/" />
  <meta property="og:title" content="üß¨ Dynamic RNA velocity model-- (3) post hoc computation | Jiyuan Liu" />
  <meta property="og:description" content="Here derives the mathematics underpinning the parameter inference of dynamic RNA velocity model, which is the second installment of our blog series to effectively apply the dynamic model in revealing the RNA velocity of single-cell RNAseq." /><meta property="og:image" content="http://localhost:1313/media/icon_hu_fd2be5a69becaf9e.png" />
    <meta property="twitter:image" content="http://localhost:1313/media/icon_hu_fd2be5a69becaf9e.png" /><meta property="og:locale" content="en-us" />
  
    
      <meta
        property="article:published_time"
        content="2025-05-28T00:00:00&#43;00:00"
      />
    
    <meta property="article:modified_time" content="2025-05-28T00:00:00&#43;00:00">
  

  



  


  <title>üß¨ Dynamic RNA velocity model-- (3) post hoc computation | Jiyuan Liu</title>

  
  
  
  
  
    
    
  
  
  <style>
    @font-face {
      font-family: 'Inter var';
      font-style: normal;
      font-weight: 100 900;
      font-display: swap;
      src: url(/dist/font/Inter.var.woff2) format(woff2);
    }
  </style>

  

  
  


  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
    
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  







<link type="text/css" rel="stylesheet" href="/dist/pagefind/pagefind-ui.be766eb419317a14ec769d216e9779bfe8f3737c80e780f4ba0dafb57a41a482.css" integrity="sha256-vnZutBkxehTsdp0hbpd5v&#43;jzc3yA54D0ug2vtXpBpII=" />


<script src="/dist/pagefind/pagefind-ui.87693d7c6f2b3b347ce359d0ede762c033419f0a32b22ce508c335a81d841f1b.js" integrity="sha256-h2k9fG8rOzR841nQ7ediwDNBnwoysizlCMM1qB2EHxs="></script>


<script>window.hbb.pagefind = {"baseUrl":"/"};</script>

<style>
  html.dark {
    --pagefind-ui-primary: #eeeeee;
    --pagefind-ui-text: #eeeeee;
    --pagefind-ui-background: #152028;
    --pagefind-ui-border: #152028;
    --pagefind-ui-tag: #152028;
  }
</style>

<script>
  window.addEventListener('DOMContentLoaded', (event) => {
    new PagefindUI({
      element: "#search",
      showSubResults: true,
      baseUrl: window.hbb.pagefind.baseUrl,
      bundlePath: window.hbb.pagefind.baseUrl + "pagefind/",
    });
  });
  document.addEventListener('DOMContentLoaded', () => {
    let element = document.getElementById('search');
    let trigger = document.getElementById('search_toggle');

    if (trigger) {
      trigger.addEventListener('click', () => {
        element.classList.toggle('hidden');
        element.querySelector("input").value = ""
        element.querySelector("input").focus()

        if (!element.classList.contains('hidden')) {
          let clear_trigger = document.querySelector('.pagefind-ui__search-clear');

          if (clear_trigger && !clear_trigger.hasAttribute('listenerOnClick')) {
            clear_trigger.setAttribute('listenerOnClick', 'true');

            clear_trigger.addEventListener('click', () => {
              element.classList.toggle('hidden');
            });
          }
        }

      });
    }
  });
</script>










  
  
  <link type="text/css" rel="stylesheet" href="/dist/lib/katex/katex.min.505d5f829022bb7b4f24dfee0aa1141cd7bba67afe411d1240335f820960b5c3.css" integrity="sha256-UF1fgpAiu3tPJN/uCqEUHNe7pnr&#43;QR0SQDNfgglgtcM=" />
  
  
  <script defer src="/dist/lib/katex/katex.min.dc84b296ec3e884de093158f760fd9d45b6c7abe58b5381557f4e138f46a58ae.js" integrity="sha256-3ISyluw&#43;iE3gkxWPdg/Z1Ftser5YtTgVV/ThOPRqWK4="></script>
  
  
  
  
  <script defer src="/js/katex-renderer.6579ec9683211cfb952064aedf3a3baea5eeb17a061775b32b70917474637c80.js" integrity="sha256-ZXnsloMhHPuVIGSu3zo7rqXusXoGF3WzK3CRdHRjfIA="></script>
  
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  






  
  
  
  
  
  
  
  <script
    defer
    src="/js/hugo-blox-en.min.js"
    integrity=""
  ></script>

  
  








  
    
      
      <script async defer src="https://buttons.github.io/buttons.js"></script>

      
    
  




</head>

  <body class="dark:bg-hb-dark dark:text-white page-wrapper" id="top">
    <div id="page-bg"></div>
    <div class="page-header sticky top-0 z-30">
      
      
      
        
        
        
          <header id="site-header" class="header">
  <nav class="navbar px-3 flex justify-left">
    <div class="order-0 h-100">
      
      <a class="navbar-brand" href="/" title="Jiyuan Liu">
        Home
      </a>
    </div>
    
    <input id="nav-toggle" type="checkbox" class="hidden" />
    <label
      for="nav-toggle"
      class="order-3 cursor-pointer flex items-center lg:hidden text-dark dark:text-white lg:order-1">
      <svg id="show-button" class="h-6 fill-current block" viewBox="0 0 20 20">
        <title>Open Menu</title>
        <path d="M0 3h20v2H0V3z m0 6h20v2H0V9z m0 6h20v2H0V0z"></path>
      </svg>
      <svg id="hide-button" class="h-6 fill-current hidden" viewBox="0 0 20 20">
        <title>Close Menu</title>
        <polygon
          points="11 9 22 9 22 11 11 11 11 22 9 22 9 11 -2 11 -2 9 9 9 9 -2 11 -2"
          transform="rotate(45 10 10)"></polygon>
      </svg>
    </label>
    

    
    
    <ul
      id="nav-menu"
      class="navbar-nav order-3 hidden lg:flex w-full pb-6 lg:order-1 lg:w-auto lg:space-x-2 lg:pb-0 xl:space-x-8 justify-left
      ">
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/"
        >Bio</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/#papers"
        >Papers</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/#talks"
        >Talks</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/post"
        >Blogs</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/experience/"
        >Experience</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/projects/"
        >Projects</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/teaching/"
        >Learning</a
        >
      </li>
      
      
      
    </ul>

    <div class="order-1 ml-auto flex items-center md:order-2 lg:ml-0">

      
      
      
      <button
        aria-label="search"
        class="text-black hover:text-primary  inline-block px-3 text-xl dark:text-white"
        id="search_toggle">
        <svg xmlns="http://www.w3.org/2000/svg" height="16" width="16" viewBox="0 0 512 512" fill="currentColor"><path d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"/></svg>
      </button>
      

      
      
      <div class="px-3 text-black hover:text-primary-700 dark:text-white dark:hover:text-primary-300
            [&.active]:font-bold [&.active]:text-black/90 dark:[&.active]:text-white">
        <button class="theme-toggle mt-1" accesskey="t" title="appearance">
          <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
               fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
               stroke-linejoin="round" class="dark:hidden">
            <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
          </svg>
          <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
               fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
               stroke-linejoin="round" class=" dark:block [&:not(dark)]:hidden">
            <circle cx="12" cy="12" r="5"></circle>
            <line x1="12" y1="1" x2="12" y2="3"></line>
            <line x1="12" y1="21" x2="12" y2="23"></line>
            <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
            <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
            <line x1="1" y1="12" x2="3" y2="12"></line>
            <line x1="21" y1="12" x2="23" y2="12"></line>
            <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
            <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
          </svg>
        </button>
      </div>
      

      
      

      
      
    </div>
  </nav>
</header>


<div id="search" class="hidden p-3"></div>


        
      
    </div>
    <div class="page-body  my-10">
      





<div class="mx-auto flex max-w-screen-xl">
  



<aside class="hb-sidebar-container max-lg:[transform:translate3d(0,-100%,0)] lg:hidden xl:block">
  
  <div class="px-4 pt-4 lg:hidden">
    
    
  </div>
  <div class="hb-scrollbar lg:h-[calc(100vh-var(--navbar-height))]">
    <ul class="flex flex-col gap-1 lg:hidden">
      
      
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/event/"
    
  >Recent &amp; Upcoming Talks
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/event/example/"
    
  >Example Talk
    </a>
              
            </li></ul>
      </div></li>
        <li class="open"><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/"
    
  >Computational Blogs
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11d.velocity_dynamic_model_derivation/"
    
  >üß¨ Dynamic RNA velocity model-- (1) math solutions
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11e.velocity_dynamic_model_inference/"
    
  >üß¨ Dynamic RNA velocity model-- (2) parameter inference
    </a>
              
            </li><li class="flex flex-col open"><a
    class="hb-sidebar-custom-link
      sidebar-active-item bg-primary-100 font-semibold text-primary-800 dark:bg-primary-300 dark:text-primary-900"
    href="/post/11f.velocity_dynamic_model_posthoc_calculation/scvelo_post-fitting_calculation/"
    
  >üß¨ Dynamic RNA velocity model-- (3) post hoc computation
    </a>
  
    <ul class="hb-sidebar-mobile-toc"><li>
              <a
                href="#"
                class="hb-docs-link"
              ></a>
            </li>
          <li>
              <a
                href="#1-standard-cosine-similarity"
                class="hb-docs-link"
              >&lt;strong&gt;1. Standard Cosine Similarity&lt;/strong&gt;</a>
            </li>
          <li>
              <a
                href="#2-variance-stabilized-transformation"
                class="hb-docs-link"
              >&lt;strong&gt;2. Variance-Stabilized Transformation&lt;/strong&gt;</a>
            </li>
          <li>
              <a
                href="#"
                class="hb-docs-link"
              ></a>
            </li>
          <li>
              <a
                href="#1-what-are-root-cells-in-differentiation"
                class="hb-docs-link"
              >&lt;strong&gt;1. What Are Root Cells in Differentiation?&lt;/strong&gt;</a>
            </li>
          <li>
              <a
                href="#2-stationary-state-equation-in-rna-velocity"
                class="hb-docs-link"
              >&lt;strong&gt;2. Stationary State Equation in RNA Velocity&lt;/strong&gt;</a>
            </li>
          <li>
              <a
                href="#3-why-does-this-identify-root-cells"
                class="hb-docs-link"
              >&lt;strong&gt;3. Why Does This Identify Root Cells?&lt;/strong&gt;</a>
            </li>
          <li>
              <a
                href="#e-example-computing-stationary-states-using-eigenvectors"
                class="hb-docs-link"
              >&lt;strong&gt;e. Example: Computing Stationary States Using Eigenvectors&lt;/strong&gt;</a>
            </li>
          <li>
              <a
                href="#is-p-gene-specific"
                class="hb-docs-link"
              >Is p Gene-Specific?</a>
            </li>
          <li>
              <a
                href="#is-p-cell-specific"
                class="hb-docs-link"
              >Is p Cell-Specific?</a>
            </li>
          <li>
              <a
                href="#is-p-root-specific"
                class="hb-docs-link"
              >Is p Root-Specific?</a>
            </li>
          <li>
              <a
                href="#summary-table"
                class="hb-docs-link"
              >Summary Table</a>
            </li>
          <li>
              <a
                href="#optimization-step"
                class="hb-docs-link"
              >Optimization Step</a>
            </li>
          <li>
              <a
                href="#-key-variables"
                class="hb-docs-link"
              >üìê Key Variables</a>
            </li>
          <li>
              <a
                href="#-formula-projected-velocity-vector"
                class="hb-docs-link"
              >üìä Formula: Projected Velocity Vector</a>
            </li>
          <li>
              <a
                href="#-intuition"
                class="hb-docs-link"
              >üß† Intuition</a>
            </li>
          <li>
              <a
                href="#-in-practice"
                class="hb-docs-link"
              >üñºÔ∏è In Practice</a>
            </li>
          <li>
              <a
                href="#-summary"
                class="hb-docs-link"
              >‚úÖ Summary</a>
            </li>
          <li>
              <a
                href="#mathematical-breakdown"
                class="hb-docs-link"
              >Mathematical Breakdown</a>
            </li>
          <li>
              <a
                href="#alternative-formulations"
                class="hb-docs-link"
              >Alternative Formulations</a>
            </li>
          <li>
              <a
                href="#implementation-notes"
                class="hb-docs-link"
              >Implementation Notes</a>
            </li>
          <li>
              <a
                href="#mathematical-definition"
                class="hb-docs-link"
              >Mathematical Definition</a>
            </li>
          <li>
              <a
                href="#what-the-reconstructability-score-tells-you"
                class="hb-docs-link"
              >What the Reconstructability Score Tells You</a>
            </li>
          <li>
              <a
                href="#mathematical-components-breakdown"
                class="hb-docs-link"
              >Mathematical Components Breakdown</a>
            </li>
          <li>
              <a
                href="#use-cases-and-significance"
                class="hb-docs-link"
              >Use Cases and Significance</a>
            </li>
          <li>
              <a
                href="#mathematical-framework"
                class="hb-docs-link"
              >Mathematical Framework</a>
            </li>
          <li>
              <a
                href="#core-concepts"
                class="hb-docs-link"
              >Core Concepts</a>
            </li>
          <li>
              <a
                href="#biological-interpretation-1"
                class="hb-docs-link"
              >Biological Interpretation</a>
            </li>
          <li>
              <a
                href="#numerical-example"
                class="hb-docs-link"
              >Numerical Example</a>
            </li>
          <li>
              <a
                href="#reverse-transport-example"
                class="hb-docs-link"
              >Reverse Transport Example</a>
            </li>
          <li>
              <a
                href="#step-by-step-explanation"
                class="hb-docs-link"
              >Step-by-step explanation</a>
            </li>
          <li>
              <a
                href="#-setup"
                class="hb-docs-link"
              >üß™ Setup</a>
            </li>
          <li>
              <a
                href="#-step-1-define-neighbors"
                class="hb-docs-link"
              >üëØ Step 1: Define neighbors</a>
            </li>
          <li>
              <a
                href="#-step-2-compute-convolved-smoothed-latent-times"
                class="hb-docs-link"
              >üîÅ Step 2: Compute convolved (smoothed) latent times</a>
            </li>
          <li>
              <a
                href="#-step-3-compare-original-vs-convolved-times"
                class="hb-docs-link"
              >üìâ Step 3: Compare original vs convolved times</a>
            </li>
          <li>
              <a
                href="#-step-4-detect-inconsistent-cells"
                class="hb-docs-link"
              >üßΩ Step 4: Detect inconsistent cells</a>
            </li>
          <li>
              <a
                href="#-step-5-replace-inconsistent-values"
                class="hb-docs-link"
              >üîÅ Step 5: Replace inconsistent values</a>
            </li>
          <li>
              <a
                href="#-final-corrected-latent-time"
                class="hb-docs-link"
              >‚úÖ Final corrected latent time:</a>
            </li>
          </ul>
  
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11c.velocity_steady_state/"
    
  >üß¨ Math derivation for steady-state RNA velocity model
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19c.bayes_variational_infer_elbo/"
    
  >üéì Math Derivation for ELBO/KL in Bayesian Inference and VAEs
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19b.scrnaseq_impute_saver/"
    
  >üßÆ Math Derivation for scRNAseq Imputation (SAVER)
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19a.conjugate_prior_bayes/"
    
  >üìä Math Derivation in Bayesian Inference-- Bypassing Marginals Using Conjugate Priors
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19.ememwithmapvariationalbayesian/"
    
  >üß¨ EM algorithm, EM with max a posterior, full Bayesian inferences (variational Bayesian) which might be used in auto-coder
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/18.what_make_neural_network_powerful_popular/"
    
  >üß¨ what makes neural network so powerful and popular
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/17.smooth.tests.of.goodness.of.fit/"
    
  >üß¨ Smooth tests of goodness of Fit used in Stijn Hawinkel ‚Äôs plos one paper
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/16.how.good.is.negative.binominal/"
    
  >üß¨ how good is negative binomial, why AI works for biology
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/15.use.seurat.for.bulk.rnaseq/"
    
  >üß¨ Seurat to analysis large sample size of bulk RNAseq. Discuss Wixcox is better for clinical samples
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/13z.scvi_integrate_scrna_spatial/"
    
  >üß¨ Deep Neural Network to integrate scRNAseq and spatial genomics
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/14.spatial_genomics_presentation/"
    
  >üß¨ Spatial genomics presentation
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/12.basian_abundance_statistics/"
    
  >üß¨ Bayesian abundance statistics in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/13.yolov8/"
    
  >üß¨ Yolov8
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11a.velocity_unraveled/"
    
  >üß¨ Math Derivation of CME-defined Stochastic Model of RNA Velocity
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11.velocity_reasoning/"
    
  >üß¨ Velocity reasoning and discussion
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/10.cca_reasoning/"
    
  >üß¨ CCA reasoning and integration discussion
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/09.sctransform/"
    
  >üß¨ sctransfrom reasoning
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/08.clinical_relevance/"
    
  >üß¨ Clinical relevance evaluation in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/07.chord_diagram/"
    
  >üß¨ Expressive cord diagram in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/06.noise_or_true_stem_cell/"
    
  >üß¨ Determine whether it‚Äôs noise or true stem like cells in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/05.xensoft_caki_paper/"
    
  >üß¨ quantify xenograft and host gene expression from the same sample
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/04.huangcong_oncogene/"
    
  >üß¨ counteract treatment-specific background in differential expressed gene analysis (huangcong paper)
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/03.well-based_scrnaseq_herv_quantification/"
    
  >üß¨ A snakemake pipline-- quantify hERV, transgenes and genome from well-based single-cell RNAseq
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/02.snakemake_quantify-herv_10xscrnaseq/"
    
  >üß¨ A snakemake pipline-- quantify hERV, transgenes and genome from 10x scRNAseq
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/01a.repeatmasker/"
    
  >üõ†Ô∏è Generate and validate ERV gtf that matches the latest reference genome
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/01.bulkrnaseq_allow_herv_quantification/"
    
  >üß¨ Quantify hERV and transgenes from Bulk-RNAseq
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/00a.seqwins/"
    
  >‚ö° SeqWins, An R Package for Flexible Base Trimming and Comprehensive FASTQ Analysis on Windows
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/00.model_aerosol_depostion/"
    
  >üí® Modeling the deposition of infectious agents suggests that aerosolized drugs offer enhanced specificity
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.get-started/"
    
  >üéâ Easily create your own simple yet highly customizable blog
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.second-brain/"
    
  >üß† Sharpen your thinking with a second brain
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.data-visualization/"
    
  >üìà Communicate your results effectively with the best data visualizations
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.teach-courses/"
    
  >üë©üèº‚Äçüè´ Teach academic courses
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.project-management/"
    
  >‚úÖ Manage your projects
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_derive_time-varying_alpha_model/"
    
  >velovi_derive_time-varying_alpha_model
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_elbo_kl_breakdown/"
    
  >velovi_elbo_kl_breakdown
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_elbo_term_with_expectation/"
    
  >velovi_elbo_term_with_expectation
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_expected_state_n_latent_time/"
    
  >velovi_expected_state_n_latent_time
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_model_n_inference/"
    
  >velovi_model_n_inference
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_parameter_initialize_n_learn/"
    
  >velovi_parameter_initialize_n_learn
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_posterior_predictive_distribution/"
    
  >velovi_posterior_predictive_distribution
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_reparameterization_trick/"
    
  >velovi_reparameterization_trick
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_uncertainty_calculation/"
    
  >velovi_uncertainty_calculation
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/"
    
  >learning
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/visualize_distribution/"
    
  >Dirichlet Distribution
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/stanford_ml_em/"
    
  >Lecture 14 - Expectation-Maximization Algorithms | Stanford CS229 Machine Learning
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/linear_algebra/"
    
  >linear algebra
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/other/home_improvement/"
    
  >linear algebra
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/method_of_charateristics/"
    
  >Method of characteristics
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/method_of_seperation_of_variables/"
    
  >Separation of Variables
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/js/"
    
  >Learn JavaScript
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/python/"
    
  >Learn Python
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/projects/"
    
  >Projects
    </a></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/"
    
  >Projects
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/pandas/"
    
  >Pandas
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/pytorch/"
    
  >PyTorch
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/scikit/"
    
  >scikit-learn
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/experience/"
    
  >Experience
    </a></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/"
    
  >Publications
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/preprint/"
    
  >An example preprint / working paper
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/journal-article/"
    
  >An example journal article
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/conference-paper/"
    
  >An example conference paper
    </a>
              
            </li></ul>
      </div></li>
    </ul>

    <div class="max-xl:hidden h-0 w-64 shrink-0"></div></div>

</aside>
  

<nav class="hb-toc order-last hidden w-64 shrink-0 xl:block print:hidden px-4" aria-label="table of contents">
  











  <div class="hb-scrollbar text-sm [hyphens:auto] sticky top-16 overflow-y-auto pr-4 pt-6 max-h-[calc(100vh-var(--navbar-height)-env(safe-area-inset-bottom))] -mr-4 rtl:-ml-4"><p class="mb-4 font-semibold tracking-tight">On this page</p><ul>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#example-setup">Example Setup</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#1-standard-cosine-similarity">1. Standard Cosine Similarity</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#2-variance-stabilized-transformation">2. Variance-Stabilized Transformation</a>
      </li></ul><ul>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#1-why-exponential-transformation">1. Why Exponential Transformation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#2-row-normalization-ensures-probabilities-sum-to-1">2. Row Normalization Ensures Probabilities Sum to 1</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#3-role-of-kernel-width-%cf%83">3. Role of Kernel Width œÉ</a>
      </li></ul><ul>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#1-what-are-root-cells-in-differentiation">1. What Are Root Cells in Differentiation?</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#2-stationary-state-equation-in-rna-velocity">2. Stationary State Equation in RNA Velocity</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#3-why-does-this-identify-root-cells">3. Why Does This Identify Root Cells?</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#a-probability-evolution-in-a-markov-process">a. Probability Evolution in a Markov Process</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#b-convergence-to-a-stable-distribution">b. Convergence to a Stable Distribution</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#c-definition-of-the-stationary-distribution">c. Definition of the Stationary Distribution</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#d-why-is-the-stationary-distribution-important">d. Why Is the Stationary Distribution Important?</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#e-example-computing-stationary-states-using-eigenvectors">e. Example: Computing Stationary States Using Eigenvectors</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#step-1-define-the-transition-probability-matrix">Step 1: Define the Transition Probability Matrix</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#step-2-find-the-left-eigenvector-for-eigenvalue-1">Step 2: Find the Left Eigenvector for Eigenvalue 1</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#4-biological-interpretation-in-rna-velocity">4. Biological Interpretation in RNA Velocity</a>
      </li></ul><ul>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#is-p-gene-specific">Is p Gene-Specific?</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#explanation">Explanation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#mathematical-illustration">Mathematical Illustration</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#is-p-cell-specific">Is p Cell-Specific?</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#explanation-1">Explanation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#mathematical-process">Mathematical Process</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#is-p-root-specific">Is p Root-Specific?</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#optimization-process">Optimization Process</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#root-selection">Root Selection</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#summary-table">Summary Table</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#optimization-step">Optimization Step</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#mathematical-formulation">Mathematical Formulation</a>
      </li></ul><ul>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#-key-variables">üìê Key Variables</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#-formula-projected-velocity-vector">üìä Formula: Projected Velocity Vector</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#-intuition">üß† Intuition</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#-in-practice">üñºÔ∏è In Practice</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#-summary">‚úÖ Summary</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#mathematical-breakdown">Mathematical Breakdown</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#step-1-compute-direction-vectors">Step 1: Compute Direction Vectors</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#step-2-get-transition-probabilities">Step 2: Get Transition Probabilities</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#step-3-weighted-average-direction">Step 3: Weighted Average Direction</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#step-4-density-correction">Step 4: Density Correction</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#step-5-final-velocity">Step 5: Final Velocity</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#alternative-formulations">Alternative Formulations</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#kernel-based-approach">Kernel-Based Approach</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#confidence-weighting">Confidence Weighting</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#implementation-notes">Implementation Notes</a>
      </li></ul><ul>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#mathematical-definition">Mathematical Definition</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#what-the-reconstructability-score-tells-you">What the Reconstructability Score Tells You</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#high-score-close-to-1">High Score (close to 1)</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#low-score">Low Score</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#mathematical-components-breakdown">Mathematical Components Breakdown</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#transition-probability-vectors">Transition Probability Vectors</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#correlation-calculation">Correlation Calculation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#median-aggregation">Median Aggregation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#use-cases-and-significance">Use Cases and Significance</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#validation-of-driver-gene-selection">Validation of Driver Gene Selection</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#identifying-essential-gene-sets">Identifying Essential Gene Sets</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#dimensionality-reductionsummarization">Dimensionality Reduction/Summarization</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#biological-interpretation">Biological Interpretation</a>
      </li></ul><ul>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#mathematical-framework">Mathematical Framework</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#original-notation">Original Notation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#core-concepts">Core Concepts</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#1-forward-transport-where-are-cells-going">1. Forward Transport: ‚ÄúWhere are cells going?‚Äù</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#2-reverse-transport-where-did-cells-come-from">2. Reverse Transport: ‚ÄúWhere did cells come from?‚Äù</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#3-indicator-based-cell-selection">3. Indicator-Based Cell Selection</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#biological-interpretation-1">Biological Interpretation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#example-use-cases">Example Use Cases</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#numerical-example">Numerical Example</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#reverse-transport-example">Reverse Transport Example</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#step-1-compute-the-ancestor-distribution">Step 1: Compute the Ancestor Distribution</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#step-2-calculate-the-ancestor-distribution">Step 2: Calculate the Ancestor Distribution</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#step-3-resulting-ancestor-distribution">Step 3: Resulting Ancestor Distribution</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#normalization">Normalization</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#why-perfect-inversion-is-impossible">Why Perfect Inversion is Impossible</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#1-non-invertibility-of-the-transport-map">1. Non-invertibility of the Transport Map</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#2-smoothing-and-mixing">2. Smoothing and Mixing</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#3-loss-of-sharpness">3. Loss of Sharpness</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#4-biological-analogy">4. Biological Analogy</a>
      </li></ul><ul>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#step-by-step-explanation">Step-by-step explanation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#1-latent-time">1. Latent time</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#2-gene-shared-latent-time-course">2. Gene-shared latent time course</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#3-neighborhood-convolution">3. Neighborhood convolution</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#4-regression--consistency-check">4. Regression &amp; consistency check</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#5-correction--replacement">5. Correction / Replacement</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#-setup">üß™ Setup</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#-step-1-define-neighbors">üëØ Step 1: Define neighbors</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#-step-2-compute-convolved-smoothed-latent-times">üîÅ Step 2: Compute convolved (smoothed) latent times</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#-step-3-compare-original-vs-convolved-times">üìâ Step 3: Compare original vs convolved times</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#-step-4-detect-inconsistent-cells">üßΩ Step 4: Detect inconsistent cells</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#-step-5-replace-inconsistent-values">üîÅ Step 5: Replace inconsistent values</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#-final-corrected-latent-time">‚úÖ Final corrected latent time:</a>
      </li></ul>

  
  
    
    
  



    












  </div>
  </nav>


  <article class="w-full break-words flex min-h-[calc(100vh-var(--navbar-height))] min-w-0 justify-center pb-8 pr-[calc(env(safe-area-inset-right)-1.5rem)]">
    <main class="w-full min-w-0 max-w-6xl px-6 pt-4 md:px-12">

      

      <h1 class="mt-2 text-4xl font-bold tracking-tight text-slate-900 dark:text-slate-100">üß¨ Dynamic RNA velocity model-- (3) post hoc computation</h1>

      <div class="mt-4 mb-16">
      <div class="text-gray-500 dark:text-gray-300 text-sm flex items-center flex-wrap gap-y-2"><span class="mr-1">May 28, 2025</span><span class="mx-1">¬∑</span>
        
          
          
          <div class="group inline-flex items-center text-current gap-x-1.5 mx-1">
            
            
            <img src="/author/jiyuan-liu/avatar_hu_98ce379ab30758a2.webp" alt="Jiyuan Liu" class="inline-block h-4 w-4 rounded-full border border-current" loading="lazy" />
            
            <div >Jiyuan Liu</div>
          </div>
          
        

        
        <span class="mx-1">¬∑</span>
        <span class="mx-1">
          21 min read
        </span>
        
        </div>

        <div class="mt-3">
          





        </div>
      </div>



      
      
      

      
      

      <div class="prose prose-slate lg:prose-xl dark:prose-invert">
        <h1 id="d-cosine-similarity-to-compute-velocity-graph">D) Cosine Similarity to compute velocity graph</h1>
<h3 id="example-setup"><strong>Example Setup</strong></h3>
<p>We have two cells \( i \) and \( j \), with gene expression vectors:</p>
 
$$ s_i = \begin{bmatrix} 2.0 \\ 3.5 \\ 1.0 \end{bmatrix}, \quad s_j = \begin{bmatrix} 3.5 \\ 2.5 \\ 1.5 \end{bmatrix} $$ 


<p>We also have the velocity vector for cell \( i \):</p>
 
$$ v_i = \begin{bmatrix} 1.2 \\ -0.5 \\ 0.3 \end{bmatrix} $$ 


<p>First, we compute the difference between the expression values of the two cells:</p>
 
$$ \delta_{ij} = s_j - s_i = \begin{bmatrix} 3.5 - 2.0 \\ 2.5 - 3.5 \\ 1.5 - 1.0 \end{bmatrix} = \begin{bmatrix} 1.5 \\ -1.0 \\ 0.5 \end{bmatrix} $$ 


<hr>
<h2 id="1-standard-cosine-similarity"><strong>1. Standard Cosine Similarity</strong></h2>
<p>The standard cosine similarity is given by:</p>
 
$$ \pi_{ij} = \cos(\delta_{ij}, v_i) = \frac{\delta_{ij}^T v_i}{\|\delta_{ij}\| \|v_i\|} $$ 


<p><strong>Step 1: Compute the dot product</strong>
 
$$ \delta_{ij}^T v_i = (1.5)(1.2) + (-1.0)(-0.5) + (0.5)(0.3) = 1.8 + 0.5 + 0.15 = 2.45 $$ 

</p>
<p><strong>Step 2: Compute the magnitudes</strong>
 
$$ \|\delta_{ij}\| = \sqrt{(1.5)^2 + (-1.0)^2 + (0.5)^2} = \sqrt{2.25 + 1.00 + 0.25} = \sqrt{3.5} \approx 1.87 $$ 

</p>
 
$$ \|v_i\| = \sqrt{(1.2)^2 + (-0.5)^2 + (0.3)^2} = \sqrt{1.44 + 0.25 + 0.09} = \sqrt{1.78} \approx 1.34 $$ 


<p><strong>Step 3: Compute cosine similarity</strong>
 
$$ \pi_{ij} = \frac{2.45}{(1.87)(1.34)} = \frac{2.45}{2.51} \approx 0.98 $$ 

</p>
<p>The vectors are strongly aligned in direction.</p>
<hr>
<h2 id="2-variance-stabilized-transformation"><strong>2. Variance-Stabilized Transformation</strong></h2>
<p>Instead of using raw values, we apply:</p>
 
$$ \text{sign}(\delta_{ij}) \sqrt{|\delta_{ij}|}, \quad \text{sign}(v_i) \sqrt{|v_i|} $$ 


<p><strong>Step 1: Transform \( \delta_{ij} \)</strong></p>
 
$$ \text{sign}(\delta_{ij}) \sqrt{|\delta_{ij}|} = \begin{bmatrix} \text{sign}(1.5) \sqrt{1.5} \\ \text{sign}(-1.0) \sqrt{1.0} \\ \text{sign}(0.5) \sqrt{0.5} \end{bmatrix} = \begin{bmatrix} 1.22 \\ -1.00 \\ 0.71 \end{bmatrix} $$ 


<p><strong>Step 2: Transform \( v_i \)</strong></p>
 
$$ \text{sign}(v_i) \sqrt{|v_i|} = \begin{bmatrix} \text{sign}(1.2) \sqrt{1.2} \\ \text{sign}(-0.5) \sqrt{0.5} \\ \text{sign}(0.3) \sqrt{0.3} \end{bmatrix} = \begin{bmatrix} 1.10 \\ -0.71 \\ 0.55 \end{bmatrix} $$ 


<p><strong>Step 3: Compute transformed cosine similarity</strong>
 
$$ \pi_{ij}^{\text{stabilized}} = \frac{\left( 1.22, -1.00, 0.71 \right) \cdot \left( 1.10, -0.71, 0.55 \right)}{\| \text{sign}(\delta_{ij}) \sqrt{|\delta_{ij}|}\| \|\text{sign}(v_i) \sqrt{|v_i|}\|} $$ 

</p>
<p><strong>Dot product</strong>
 
$$ 1.22(1.10) + (-1.00)(-0.71) + (0.71)(0.55) = 1.342 + 0.71 + 0.391 = 2.443 $$ 

</p>
<p><strong>Compute magnitudes</strong>
 
$$ \sqrt{(1.22)^2 + (-1.00)^2 + (0.71)^2} = \sqrt{1.488 + 1.000 + 0.504} = \sqrt{2.992} \approx 1.73 $$ 

</p>
 
$$ \sqrt{(1.10)^2 + (-0.71)^2 + (0.55)^2} = \sqrt{1.210 + 0.504 + 0.302} = \sqrt{2.016} \approx 1.42 $$ 


<p><strong>Final computation</strong>
 
$$ \pi_{ij}^{\text{stabilized}} = \frac{2.443}{(1.73)(1.42)} = \frac{2.443}{2.46} \approx 0.99 $$ 

</p>
<hr>
<h1 id="e-exponential-kernel-transformation-of-cosine-correlation">E) Exponential Kernel Transformation of Cosine Correlation</h1>
<p>The raw cosine similarity \( \pi_{ij} \) measures directional alignment between velocity vectors, but to obtain meaningful <strong>transition probabilities</strong>, we apply an <strong>exponential kernel</strong>:</p>
 
$$ \tilde{\pi}_{ij} = \frac{1}{z_i} \exp \left(\frac{\pi_{ij}}{\sigma^2} \right) $$ 


<p>where:</p>
<ul>
<li> $ \pi_{ij} $ 
 is the <strong>cosine similarity</strong> between cell  $ i $ 
 and  $ j $ 
.</li>
<li> $ \sigma $ 
 is the <strong>kernel width parameter</strong>, which controls the spread of probabilities.</li>
<li> $ z_i $ 
 is a <strong>row normalization factor</strong>, ensuring probabilities sum to 1:</li>
</ul>
 
$$ z_i = \sum_j \exp \left(\frac{\pi_{ij}}{\sigma^2} \right) $$ 


<hr>
<h3 id="1-why-exponential-transformation"><strong>1. Why Exponential Transformation</strong></h3>
<p>The <strong>exponential function</strong> ensures:
‚úÖ <strong>Amplification of strong correlations</strong>: Large values of  $ \pi_{ij} $ 
 get boosted.<br>
‚úÖ <strong>Suppression of weak correlations</strong>: Small or negative values decay rapidly, reducing their transition influence.<br>
‚úÖ <strong>Nonlinear scaling</strong>: Instead of treating all similarity scores equally, this transformation <strong>sharpens distinctions</strong> between stronger and weaker connections.</p>
<p>Think of it as a <strong>softmax-like weighting</strong> that enhances <strong>directional flow probability</strong> based on cosine similarity.</p>
<hr>
<h3 id="2-row-normalization-ensures-probabilities-sum-to-1"><strong>2. Row Normalization Ensures Probabilities Sum to 1</strong></h3>
<p>The <strong>normalization factor</strong>  $ z_i $ 
 guarantees valid probability distributions:</p>
 
$$ \sum_j \tilde{\pi}_{ij} = 1 $$ 


<p>Without this step, raw exponentiated values might grow arbitrarily large, leading to <strong>improper probability distributions</strong>.</p>
<hr>
<h3 id="3-role-of-kernel-width-œÉ"><strong>3. Role of Kernel Width œÉ</strong></h3>
<ul>
<li><strong>Large  $ \sigma $ 
</strong> ‚Üí Softens probability differences, making transitions smoother.</li>
<li><strong>Small  $ \sigma $ 
</strong> ‚Üí Sharper transitions, favoring <strong>strong</strong> directional alignments.</li>
</ul>
<p>Choosing an appropriate  $ \sigma $ 
 ensures <strong>biological interpretability</strong> in applications like RNA velocity and cell fate prediction.</p>
<p>Would you like a <strong>numerical example</strong> demonstrating how this transformation works with actual cosine similarity values? üöÄ</p>
<h1 id="f-how-scvelo-identifies-the-starting-points-of-differentiation-using-markov-chain-analysis">F) How scVelo Identifies the Starting Points of Differentiation Using Markov Chain Analysis</h1>
<h2 id="1-what-are-root-cells-in-differentiation"><strong>1. What Are Root Cells in Differentiation?</strong></h2>
<p>In single-cell RNA velocity analysis, root cells are the <strong>initial states</strong> in a developmental trajectory. These are cells that:<br>
‚úÖ Have <strong>low latent time</strong> (early progenitors).<br>
‚úÖ Are <strong>unlikely</strong> to have transitioned from other states.<br>
‚úÖ Serve as the <strong>starting points</strong> for differentiation pathways.</p>
<p>Identifying these root cells is crucial for understanding <strong>how differentiation originates</strong>.</p>
<hr>
<h2 id="2-stationary-state-equation-in-rna-velocity"><strong>2. Stationary State Equation in RNA Velocity</strong></h2>
<p>The root cells are found by solving the <strong>stationary distribution equation</strong> for the <strong>transition probability matrix</strong>  $ \tilde{\pi} $ 
:</p>
 
$$ \mu^* = \mu^* \tilde{\pi}^T $$ 


<p>where:</p>
<ul>
<li> $ \tilde{\pi}^T $ 
 is the <strong>transpose of the transition probability matrix</strong> that encodes RNA velocity-driven transitions.</li>
<li> $ \mu^* $ 
 represents the <strong>stationary distribution</strong>, which describes the <strong>long-term probabilities</strong> of cells staying in each state.</li>
<li>The <strong>left eigenvectors</strong> of  $ \tilde{\pi}^T $ 
 corresponding to the <strong>eigenvalue of 1</strong> define these <strong>stationary states</strong>.</li>
</ul>
<hr>
<h2 id="3-why-does-this-identify-root-cells"><strong>3. Why Does This Identify Root Cells?</strong></h2>
<h3 id="a-probability-evolution-in-a-markov-process"><strong>a. Probability Evolution in a Markov Process</strong></h3>
<p>In a discrete-time <strong>Markov chain</strong>, the state of the system at time  $ t $ 
 is represented by a <strong>probability distribution</strong>  $ \mu_t $ 
, which evolves according to the <strong>transition probability matrix</strong>  $ P $ 
:</p>
 
$$ \mu_{t+1} = \mu_t P $$ 


<p>where:</p>
<ul>
<li> $ \mu_t $ 
 is a row vector of probabilities over all possible states at time  $ t $ 
.</li>
<li> $ P $ 
 is an  $ n \times n $ 
 <strong>transition matrix</strong>, where  $ P_{ij} $ 
 represents the probability of transitioning from state  $ i $ 
 to state  $ j $ 
.</li>
</ul>
<p>Each step updates the probability distribution, progressively altering the likelihood of being in each state.</p>
<hr>
<h3 id="b-convergence-to-a-stable-distribution"><strong>b. Convergence to a Stable Distribution</strong></h3>
<p>As the Markov chain progresses, repeated multiplications by  $ P $ 
 lead the probability distribution toward a <strong>stable equilibrium</strong>:</p>
 
$$ \mu_{t+k} = \mu_t P^k, \quad \text{as } k \to \infty $$ 


<p>After many steps, the probabilities stabilize, meaning <strong>the system reaches a final long-term behavior</strong> where further transitions do not significantly change the probabilities.</p>
<hr>
<h3 id="c-definition-of-the-stationary-distribution"><strong>c. Definition of the Stationary Distribution</strong></h3>
<p>The <strong>stationary distribution</strong>  $ \mu^* $ 
 is a probability vector that remains <strong>unchanged</strong> under transitions:</p>
 
$$ \mu^* = \mu^* P $$ 


<p>This implies that if a system starts in  $ \mu^* $ 
, applying the transition matrix does <strong>not</strong> alter its probabilities‚Äîit stays in equilibrium.</p>
<hr>
<h3 id="d-why-is-the-stationary-distribution-important"><strong>d. Why Is the Stationary Distribution Important?</strong></h3>
<p>‚úÖ <strong>Predicts long-term state occupancy</strong>‚Äîthe fraction of time spent in each state.<br>
‚úÖ <strong>Defines steady-state probabilities</strong> in applications like RNA velocity analysis.<br>
‚úÖ <strong>Identifies stable states (e.g., progenitor/root cells in differentiation)</strong> when applied to biological modeling.</p>
<h2 id="e-example-computing-stationary-states-using-eigenvectors"><strong>e. Example: Computing Stationary States Using Eigenvectors</strong></h2>
<h3 id="step-1-define-the-transition-probability-matrix"><strong>Step 1: Define the Transition Probability Matrix</strong></h3>
<p>Consider a <strong>3-state system</strong> with transition probabilities:</p>
 
$$ \tilde{\pi} = \begin{bmatrix} 
0.8 & 0.1 & 0.1 \\ 
0.2 & 0.7 & 0.1 \\ 
0.1 & 0.2 & 0.7 
\end{bmatrix} $$ 


<p>This matrix describes <strong>how probabilities shift</strong> between states.</p>
<hr>
<h3 id="step-2-find-the-left-eigenvector-for-eigenvalue-1"><strong>Step 2: Find the Left Eigenvector for Eigenvalue 1</strong></h3>
<p>We solve:</p>
 
$$ v^T \tilde{\pi}^T = v^T $$ 


<p>This means we need the <strong>eigenvector corresponding to eigenvalue \( \lambda = 1 \)</strong>.</p>
<p>Computing the eigenvectors of  $ \tilde{\pi}^T $ 
, we obtain:</p>
 
$$ v = \begin{bmatrix} 0.45 \\ 0.35 \\ 0.20 \end{bmatrix} $$ 


<p>This <strong>left eigenvector</strong> represents the <strong>stationary distribution</strong>, meaning:
‚úÖ <strong>State 1 holds 45% of the long-term probability</strong>.<br>
‚úÖ <strong>State 2 holds 35%</strong>.<br>
‚úÖ <strong>State 3 holds 20%</strong>.</p>
<p>These <strong>steady-state values</strong> describe <strong>how the system behaves in the long run</strong>, identifying the states <strong>least likely to transition further</strong>.</p>
<hr>
<h3 id="4-biological-interpretation-in-rna-velocity"><strong>4. Biological Interpretation in RNA Velocity</strong></h3>
<p>In <strong>single-cell analysis</strong>:
‚úÖ <strong>Cells corresponding to the stationary distribution</strong> act as <strong>root cells</strong>, initiating differentiation.<br>
‚úÖ This eigenvector-driven method ensures an <strong>objective, data-driven way</strong> to define <strong>progenitor states</strong>.<br>
‚úÖ <strong>Eigenvalue 1 captures states that remain stable over differentiation</strong>, meaning they are <strong>starting points</strong> in biological development.</p>
<h1 id="g-is-p-in-eq16-dependent-on-gene-or-cell">G) Is p in eq(16) dependent on gene or cell?</h1>
<p>In scVelo&rsquo;s global time normalization, we have:</p>
 $$t_{i;o} = Q^p_g \left( t_{i;g} - t_{o;g} \right)$$ 

<p>Where:</p>
<ul>
<li> $t_{i;o}$ 
: normalized time for cell  $i$ 
 relative to root cell  $o$ 
</li>
<li> $Q^p_g$ 
:  $p$ 
-quantile computed for gene  $g$ 
</li>
<li> $t_{i;g} - t_{o;g}$ 
: time shift between cell  $i$ 
 and root cell  $o$ 
 for gene  $g$ 
</li>
</ul>
<h2 id="is-p-gene-specific">Is p Gene-Specific?</h2>
<p><strong>‚ùå No</strong> ‚Äî  $p$ 
 is <strong>not gene-specific</strong>.</p>
<h3 id="explanation">Explanation</h3>
<p> $Q^p_g$ 
 means: for <strong>each gene</strong>  $g$ 
, you compute the <strong> $p$ 
-quantile</strong> over the time shifts  $t_{i;g} - t_{o;g}$ 
, across all cells  $i$ 
.</p>
<p>But the value of <strong> $p$ 
</strong> itself is the <strong>same for all genes</strong> ‚Äî it is <strong>not gene-specific</strong>.</p>
<p>So even though the quantile is applied <strong>per gene</strong> (i.e., each gene gets its own  $Q^p_g$ 
), the <strong> $p$ 
 used in all those calculations is shared</strong>.</p>
<h3 id="mathematical-illustration">Mathematical Illustration</h3>
<p>For a fixed value  $p = 0.5$ 
 (median):</p>
 $$Q^{0.5}_{gene1} = \text{median}(\{t_{i;gene1} - t_{o;gene1}\}_{i=1}^N)$$ 

 $$Q^{0.5}_{gene2} = \text{median}(\{t_{i;gene2} - t_{o;gene2}\}_{i=1}^N)$$ 

 $$\vdots$$ 

 $$Q^{0.5}_{geneM} = \text{median}(\{t_{i;geneM} - t_{o;geneM}\}_{i=1}^N)$$ 

<p>The <strong>same</strong>  $p = 0.5$ 
 is used for all genes, but each gene gets its own quantile value.</p>
<h2 id="is-p-cell-specific">Is p Cell-Specific?</h2>
<p><strong>‚ùå Also no</strong> ‚Äî  $p$ 
 is not adjusted for each cell. It is only used to <strong>transform the gene-wise temporal shifts</strong> into a consistent time estimate <strong>for each cell</strong>, <strong>relative to the root cell</strong>  $o$ 
.</p>
<h3 id="explanation-1">Explanation</h3>
<p>The only place where indexing over cells happens is in computing the latent time  $t_{i;o}$ 
, but the <strong>value of  $p$ 
 stays fixed</strong> for all cells.</p>
<h3 id="mathematical-process">Mathematical Process</h3>
<ol>
<li>
<p><strong>Fixed  $p$ 
 across all computations</strong>:  $p = p_{\text{fixed}}$ 
</p>
</li>
<li>
<p><strong>Gene-wise quantile computation</strong>:
 $$Q^{p_{\text{fixed}}}_g = \text{quantile}_{p_{\text{fixed}}}(\{t_{i;g} - t_{o;g}\}_{i=1}^N)$$ 
</p>
</li>
<li>
<p><strong>Cell-specific time assignment</strong>:
 $$t_{1;o} = Q^{p_{\text{fixed}}}_g \left( t_{1;g} - t_{o;g} \right)$$ 

 $$t_{2;o} = Q^{p_{\text{fixed}}}_g \left( t_{2;g} - t_{o;g} \right)$$ 

 $$\vdots$$ 

 $$t_{N;o} = Q^{p_{\text{fixed}}}_g \left( t_{N;g} - t_{o;g} \right)$$ 
</p>
</li>
</ol>
<h2 id="is-p-root-specific">Is p Root-Specific?</h2>
<p><strong>‚úÖ Yes</strong> ‚Äî scVelo searches for the best  $p$ 
 <strong>per root candidate</strong> to optimize time smoothness.</p>
<h3 id="optimization-process">Optimization Process</h3>
<p>For each potential root cell  $o$ 
, scVelo optimizes:</p>
 $$p^*_o = \arg\min_p \mathcal{L}_{\text{smoothness}}(p, o)$$ 

<p>Where the loss function might be:</p>
 $$\mathcal{L}_{\text{smoothness}}(p, o) = \sum_{i,j \in \text{neighbors}} \left| t_{i;o}(p) - t_{j;o}(p) \right|^2$$ 

<h3 id="root-selection">Root Selection</h3>
<p>The final root and  $p$ 
 are chosen as:</p>
 $$o^*, p^* = \arg\min_{o,p} \mathcal{L}_{\text{smoothness}}(p, o)$$ 

<h2 id="summary-table">Summary Table</h2>
<table>
  <thead>
      <tr>
          <th>Aspect</th>
          <th>Is  $p$ 
 dependent?</th>
          <th>Explanation</th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td><strong>Per gene</strong>  $g$ 
</td>
          <td>‚ùå No</td>
          <td>The quantile  $Q^p$ 
 is computed per gene, but <strong> $p$ 
 itself is fixed</strong></td>
      </tr>
      <tr>
          <td><strong>Per cell</strong>  $i$ 
</td>
          <td>‚ùå No</td>
          <td>All cells use the same  $p$ 
; only their computed values differ</td>
      </tr>
      <tr>
          <td><strong>Per root</strong>  $o$ 
</td>
          <td>‚úÖ Yes</td>
          <td>scVelo searches for the best  $p$ 
 <strong>per root candidate</strong> to optimize time smoothness</td>
      </tr>
  </tbody>
</table>
<h2 id="optimization-step">Optimization Step</h2>
<p>Once <strong>scVelo</strong> computes all  $ t_{i,o} $ 
 for different candidate root cells  $ o $ 
, it looks for the root and quantile  $ p $ 
 that maximize correlation between the <strong>latent time series</strong><br>
 
$$ \{ t_{i,o} \} $$ 

<br>
and its <strong>convolution over a local neighborhood graph</strong> (e.g., KNN graph of cells).</p>
<h3 id="mathematical-formulation"><strong>Mathematical Formulation</strong></h3>
<p>The optimization step seeks:</p>
 
$$ \arg\max_{o,p} \text{corr} \left( t_{i,o}, \sum_{j \in N(i)} w_{ij} t_{j,o} \right) $$ 


<p>where:</p>
<ul>
<li> $ N(i) $ 
 is the <strong>neighborhood of cell</strong>  $ i $ 
.</li>
<li> $ w_{ij} $ 
 represents <strong>graph weights</strong>, determining connectivity between cells.</li>
</ul>
<h1 id="h-project-rna-velocities-into-an-embedding-eg-umap">H) Project RNA Velocities into an Embedding (e.g., UMAP)</h1>
<p>In scVelo, you have:</p>
<ul>
<li><strong>High-dimensional RNA velocities</strong> (in gene expression space)</li>
<li><strong>Low-dimensional embeddings</strong> (e.g., UMAP, t-SNE)</li>
</ul>
<p>The challenge is to project velocities into the embedding so that arrows in UMAP space reflect true dynamics.</p>
<h2 id="-key-variables">üìê Key Variables</h2>
<p>Let  $\vec{s}_i$ 
 be the embedding position of cell  $i$ 
 (e.g., in 2D UMAP space)</p>
<p>Let  $\vec{\delta}_{ij} = \frac{\vec{s}_j - \vec{s}_i}{\|\vec{s}_j - \vec{s}_i\|}$ 
 be the normalized direction vector from cell  $i$ 
 to cell  $j$ 
</p>
<p>Let  $\pi_{ij}$ 
 be the transition probability from cell  $i$ 
 to  $j$ 
 in the velocity-derived transition matrix</p>
<p>Let  $\vec{\nu}_i$ 
 be the velocity vector in the embedding space for cell  $i$ 
</p>
<h2 id="-formula-projected-velocity-vector">üìä Formula: Projected Velocity Vector</h2>
<p>The projected velocity for cell  $i$ 
 in the embedding space is:</p>
 $$\vec{\nu}_i = E_{\vec{\pi}_i}[\vec{\delta}_{ij}] = \sum_{j \neq i} \pi_{ij} \vec{\delta}_{ij} - \frac{1}{n} \sum_{j \neq i} \vec{\delta}_{ij}$$ 

<p>Or in compact form:</p>
 $$\vec{\nu}_i = \vec{\delta}_{i \cdot}^{\top} \vec{\pi}_i - \frac{1}{n} \sum_{j \neq i} \vec{\delta}_{ij}$$ 

<h2 id="-intuition">üß† Intuition</h2>
<p>The <strong>first term</strong> is the expected movement direction in embedding space, weighted by the transition probabilities derived from RNA velocity.</p>
<p>The <strong>second term</strong>,  $\frac{1}{n} \sum \vec{\delta}_{ij}$ 
, corrects for non-uniform density in the embedding. Without it, areas with more cells would bias the velocity field.</p>
<p>This ensures that the resulting velocity arrows:</p>
<ul>
<li>Follow the dynamics inferred from spliced/unspliced RNA</li>
<li>Are not artifacts of cell density in UMAP</li>
<li>Represent true directionality in cellular state transitions</li>
</ul>
<h2 id="-in-practice">üñºÔ∏è In Practice</h2>
<p>This velocity vector  $\vec{\nu}_i$ 
 is:</p>
<ul>
<li>Computed for each cell</li>
<li>Overlaid as arrows on UMAP or t-SNE plots using <code>scv.pl.velocity_embedding()</code></li>
</ul>
<h2 id="-summary">‚úÖ Summary</h2>
<table>
  <thead>
      <tr>
          <th>Term</th>
          <th>Meaning</th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td> $\vec{s}_i$ 
</td>
          <td>Position of cell  $i$ 
 in embedding (e.g., UMAP)</td>
      </tr>
      <tr>
          <td> $\vec{\delta}_{ij}$ 
</td>
          <td>Normalized direction vector from  $i$ 
 to  $j$ 
</td>
      </tr>
      <tr>
          <td> $\pi_{ij}$ 
</td>
          <td>Velocity-based transition probability from  $i$ 
 to  $j$ 
</td>
      </tr>
      <tr>
          <td> $\vec{\nu}_i$ 
</td>
          <td>Estimated velocity vector in embedding for cell  $i$ 
</td>
      </tr>
      <tr>
          <td> $\frac{1}{n} \sum \vec{\delta}_{ij}$ 
</td>
          <td>Density correction term</td>
      </tr>
  </tbody>
</table>
<h2 id="mathematical-breakdown">Mathematical Breakdown</h2>
<h3 id="step-1-compute-direction-vectors">Step 1: Compute Direction Vectors</h3>
<p>For each cell  $i$ 
 and all its neighbors  $j$ 
:</p>
 $$\vec{\delta}_{ij} = \frac{\vec{s}_j - \vec{s}_i}{\|\vec{s}_j - \vec{s}_i\|_2}$$ 

<h3 id="step-2-get-transition-probabilities">Step 2: Get Transition Probabilities</h3>
<p>From the velocity graph computed by scVelo:</p>
 $$\pi_{ij} = P(\text{cell } i \rightarrow \text{cell } j | \text{RNA velocity})$$ 

<h3 id="step-3-weighted-average-direction">Step 3: Weighted Average Direction</h3>
 $$\text{Expected direction} = \sum_{j \neq i} \pi_{ij} \vec{\delta}_{ij}$$ 

<h3 id="step-4-density-correction">Step 4: Density Correction</h3>
 $$\text{Uniform correction} = \frac{1}{n} \sum_{j \neq i} \vec{\delta}_{ij}$$ 

<h3 id="step-5-final-velocity">Step 5: Final Velocity</h3>
 $$\vec{\nu}_i = \text{Expected direction} - \text{Uniform correction}$$ 

<h2 id="alternative-formulations">Alternative Formulations</h2>
<h3 id="kernel-based-approach">Kernel-Based Approach</h3>
<p>Some implementations use a kernel-weighted version:</p>
 $$\vec{\nu}_i = \frac{\sum_j K(\vec{s}_i, \vec{s}_j) \pi_{ij} \vec{\delta}_{ij}}{\sum_j K(\vec{s}_i, \vec{s}_j)}$$ 

<p>where  $K(\vec{s}_i, \vec{s}_j)$ 
 is a distance-based kernel (e.g., Gaussian).</p>
<h3 id="confidence-weighting">Confidence Weighting</h3>
<p>Incorporate velocity confidence scores:</p>
 $$\vec{\nu}_i = \frac{\sum_j c_{ij} \pi_{ij} \vec{\delta}_{ij}}{\sum_j c_{ij}}$$ 

<p>where  $c_{ij}$ 
 represents the confidence in the transition from  $i$ 
 to  $j$ 
.</p>
<h2 id="implementation-notes">Implementation Notes</h2>
<p>The projected velocities are typically stored in:</p>
<ul>
<li><code>adata.obsm['velocity_umap']</code> for UMAP projections</li>
<li><code>adata.obsm['velocity_tsne']</code> for t-SNE projections</li>
</ul>
<p>These can be accessed and visualized using scVelo&rsquo;s plotting functions to show the directional flow of cellular development in low-dimensional space.</p>
<h1 id="i-the-reconstructability-score-r">I) The Reconstructability Score (r)</h1>
<h2 id="mathematical-definition">Mathematical Definition</h2>
 $$r = \text{median}_i \, \text{corr}(\pi_i, \pi_i')$$ 

<p>Where:</p>
<ul>
<li>
<p> $\pi_i$ 
: The row vector of the full velocity graph for cell  $i$ 
, representing its outgoing transition probabilities to all other cells.</p>
</li>
<li>
<p> $\pi_i'$ 
: The row vector of the reduced velocity graph for cell  $i$ 
, representing its outgoing transition probabilities based only on the selected genes.</p>
</li>
<li>
<p> $\text{corr}(\pi_i, \pi_i')$ 
: This calculates the correlation between the full outgoing transition probabilities for cell  $i$ 
 and the outgoing transition probabilities derived from the subset of genes for cell  $i$ 
. A high correlation means that the subset of genes largely reproduces the directional information provided by all genes for that particular cell.</p>
</li>
<li>
<p> $\text{median}_i$ 
: The score is then computed as the median of these per-cell correlations across all cells  $i$ 
. Taking the median makes the score robust to outliers (e.g., a few cells where the subset of genes might perform poorly due to local noise).</p>
</li>
</ul>
<h2 id="what-the-reconstructability-score-tells-you">What the Reconstructability Score Tells You</h2>
<p>The reconstructability score quantifies how well a specific subset of genes can &ldquo;reconstruct&rdquo; or recapitulate the overall RNA velocity dynamics inferred from the full set of genes.</p>
<h3 id="high-score-close-to-1">High Score (close to 1)</h3>
<p>If  $r$ 
 is high (e.g.,  $> 0.8$ 
), it suggests that the selected subset of genes (e.g., top likelihood genes) are indeed the primary drivers of the observed cellular transitions. Their dynamics alone are largely sufficient to explain the overall directionality and speed of differentiation. This validates their &ldquo;driver&rdquo; status and indicates that focusing on these genes provides a good summary of the system&rsquo;s dynamics.</p>
<h3 id="low-score">Low Score</h3>
<p>If  $r$ 
 is low, it means that the selected subset of genes does not adequately capture the full dynamic picture. The overall cellular transitions are likely influenced by a broader set of genes not included in your selection, or the selected genes might not be truly representative of the overall dynamics.</p>
<h2 id="mathematical-components-breakdown">Mathematical Components Breakdown</h2>
<h3 id="transition-probability-vectors">Transition Probability Vectors</h3>
<p>For each cell  $i$ 
, the transition probabilities are derived from the velocity graph:</p>
 $$\pi_i = [\pi_{i,1}, \pi_{i,2}, \ldots, \pi_{i,N}]$$ 

<p>where  $\pi_{i,j}$ 
 represents the probability of cell  $i$ 
 transitioning to cell  $j$ 
, and  $N$ 
 is the total number of cells.</p>
<h3 id="correlation-calculation">Correlation Calculation</h3>
<p>The correlation between full and reduced transition probabilities:</p>
 $$\text{corr}(\pi_i, \pi_i') = \frac{\text{cov}(\pi_i, \pi_i')}{\sigma_{\pi_i} \sigma_{\pi_i'}}$$ 

<p>where:</p>
<ul>
<li> $\text{cov}(\pi_i, \pi_i')$ 
 is the covariance</li>
<li> $\sigma_{\pi_i}$ 
 and  $\sigma_{\pi_i'}$ 
 are the standard deviations</li>
</ul>
<h3 id="median-aggregation">Median Aggregation</h3>
<p>The final score uses median aggregation for robustness:</p>
 $$r = \text{median}\{c_1, c_2, \ldots, c_N\}$$ 

<p>where  $c_i = \text{corr}(\pi_i, \pi_i')$ 
 for each cell  $i$ 
.</p>
<h2 id="use-cases-and-significance">Use Cases and Significance</h2>
<h3 id="validation-of-driver-gene-selection">Validation of Driver Gene Selection</h3>
<p>It&rsquo;s commonly used to validate that the &ldquo;top likelihood genes&rdquo; (which scVelo identifies as dynamically strong) are indeed the key players shaping the global trajectory.</p>
<h3 id="identifying-essential-gene-sets">Identifying Essential Gene Sets</h3>
<p>You can test specific gene modules or pathways to see if they are collectively responsible for a particular fate decision or developmental progression.</p>
<h3 id="dimensionality-reductionsummarization">Dimensionality Reduction/Summarization</h3>
<p>If a small subset of genes yields a high reconstructability score, it suggests that these genes form a powerful, low-dimensional representation of the system&rsquo;s dynamics.</p>
<h3 id="biological-interpretation">Biological Interpretation</h3>
<p>High reconstructability from a specific gene set points to their significant biological role in driving the observed cellular changes.</p>
<h1 id="j-transport-maps-and-cell-distribution-analysis">J) Transport Maps and Cell Distribution Analysis</h1>
<h2 id="mathematical-framework">Mathematical Framework</h2>
<h3 id="original-notation">Original Notation</h3>
<p> $\mu = (\mu_1, \ldots, \mu_n)$ 
: a distribution over cells</p>
<p> $\pi_e$ 
: a transport map, e.g., inferred from RNA velocity</p>
<p> $\tilde{\pi}^T$ 
: the transpose (or reverse) of the transport map</p>
<p> $\mu_{\text{des}} = \mu \cdot \pi_e$ 
: the descendant distribution</p>
<p> $\mu_{\text{anc}} = \mu \cdot \tilde{\pi}^T$ 
: the ancestor distribution</p>
<h2 id="core-concepts">Core Concepts</h2>
<h3 id="1-forward-transport-where-are-cells-going">1. Forward Transport: &ldquo;Where are cells going?&rdquo;</h3>
<p>A distribution of cells  $\mu$ 
 can be pushed through the transport map to determine future cell states.</p>
<p>Starting with a distribution over some cells  $\mu$ 
, for example:</p>
<ul>
<li>All cells in a particular cluster or cell type</li>
<li>Or a uniform distribution over some selected cells</li>
</ul>
<p>Pushing forward this distribution through the transport map  $\pi_e$ 
 gives the distribution of where these cells go in the future:</p>

$$\mu_{\text{des}} = \mu \cdot \pi_e \tag{1}$$


<p>This answers the question: &ldquo;Given that my cells are here now, where are they going?&rdquo;</p>
<h3 id="2-reverse-transport-where-did-cells-come-from">2. Reverse Transport: &ldquo;Where did cells come from?&rdquo;</h3>
<p>Conversely, a distribution  $\mu$ 
 can be pulled back to determine cell origins.</p>
<p>You take a set of cells at a later time or later state, and ask: &ldquo;Where did they come from?&rdquo;</p>
<p>You pull back the distribution via the transpose of the transport matrix:</p>

$$\mu_{\text{anc}} = \mu \cdot \tilde{\pi}^T \tag{2}$$


<p>The transpose reverses the direction of flow ‚Äî it computes how much each earlier state contributed to the current cell distribution.</p>
<h3 id="3-indicator-based-cell-selection">3. Indicator-Based Cell Selection</h3>
<p>For a set of cells  $S = \{s_1, \ldots, s_n\}$ 
, the distribution  $\mu$ 
 is defined using an indicator vector:</p>

$$\mu_i = \mathbf{1}_{s_i \in S} \tag{3}$$


<p>This means:</p>
<ul>
<li> $\mu_i = 1$ 
 if cell  $i$ 
 is in the set  $S$ 
</li>
<li> $\mu_i = 0$ 
 otherwise</li>
</ul>
<p>The input distribution puts mass 1 on each selected cell, and 0 elsewhere.</p>
<h2 id="biological-interpretation-1">Biological Interpretation</h2>
<h3 id="example-use-cases">Example Use Cases</h3>
<p><strong>Progenitor Analysis:</strong></p>
<ol>
<li>Select a set of progenitor cells (set  $S$ 
)</li>
<li>Compute  $\mu_{\text{des}} = \mu \cdot \pi_e$ 
 ‚Üí get the expected distribution of their descendants</li>
</ol>
<p><strong>Lineage Tracing:</strong></p>
<ol>
<li>Select differentiated cells</li>
<li>Compute  $\mu_{\text{anc}} = \mu \cdot \tilde{\pi}^T$ 
 ‚Üí get their likely origins</li>
</ol>
<p>This framework helps answer:</p>
<ul>
<li>&ldquo;Where are these cells going?&rdquo;</li>
<li>&ldquo;Where did these cells come from?&rdquo;</li>
</ul>
<h2 id="numerical-example">Numerical Example</h2>
<p>Consider 4 cells ( $n=4$ 
), selecting cells 1 and 3:</p>

$$\mu = [1, 0, 1, 0] \tag{4}$$


<p>Assume a learned transport map:</p>

$$\pi_e = \begin{bmatrix}
0.6 & 0.2 & 0.1 & 0.1 \\
0.1 & 0.7 & 0.1 & 0.1 \\
0.2 & 0.1 & 0.6 & 0.1 \\
0.3 & 0.2 & 0.1 & 0.4
\end{bmatrix} \tag{5}$$


<p>Then the descendant distribution is:</p>

$$\mu_{\text{des}} = \mu \cdot \pi_e = [1, 0, 1, 0] \cdot \pi_e = \text{row 1} + \text{row 3} \tag{6}$$


<p>Computing the result:</p>

$$\mu_{\text{des}} = [0.6 + 0.2, 0.2 + 0.1, 0.1 + 0.6, 0.1 + 0.1] = [0.8, 0.3, 0.7, 0.2] \tag{7}$$


<p>This gives a distribution over where the selected cells (1 &amp; 3) are headed ‚Äî their descendant cell states.</p>
<h2 id="reverse-transport-example">Reverse Transport Example</h2>
<h3 id="step-1-compute-the-ancestor-distribution">Step 1: Compute the Ancestor Distribution</h3>
<p>To find the ancestor distribution, we use the transpose of  $\pi_e$ 
:</p>
 
$$(\pi_e)^T = \begin{bmatrix}
0.6 & 0.1 & 0.2 & 0.3 \\
0.2 & 0.7 & 0.1 & 0.2 \\
0.1 & 0.1 & 0.6 & 0.1 \\
0.1 & 0.1 & 0.1 & 0.4
\end{bmatrix} \tag{3}$$


<h3 id="step-2-calculate-the-ancestor-distribution">Step 2: Calculate the Ancestor Distribution</h3>
<p>The ancestor distribution is computed as:</p>
 
$$\mu_{\text{anc}} = \mu_{\text{des}} \cdot (\pi_e)^T \tag{4}$$


<p>Multiplying the row vector  $\mu_{\text{des}} = [0.8, 0.3, 0.7, 0.2]$ 
 by the matrix  $(\pi_e)^T$ 
:</p>
 
$$\mu_{\text{anc},1} = 0.8 \times 0.6 + 0.3 \times 0.1 + 0.7 \times 0.2 + 0.2 \times 0.3 = 0.48 + 0.03 + 0.14 + 0.06 = 0.71 $$


 
$$\mu_{\text{anc},2} = 0.8 \times 0.2 + 0.3 \times 0.7 + 0.7 \times 0.1 + 0.2 \times 0.2 = 0.16 + 0.21 + 0.07 + 0.04 = 0.48 $$


 
$$\mu_{\text{anc},3} = 0.8 \times 0.1 + 0.3 \times 0.1 + 0.7 \times 0.6 + 0.2 \times 0.1 = 0.08 + 0.03 + 0.42 + 0.02 = 0.55 $$


 
$$\mu_{\text{anc},4} = 0.8 \times 0.1 + 0.3 \times 0.1 + 0.7 \times 0.1 + 0.2 \times 0.4 = 0.08 + 0.03 + 0.07 + 0.08 = 0.26 $$


<h3 id="step-3-resulting-ancestor-distribution">Step 3: Resulting Ancestor Distribution</h3>
 
$$\mu_{\text{anc}} = [0.71, 0.48, 0.55, 0.26] \tag{9}$$


<p><strong>Interpretation</strong>: Given that the descendant distribution was  $[0.8, 0.3, 0.7, 0.2]$ 
, pulling it back through the transpose of the transport map estimates the likely ancestor distribution of these descendants. The weights show which earlier states contributed more to this descendant cell population.</p>
<h3 id="normalization">Normalization</h3>
<p>We can normalize  $\mu_{\text{anc}}$ 
 to turn it into a proper probability distribution:</p>
 
$$\text{Sum} = 0.71 + 0.48 + 0.55 + 0.26 = 2.00 $$


 
$$\mu_{\text{anc,normalized}} = [0.355, 0.24, 0.275, 0.13] $$


<h3 id="why-perfect-inversion-is-impossible">Why Perfect Inversion is Impossible</h3>
<p>The descendant distribution  $[0.8, 0.3, 0.7, 0.2]$ 
 cannot lead exactly back to the original indicator vector  $[1, 0, 1, 0]$ 
 when using the transpose of the transport map for several fundamental reasons:</p>
<h4 id="1-non-invertibility-of-the-transport-map">1. Non-invertibility of the Transport Map</h4>
<p>The transport matrix  $\pi_e$ 
 typically represents a probabilistic, many-to-many transition between cell states. It&rsquo;s a stochastic matrix where rows sum to 1, but it&rsquo;s often not invertible or the inverse is not unique. The operation of pushing forward loses information through mixing and averaging, and pulling back is not a perfect inverse.</p>
<h4 id="2-smoothing-and-mixing">2. Smoothing and Mixing</h4>
<p>When we multiplied  $[1, 0, 1, 0]$ 
 by  $\pi_e$ 
, we obtained a weighted average of rows 1 and 3, resulting in the smoothed vector  $[0.8, 0.3, 0.7, 0.2]$ 
. This vector is continuous-valued and represents a distribution over many possible descendant states. The reverse step with the transpose spreads these probabilities back to ancestors but cannot undo the smoothing perfectly.</p>
<h4 id="3-loss-of-sharpness">3. Loss of Sharpness</h4>
<p>The original vector  $[1, 0, 1, 0]$ 
 is a sharp indicator (only cells 1 and 3 are selected). After forward mapping, the distribution reflects uncertainty or spreading. Pulling back cannot perfectly re-sharpen because the transport matrix mixes information.</p>
<h4 id="4-biological-analogy">4. Biological Analogy</h4>
<p>In biological systems, a population of cells can differentiate into multiple descendant states probabilistically. The future cell states reflect a blend of many ancestors. Hence, you cannot simply invert descendants back to a unique set of ancestors ‚Äî instead, you get a probabilistic ancestor distribution that represents the uncertainty inherent in the biological process.</p>
<h1 id="k-calculating-the-gene-shared-latent-time-in-scvelo">K) Calculating the gene shared latent time in scVelo</h1>
<h2 id="step-by-step-explanation">Step-by-step explanation</h2>
<h3 id="1-latent-time">1. <strong>Latent time</strong></h3>
<ul>
<li>scVelo computes a <strong>latent time</strong>  $t_n$ 
 for each cell  $n$ 
, indicating <strong>how far along a differentiation trajectory</strong> it is, starting from the inferred root cells.</li>
<li>This time is estimated based on how consistent a cell&rsquo;s transcriptional state is with the <strong>direction of gene expression changes (RNA velocity)</strong>.</li>
</ul>
<h3 id="2-gene-shared-latent-time-course">2. <strong>Gene-shared latent time course</strong></h3>
<ul>
<li>This is the original latent time vector computed across all cells.</li>
<li>It&rsquo;s &ldquo;gene-shared&rdquo; because it&rsquo;s not just one gene&rsquo;s trajectory ‚Äî it aggregates information across genes to produce a <strong>common timeline</strong>.</li>
</ul>
<h3 id="3-neighborhood-convolution">3. <strong>Neighborhood convolution</strong></h3>
<ul>
<li>scVelo smooths the latent time values using <strong>k-nearest neighbor (KNN)</strong> information (e.g., based on transcriptomic similarity).</li>
<li>The <strong>convolved time</strong> for a cell is the <strong>average of the latent times of its neighboring cells</strong>.</li>
</ul>
<p>Think of it as:</p>
 
$$\tilde{t}_n = \frac{1}{|N(n)|} \sum_{m \in N(n)} t_m \tag{1}$$


<p>where  $N(n)$ 
 is the neighborhood of cell  $n$ 
.</p>
<h3 id="4-regression--consistency-check">4. <strong>Regression &amp; consistency check</strong></h3>
<ul>
<li>scVelo performs <strong>linear regression</strong>: it tries to predict each cell&rsquo;s latent time using its convolved value.</li>
<li>If a cell&rsquo;s latent time significantly <strong>deviates from the trend of its neighbors</strong>, it is likely noisy or unreliable.</li>
</ul>
<h3 id="5-correction--replacement">5. <strong>Correction / Replacement</strong></h3>
<ul>
<li>For these <strong>outlier cells</strong>, scVelo <strong>replaces their latent time</strong> with the convolved (smoothed) version.</li>
<li>This acts like a denoising step ‚Äî ensuring latent time values are <strong>locally smooth and robust</strong> to noise.</li>
</ul>
<p><strong>scVelo uses neighborhood convolution</strong> to detect and fix inconsistent latent times using regression.</p>
<h2 id="-setup">üß™ Setup</h2>
<p>Let&rsquo;s say we have <strong>5 cells</strong>, and we already computed latent times for them as:</p>
<table>
  <thead>
      <tr>
          <th>Cell</th>
          <th>Latent Time  $t_n$ 
</th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td>A</td>
          <td>0.10</td>
      </tr>
      <tr>
          <td>B</td>
          <td>0.15</td>
      </tr>
      <tr>
          <td>C</td>
          <td>0.85 ‚ùó</td>
      </tr>
      <tr>
          <td>D</td>
          <td>0.20</td>
      </tr>
      <tr>
          <td>E</td>
          <td>0.25</td>
      </tr>
  </tbody>
</table>
<p>Clearly, cell <strong>C</strong> looks suspicious ‚Äî it&rsquo;s way ahead in time compared to its neighbors.</p>
<h2 id="-step-1-define-neighbors">üëØ Step 1: Define neighbors</h2>
<p>Let&rsquo;s say the <strong>neighborhoods (e.g., via KNN)</strong> are:</p>
<ul>
<li>A ‚Üí neighbors: [B, D]</li>
<li>B ‚Üí neighbors: [A, D]</li>
<li>C ‚Üí neighbors: [B, D, E]</li>
<li>D ‚Üí neighbors: [A, B, E]</li>
<li>E ‚Üí neighbors: [C, D]</li>
</ul>
<h2 id="-step-2-compute-convolved-smoothed-latent-times">üîÅ Step 2: Compute convolved (smoothed) latent times</h2>
 
$$\tilde{t}_n = \text{average of neighbor latent times} \tag{2}$$


<table>
  <thead>
      <tr>
          <th>Cell</th>
          <th>Neighbors</th>
          <th>Neighbor Times</th>
          <th>Convolved Time  $\tilde{t}_n$ 
</th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td>A</td>
          <td>B, D</td>
          <td>0.15, 0.20</td>
          <td>0.175</td>
      </tr>
      <tr>
          <td>B</td>
          <td>A, D</td>
          <td>0.10, 0.20</td>
          <td>0.15</td>
      </tr>
      <tr>
          <td>C</td>
          <td>B, D, E</td>
          <td>0.15, 0.20, 0.25</td>
          <td>0.20</td>
      </tr>
      <tr>
          <td>D</td>
          <td>A, B, E</td>
          <td>0.10, 0.15, 0.25</td>
          <td>0.167</td>
      </tr>
      <tr>
          <td>E</td>
          <td>C, D</td>
          <td>0.85, 0.20</td>
          <td>0.525 ‚ùó</td>
      </tr>
  </tbody>
</table>
<h2 id="-step-3-compare-original-vs-convolved-times">üìâ Step 3: Compare original vs convolved times</h2>
<table>
  <thead>
      <tr>
          <th>Cell</th>
          <th> $t_n$ 
</th>
          <th> $\tilde{t}_n$ 
</th>
          <th>Difference</th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td>A</td>
          <td>0.10</td>
          <td>0.175</td>
          <td>0.075</td>
      </tr>
      <tr>
          <td>B</td>
          <td>0.15</td>
          <td>0.15</td>
          <td>0.000</td>
      </tr>
      <tr>
          <td>C</td>
          <td>0.85 ‚ùó</td>
          <td>0.20</td>
          <td>0.65 ‚ùó</td>
      </tr>
      <tr>
          <td>D</td>
          <td>0.20</td>
          <td>0.167</td>
          <td>0.033</td>
      </tr>
      <tr>
          <td>E</td>
          <td>0.25</td>
          <td>0.525 ‚ùó</td>
          <td>0.275 ‚ùó</td>
      </tr>
  </tbody>
</table>
<h2 id="-step-4-detect-inconsistent-cells">üßΩ Step 4: Detect inconsistent cells</h2>
<ul>
<li>scVelo runs a <strong>regression of</strong>  $t_n \sim \tilde{t}_n$ 
 and checks for large residuals.</li>
<li>Cells <strong>C</strong> and <strong>E</strong> have large mismatches between original and smoothed values ‚Üí likely noisy or incorrect.</li>
</ul>
<h2 id="-step-5-replace-inconsistent-values">üîÅ Step 5: Replace inconsistent values</h2>
<p>Let&rsquo;s say we define a cutoff: if difference &gt; 0.3, replace with  $\tilde{t}_n$ 
.</p>
<table>
  <thead>
      <tr>
          <th>Cell</th>
          <th>New  $t_n$ 
</th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td>A</td>
          <td>0.10 (unchanged)</td>
      </tr>
      <tr>
          <td>B</td>
          <td>0.15 (unchanged)</td>
      </tr>
      <tr>
          <td>C</td>
          <td><strong>0.20</strong> (replaced!)</td>
      </tr>
      <tr>
          <td>D</td>
          <td>0.20 (unchanged)</td>
      </tr>
      <tr>
          <td>E</td>
          <td><strong>0.525</strong> (replaced!)</td>
      </tr>
  </tbody>
</table>
<h2 id="-final-corrected-latent-time">‚úÖ Final corrected latent time:</h2>
 
$$\mathbf{t} = [0.10, 0.15, \mathbf{0.20}, 0.20, \mathbf{0.525}] \tag{3}$$


<p>Cell C is now brought <strong>back in line</strong> with its local trajectory.</p>

      </div>

      
  <time class="mt-12 mb-8 block text-xs text-gray-500 ltr:text-right rtl:text-left dark:text-gray-400" datetime="2025-05-28T00:00:00.000Z">
    <span>Last updated on</span>
    May 28, 2025</time>

      <div class="container mx-auto prose prose-slate lg:prose-xl dark:prose-invert mt-5">
        
        <div class="max-w-prose print:hidden">
  
  

  

<div class="flex justify-center">
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/scrnaseq-rna-velocity-parameter-inference-em-algorithm/">ScRNAseq RNA Velocity, Parameter Inference, EM Algorithm</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/dynamic-model/">Dynamic Model</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/scvelo/">ScVelo</a>
  
  <a class="no-underline bg-primary-100 hover:bg-primary-300 text-primary-800 text-xs font-medium mr-2 px-2.5 py-0.5 lg:px-5 lg:py-2 rounded dark:bg-primary-900 dark:hover:bg-primary-700 dark:text-primary-300" href="/tags/differential-equations/">Differential Equations</a>
  
</div>


  
<section class="flex flex-row flex-wrap justify-center pt-4 text-xl">
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F11f.velocity_dynamic_model_posthoc_calculation%2Fscvelo_post-fitting_calculation%2F&amp;text=%F0%9F%A7%AC&#43;Dynamic&#43;RNA&#43;velocity&#43;model--&#43;%283%29&#43;post&#43;hoc&#43;computation"
    title="X"
    aria-label="X"
    id="share-link-x"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://www.facebook.com/sharer.php?u=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F11f.velocity_dynamic_model_posthoc_calculation%2Fscvelo_post-fitting_calculation%2F&amp;t=%F0%9F%A7%AC&#43;Dynamic&#43;RNA&#43;velocity&#43;model--&#43;%283%29&#43;post&#43;hoc&#43;computation"
    title="Facebook"
    aria-label="Facebook"
    id="share-link-facebook"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="currentColor" d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12c0 4.84 3.44 8.87 8 9.8V15H8v-3h2V9.5C10 7.57 11.57 6 13.5 6H16v3h-2c-.55 0-1 .45-1 1v2h3v3h-3v6.95c5.05-.5 9-4.76 9-9.95z"/></svg>
  </a>
  

  
  
  
  
  
  
    
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="mailto:?subject=%F0%9F%A7%AC%20Dynamic%20RNA%20velocity%20model--%20%283%29%20post%20hoc%20computation&amp;body=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F11f.velocity_dynamic_model_posthoc_calculation%2Fscvelo_post-fitting_calculation%2F"
    title="Email"
    aria-label="Email"
    id="share-link-email"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-width="1.5" d="M16.5 12a4.5 4.5 0 1 1-9 0a4.5 4.5 0 0 1 9 0Zm0 0c0 1.657 1.007 3 2.25 3S21 13.657 21 12a9 9 0 1 0-2.636 6.364M16.5 12V8.25"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://www.linkedin.com/shareArticle?url=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F11f.velocity_dynamic_model_posthoc_calculation%2Fscvelo_post-fitting_calculation%2F&amp;title=%F0%9F%A7%AC&#43;Dynamic&#43;RNA&#43;velocity&#43;model--&#43;%283%29&#43;post&#43;hoc&#43;computation"
    title="LinkedIn"
    aria-label="LinkedIn"
    id="share-link-linkedin"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" height="1em" viewBox="0 0 448 512"><path fill="currentColor" d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="whatsapp://send?text=%F0%9F%A7%AC&#43;Dynamic&#43;RNA&#43;velocity&#43;model--&#43;%283%29&#43;post&#43;hoc&#43;computation%20http%3A%2F%2Flocalhost%3A1313%2Fpost%2F11f.velocity_dynamic_model_posthoc_calculation%2Fscvelo_post-fitting_calculation%2F"
    title="WhatsApp"
    aria-label="WhatsApp"
    id="share-link-whatsapp"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256" fill="currentColor"><path d="m187.58 144.84-32-16a8 8 0 0 0-8 .5l-14.69 9.8a40.55 40.55 0 0 1-16-16l9.8-14.69a8 8 0 0 0 .5-8l-16-32A8 8 0 0 0 104 64a40 40 0 0 0-40 40 88.1 88.1 0 0 0 88 88 40 40 0 0 0 40-40 8 8 0 0 0-4.42-7.16ZM152 176a72.08 72.08 0 0 1-72-72 24 24 0 0 1 19.29-23.54l11.48 23L101 118a8 8 0 0 0-.73 7.51 56.47 56.47 0 0 0 30.15 30.15A8 8 0 0 0 138 155l14.61-9.74 23 11.48A24 24 0 0 1 152 176ZM128 24a104 104 0 0 0-91.82 152.88l-11.35 34.05a16 16 0 0 0 20.24 20.24l34.05-11.35A104 104 0 1 0 128 24Zm0 192a87.87 87.87 0 0 1-44.06-11.81 8 8 0 0 0-6.54-.67L40 216l12.47-37.4a8 8 0 0 0-.66-6.54A88 88 0 1 1 128 216Z"/></svg>
  </a>
  
</section>


  








  
  
    



  
  
  
    
  
  
  

<div class="flex pt-12 pb-4">
  
  
  <img
    class="mr-4 h-24 w-24 rounded-full"
    width="96"
    height="96"
    alt="Jiyuan Liu"
  src="/author/jiyuan-liu/avatar_hu_6297b4a1a274518.jpg"
  loading="lazy"
  />
  
  <div class="place-self-center">
    <div class="text-[0.6rem] uppercase leading-3 text-neutral-500 dark:text-neutral-400">
      Authors
    </div>
    <div class="font-semibold leading-6 text-neutral-800 dark:text-neutral-300">
      <a href="http://localhost:1313/" class="no-underline">
      Jiyuan Liu
      </a>
    </div>

    
    <div class="text-sm font-bold text-neutral-700 dark:text-neutral-300">
    Postdoctoral Scientist
    </div>
    


    

    <div class="text-2xl sm:text-lg pt-1">

      
<div class="flex flex-wrap text-neutral-500 dark:text-neutral-300">
  
    
    
    
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="mailto:jliu78@mgh.harvard.edu"
      
      aria-label="At-Symbol"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-width="1.5" d="M16.5 12a4.5 4.5 0 1 1-9 0a4.5 4.5 0 0 1 9 0Zm0 0c0 1.657 1.007 3 2.25 3S21 13.657 21 12a9 9 0 1 0-2.636 6.364M16.5 12V8.25"/></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://github.com/stars/jliu678/lists/owned-contributed-by-me"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Brands/Github"
    ><svg style="height: 1em;" fill="currentColor" viewBox="3 3 18 18"><path d="M12 3C7.0275 3 3 7.12937 3 12.2276C3 16.3109 5.57625 19.7597 9.15374 20.9824C9.60374 21.0631 9.77249 20.7863 9.77249 20.5441C9.77249 20.3249 9.76125 19.5982 9.76125 18.8254C7.5 19.2522 6.915 18.2602 6.735 17.7412C6.63375 17.4759 6.19499 16.6569 5.8125 16.4378C5.4975 16.2647 5.0475 15.838 5.80124 15.8264C6.51 15.8149 7.01625 16.4954 7.18499 16.7723C7.99499 18.1679 9.28875 17.7758 9.80625 17.5335C9.885 16.9337 10.1212 16.53 10.38 16.2993C8.3775 16.0687 6.285 15.2728 6.285 11.7432C6.285 10.7397 6.63375 9.9092 7.20749 9.26326C7.1175 9.03257 6.8025 8.08674 7.2975 6.81794C7.2975 6.81794 8.05125 6.57571 9.77249 7.76377C10.4925 7.55615 11.2575 7.45234 12.0225 7.45234C12.7875 7.45234 13.5525 7.55615 14.2725 7.76377C15.9937 6.56418 16.7475 6.81794 16.7475 6.81794C17.2424 8.08674 16.9275 9.03257 16.8375 9.26326C17.4113 9.9092 17.76 10.7281 17.76 11.7432C17.76 15.2843 15.6563 16.0687 13.6537 16.2993C13.98 16.5877 14.2613 17.1414 14.2613 18.0065C14.2613 19.2407 14.25 20.2326 14.25 20.5441C14.25 20.7863 14.4188 21.0746 14.8688 20.9824C16.6554 20.364 18.2079 19.1866 19.3078 17.6162C20.4077 16.0457 20.9995 14.1611 21 12.2276C21 7.12937 16.9725 3 12 3Z"></path></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://www.linkedin.com/in/jiyuan-liu-7608a42a6/"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Brands/Linkedin"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" height="1em" viewBox="0 0 448 512"><path fill="currentColor" d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://scholar.google.com/citations?view_op=list_works&amp;hl=en&amp;hl=en&amp;user=5c1kGO4AAAAJ&amp;sortby=pubdate"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Academicons/Google-Scholar"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M343.759 106.662V79.43L363.524 64h-213.89L20.476 176.274h85.656a82.339 82.339 0 0 0-.219 6.225c0 20.845 7.22 38.087 21.672 51.861c14.453 13.797 32.252 20.648 53.327 20.648c4.923 0 9.75-.368 14.438-1.024c-2.907 6.5-4.374 12.523-4.374 18.142c0 9.875 4.499 20.43 13.467 31.642c-39.234 2.67-68.061 9.732-86.437 21.163c-10.531 6.5-19 14.704-25.39 24.531c-6.391 9.9-9.578 20.515-9.578 31.962c0 9.648 2.062 18.336 6.219 26.062c4.156 7.726 9.578 14.07 16.312 18.984c6.718 4.968 14.469 9.101 23.219 12.469c8.734 3.344 17.406 5.718 26.061 7.062A167.052 167.052 0 0 0 180.555 448c13.469 0 26.953-1.734 40.547-5.187c13.562-3.485 26.28-8.642 38.171-15.493c11.86-6.805 21.515-16.086 28.922-27.718c7.39-11.68 11.094-24.805 11.094-39.336c0-11.016-2.25-21.039-6.75-30.14c-4.468-9.073-9.938-16.542-16.452-22.345c-6.501-5.813-13-11.155-19.516-15.968c-6.5-4.845-12-9.75-16.468-14.813c-4.485-5.046-6.735-10.054-6.735-14.984c0-4.921 1.734-9.672 5.216-14.265c3.455-4.61 7.674-9.048 12.61-13.306c4.937-4.25 9.875-8.968 14.796-14.133c4.922-5.147 9.141-11.827 12.61-20.008c3.485-8.18 5.203-17.445 5.203-27.757c0-13.453-2.547-24.46-7.547-33.314c-.594-1.022-1.218-1.803-1.875-3.022l56.907-46.672v17.119c-7.393.93-6.624 5.345-6.624 10.635V245.96c0 5.958 4.875 10.834 10.834 10.834h3.989c5.958 0 10.833-4.875 10.833-10.834V117.293c0-5.277.778-9.688-6.561-10.63zm-107.36 222.48c1.14.75 3.704 2.78 7.718 6.038c4.05 3.243 6.797 5.695 8.266 7.414a443.553 443.553 0 0 1 6.376 7.547c2.813 3.375 4.718 6.304 5.718 8.734c1 2.477 2.016 5.461 3.047 8.946a38.27 38.27 0 0 1 1.485 10.562c0 17.048-6.564 29.68-19.656 37.859c-13.125 8.18-28.767 12.274-46.938 12.274c-9.187 0-18.203-1.093-27.063-3.196c-8.843-2.116-17.311-5.336-25.39-9.601c-8.078-4.258-14.577-10.204-19.5-17.797c-4.938-7.64-7.407-16.415-7.407-26.25c0-10.32 2.797-19.29 8.422-26.906c5.594-7.625 12.938-13.391 22.032-17.315c9.063-3.946 18.25-6.742 27.562-8.398a157.865 157.865 0 0 1 28.438-2.555c4.47 0 7.936.25 10.405.696c.455.219 3.032 2.07 7.735 5.563c4.704 3.462 7.625 5.595 8.75 6.384zm-3.359-100.579c-7.406 8.86-17.734 13.288-30.953 13.288c-11.86 0-22.298-4.764-31.266-14.312c-9-9.523-15.422-20.328-19.344-32.43c-3.937-12.109-5.906-23.984-5.906-35.648c0-13.694 3.596-25.352 10.781-34.976c7.187-9.65 17.5-14.485 30.938-14.485c11.875 0 22.374 5.038 31.437 15.157c9.094 10.085 15.61 21.413 19.517 33.968c3.922 12.54 5.873 24.53 5.873 35.984c0 13.446-3.702 24.61-11.076 33.454z"/></svg></a>
  
</div>



    </div>
  </div>
</div>



  




  
  
    
    
    
      
      
    
<div class="pt-1 no-prose w-full">
  <hr class="border-dotted border-neutral-300 dark:border-neutral-600" />
  <div class="flex flex-col md:flex-row flex-nowrap justify-between gap-5 pt-2">
    <div class="">
      
        <a class="group flex no-underline" href="/post/11e.velocity_dynamic_model_inference/">
          <span
            class="mt-[-0.3rem] me-2 text-neutral-700 transition-transform group-hover:-translate-x-[2px] group-hover:text-primary-600 dark:text-neutral dark:group-hover:text-primary-400"
          ><span class="ltr:inline rtl:hidden">&larr;</span></span>
          <span class="flex flex-col">
            <span
              class="mt-[0.1rem] leading-6 group-hover:underline group-hover:decoration-primary-500"
            >üß¨ Dynamic RNA velocity model-- (2) parameter inference</span>
            <span class="mt-[0.1rem] text-xs text-neutral-500 dark:text-neutral-400">
              
                May 28, 2025
              
            </span>
          </span>
        </a>
      
    </div>
    <div class="">
      
        <a class="group flex text-right no-underline" href="/post/11c.velocity_steady_state/">
          <span class="flex flex-col">
            <span
              class="mt-[0.1rem] leading-6 group-hover:underline group-hover:decoration-primary-500"
            >üß¨ Math derivation for steady-state RNA velocity model</span
            >
            <span class="mt-[0.1rem] text-xs text-neutral-500 dark:text-neutral-400">
              
                May 28, 2025
              
            </span>
          </span>
          <span
            class="mt-[-0.3rem] ms-2 text-neutral-700 transition-transform group-hover:-translate-x-[2px] group-hover:text-primary-600 dark:text-neutral dark:group-hover:text-primary-400"
          ><span class="ltr:inline">&rarr;</span></span>
        </a>
      
    </div>
  </div>
</div>



  


  



</div>

      </div>

    </main>
  </article>
</div>

    </div>
    <div class="page-footer">
      <footer class="container mx-auto flex flex-col justify-items-center text-sm leading-6 mt-24 mb-4 text-slate-700 dark:text-slate-200">

  












  
  
  
  
  














  
  
  

  
  
    
  
  
    
  

  

  
  <p class="powered-by text-center">
    ¬© 2025 Me. This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-nd/4.0" rel="noopener noreferrer" target="_blank">CC BY NC ND 4.0</a>
  </p>
  

  <p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-nd/4.0" rel="noopener noreferrer" target="_blank" aria-label="Creative Commons">
      <i class="fab fa-creative-commons fa-2x" aria-hidden="true"></i>
      <i class="fab fa-creative-commons-by fa-2x" aria-hidden="true"></i>
      
        <i class="fab fa-creative-commons-nc fa-2x" aria-hidden="true"></i>
      
      
        <i class="fab fa-creative-commons-nd fa-2x" aria-hidden="true"></i>
      
    </a>
  </p>





  <p class="powered-by text-center">
    
    
    
      
      
      
      
      
      
      Published with <a href="https://hugoblox.com/?utm_campaign=poweredby" target="_blank" rel="noopener">Hugo Blox Builder</a> ‚Äî the free, <a href="https://github.com/HugoBlox/hugo-blox-builder" target="_blank" rel="noopener">open source</a> website builder that empowers creators.
    
  </p>

</footer>

    </div>

    
    











  </body>
</html>
