<!doctype html>
<!-- This site was created with Hugo Blox. https://hugoblox.com -->
<!-- Last Published: July 7, 2025 --><html lang="en-us" dir="ltr"
      data-wc-theme-default="system">
  
<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <meta name="generator" content="Hugo Blox Builder 0.3.1" />

  
  












  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Jiyuan Liu" />

  
  
  
    
  
  <meta name="description" content="VeloVI Posterior Predictive Distribution: Theory and Numerical Example Introduction Equation (28) describes the concept of a posterior predictive distribution for the unspliced abundance ($u_n^*$ ) of a cell $n$ in the VeloVI model. It&rsquo;s a powerful idea that allows us to understand what new, unobserved data would look like, given our current model and the observed data.
" />

  
  <link rel="alternate" hreflang="en-us" href="http://localhost:1313/post/11b.velocity_velovi/velovi_posterior_predictive_distribution/" />

  
  
  
  
    
    <link rel="stylesheet" href="/css/themes/emerald.min.css" />
  

  
  
    
    <link href="/dist/wc.min.css" rel="stylesheet" />
  

  
  
  

  
    
    <link href="/css/custom.min.9a8c1508a999d285edca71446056519c3773544bdbe9fd11396354c75238a016.css" rel="stylesheet" />
  

  <script>
     
    window.hbb = {
       defaultTheme: document.documentElement.dataset.wcThemeDefault,
       setDarkTheme: () => {
        document.documentElement.classList.add("dark");
        document.documentElement.style.colorScheme = "dark";
      },
       setLightTheme: () => {
        document.documentElement.classList.remove("dark");
        document.documentElement.style.colorScheme = "light";
      }
    }

    console.debug(`Default Hugo Blox Builder theme is ${window.hbb.defaultTheme}`);

    if ("wc-color-theme" in localStorage) {
      localStorage.getItem("wc-color-theme") === "dark" ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
    } else {
      window.hbb.defaultTheme === "dark" ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
      if (window.hbb.defaultTheme === "system") {
        window.matchMedia("(prefers-color-scheme: dark)").matches ? window.hbb.setDarkTheme() : window.hbb.setLightTheme();
      }
    }
  </script>

  <script>
    
    document.addEventListener('DOMContentLoaded', function () {
      
      let checkboxes = document.querySelectorAll('li input[type=\'checkbox\'][disabled]');
      checkboxes.forEach(e => {
        e.parentElement.parentElement.classList.add('task-list');
      });

      
      const liNodes = document.querySelectorAll('.task-list li');
      liNodes.forEach(nodes => {
        let textNodes = Array.from(nodes.childNodes).filter(node => node.nodeType === 3 && node.textContent.trim().length > 1);
        if (textNodes.length > 0) {
          const span = document.createElement('label');
          textNodes[0].after(span);  
          span.appendChild(nodes.querySelector('input[type=\'checkbox\']'));
          span.appendChild(textNodes[0]);
        }
      });
    });
  </script>

  
  
  




































  
  

  
  <link rel="icon" type="image/png" href="/media/icon_hu16496738209820128384.png" />
  <link rel="apple-touch-icon" type="image/png" href="/media/icon_hu17763857046462676418.png" />

  <link rel="canonical" href="http://localhost:1313/post/11b.velocity_velovi/velovi_posterior_predictive_distribution/" />

  
  
  
  
  
  
  
  
    
    
  
  

  
  
    
    
  
  <meta property="twitter:card" content="summary" />
  
    <meta property="twitter:site" content="@GetResearchDev" />
    <meta property="twitter:creator" content="@GetResearchDev" />
  
  <meta property="og:site_name" content="Jiyuan Liu" />
  <meta property="og:url" content="http://localhost:1313/post/11b.velocity_velovi/velovi_posterior_predictive_distribution/" />
  <meta property="og:title" content="Jiyuan Liu" />
  <meta property="og:description" content="VeloVI Posterior Predictive Distribution: Theory and Numerical Example
Introduction
Equation (28) describes the concept of a posterior predictive distribution for the unspliced abundance ($u_n^*$
) of a cell $n$
 in the VeloVI model. It&rsquo;s a powerful idea that allows us to understand what new, unobserved data would look like, given our current model and the observed data." /><meta property="og:image" content="http://localhost:1313/media/icon_hu6671943710606299486.png" />
    <meta property="twitter:image" content="http://localhost:1313/media/icon_hu6671943710606299486.png" /><meta property="og:locale" content="en-us" />
  
    
    
  

  



  


  <title>Jiyuan Liu</title>

  
  
  
  
  
    
    
  
  
  <style>
    @font-face {
      font-family: 'Inter var';
      font-style: normal;
      font-weight: 100 900;
      font-display: swap;
      src: url(/dist/font/Inter.var.woff2) format(woff2);
    }
  </style>

  

  
  


  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
  
  
  
  
  
  
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
        
        
          
        
          
        
          
        
          
        
          
        
        
        
      
    
      
      
    
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  







<link type="text/css" rel="stylesheet" href="/dist/pagefind/pagefind-ui.be766eb419317a14ec769d216e9779bfe8f3737c80e780f4ba0dafb57a41a482.css" integrity="sha256-vnZutBkxehTsdp0hbpd5v&#43;jzc3yA54D0ug2vtXpBpII=" />


<script src="/dist/pagefind/pagefind-ui.87693d7c6f2b3b347ce359d0ede762c033419f0a32b22ce508c335a81d841f1b.js" integrity="sha256-h2k9fG8rOzR841nQ7ediwDNBnwoysizlCMM1qB2EHxs="></script>


<script>window.hbb.pagefind = {"baseUrl":"/"};</script>

<style>
  html.dark {
    --pagefind-ui-primary: #eeeeee;
    --pagefind-ui-text: #eeeeee;
    --pagefind-ui-background: #152028;
    --pagefind-ui-border: #152028;
    --pagefind-ui-tag: #152028;
  }
</style>

<script>
  window.addEventListener('DOMContentLoaded', (event) => {
    new PagefindUI({
      element: "#search",
      showSubResults: true,
      baseUrl: window.hbb.pagefind.baseUrl,
      bundlePath: window.hbb.pagefind.baseUrl + "pagefind/",
    });
  });
  document.addEventListener('DOMContentLoaded', () => {
    let element = document.getElementById('search');
    let trigger = document.getElementById('search_toggle');

    if (trigger) {
      trigger.addEventListener('click', () => {
        element.classList.toggle('hidden');
        element.querySelector("input").value = ""
        element.querySelector("input").focus()

        if (!element.classList.contains('hidden')) {
          let clear_trigger = document.querySelector('.pagefind-ui__search-clear');

          if (clear_trigger && !clear_trigger.hasAttribute('listenerOnClick')) {
            clear_trigger.setAttribute('listenerOnClick', 'true');

            clear_trigger.addEventListener('click', () => {
              element.classList.toggle('hidden');
            });
          }
        }

      });
    }
  });
</script>










  
  
  <link type="text/css" rel="stylesheet" href="/dist/lib/katex/katex.min.505d5f829022bb7b4f24dfee0aa1141cd7bba67afe411d1240335f820960b5c3.css" integrity="sha256-UF1fgpAiu3tPJN/uCqEUHNe7pnr&#43;QR0SQDNfgglgtcM=" />
  
  
  <script defer src="/dist/lib/katex/katex.min.dc84b296ec3e884de093158f760fd9d45b6c7abe58b5381557f4e138f46a58ae.js" integrity="sha256-3ISyluw&#43;iE3gkxWPdg/Z1Ftser5YtTgVV/ThOPRqWK4="></script>
  
  
  
  
  <script defer src="/js/katex-renderer.6579ec9683211cfb952064aedf3a3baea5eeb17a061775b32b70917474637c80.js" integrity="sha256-ZXnsloMhHPuVIGSu3zo7rqXusXoGF3WzK3CRdHRjfIA="></script>
  
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  






  
  
  
  
  
  
  
  <script
    defer
    src="/js/hugo-blox-en.min.js"
    integrity=""
  ></script>

  
  








  
    
      
      <script async defer src="https://buttons.github.io/buttons.js"></script>

      
    
  




</head>

  <body class="dark:bg-hb-dark dark:text-white page-wrapper" id="top">
    <div id="page-bg"></div>
    <div class="page-header sticky top-0 z-30">
      
      
      
        
        
        
          <header id="site-header" class="header">
  <nav class="navbar px-3 flex justify-left">
    <div class="order-0 h-100">
      
      <a class="navbar-brand" href="/" title="Jiyuan Liu">
        Home
      </a>
    </div>
    
    <input id="nav-toggle" type="checkbox" class="hidden" />
    <label
      for="nav-toggle"
      class="order-3 cursor-pointer flex items-center lg:hidden text-dark dark:text-white lg:order-1">
      <svg id="show-button" class="h-6 fill-current block" viewBox="0 0 20 20">
        <title>Open Menu</title>
        <path d="M0 3h20v2H0V3z m0 6h20v2H0V9z m0 6h20v2H0V0z"></path>
      </svg>
      <svg id="hide-button" class="h-6 fill-current hidden" viewBox="0 0 20 20">
        <title>Close Menu</title>
        <polygon
          points="11 9 22 9 22 11 11 11 11 22 9 22 9 11 -2 11 -2 9 9 9 9 -2 11 -2"
          transform="rotate(45 10 10)"></polygon>
      </svg>
    </label>
    

    
    
    <ul
      id="nav-menu"
      class="navbar-nav order-3 hidden lg:flex w-full pb-6 lg:order-1 lg:w-auto lg:space-x-2 lg:pb-0 xl:space-x-8 justify-left
      ">
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/"
        >Bio</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/#papers"
        >Papers</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/#talks"
        >Talks</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/post"
        >Blogs</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/experience/"
        >Experience</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/projects/"
        >Projects</a
        >
      </li>
      
      
      
      
      
      
      <li class="nav-item">
        <a
          class="nav-link "
          
          href="/teaching/"
        >Learning</a
        >
      </li>
      
      
      
    </ul>

    <div class="order-1 ml-auto flex items-center md:order-2 lg:ml-0">

      
      
      
      <button
        aria-label="search"
        class="text-black hover:text-primary  inline-block px-3 text-xl dark:text-white"
        id="search_toggle">
        <svg xmlns="http://www.w3.org/2000/svg" height="16" width="16" viewBox="0 0 512 512" fill="currentColor"><path d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"/></svg>
      </button>
      

      
      
      <div class="px-3 text-black hover:text-primary-700 dark:text-white dark:hover:text-primary-300
            [&.active]:font-bold [&.active]:text-black/90 dark:[&.active]:text-white">
        <button class="theme-toggle mt-1" accesskey="t" title="appearance">
          <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
               fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
               stroke-linejoin="round" class="dark:hidden">
            <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
          </svg>
          <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
               fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
               stroke-linejoin="round" class=" dark:block [&:not(dark)]:hidden">
            <circle cx="12" cy="12" r="5"></circle>
            <line x1="12" y1="1" x2="12" y2="3"></line>
            <line x1="12" y1="21" x2="12" y2="23"></line>
            <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
            <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
            <line x1="1" y1="12" x2="3" y2="12"></line>
            <line x1="21" y1="12" x2="23" y2="12"></line>
            <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
            <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
          </svg>
        </button>
      </div>
      

      
      

      
      
    </div>
  </nav>
</header>


<div id="search" class="hidden p-3"></div>


        
      
    </div>
    <div class="page-body  my-10">
      





<div class="mx-auto flex max-w-screen-xl">
  



<aside class="hb-sidebar-container max-lg:[transform:translate3d(0,-100%,0)] lg:hidden xl:block">
  
  <div class="px-4 pt-4 lg:hidden">
    
    
  </div>
  <div class="hb-scrollbar lg:h-[calc(100vh-var(--navbar-height))]">
    <ul class="flex flex-col gap-1 lg:hidden">
      
      
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/event/"
    
  >Recent &amp; Upcoming Talks
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/event/example/"
    
  >Example Talk
    </a>
              
            </li></ul>
      </div></li>
        <li class="open"><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/"
    
  >Computational Blogs
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11c.velocity_steady_state/"
    
  >🧬 Math derivation for steady-state RNA velocity model
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19c.bayes_variational_infer_elbo/"
    
  >🎓 Math Derivation for ELBO/KL in Bayesian Inference and VAEs
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19b.scrnaseq_impute_saver/"
    
  >🧮 Math Derivation for scRNAseq Imputation (SAVER)
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19a.conjugate_prior_bayes/"
    
  >📊 Math Derivation in Bayesian Inference-- Bypassing Marginals Using Conjugate Priors
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/19.ememwithmapvariationalbayesian/"
    
  >🧬 EM algorithm, EM with max a posterior, full Bayesian inferences (variational Bayesian) which might be used in auto-coder
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/18.what_make_neural_network_powerful_popular/"
    
  >🧬 what makes neural network so powerful and popular
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/17.smooth.tests.of.goodness.of.fit/"
    
  >🧬 Smooth tests of goodness of Fit used in Stijn Hawinkel ’s plos one paper
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/16.how.good.is.negative.binominal/"
    
  >🧬 how good is negative binomial, why AI works for biology
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/15.use.seurat.for.bulk.rnaseq/"
    
  >🧬 Seurat to analysis large sample size of bulk RNAseq. Discuss Wixcox is better for clinical samples
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/13z.scvi_integrate_scrna_spatial/"
    
  >🧬 Deep Neural Network to integrate scRNAseq and spatial genomics
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/14.spatial_genomics_presentation/"
    
  >🧬 Spatial genomics presentation
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/12.basian_abundance_statistics/"
    
  >🧬 Bayesian abundance statistics in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/13.yolov8/"
    
  >🧬 Yolov8
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11a.velocity_unraveled/"
    
  >🧬 Math Derivation of CME-defined Stochastic Model of RNA Velocity
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11.velocity_reasoning/"
    
  >🧬 Velocity reasoning and discussion
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/10.cca_reasoning/"
    
  >🧬 CCA reasoning and integration discussion
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/09.sctransform/"
    
  >🧬 sctransfrom reasoning
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/08.clinical_relevance/"
    
  >🧬 Clinical relevance evaluation in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/07.chord_diagram/"
    
  >🧬 Expressive cord diagram in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/06.noise_or_true_stem_cell/"
    
  >🧬 Determine whether it’s noise or true stem like cells in PRCC
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/05.xensoft_caki_paper/"
    
  >🧬 quantify xenograft and host gene expression from the same sample
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/04.huangcong_oncogene/"
    
  >🧬 counteract treatment-specific background in differential expressed gene analysis (huangcong paper)
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/03.well-based_scrnaseq_herv_quantification/"
    
  >🧬 A snakemake pipline-- quantify hERV, transgenes and genome from well-based single-cell RNAseq
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/02.snakemake_quantify-herv_10xscrnaseq/"
    
  >🧬 A snakemake pipline-- quantify hERV, transgenes and genome from 10x scRNAseq
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/01a.repeatmasker/"
    
  >🛠️ Generate and validate ERV gtf that matches the latest reference genome
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/01.bulkrnaseq_allow_herv_quantification/"
    
  >🧬 Quantify hERV and transgenes from Bulk-RNAseq
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/00a.seqwins/"
    
  >⚡ SeqWins, An R Package for Flexible Base Trimming and Comprehensive FASTQ Analysis on Windows
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/00.model_aerosol_depostion/"
    
  >💨 Modeling the deposition of infectious agents suggests that aerosolized drugs offer enhanced specificity
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.get-started/"
    
  >🎉 Easily create your own simple yet highly customizable blog
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.second-brain/"
    
  >🧠 Sharpen your thinking with a second brain
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.data-visualization/"
    
  >📈 Communicate your results effectively with the best data visualizations
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.teach-courses/"
    
  >👩🏼‍🏫 Teach academic courses
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/99.project-management/"
    
  >✅ Manage your projects
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_derive_time-varying_alpha_model/"
    
  >velovi_derive_time-varying_alpha_model
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_elbo_kl_breakdown/"
    
  >velovi_elbo_kl_breakdown
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_elbo_term_with_expectation/"
    
  >velovi_elbo_term_with_expectation
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_expected_state_n_latent_time/"
    
  >velovi_expected_state_n_latent_time
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_model_n_inference/"
    
  >velovi_model_n_inference
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_parameter_initialize_n_learn/"
    
  >velovi_parameter_initialize_n_learn
    </a>
              
            </li><li class="flex flex-col open"><a
    class="hb-sidebar-custom-link
      sidebar-active-item bg-primary-100 font-semibold text-primary-800 dark:bg-primary-300 dark:text-primary-900"
    href="/post/11b.velocity_velovi/velovi_posterior_predictive_distribution/"
    
  >velovi_posterior_predictive_distribution
    </a>
  
    <ul class="hb-sidebar-mobile-toc"><li>
              <a
                href="#introduction"
                class="hb-docs-link"
              >Introduction</a>
            </li>
          <li>
              <a
                href="#understanding-the-components"
                class="hb-docs-link"
              >Understanding the Components</a>
            </li>
          <li>
              <a
                href="#putting-it-all-together-the-intuition"
                class="hb-docs-link"
              >Putting it All Together: The Intuition</a>
            </li>
          <li>
              <a
                href="#why-is-this-important"
                class="hb-docs-link"
              >Why is this Important?</a>
            </li>
          <li>
              <a
                href="#monte-carlo-approximation"
                class="hb-docs-link"
              >Monte Carlo Approximation</a>
            </li>
          <li>
              <a
                href="#numerical-example"
                class="hb-docs-link"
              >Numerical Example</a>
            </li>
          <li>
              <a
                href="#result"
                class="hb-docs-link"
              >Result</a>
            </li>
          </ul>
  
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_reparameterization_trick/"
    
  >velovi_reparameterization_trick
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/post/11b.velocity_velovi/velovi_uncertainty_calculation/"
    
  >velovi_uncertainty_calculation
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/"
    
  >learning
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/visualize_distribution/"
    
  >Dirichlet Distribution
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/stanford_ml_em/"
    
  >Lecture 14 - Expectation-Maximization Algorithms | Stanford CS229 Machine Learning
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/linear_algebra/"
    
  >linear algebra
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/other/home_improvement/"
    
  >linear algebra
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/method_of_charateristics/"
    
  >Method of characteristics
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/method_of_seperation_of_variables/"
    
  >Separation of Variables
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/js/"
    
  >Learn JavaScript
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/teaching/python/"
    
  >Learn Python
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/projects/"
    
  >Projects
    </a></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/"
    
  >Projects
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/pandas/"
    
  >Pandas
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/pytorch/"
    
  >PyTorch
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/project/scikit/"
    
  >scikit-learn
    </a>
              
            </li></ul>
      </div></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/experience/"
    
  >Experience
    </a></li>
        <li class=""><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/"
    
  >Publications
        <span data-hb-sidebar-toggle>
            <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" class="h-[18px] min-w-[18px] rounded-sm p-0.5 hover:bg-gray-800/5 dark:hover:bg-gray-100/5"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" class="origin-center transition-transform rtl:-rotate-180"></path></svg>
        </span>
    </a><div class="ltr:pr-0 overflow-hidden">
        <ul class="hb-sidebar-list"><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/preprint/"
    
  >An example preprint / working paper
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/journal-article/"
    
  >An example journal article
    </a>
              
            </li><li class="flex flex-col "><a
    class="hb-sidebar-custom-link
      text-gray-500 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-300 dark:hover:bg-primary-800 dark:hover:text-gray-50"
    href="/publication/conference-paper/"
    
  >An example conference paper
    </a>
              
            </li></ul>
      </div></li>
    </ul>

    <div class="max-xl:hidden h-0 w-64 shrink-0"></div></div>

</aside>
  

<nav class="hb-toc order-last hidden w-64 shrink-0 xl:block print:hidden px-4" aria-label="table of contents">
  











  <div class="hb-scrollbar text-sm [hyphens:auto] sticky top-16 overflow-y-auto pr-4 pt-6 max-h-[calc(100vh-var(--navbar-height)-env(safe-area-inset-bottom))] -mr-4 rtl:-ml-4"><p class="mb-4 font-semibold tracking-tight">On this page</p><ul>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#introduction">Introduction</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#understanding-the-components">Understanding the Components</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#unobserved-random-variables">Unobserved Random Variables</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#the-posterior-predictive-distribution">The Posterior Predictive Distribution</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#the-integral-marginalization">The Integral (Marginalization)</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#the-generative-model--decoder">The Generative Model / Decoder</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#the-approximate-posterior--encoder">The Approximate Posterior / Encoder</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#putting-it-all-together-the-intuition">Putting it All Together: The Intuition</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#why-is-this-important">Why is this Important?</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#monte-carlo-approximation">Monte Carlo Approximation</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#numerical-example">Numerical Example</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#our-goal">Our Goal</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#what-we-assume-we-have">What We Assume We Have</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#observed-data-for-cell-1-gene-a">Observed Data for Cell 1, Gene A:</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#learned-biophysical-parameters-hahahugoshortcode15s67hbhb-for-gene-a">Learned Biophysical Parameters (HAHAHUGOSHORTCODE15s67HBHB) for Gene A:</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#learned-neural-network-parameters-hahahugoshortcode15s76hbhb-for-the-encoder-and-decoder">Learned Neural Network Parameters (HAHAHUGOSHORTCODE15s76HBHB) for the Encoder and Decoder:</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#the-monte-carlo-approximation-process-step-by-step-numerical-example">The Monte Carlo Approximation Process (Step-by-Step Numerical Example)</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-8 rtl:pr-8 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#iteration-1-of-m-sample-1">Iteration 1 of M (Sample #1):</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="pl-4 rtl:pr-4 inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#repeat-for-m-samples-conceptual">Repeat for M Samples (Conceptual)</a>
      </li>
      <li class="my-2 scroll-my-6 scroll-py-6">
        <a class="font-semibold inline-block text-gray-500 hover:text-gray-900 dark:text-gray-400 dark:hover:text-gray-300 w-full break-words" href="#result">Result</a>
      </li></ul>

  
  
    
    
  



    












  </div>
  </nav>


  <article class="w-full break-words flex min-h-[calc(100vh-var(--navbar-height))] min-w-0 justify-center pb-8 pr-[calc(env(safe-area-inset-right)-1.5rem)]">
    <main class="w-full min-w-0 max-w-6xl px-6 pt-4 md:px-12">

      

      <h1 class="mt-2 text-4xl font-bold tracking-tight text-slate-900 dark:text-slate-100"></h1>

      <div class="mt-4 mb-16">
      <div class="text-gray-500 dark:text-gray-300 text-sm flex items-center flex-wrap gap-y-2">
        

        
        <span class="mx-1">·</span>
        <span class="mx-1">
          9 min read
        </span>
        
        </div>

        <div class="mt-3">
          





        </div>
      </div>



      
      
      

      
      

      <div class="prose prose-slate lg:prose-xl dark:prose-invert">
        <h1 id="velovi-posterior-predictive-distribution-theory-and-numerical-example">VeloVI Posterior Predictive Distribution: Theory and Numerical Example</h1>
<h2 id="introduction">Introduction</h2>
<p>Equation (28) describes the concept of a posterior predictive distribution for the unspliced abundance ($u_n^*$
) of a cell $n$
 in the VeloVI model. It&rsquo;s a powerful idea that allows us to understand what new, unobserved data would look like, given our current model and the observed data.</p>
<h2 id="understanding-the-components">Understanding the Components</h2>
<h3 id="unobserved-random-variables">Unobserved Random Variables</h3>
<p>$u_n^*$
 and $s_n^*$
 are unobserved random variables representing posterior predictive values of unspliced and spliced abundances for cell $n$
.</p>
<p>This means we&rsquo;re not talking about the actual observed $u_n, s_n$
 that we fed into the model. Instead, we&rsquo;re thinking about hypothetical, new (unseen) unspliced and spliced RNA counts for this same cell $n$
, assuming its underlying biological state (represented by $z_n$
 and $\pi_n$
) is consistent with what we observed.</p>
<p>&ldquo;Posterior predictive&rdquo; means we&rsquo;re predicting after having observed data ($u_n, s_n$
) and after having inferred the likely latent states ($z_n, \pi_n$
) for that data.</p>
<h3 id="the-posterior-predictive-distribution">The Posterior Predictive Distribution</h3>
<p>The main equation we want to calculate is:</p>

$$
p(u_n^* \mid u_n, s_n) = \int p_\theta(u_n^* \mid z_n, \pi_n) q_\phi(z_n, \pi_n \mid u_n, s_n) d\pi_n dz_n \tag{28}
$$


<p>This is what we want to calculate. It&rsquo;s the probability of observing a new unspliced count $u_n^*$
, given the original observed counts $u_n$
 and $s_n$
 for cell $n$
.</p>
<p>It essentially answers: &ldquo;If I were to measure cell $n$
 again, what kind of unspliced count would I expect to see, given what I already know about it?&rdquo;</p>
<h3 id="the-integral-marginalization">The Integral (Marginalization)</h3>
<p>The integral sign $\int \ldots d\pi_n dz_n$
 means we are marginalizing out (summing over all possible values of) the latent variables $z_n$
 and $\pi_n$
.</p>
<p>Why? Because $u_n^*$
 doesn&rsquo;t directly depend on $u_n, s_n$
. Instead, $u_n^*$
 depends on the cell&rsquo;s underlying biological state ($z_n, \pi_n$
), which we inferred from $u_n, s_n$
. To get the total probability of $u_n^*$
, we need to consider all possible values of $z_n$
 and $\pi_n$
 that could have led to the observed $u_n, s_n$
, weighted by how likely they are.</p>
<h3 id="the-generative-model--decoder">The Generative Model / Decoder</h3>
<p>$p_\theta(u_n^* \mid z_n, \pi_n)$
 (The Generative Model / Decoder):</p>
<p>This term is part of the decoder or generative model of VeloVI. It describes the probability of generating an unspliced count $u_n^*$
, given specific values of the latent variable $z_n$
 and the kinetic state probabilities $\pi_n$
, and using the biophysical parameters $\theta$
 (like $\alpha, \beta, \gamma$
) that govern the RNA dynamics.</p>
<p>In VeloVI, $z_n$
 helps determine the latent time, and $\pi_n$
 determines the kinetic state (e.g., induction, repression). These then feed into the kinetic equations (which use $\theta$
) to predict the mean expected abundance $\bar{u}^{(g)}$
. $p_\theta(u_n^* \mid z_n, \pi_n)$
 would then be a distribution (like a Normal distribution) centered around that predicted mean, with variance $\sigma_{gu}$
.</p>
<h3 id="the-approximate-posterior--encoder">The Approximate Posterior / Encoder</h3>
<p>$q_\phi(z_n, \pi_n \mid u_n, s_n)$
 (The Approximate Posterior / Encoder):</p>
<p>This is the approximate posterior distribution provided by the encoder network. It represents VeloVI&rsquo;s learned understanding of the probability of a cell having latent state ($z_n, \pi_n$
), given its observed unspliced and spliced counts ($u_n, s_n$
).</p>
<p>This is the distribution from which we draw samples ($z_n^*, \pi_n^*$
) during training. It&rsquo;s parameterized by the neural network parameters $\phi$
.</p>
<h2 id="putting-it-all-together-the-intuition">Putting it All Together: The Intuition</h2>
<p>Equation (28) essentially says:</p>
<p>&ldquo;To predict a new, unobserved unspliced count ($u_n^*$
) for a cell ($n$
) that we&rsquo;ve already measured ($u_n, s_n$
):</p>
<ol>
<li>
<p>Consider all possible underlying biological states ($z_n, \pi_n$
) that could explain the observed data ($u_n, s_n$
). The encoder ($q_\phi$
) tells us how likely each of these states is.</p>
</li>
<li>
<p>For each of these possible states, use the generative model ($p_\theta$
) to predict what unspliced count ($u_n^*$
) would be produced.</p>
</li>
<li>
<p>Sum up these predictions, weighted by how likely each underlying state is.&rdquo;</p>
</li>
</ol>
<h2 id="why-is-this-important">Why is this Important?</h2>
<ul>
<li>
<p><strong>Model Evaluation</strong>: It allows you to check if your model makes reasonable predictions for new data. If the predicted $u_n^*$
 looks very different from the original $u_n$
, it suggests a problem with your model&rsquo;s fit or assumptions.</p>
</li>
<li>
<p><strong>Missing Data Imputation (conceptual)</strong>: While not explicitly used for imputation in VeloVI&rsquo;s core, posterior predictive distributions are the foundation for imputing missing data or generating synthetic data that resembles the observed data.</p>
</li>
<li>
<p><strong>Uncertainty Quantification</strong>: The posterior predictive distribution gives you not just a single predicted value, but a full distribution, allowing you to quantify the uncertainty in your predictions.</p>
</li>
</ul>
<h2 id="monte-carlo-approximation">Monte Carlo Approximation</h2>
<p>In practice, this integral is usually intractable (cannot be solved analytically). So, in VAEs, it&rsquo;s often approximated using Monte Carlo sampling:</p>
<ol>
<li>
<p>Draw many samples of ($z_n, \pi_n$
) from the approximate posterior $q_\phi(z_n, \pi_n \mid u_n, s_n)$
.</p>
</li>
<li>
<p>For each sample, draw a new $u_n^*$
 from $p_\theta(u_n^* \mid z_n, \pi_n)$
.</p>
</li>
<li>
<p>The collection of these $u_n^*$
 samples forms an empirical approximation of the posterior predictive distribution.</p>
</li>
</ol>
<h2 id="numerical-example">Numerical Example</h2>
<p>Let&rsquo;s calculate (or more accurately, approximate) Equation (28) using numeric examples. We&rsquo;ll use our scenario of Cell 1 and Gene A for the numeric example.</p>
<h3 id="our-goal">Our Goal</h3>
<p>Generate samples of $u_{1A}^*$
 to approximate the distribution $p(u_{1A}^* \mid u_{1A}, s_{1A})$
.</p>
<h3 id="what-we-assume-we-have">What We Assume We Have</h3>
<p>After VeloVI has been trained and its parameters $\theta$
 and $\phi$
 have converged:</p>
<h4 id="observed-data-for-cell-1-gene-a">Observed Data for Cell 1, Gene A:</h4>
<ul>
<li>$u_{1A} = 0.8$
</li>
<li>$s_{1A} = 0.2$
 (These are the inputs to our encoder)</li>
</ul>
<h4 id="learned-biophysical-parameters-hahahugoshortcode15s67hbhb-for-gene-a">Learned Biophysical Parameters ($\theta$
) for Gene A:</h4>
<p>(These are fixed values after training)</p>
<ul>
<li>Transcription Rate ($\alpha_{A1}$
) = 1.2</li>
<li>Splicing Rate ($\beta_A$
) = 0.2</li>
<li>Degradation Rate ($\gamma_A$
) = 0.1</li>
<li>Switching Time ($t_A^s$
) = 8.0</li>
<li>Unspliced Std. Dev. ($\sigma_{Au}$
) = 0.15</li>
<li>Spliced Std. Dev. ($\sigma_{As}$
) = 0.15</li>
</ul>
<p>(For simplicity, let&rsquo;s assume all state scaling factors $c_k = 1$
 for this example, though VeloVI uses a specific $c_4 = 0.1$
)</p>
<h4 id="learned-neural-network-parameters-hahahugoshortcode15s76hbhb-for-the-encoder-and-decoder">Learned Neural Network Parameters ($\phi$
) for the Encoder and Decoder:</h4>
<p>(These are the weights and biases of the networks. We don&rsquo;t see the numbers directly, but we know their effect on the outputs.)</p>
<p><strong>Encoder&rsquo;s Output for $z_1$
 (given $u_1, s_1$
)</strong>: This defines the approximate posterior $q_\phi(z_1 \mid u_1, s_1)$
, which is a Gaussian.</p>
<ul>
<li>Mean ($\mu_{z_1}$
) = (0.5, -0.3)</li>
<li>Log-Variance ($\log\Sigma_{z_1}$
) = (-0.5, -0.5)</li>
</ul>
<p>This implies $\Sigma_{z_1} = \text{diag}(e^{-0.5}, e^{-0.5}) \approx \text{diag}(0.6065, 0.6065)$
 (Let&rsquo;s assume $z_n$
 is 2-dimensional for this example)</p>
<p><strong>Encoder&rsquo;s Output for $\pi_{1A}$
 (given $z_1$
)</strong>: This defines $q_\phi(\pi_{1A} \mid z_1)$
, which is a Dirichlet.</p>
<p>Example concentration parameters (output for a specific $z_1$
): $\text{conc}_{1A} = (5.0, 0.1, 0.2, 0.1)$
 (This strongly favors State 1: Induction)</p>
<p><strong>Decoder&rsquo;s Output for $\rho_{1A}^{(k)}$
 (given $z_1$
)</strong>: These networks take $z_1$
 and output scaled latent times.</p>
<ul>
<li>Example output for a specific $z_1$
: $\rho_{1A}^{(1)} = 0.5$
 (for state 1)</li>
<li>Example output for a specific $z_1$
: $\rho_{1A}^{(3)} = 0.8$
 (for state 3)</li>
</ul>
<h3 id="the-monte-carlo-approximation-process-step-by-step-numerical-example">The Monte Carlo Approximation Process (Step-by-Step Numerical Example)</h3>
<p>We will repeat the following steps many times (say, $M = 1000$
 times) to get enough samples of $u_{1A}^*$
. For this explanation, we&rsquo;ll trace just one single sample to illustrate the process.</p>
<h4 id="iteration-1-of-m-sample-1">Iteration 1 of M (Sample #1):</h4>
<p><strong>Step 1</strong>: Sample latent variables ($z_1^*, \pi_{1A}^*$
, and determine $k_{1A}$
) from the approximate posterior $q_\phi(z_1, \pi_{1A} \mid u_{1A}, s_{1A})$
.</p>
<p><strong>1a. Sample $z_1^*$
 from $q_\phi(z_1 \mid u_{1A}, s_{1A})$
 using the reparameterization trick:</strong></p>
<p>Recall $q_\phi(z_1 \mid u_{1A}, s_{1A})$
 is $\mathcal{N}(\mu_{z_1}, \Sigma_{z_1})$
, where $\mu_{z_1} = (0.5, -0.3)$
 and $\Sigma_{z_1} = \text{diag}(0.6065, 0.6065)$
.</p>
<ul>
<li>Draw a random standard normal noise vector $\epsilon_z$
: Let $\epsilon_z = (0.2, -0.1)$
.</li>
<li>Calculate $z_1^* = \mu_{z_1} + \sqrt{\Sigma_{z_1}} \odot \epsilon_z$
 (where $\odot$
 is element-wise multiplication):</li>
</ul>

$$
z_1^* = (0.5, -0.3) + (\sqrt{0.6065}, \sqrt{0.6065}) \odot (0.2, -0.1) \tag{29}
$$



$$
z_1^* = (0.5, -0.3) + (0.7788, 0.7788) \odot (0.2, -0.1) \tag{30}
$$



$$
z_1^* = (0.5, -0.3) + (0.1558, -0.0779) \tag{31}
$$



$$
z_1^* = (0.6558, -0.3779) \tag{32}
$$


<p><strong>1b. Given $z_1^*$
, sample $\pi_{1A}^*$
 from $q_\phi(\pi_{1A} \mid z_1^*)$
:</strong></p>
<ul>
<li>The encoder (a specific part of it) takes this sampled $z_1^* = (0.6558, -0.3779)$
 as input and outputs the concentration parameters for $\pi_{1A}$
.</li>
<li>Let&rsquo;s assume for $z_1^* = (0.6558, -0.3779)$
, the encoder outputs: $\text{conc}_{1A} = (5.0, 0.1, 0.2, 0.1)$
.</li>
<li>Sample $\pi_{1A}^*$
 from $\text{Dirichlet}(5.0, 0.1, 0.2, 0.1)$
. This is a stochastic step.</li>
<li>Let&rsquo;s say we get the sample: $\pi_{1A}^* = (0.95, 0.02, 0.02, 0.01)$
.</li>
</ul>
<p><strong>1c. Determine the kinetic state $k_{1A}$
 from $\pi_{1A}^*$
:</strong></p>
<ul>
<li>We choose the state with the highest probability. Here, 0.95 is the highest, corresponding to the first state.</li>
<li>So, $k_{1A} = 1$
 (Induction).</li>
</ul>
<p><strong>Step 2</strong>: Generate a new unspliced abundance $u_{1A}^*$
 from the generative model $p_\theta(u_{1A}^* \mid z_1^*, \pi_{1A}^*)$
.</p>
<p><strong>2a. Decode $z_1^*$
 to latent time ($t_{1A}^{(k)}$
):</strong></p>
<ul>
<li>The decoder network (specifically, $h_{\text{ind}}$
 for Induction states) takes $z_1^* = (0.6558, -0.3779)$
 and outputs a scaled time $\rho_{1A}^{(1)}$
.</li>
<li>Let&rsquo;s assume the decoder outputs $\rho_{1A}^{(1)} = 0.5$
.</li>
<li>Calculate the actual latent time: $t_{1A}^{(1)} = \rho_{1A}^{(1)} \times t_A^s = 0.5 \times 8.0 = 4.0$
.</li>
</ul>
<p><strong>2b. Calculate the predicted mean unspliced abundance $\bar{u}^{(A)}$
 using the kinetic equations and $\theta$
:</strong></p>
<p>Since $k_{1A} = 1$
 (Induction), we use the kinetic equation for the induction phase.</p>

$$
\bar{u}^{(A)}(t=4.0, k=1) = \frac{\alpha_{A1}}{\beta_A}(1 - e^{-\beta_A t}) \tag{33}
$$



$$
\bar{u}^{(A)}(4.0, 1) = \frac{1.2}{0.2}(1 - e^{-0.2 \times 4.0}) \tag{34}
$$



$$
\bar{u}^{(A)}(4.0, 1) = 6(1 - e^{-0.8}) \tag{35}
$$



$$
\bar{u}^{(A)}(4.0, 1) = 6(1 - 0.4493) \tag{36}
$$



$$
\bar{u}^{(A)}(4.0, 1) = 6 \times 0.5507 = 3.3042 \tag{37}
$$


<p><strong>2c. Sample $u_{1A}^*$
 from $p_\theta(u_{1A}^* \mid z_1^*, \pi_{1A}^*)$
:</strong></p>
<ul>
<li>This distribution is a Normal distribution: $\mathcal{N}(\text{mean} = \bar{u}^{(A)}, \text{variance} = \sigma_{Au}^2)$
.</li>
<li>So, $u_{1A}^* \sim \mathcal{N}(3.3042, 0.15^2)$
.</li>
<li>Draw one sample from this Normal distribution (this is again a stochastic step).</li>
<li>Let&rsquo;s say we sample: $u_{1A}^* = 3.25$
. This is our first sample for the posterior predictive distribution.</li>
</ul>
<h3 id="repeat-for-m-samples-conceptual">Repeat for M Samples (Conceptual)</h3>
<p>You would repeat these two main steps (Step 1 and Step 2) many times (e.g., $M = 1000$
). Each repetition would involve new random draws for $\epsilon_z$
, new Dirichlet samples for $\pi_{1A}^*$
, and new Normal samples for $u_{1A}^*$
.</p>
<p>For instance, if we ran a second iteration, we might get:</p>
<p><strong>Sample #2</strong>: $\epsilon_z = (-0.5, 0.3) \rightarrow z_1^* = (0.0106, -0.0663) \rightarrow \text{conc}_{1A} = (4.0, 0.3, 0.1, 0.1) \rightarrow \pi_{1A}^* = (0.90, 0.07, 0.02, 0.01) \rightarrow k_{1A} = 1 \rightarrow \rho_{1A}^{(1)} = 0.4 \rightarrow t_{1A}^{(1)} = 3.2 \rightarrow \bar{u}^{(A)} = 2.75 \rightarrow u_{1A}^* = 2.80$
</p>
<h2 id="result">Result</h2>
<p>After $M$
 iterations, you would have a collection of $M$
 sampled $u_{1A}^*$
 values: $\{u_{1A}^{*(1)}, u_{1A}^{*(2)}, \ldots, u_{1A}^{*(M)}\}$
.</p>
<p>This collection of samples empirically approximates the posterior predictive distribution $p(u_{1A}^* \mid u_{1A}, s_{1A})$
. You can then:</p>
<ol>
<li>Plot a histogram of these samples to visualize the distribution.</li>
<li>Calculate the mean of these samples (e.g., $(3.25 + 2.80 + \ldots)/M$
) as your model&rsquo;s best prediction for a new unspliced measurement.</li>
<li>Calculate the standard deviation or percentiles to understand the uncertainty in the prediction.</li>
</ol>
<p>This entire process allows VeloVI to &ldquo;imagine&rdquo; what the unspliced counts should be, given its understanding of the cell&rsquo;s underlying biological state and the learned kinetic parameters, thereby providing a powerful way to interpret its learned representations.</p>

      </div>

      <div class="mt-16"></div>

      <div class="container mx-auto prose prose-slate lg:prose-xl dark:prose-invert mt-5">
        
        <div class="max-w-prose print:hidden">
  
  

  


  
<section class="flex flex-row flex-wrap justify-center pt-4 text-xl">
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F11b.velocity_velovi%2Fvelovi_posterior_predictive_distribution%2F&amp;text="
    title="X"
    aria-label="X"
    id="share-link-x"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://www.facebook.com/sharer.php?u=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F11b.velocity_velovi%2Fvelovi_posterior_predictive_distribution%2F&amp;t="
    title="Facebook"
    aria-label="Facebook"
    id="share-link-facebook"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="currentColor" d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12c0 4.84 3.44 8.87 8 9.8V15H8v-3h2V9.5C10 7.57 11.57 6 13.5 6H16v3h-2c-.55 0-1 .45-1 1v2h3v3h-3v6.95c5.05-.5 9-4.76 9-9.95z"/></svg>
  </a>
  

  
  
  
  
  
  
    
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="mailto:?subject=&amp;body=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F11b.velocity_velovi%2Fvelovi_posterior_predictive_distribution%2F"
    title="Email"
    aria-label="Email"
    id="share-link-email"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-width="1.5" d="M16.5 12a4.5 4.5 0 1 1-9 0a4.5 4.5 0 0 1 9 0Zm0 0c0 1.657 1.007 3 2.25 3S21 13.657 21 12a9 9 0 1 0-2.636 6.364M16.5 12V8.25"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="https://www.linkedin.com/shareArticle?url=http%3A%2F%2Flocalhost%3A1313%2Fpost%2F11b.velocity_velovi%2Fvelovi_posterior_predictive_distribution%2F&amp;title="
    title="LinkedIn"
    aria-label="LinkedIn"
    id="share-link-linkedin"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" height="1em" viewBox="0 0 448 512"><path fill="currentColor" d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
  </a>
  

  
  
  
  
  
  
  <a
    target="_blank" rel="noopener"
    class="m-1 rounded-md bg-neutral-300 p-1.5 text-neutral-700 hover:bg-primary-500 hover:text-neutral-300 dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800"
    href="whatsapp://send?text=%20http%3A%2F%2Flocalhost%3A1313%2Fpost%2F11b.velocity_velovi%2Fvelovi_posterior_predictive_distribution%2F"
    title="WhatsApp"
    aria-label="WhatsApp"
    id="share-link-whatsapp"
  >
    <svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256" fill="currentColor"><path d="m187.58 144.84-32-16a8 8 0 0 0-8 .5l-14.69 9.8a40.55 40.55 0 0 1-16-16l9.8-14.69a8 8 0 0 0 .5-8l-16-32A8 8 0 0 0 104 64a40 40 0 0 0-40 40 88.1 88.1 0 0 0 88 88 40 40 0 0 0 40-40 8 8 0 0 0-4.42-7.16ZM152 176a72.08 72.08 0 0 1-72-72 24 24 0 0 1 19.29-23.54l11.48 23L101 118a8 8 0 0 0-.73 7.51 56.47 56.47 0 0 0 30.15 30.15A8 8 0 0 0 138 155l14.61-9.74 23 11.48A24 24 0 0 1 152 176ZM128 24a104 104 0 0 0-91.82 152.88l-11.35 34.05a16 16 0 0 0 20.24 20.24l34.05-11.35A104 104 0 1 0 128 24Zm0 192a87.87 87.87 0 0 1-44.06-11.81 8 8 0 0 0-6.54-.67L40 216l12.47-37.4a8 8 0 0 0-.66-6.54A88 88 0 1 1 128 216Z"/></svg>
  </a>
  
</section>


  








  
  



  
  
  
    
  
  
  

<div class="flex pt-12 pb-4">
  
  
  <img
    class="mr-4 h-24 w-24 rounded-full"
    width="96"
    height="96"
    alt="Jiyuan Liu"
  src="/author/jiyuan-liu/avatar_hu6328122812819896191.jpg"
  loading="lazy"
  />
  
  <div class="place-self-center">
    <div class="text-[0.6rem] uppercase leading-3 text-neutral-500 dark:text-neutral-400">
      Authors
    </div>
    <div class="font-semibold leading-6 text-neutral-800 dark:text-neutral-300">
      <a href="http://localhost:1313/" class="no-underline">
      Jiyuan Liu
      </a>
    </div>

    
    <div class="text-sm font-bold text-neutral-700 dark:text-neutral-300">
    Postdoctoral Scientist
    </div>
    


    

    <div class="text-2xl sm:text-lg pt-1">

      
<div class="flex flex-wrap text-neutral-500 dark:text-neutral-300">
  
    
    
    
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="mailto:jliu78@mgh.harvard.edu"
      
      aria-label="At-Symbol"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-width="1.5" d="M16.5 12a4.5 4.5 0 1 1-9 0a4.5 4.5 0 0 1 9 0Zm0 0c0 1.657 1.007 3 2.25 3S21 13.657 21 12a9 9 0 1 0-2.636 6.364M16.5 12V8.25"/></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://github.com/stars/jliu678/lists/owned-contributed-by-me"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Brands/Github"
    ><svg style="height: 1em;" fill="currentColor" viewBox="3 3 18 18"><path d="M12 3C7.0275 3 3 7.12937 3 12.2276C3 16.3109 5.57625 19.7597 9.15374 20.9824C9.60374 21.0631 9.77249 20.7863 9.77249 20.5441C9.77249 20.3249 9.76125 19.5982 9.76125 18.8254C7.5 19.2522 6.915 18.2602 6.735 17.7412C6.63375 17.4759 6.19499 16.6569 5.8125 16.4378C5.4975 16.2647 5.0475 15.838 5.80124 15.8264C6.51 15.8149 7.01625 16.4954 7.18499 16.7723C7.99499 18.1679 9.28875 17.7758 9.80625 17.5335C9.885 16.9337 10.1212 16.53 10.38 16.2993C8.3775 16.0687 6.285 15.2728 6.285 11.7432C6.285 10.7397 6.63375 9.9092 7.20749 9.26326C7.1175 9.03257 6.8025 8.08674 7.2975 6.81794C7.2975 6.81794 8.05125 6.57571 9.77249 7.76377C10.4925 7.55615 11.2575 7.45234 12.0225 7.45234C12.7875 7.45234 13.5525 7.55615 14.2725 7.76377C15.9937 6.56418 16.7475 6.81794 16.7475 6.81794C17.2424 8.08674 16.9275 9.03257 16.8375 9.26326C17.4113 9.9092 17.76 10.7281 17.76 11.7432C17.76 15.2843 15.6563 16.0687 13.6537 16.2993C13.98 16.5877 14.2613 17.1414 14.2613 18.0065C14.2613 19.2407 14.25 20.2326 14.25 20.5441C14.25 20.7863 14.4188 21.0746 14.8688 20.9824C16.6554 20.364 18.2079 19.1866 19.3078 17.6162C20.4077 16.0457 20.9995 14.1611 21 12.2276C21 7.12937 16.9725 3 12 3Z"></path></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://www.linkedin.com/in/jiyuan-liu-7608a42a6/"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Brands/Linkedin"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" height="1em" viewBox="0 0 448 512"><path fill="currentColor" d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg></a>
  
    
    
    
    
      
    
    <a
      class="pr-2 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400"
      style="will-change:transform;"
      href="https://scholar.google.com/citations?view_op=list_works&amp;hl=en&amp;hl=en&amp;user=5c1kGO4AAAAJ&amp;sortby=pubdate"
      target="_blank" rel="noopener" rel="me noopener noreferrer"
      aria-label="Academicons/Google-Scholar"
    ><svg style="height: 1em;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M343.759 106.662V79.43L363.524 64h-213.89L20.476 176.274h85.656a82.339 82.339 0 0 0-.219 6.225c0 20.845 7.22 38.087 21.672 51.861c14.453 13.797 32.252 20.648 53.327 20.648c4.923 0 9.75-.368 14.438-1.024c-2.907 6.5-4.374 12.523-4.374 18.142c0 9.875 4.499 20.43 13.467 31.642c-39.234 2.67-68.061 9.732-86.437 21.163c-10.531 6.5-19 14.704-25.39 24.531c-6.391 9.9-9.578 20.515-9.578 31.962c0 9.648 2.062 18.336 6.219 26.062c4.156 7.726 9.578 14.07 16.312 18.984c6.718 4.968 14.469 9.101 23.219 12.469c8.734 3.344 17.406 5.718 26.061 7.062A167.052 167.052 0 0 0 180.555 448c13.469 0 26.953-1.734 40.547-5.187c13.562-3.485 26.28-8.642 38.171-15.493c11.86-6.805 21.515-16.086 28.922-27.718c7.39-11.68 11.094-24.805 11.094-39.336c0-11.016-2.25-21.039-6.75-30.14c-4.468-9.073-9.938-16.542-16.452-22.345c-6.501-5.813-13-11.155-19.516-15.968c-6.5-4.845-12-9.75-16.468-14.813c-4.485-5.046-6.735-10.054-6.735-14.984c0-4.921 1.734-9.672 5.216-14.265c3.455-4.61 7.674-9.048 12.61-13.306c4.937-4.25 9.875-8.968 14.796-14.133c4.922-5.147 9.141-11.827 12.61-20.008c3.485-8.18 5.203-17.445 5.203-27.757c0-13.453-2.547-24.46-7.547-33.314c-.594-1.022-1.218-1.803-1.875-3.022l56.907-46.672v17.119c-7.393.93-6.624 5.345-6.624 10.635V245.96c0 5.958 4.875 10.834 10.834 10.834h3.989c5.958 0 10.833-4.875 10.833-10.834V117.293c0-5.277.778-9.688-6.561-10.63zm-107.36 222.48c1.14.75 3.704 2.78 7.718 6.038c4.05 3.243 6.797 5.695 8.266 7.414a443.553 443.553 0 0 1 6.376 7.547c2.813 3.375 4.718 6.304 5.718 8.734c1 2.477 2.016 5.461 3.047 8.946a38.27 38.27 0 0 1 1.485 10.562c0 17.048-6.564 29.68-19.656 37.859c-13.125 8.18-28.767 12.274-46.938 12.274c-9.187 0-18.203-1.093-27.063-3.196c-8.843-2.116-17.311-5.336-25.39-9.601c-8.078-4.258-14.577-10.204-19.5-17.797c-4.938-7.64-7.407-16.415-7.407-26.25c0-10.32 2.797-19.29 8.422-26.906c5.594-7.625 12.938-13.391 22.032-17.315c9.063-3.946 18.25-6.742 27.562-8.398a157.865 157.865 0 0 1 28.438-2.555c4.47 0 7.936.25 10.405.696c.455.219 3.032 2.07 7.735 5.563c4.704 3.462 7.625 5.595 8.75 6.384zm-3.359-100.579c-7.406 8.86-17.734 13.288-30.953 13.288c-11.86 0-22.298-4.764-31.266-14.312c-9-9.523-15.422-20.328-19.344-32.43c-3.937-12.109-5.906-23.984-5.906-35.648c0-13.694 3.596-25.352 10.781-34.976c7.187-9.65 17.5-14.485 30.938-14.485c11.875 0 22.374 5.038 31.437 15.157c9.094 10.085 15.61 21.413 19.517 33.968c3.922 12.54 5.873 24.53 5.873 35.984c0 13.446-3.702 24.61-11.076 33.454z"/></svg></a>
  
</div>



    </div>
  </div>
</div>







  
  
    
    
    
      
      
    
<div class="pt-1 no-prose w-full">
  <hr class="border-dotted border-neutral-300 dark:border-neutral-600" />
  <div class="flex flex-col md:flex-row flex-nowrap justify-between gap-5 pt-2">
    <div class="">
      
        <a class="group flex no-underline" href="/post/11b.velocity_velovi/velovi_parameter_initialize_n_learn/">
          <span
            class="mt-[-0.3rem] me-2 text-neutral-700 transition-transform group-hover:-translate-x-[2px] group-hover:text-primary-600 dark:text-neutral dark:group-hover:text-primary-400"
          ><span class="ltr:inline rtl:hidden">&larr;</span></span>
          <span class="flex flex-col">
            <span
              class="mt-[0.1rem] leading-6 group-hover:underline group-hover:decoration-primary-500"
            ></span>
            <span class="mt-[0.1rem] text-xs text-neutral-500 dark:text-neutral-400">
              
                Jan 1, 0001
              
            </span>
          </span>
        </a>
      
    </div>
    <div class="">
      
        <a class="group flex text-right no-underline" href="/post/11b.velocity_velovi/velovi_reparameterization_trick/">
          <span class="flex flex-col">
            <span
              class="mt-[0.1rem] leading-6 group-hover:underline group-hover:decoration-primary-500"
            ></span
            >
            <span class="mt-[0.1rem] text-xs text-neutral-500 dark:text-neutral-400">
              
                Jan 1, 0001
              
            </span>
          </span>
          <span
            class="mt-[-0.3rem] ms-2 text-neutral-700 transition-transform group-hover:-translate-x-[2px] group-hover:text-primary-600 dark:text-neutral dark:group-hover:text-primary-400"
          ><span class="ltr:inline">&rarr;</span></span>
        </a>
      
    </div>
  </div>
</div>



  


  



</div>

      </div>

    </main>
  </article>
</div>

    </div>
    <div class="page-footer">
      <footer class="container mx-auto flex flex-col justify-items-center text-sm leading-6 mt-24 mb-4 text-slate-700 dark:text-slate-200">

  












  
  
  
  
  














  
  
  

  
  
    
  
  
    
  

  

  
  <p class="powered-by text-center">
    © 2025 Me. This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-nd/4.0" rel="noopener noreferrer" target="_blank">CC BY NC ND 4.0</a>
  </p>
  

  <p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-nd/4.0" rel="noopener noreferrer" target="_blank" aria-label="Creative Commons">
      <i class="fab fa-creative-commons fa-2x" aria-hidden="true"></i>
      <i class="fab fa-creative-commons-by fa-2x" aria-hidden="true"></i>
      
        <i class="fab fa-creative-commons-nc fa-2x" aria-hidden="true"></i>
      
      
        <i class="fab fa-creative-commons-nd fa-2x" aria-hidden="true"></i>
      
    </a>
  </p>





  <p class="powered-by text-center">
    
    
    
      
      
      
      
      
      
      Published with <a href="https://hugoblox.com/?utm_campaign=poweredby" target="_blank" rel="noopener">Hugo Blox Builder</a> — the free, <a href="https://github.com/HugoBlox/hugo-blox-builder" target="_blank" rel="noopener">open source</a> website builder that empowers creators.
    
  </p>

</footer>

    </div>

    
    











  </body>
</html>
